

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Configuration parameters &mdash; Training Tutorials  documentation</title>
  

  
  
    <link rel="shortcut icon" href="_static/favicon.ico"/>
  

  

  
  
    
      <link rel="search" type="application/opensearchdescription+xml" title="Search within Training Tutorials  documentation" href="_static/opensearch.xml"/>
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/tipuesearch/tipuesearch.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Training Tutorials  documentation" href="index.html"/>
        <link rel="up" title="EMOD parameter reference" href="parameter-overview.html"/>
        <link rel="next" title="Campaign parameters" href="parameter-campaign.html"/>
        <link rel="prev" title="Demographics file structure and parameters" href="parameter-demographics.html"/>

<link href="https://fonts.googleapis.com/css?family=Lato|Montserrat|Open+Sans" rel="stylesheet">



  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Training Tutorials
          

          
            
            <img src="_static/IDM_white.png" class="logo" />
          
          </a>

          
            
            
              <div class="version">
                2.5
              </div>
            
          

          <div role="search">
  <form id='rtd-search-form' class='wy-form' action="search-local.html" method="get">
    <input type="text" name="q" id="tipue_search_input" autocomplete="off"
           placeholder="Search docs" required>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro-overview.html">Introduction to the software</a></li>
<li class="toctree-l1"><a class="reference internal" href="install-overview.html">Install the Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="using-quickstart.html">How to run Quick Start simulations</a></li>
<li class="toctree-l1"><a class="reference internal" href="intro-modeling.html">Introduction to disease modeling</a></li>
<li class="toctree-l1"><a class="reference internal" href="generic-simulation-tutorial-overview.html">Generic simulation tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="malaria-VM-tutorials-overview.html">Vector and malaria tutorials</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="parameter-overview.html">EMOD parameter reference</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="parameter-demographics.html">Demographics file structure and parameters</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Configuration parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="parameter-campaign.html">Campaign parameters</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="troubleshooting.html">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="glossary.html">Glossary</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Training Tutorials</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          





<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
  <li><a href="../index.html">IDM docs</a> &raquo;</li>
    <li><a href="index.html">Training</a> &raquo;</li>
      
          <li><a href="parameter-overview.html">EMOD parameter reference</a> &raquo;</li>
      
    <li>Configuration parameters</li>
      <li class="wy-breadcrumbs-aside">
        
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
<!-- From _templates/page.html -->

  <div class="section" id="configuration-parameters">
<h1>Configuration parameters<a class="headerlink" href="#configuration-parameters" title="Permalink to this headline">¶</a></h1>
<p>The parameters described in this reference section determine the behavior of a simulation.
Simulations are configured in a <a class="reference internal" href="glossary.html#term-configuration-file"><span class="xref std std-term">configuration file</span></a>. This file is a <a class="reference internal" href="glossary.html#term-javascript-object-notation-json"><span class="xref std std-term">JavaScript Object
Notation (JSON)</span></a> formatted file that contains mostly a flat list of JSON key-value pairs. For
information on JSON, see <a class="reference internal" href="parameter-overview.html"><span class="doc">EMOD parameter reference</span></a>.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false.</p>
</div>
<p>The tables below contain only parameters available when using the training tutorials.</p>
<p><strong>Contents</strong></p>
<ul class="simple">
<li><a class="reference internal" href="#config-clusteroptions"><span class="std std-ref">Cluster options</span></a></li>
<li><a class="reference internal" href="#config-demographics"><span class="std std-ref">Demographics</span></a></li>
<li><a class="reference internal" href="#config-eventrecorder"><span class="std std-ref">Event recorder report settings</span></a></li>
<li><a class="reference internal" href="#config-generaldisease"><span class="std std-ref">General disease</span></a></li>
<li><a class="reference internal" href="#config-geography"><span class="std std-ref">Geography and environment</span></a></li>
<li><a class="reference internal" href="#config-inputfiles"><span class="std std-ref">Input files</span></a></li>
<li><a class="reference internal" href="#config-malaria"><span class="std std-ref">Malaria</span></a></li>
<li><a class="reference internal" href="#config-malariadrug"><span class="std std-ref">Malaria drug efficacy</span></a></li>
<li><a class="reference internal" href="#config-migration"><span class="std std-ref">Migration</span></a></li>
<li><a class="reference internal" href="#config-outputoptions"><span class="std std-ref">Output options</span></a></li>
<li><a class="reference internal" href="#config-sampling"><span class="std std-ref">Sampling</span></a></li>
<li><a class="reference internal" href="#config-scalingfactors"><span class="std std-ref">Scaling factors</span></a></li>
<li><a class="reference internal" href="#config-simulationsetup"><span class="std std-ref">Simulation setup</span></a></li>
<li><a class="reference internal" href="#config-spatialoutput"><span class="std std-ref">Spatial output channels</span></a></li>
<li><a class="reference internal" href="#config-vector"><span class="std std-ref">Vector</span></a></li>
<li><a class="reference internal" href="#config-vectorspecies"><span class="std std-ref">Vector species</span></a></li>
</ul>
<div class="section" id="cluster-options">
<span id="config-clusteroptions"></span><h2><a class="toc-backref" href="#id1">Cluster options</a><a class="headerlink" href="#cluster-options" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Load_Balance_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>Path to input file used when a static load balancing scheme is selected.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Num_Cores</strong></td>
<td>integer</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>Number of cores used to run a simulation. This is used by the infrastructure that runs the job, such as an HPC cluster or the Regression scripts, and is not used by the DTK.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="demographics">
<span id="config-demographics"></span><h2><a class="toc-backref" href="#id2">Demographics</a><a class="headerlink" href="#demographics" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Age_Initialization_Distribution_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>DISTRIBUTION_OFF</td>
<td>Method for initializing the age distribution in the simulated population. Possible values are: DISTRIBUTION_OFF - All individuals default to age 20-years-old. DISTRIBUTION_SIMPLE - Individual ages are drawn from a distribution whose functional form and parameters are specified in the IndividualAttributes demographics input layers: AgeDistributionFlag, AgeDistribution1, and AgeDistribution2. DISTRIBUTION_COMPLEX - Individual ages are drawn from a piecewise linear distribution specified in the IndividualAttributes demographics input layer AgeDistribution.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Population_Scale_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>If Population_Scale_Type is set to FIXED_SCALING, the initial simulation population is uniformly scaled over the entire area to adjust for historical or future population density.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Birth_Rate_Dependence</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>FIXED_BIRTH_RATE</td>
<td>This parameter determines how the birth rate specified in the demographics file is used. Possible values are: FIXED_BIRTH_RATE- The absolute rate at which new individuals are born. POPULATION_DEP_RATE - Scales the node population to determine the birth rate. DEMOGRAPHIC_DEP_RATE - Scales the female population within fertility age ranges to determine the birth rate. INDIVIDUAL_PREGNANCIES - Results in a similar birth rate as DEMOGRAPHIC_DEP_RATE, but pregnancies are assigned on an individual basis and result  in a 40-week pregnancy for a specific individual with a birth at the end, INDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE, and INDIVIDUAL_PREGNANCIES_BY_AGE_AND_YEAR.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Death_Rate_Dependence</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONDISEASE_MORTALITY_OFF</td>
<td>This parameter determines how the death rate specified in the demographics file is used. Possible values are: NONDISEASE_MORTALITY_OFF, NONDISEASE_MORTALITY_BY_AGE_AND_GENDER and NONDISEASE_MORTALITY_BY_YEAR_AND_AGE_FOR_EACH_GENDER.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Aging</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 to account for aging in the population. Set to 0 to assume that the population does not age.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Birth</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 to enable individuals to be added to the simulation by birth.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Demographics_Birth</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Assumes that newborn characteristics such as sickle-cell status are heterogeneous. Set to 0 to assume by default that all newborns have identical characteristics.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Demographics_Builtin</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Uses demographics input files to configure the initial population. Enter 0 to use a standard testing sandbox which is useful for testing and validating code pathways.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Demographics_Gender</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Enable_Demographics_Gender=0 assigns gender based on a 50/50 draw, while Enable_Demographics_Gender=1 draws from a male/female ratio that is randomly smeared by a Gaussian of width 1%.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Demographics_Other</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Includes the impact of other relevant demographic factors. For example, fraction of individuals above poverty, urban/rural characteristics, heterogeneous initial immunity and risk. Set to 0 to run the simulation without other relevant demographic factors.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Disease_Mortality</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 to enable disease mortality. Set to 0 to disable disease mortality.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Heterogeneous_Intranode_Transmission</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to enable heterogeneous intra-node disease transmission. Requires individual property definitions and beta matrix to be specified. Set to 0 to disable.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Vital_Dynamics</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 as a master switch to enable vital dynamics (births and deaths). Set to 0 to disable vital dynamics. Even when set to 1, the individual toggles for births and deaths can still disable these.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Population_Scale_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>USE_INPUT_FILE</td>
<td>Either use the initial population specified in the demographics input file or a fixed scaling of this value based on the population scaling factor parameter. Possible values are: USE_INPUT_FILE and FIXED_SCALING.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="event-recorder-report-settings">
<span id="config-eventrecorder"></span><h2><a class="toc-backref" href="#id3">Event recorder report settings</a><a class="headerlink" href="#event-recorder-report-settings" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Report_Event_Recorder</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Enables or disables the ReportEventRecorder.csv report.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Report_Event_Recorder_Ignore_Events_In_List</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>f this flag is set to true, all events listed in Report_Event_Recorder_Events will be ignored. If it is set to false, all events will be recorded.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Report_Event_Recorder_Individual_Properties</strong></td>
<td>Dynamic String Set</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>Fraction of individuals in the target demographic that will receive this intervention.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="general-disease">
<span id="config-generaldisease"></span><h2><a class="toc-backref" href="#id4">General disease</a><a class="headerlink" href="#general-disease" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Acquisition_Blocking_Immunity_Duration_Before_Decay</strong></td>
<td>float</td>
<td>0</td>
<td>45000</td>
<td>0</td>
<td>Number of days after infection until acquisition-blocking immunity begins to decay. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Animal_Reservoir_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_ZOONOSIS</td>
<td>Configures whether there is an animal reservoir and how the risk of zoonosis in configured. Use of the animal reservoir sets a low constant baseline of infectivity beyond what is present in the human population. It allows a more random introduction of cases in continuous time, which is more applicable for various situations such as zoonosis. Possible values are: NO_ZOONOSIS, CONSTANT_ZOONOSIS where the daily rate of zoonotic infection is configured by the parameter Zoonosis_Rate, and ZOONOSIS_FROM_DEMOGRAPHICS where the zoonosis rate is additionally scaled by the node-specific Zoonosis value in the NodeAttributes section of a demographics overlay file.  See Zoonosis_Rate.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Base_Incubation_Period</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>Average duration, in days, of the incubation period before infected individuals becomes infectious.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Infectious_Period</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>Average duration, in days, of the infectious period before the infection is cleared.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Base_Infectivity</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.3</td>
<td>The Base_Infectivity parameter determines the base infectiousness of individuals before accounting for transmission-blocking effects of acquired immunity and/or campaign interventions. It has a slightly different meaning for each of the disease types. GENERIC_SIM: The average number of individuals per time step who will be exposed to infection by one infectious individual. In the case of super-infection, the infectiousness is summed over all infections. VECTOR_SIM: The probability of infecting a mosquito during a successful blood meal (modulated by the vector parameter Acquire_Modifier). For vector simulations, the sum infectiousness of an individual is not allowed to exceed 100%. MALARIA_SIM: This simple scale factor is not used. Instead, gametocyte abundances and cytokine-mediated infectiousness are modeled explicitly.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Mortality</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>The base mortality of the infection before accounting for individual immune modification factors. Depending on the setting of Mortality_Time_Course, this is either the daily probability of the disease being fatal (DAILY_MORTALITY) or the probability of death at the end of the infection duration (MORTALITY_AFTER_INFECTIOUS).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Immune_Decay</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 if immunity decays after an infection clears. Set to 0 if recovery from the disease confers complete immunity for life.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Immunity</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 if an individual has protective immunity after an infection clears. Set to 0 if recovery from the infection does not affect the individual&#8217;s future immunity.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Maternal_Transmission</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to account for infection of infants at birth by infected mothers. Set to 0 to assume that infection is not transmitted by mothers to infants at birth.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Superinfection</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 if an individual can have multiple infections with the same agent simultaneously. Set to 0 if multiple infections are not possible. See the Max_Individual_Infections parameter.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Immunity_Acquisition_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>Multiplicative reduction in probability of reacquiring the disease. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Immunity_Initialization_Distribution_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>DISTRIBUTION_OFF</td>
<td>Method for initializing the immunity distribution in the simulated population. Possible values are: DISTRIBUTION_OFF - All individuals default to no immunity. DISTRIBUTION_SIMPLE - Individual immunities are drawn from a distribution whose functional form and parameters are specified in the IndividualAttributes demographics input layers: ImmunityDistributionFlag, ImmunityDistribution1, and ImmunityDistribution2. DISTRIBUTION_COMPLEX - Individual immnunities are drawn from an age-dependent piecewise l.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Immunity_Mortality_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>Multiplicative reduction in probability of dying from infection after getting re-infected. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Immunity_Transmission_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>Multiplicative reduction in probability of transmitting infection after getting re-infected. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Incubation_Period_Distribution</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NOT_INITIALIZED</td>
<td>Distribution of duration of incubation period. Can be fixed or exponentially distributed with an average duration from Base_Incubation_Period. Possible values are: FIXED_DURATION and EXPONENTIAL_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Incubation_Period_Max</strong></td>
<td>float</td>
<td>0.6</td>
<td>3.40E+38</td>
<td>0</td>
<td>The maximum length of the incubation period. Used when Incubation_Period_Distribution is UNIFORM_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Incubation_Period_Mean</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>The standard deviation used when Incubation_Period_Distribution is either GAUSSIAN_DURATION or POISSON_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Incubation_Period_Min</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The minimum length of the incubation period. Used when the Incubation_Period_Distribution is UNIFORM_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Incubation_Period_Std_Dev</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>The standard deviation incubation period. Used when the Incubation_Period_Distribution is GAUSSIAN_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Infection_Updates_Per_Timestep</strong></td>
<td>integer</td>
<td>0</td>
<td>144</td>
<td>1</td>
<td>Number of infection updates executed during each timestep. A timestep defaults to one day.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Infectious_Period_Distribution</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NOT_INITIALIZED</td>
<td>Distribution of duration of infectious period. Can be fixed or exponentially distributed with an average duration from Base_Infectious_Period. Possible values are: FIXED_DURATION, EXPONENTIAL_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Infectious_Period_Max</strong></td>
<td>float</td>
<td>0.6</td>
<td>3.40E+38</td>
<td>0</td>
<td>The maximum length of the infectious period. Used when Infectious_Period_Distribution is UNIFORM_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Infectious_Period_Mean</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>The mean of the infectious period. Used when Infectious_Period_Distribution is either GAUSSIAN_DURATION or POISSON_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Infectious_Period_Min</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The minimum length of the infectious period. Used when the Infectious_Period_Distribution is UNIFORM_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Infectious_Period_Std_Dev</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>The standard deviation of the infectious period. Used when the Infectious_Period_Distribution is GAUSSIAN_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Infectivity_Scale_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_INFECTIVITY</td>
<td>Alter infectivity by time or season.  Possible values are: CONSTANT_INFECTIVITY - No infectivity correction is applied. FUNCTION_OF_TIME_AND_LATITUDE - Infectivity is corrected for approximate seasonal forcing. The use of a seasonal infectivity correction is a proxy for the effects of varying climate. From October through March, infectivity increases in the Northern Hemisphere and decreases in the Southern Hemisphere. From April through September, the trend reverses. Regions closer to the equator have reduced forcing compared to temperate regions.  This is not a substitute for the weather-driven vector dynamics of vector-borne and malaria simulations.  FUNCTION_OF_CLIMATE - Allows infectivity to be modulated by weather directly, for example, relative humidity in airborne simulations or rainfall in environmental simulations.  There is no default climate dependence enabled for generic simulations.  EXPONENTIAL_FUNCTION_OF_TIME - To facilitate certain burn-in scenarios, infectivity ramps up from zero at the beginning of the simulation according to the functional form, 1-exp(-rate*time), where the rate is specified by the parameter Infectivity_Scaling_Rate.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Maternal_Transmission_Probability</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Probability of transmission of infection from mother to infant at birth. Relevant only if Enable_Maternal_Transmission is set to 1. Note: This parameter should be set to 0 and ignored for malaria and vector simulations.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Max_Individual_Infections</strong></td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>Limit on the number of infections that an individual can have simultaneously. Note: Relevant only if the Enable_Superinfection parameter is set to 1 to allow multiple infections.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Mortality_Blocking_Immunity_Decay_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>Rate at which mortality-blocking immunity decays after the mortality-blocking immunity offset period. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Mortality_Time_Course</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>DAILY_MORTALITY</td>
<td>Determines whether disease deaths are calculated on every time step or once at the end of the disease duration. Possible values are: DAILY_MORTALITY and MORTALITY_AFTER_INFECTIOUS.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Number_Basestrains</strong></td>
<td>integer</td>
<td>1</td>
<td>10</td>
<td>1</td>
<td>The number of base strains in the simulation, such as antigenic variants.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Number_Substrains</strong></td>
<td>integer</td>
<td>1</td>
<td>16777200</td>
<td>256</td>
<td>The number of disease substrains for each base strain, such as genetic variants.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Population_Density_Infectivity_Correction</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_INFECTIVITY</td>
<td>Correction to alter infectivity by population density set in the Population_Density_C50 parameter. Measured in people per sq. km. Possible values are:  CONSTANT_INFECTIVITY and SATURATING_FUNCTION_OF_DENSITY.  Note: Sparsely populated areas have a lower infectivity, while densely populated areas have a higher infectivity which rises to saturate at the Base_Infectivity value.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Transmission_Blocking_Immunity_Decay_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>Rate at which transmission-blocking immunity decays after the base transmission-blocking immunity offset period. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Transmission_Blocking_Immunity_Duration_Before_Decay</strong></td>
<td>float</td>
<td>0</td>
<td>45000</td>
<td>0</td>
<td>Number of days after infection until transmission-blocking immunity begins to decay. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Zoonosis_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The daily rate of zoonotic infection per individual when the Animal_Reservoir_Type parameter is set to either CONSTANT_ZOONOSIS or ZOONOSIS_FROM_DEMOGRAPHICS.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="geography-and-environment">
<span id="config-geography"></span><h2><a class="toc-backref" href="#id5">Geography and environment</a><a class="headerlink" href="#geography-and-environment" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Air_Temperature_Variance</strong></td>
<td>float</td>
<td>0</td>
<td>5</td>
<td>2</td>
<td>The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily air temperature values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA.  Only used if the Enable_Climate_Stochasticity is set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Air_Temperature</strong></td>
<td>float</td>
<td>-55</td>
<td>45</td>
<td>22</td>
<td>The value of air temperature where Climate_Model is set to CLIMATE_CONSTANT.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Base_Land_Temperature</strong></td>
<td>float</td>
<td>-55</td>
<td>60</td>
<td>26</td>
<td>The value of land temperature where Climate_Model is set to CLIMATE_CONSTANT.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Rainfall</strong></td>
<td>float</td>
<td>0</td>
<td>150</td>
<td>10</td>
<td>The value of rainfall per day in mm when Climate_Model is set to CLIMATE_CONSTANT.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Base_Relative_Humidity</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.75</td>
<td>The value of humidity where Climate_Model is set to CLIMATE_CONSTANT.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Climate_Model</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CLIMATE_OFF</td>
<td>Determines how the climate of a simulation is configured and from what file(s). Possible values are: CLIMATE_OFF, CLIMATE_CONSTANT - Uses the conditional parameters that give the fixed values of temperature or rain for land temperature, air temperature, rainfall and humidity. CLIMATE_KOPPEN - Uses an input file that decodes Koppen codes by geographic location. CLIMATE_BY_DATA - Reads everything out of several input files with additional parameters that allow the addition of stochasticity or scale offsets.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Climate_Update_Resolution</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CLIMATE_UPDATE_YEAR</td>
<td>Update resolution for data in climate files. Possible values are: CLIMATE_UPDATE_YEAR, CLIMATE_UPDATE_MONTH, CLIMATE_UPDATE_WEEK, CLIMATE_UPDATE_DAY and CLIMATE_UPDATE_HOUR.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Default_Geography_Initial_Node_Population</strong></td>
<td>integer</td>
<td>0</td>
<td>1000000</td>
<td>1000</td>
<td>When using the default geography (i.e. Enable_Demographics_Initial = 0), this is the initial number of individuals in each node.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Default_Geography_Torus_Size</strong></td>
<td>integer</td>
<td>3</td>
<td>100</td>
<td>10</td>
<td>When using the default geography (i.e. Enable_Demographics_Initial = 0), this is the square root of the number of nodes in the simulation.  For migration, the nodes are assumed to be a torus.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Climate_Stochasticity</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls overall stochasticity for climate. Use value 0 to disable all additional climate stochasticity. Use value 1 to enable additional variation as specified by the parameters: Air_Temperature_Variance, Land_Temperature_Variance, Enable_Rainfall_Stochasticity and Relative_Humidity_Variance.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Rainfall_Stochasticity</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 0 to disable rainfall stochasticity.  Set to 1 to enable stochastic variation of rainfall drawn from an exponential distribution (with a mean value as the daily rainfall from the  Climate_Model values, CLIMATE_CONSTANT or CLIMATE_BY_DATA).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Land_Temperature_Offset</strong></td>
<td>float</td>
<td>-20</td>
<td>20</td>
<td>0</td>
<td>The linear shift of land surface temperature in degrees Celsius. This is only used or needed when Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Land_Temperature_Variance</strong></td>
<td>float</td>
<td>0</td>
<td>7</td>
<td>2</td>
<td>The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily land temperature values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA.  Only used if the Enable_Climate_Stochasticity is set to 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Node_Grid_Size</strong></td>
<td>float</td>
<td>0.004167</td>
<td>90</td>
<td>0.004167</td>
<td>Spatial resolution indicating the node grid size for a simulation in degrees.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Rainfall_In_mm_To_Fill_Swamp</strong></td>
<td>float</td>
<td>1</td>
<td>10000</td>
<td>1000</td>
<td>Millimeters of rain to fill larval habitat to capacity. Only used for vector species with Habitat_Type set to BRACKISH_SWAMP.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Rainfall_Scale_Factor</strong></td>
<td>float</td>
<td>0.1</td>
<td>10</td>
<td>1</td>
<td>The scalar used in multiplying rainfall value(s). This is only used or needed when Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Relative_Humidity_Scale_Factor</strong></td>
<td>float</td>
<td>0.1</td>
<td>10</td>
<td>1</td>
<td>The scalar used in multiplying relative humidity value(s). This is only used or needed when Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Relative_Humidity_Variance</strong></td>
<td>float</td>
<td>0</td>
<td>0.12</td>
<td>0.05</td>
<td>The standard deviation (in percentage, %) for normally distributed noise applied to the daily relative humidity values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA.  Only used if the Enable_Climate_Stochasticity is set to 1.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="input-files">
<span id="config-inputfiles"></span><h2><a class="toc-backref" href="#id6">Input files</a><a class="headerlink" href="#input-files" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Air_Migration_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>Path of input file defining patterns of migration by airplane.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Air_Temperature_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>air_temp.json</td>
<td>Path of input file defining air temperature data measured 2 meters above ground. Relevant only if Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Campaign_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>Path of the simulation campaign file.  This configuration file defines campaign related settings and thus, is an optional file, only required when interventions are part of the simulation.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Demographics_Filenames</strong></td>
<td>Vector String</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>Array of demographics files containing information on the identity and demographics of the region to simulate.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Koppen_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>Path to input file used to specify Koppen climate classifications. This is only used or needed when Climate_Model is set to CLIMATE_KOPPEN.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Land_Temperature_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>land_temp.json</td>
<td>Path of input file defining temperature data measured at land surface. Relevant only if Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Local_Migration_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>Path of input file defining patterns of migration to adjacent nodes by foot travel.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Rainfall_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>rainfall.json</td>
<td>Path of input file defining rainfall data. Relevant only if Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Regional_Migration_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>Path of input file defining patterns of migration by vehicle via road or rail network. If the node is not on a road or rail network, regional migration focuses on the closest hub city in the network.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Relative_Humidity_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>rel_hum.json</td>
<td>Path of input file defining relative humidity data measured 2 meters above ground. Relevant only if Climate_Model is set to CLIMATE_BY_DATA.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Sea_Migration_Filename</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>Path of input file defining patterns of migration by ship.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="malaria">
<span id="config-malaria"></span><h2>Malaria<a class="headerlink" href="#malaria" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Anemia_Mortality_Inverse_Width</strong></td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td>The inverse width relative to threshold value of severe disease turn-on around threshold for anemia.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Anemia_Mortality_Threshold</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>3</td>
<td>The threshold value on hemoglobin count in grams per deciliter (g/dL) at which 50% of individuals die per day.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Anemia_Severe_Inverse_Width</strong></td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td>The inverse width relative to threshold value of severe disease turn-on around threshold for anemia.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Anemia_Severe_Threshold</strong></td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>5</td>
<td>The severe disease threshold level for anemia. Threshold units are in grams per deciliter (g/dL).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Antibody_Capacity_Growth_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>Maximum daily rate of specific antibody capacity increase. Antibody production begins at capacity of 0.3. Hyperimmunity results at 0.4. The actual growth rate will tend to be lower and is a function of antigen concentrations.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Antibody_CSP_Decay_Days</strong></td>
<td>float</td>
<td>1</td>
<td>3.40E+38</td>
<td>90</td>
<td>Exponential decay time for CSP antibody concentration (in days).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Antibody_CSP_Killing_Inverse_Width</strong></td>
<td>float</td>
<td>1.00E-06</td>
<td>1000000</td>
<td>1.5</td>
<td>Inverse width of sigmoidal sporozoite killing function of CSP antibody concentration.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Antibody_CSP_Killing_Threshold</strong></td>
<td>float</td>
<td>1.00E-06</td>
<td>1000000</td>
<td>10</td>
<td>Threshold value on CSP antibody concentration for sporozoite killing.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Antibody_IRBC_Kill_Rate</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2</td>
<td>Factor multiplied by antibody level to produce the rate of clearance of the infected red blood cell (IRBC) population.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Antibody_Memory_Level</strong></td>
<td>float</td>
<td>0</td>
<td>0.35</td>
<td>0.2</td>
<td>Limiting level of antibody capacity that remains after a prolonged absence of a specific antigen. The antibody capacity decays to this level gradually after infection is cleared. The decay rate in antibody capacity is set so that hyperimmunity is lost within 4 months, and capacity continues to decay to this level. The antibody memory level is relevant for year-scale dynamics, but not for long dynamics (10-20 years).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Antibody_Stimulation_C50</strong></td>
<td>float</td>
<td>0.1</td>
<td>10000</td>
<td>10</td>
<td>Concentration of an antigen, measured in IRBC/ul at which growth in antibody capacity against the antigen increases at half the maximum rate specified by Antibody_Capacity_Growth_Rate.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Antigen_Switch_Rate</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2.00E-09</td>
<td>Antigenic switching rate per infected red blood cell per asexual cycle. See Parasite_Switch_Type for different switching patterns.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Base_Gametocyte_Fraction_Male</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.2</td>
<td>Fraction of gametocytes that are male.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Gametocyte_Mosquito_Survival_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.01</td>
<td>Average fraction of gametocytes in blood meal that are successful in infecting mosquito in the absence of other modulating effects, such as fever.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Base_Gametocyte_Production_Rate</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.02</td>
<td>Fraction of IRBCs producing gametocytes.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Base_Sporozoite_Survival_Fraction</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.25</td>
<td>Fraction of sporozoites that survive to infect a hepatocyte in the absence of anti-CSP protection.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Clinical_Fever_Threshold_High</strong></td>
<td>float</td>
<td>0</td>
<td>15</td>
<td>1</td>
<td>The fever threshold (in degrees Celsius above normal) to start a clinical incident.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Clinical_Fever_Threshold_Low</strong></td>
<td>float</td>
<td>0</td>
<td>5</td>
<td>1</td>
<td>The fever threshold (in degrees Celsius above normal) to end a clinical incident.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Cytokine_Gametocyte_Inactivation</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>The strength of inflammatory response inactivation of gametocytes.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Sexual_Combination</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to enable sexual combination of male and female gametocytes. Set to 0 for transmission of each strain only. Note: This parameter is currently not in use.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Erythropoiesis_Anemia_Effect</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>3.5</td>
<td>Exponential rate of increased red-blood-cell production from reduced red-blood-cell availability.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Falciparum_MSP_Variants</strong></td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>100</td>
<td>The number of distinct MSP variants for P. falciparum malaria in the overall parasite population in the simulation, not necessarily in an individual.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Falciparum_Nonspecific_Types</strong></td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>20</td>
<td>The number of distinct non-specific types of P. falciparum malaria.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Falciparum_PfEMP1_Variants</strong></td>
<td>integer</td>
<td>0</td>
<td>100000</td>
<td>1000</td>
<td>The number of distinct PfEMP1 variants for P. falciparum malaria in the overall parasite population in the simulation.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Fever_Detection_Threshold</strong></td>
<td>float</td>
<td>0.5</td>
<td>5</td>
<td>1</td>
<td>Level of body temperature above normal, in degrees Celsius, corresponding to detectable fever. Normal body temperature is defined as 37 degrees Celsius.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Fever_IRBC_Kill_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.15</td>
<td>Factor multiplied by fever level to produce a decay rate of infected red blood cell population due to inflammatory innate immune response.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Fever_Mortality_Inverse_Width</strong></td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td>The inverse width relative to threshold value of mortality turn-on around threshold for fever.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Fever_Mortality_Threshold</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>3</td>
<td>The fever mortality threshold for a simulation in units of degrees Celsius above normal body temperature. Sets the midpoint in the sigmoidal response for mortality rate associated with fever.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Fever_Severe_Inverse_Width</strong></td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td>The inverse width relative to threshold value of severe disease turn-on around threshold for fever.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Fever_Severe_Threshold</strong></td>
<td>float</td>
<td>0</td>
<td>10</td>
<td>1.5</td>
<td>The severe disease threshold level for fever.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Gametocyte_Stage_Survival_Rate</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>1</td>
<td>Rate of gametocyte survival from one development stage to the next in the absence of drugs or inflammatory immune response.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Innate_Immune_Variation_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONE</td>
<td>Type of individual innate immune variation.  Possible values are: NONE (homogeneous), PYROGENIC_THRESHOLD (parasite threshold resulting in cytokine production), PYROGENIC_THRESHOLD_VS_AGE (age-dependent threshold), CYTOKINE_KILLING (effectiveness of cytokines in killing infected red-blood cells).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Malaria_Model</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>MALARIA_MECHANISTIC_MODEL</td>
<td>The type of malaria model used in a simulation. Possible values are:  MALARIA_FIXED_DURATION, MALARIA_EXPONENTIAL_DURATION, MALARIA_REDUCEDSTATE_MODEL and MALARIA_MECHANISTIC_MODEL. Important: All malaria simulations should use MALARIA_MECHANISTIC_MODEL. The others are placeholder values.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Malaria_Strain_Model</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>FALCIPARUM_NONRANDOM_STRAIN</td>
<td>The generator that is used to construct the antigenic repertoire of a malaria infection. Possible values are: FALCIPARUM_NONRANDOM_STRAIN, FALCIPARUM_RANDOM50_STRAIN, FALCIPARUM_RANDOM_STRAIN and FALCIPARUM_STRAIN_GENERATOR.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Maternal_Antibodies_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>OFF</td>
<td>Type of maternal antibody protection.  Possible values are: OFF, SIMPLE_WANING, CONSTANT_INITIAL_IMMUNITY (independent of acquired immunity in possible mothers).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Maternal_Antibody_Decay_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0.01</td>
<td>Decay rate per day in protection from maternal antibodies.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Maternal_Antibody_Protection</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>Strength of protection from maternal antibodies as a multiple of full antibody killing effect.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Max_MSP1_Antibody_Growthrate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>Maximum increase in MSP1 antibody capacity during each asexual cycle. The higher this value, the sooner early clearances are observed and the earlier the parasite density envelope decreases.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Mean_Sporozoites_Per_Bite</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>11</td>
<td>Mean number of sporozoites per infectious mosquito bite.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Merozoites_Per_Hepatocyte</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>15000</td>
<td>Number of IRBCs caused by a single infected hepatocyte at the start of infection.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Merozoites_Per_Schizont</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>16</td>
<td>Number of merozoites released by a single infected schizont after each asexual cycle. The number of resulting IRBC&#8217;s depends on the RBC availability and merozoite-specific immunity.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Min_Adapted_Response</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>Minimum level of antibody stimulation to novel antigen. The value sets the low-range asymptote for antibody capacity growth, which is calculated from Antibody_Capacity_Growth_Rate and antigen density, in the presence of any nonzero antigen level.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Min_Days_Between_Clinical_Incidents</strong></td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>3</td>
<td>The number of days with fever below the low threshold before a new incident can start when the high fever threshold is again exceeded.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>MSP1_Merozoite_Kill_Fraction</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.5</td>
<td>Fraction of merozoites inhibited from invading new erythrocytes when MSP1-specific antibody level is 1.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>New_Diagnostic_Sensitivity</strong></td>
<td>float</td>
<td>0.0001</td>
<td>100000</td>
<td>0.01</td>
<td>The number of microliters of blood tested to find single parasites in a new diagnostic (corresponds to inverse parasites/microliters sensitivity).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Nonspecific_Antibody_Growth_Rate_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.5</td>
<td>Factor that adjusts Antibody_Capacity_Growth_Rate for less immunogenic surface proteins called the minor epitopes.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Nonspecific_Antigenicity_Factor</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.2</td>
<td>Nonspecific antigenicity factor that adjusts antibody IRBC killrate to account for IRBCs caused by antibody responses to antigenically weak surface proteins.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Number_Of_Asexual_Cycles_Without_Gametocytes</strong></td>
<td>integer</td>
<td>0</td>
<td>500</td>
<td>1</td>
<td>Number of asexual reproduction cycles that do not produce gametocytes. All later cycles will produce gametocytes according to Base_Gametocyte_Production_Rate.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Parasite_Mortality_Inverse_Width</strong></td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td>The inverse width relative to threshold value of mortality turn-on around threshold for parasite density.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Parasite_Mortality_Threshold</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2000000</td>
<td>The parasite-density mortality threshold for a simulation.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Parasite_Severe_Inverse_Width</strong></td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td>The inverse width relative to threshold value of severe disease turn-on around threshold for parasite density.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Parasite_Severe_Threshold</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1000000</td>
<td>The severe disease threshold level for parasite density.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Parasite_Smear_Sensitivity</strong></td>
<td>float</td>
<td>0.0001</td>
<td>100</td>
<td>0.1</td>
<td>The number of microliters of blood tested to find single parasites in a traditional smear (corresponds to inverse parasites/microliters sensitivity).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Parasite_Switch_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_SWITCH_RATE_2VARS</td>
<td>Parasite switch type for erythrocyte surface antigens. Possible values are: CONSTANT_SWITCH_RATE_2VARS, RATE_PER_PARASITE_7VARS - this is the one that is best parameterized - and finally,  RATE_PER_PARASITE_5VARS_DECAYING.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>PKPD_Model</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>FIXED_DURATION_CONSTANT_EFFECT</td>
<td>Determines which pharmacokinetic pharmacodynamic module to use. Possible values are: FIXED_DURATION_CONSTANT_EFFECT and CONCENTRATION_VERSUS_TIME.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Pyrogenic_Threshold</strong></td>
<td>float</td>
<td>0.1</td>
<td>20000</td>
<td>1000</td>
<td>Level of bloodstream infection, measured in IRBC per microliter, at which stimulation of the innate inflammatory immune response is half its maximum value.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>RBC_Destruction_Multiplier</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>9.5</td>
<td>Number of total RBCs destroyed per infected rupturing schizont.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="malaria-drug-efficacy">
<span id="config-malariadrug"></span><h2>Malaria drug efficacy<a class="headerlink" href="#malaria-drug-efficacy" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Bodyweight_Exponent</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>Effect of body weight on maximum drug concentration in an individual patient.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Drug_Cmax</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1000</td>
<td>Maximum drug concentration in the same units as Drug_PKPD_C50. Note: Since drug concentrations and C50 values are often specified with different unit systems, these two parameters have been designed to be flexible, provided that the units are the same for each.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Drug_Decay_T1</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1</td>
<td>Primary drug decay rate, in days.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Drug_Decay_T2</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1</td>
<td>Secondary drug decay rate, in days.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Drug_Dose_Interval</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1</td>
<td>Interval between doses, in days.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Drug_Fulltreatment_Doses</strong></td>
<td>float</td>
<td>1</td>
<td>100000</td>
<td>3</td>
<td>Number of doses for a full treatment.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Drug_Gametocyte02_Killrate</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>Log reduction per day in early-stage gametocytes.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Drug_Gametocyte34_Killrate</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>Log reduction per day in late-stage gametocytes.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Drug_GametocyteM_Killrate</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>Log reduction per day in mature gametocytes at saturated drug concentration.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Drug_Hepatocyte_Killrate</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>Log reduction in hepatocytes per day.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Drug_PKPD_C50</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>100</td>
<td>Concentration at which drug killing rates are half of maximum. Note: This parameter and the Drug_Cmax parameter must use the same units.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Drug_Vd</strong></td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>10</td>
<td>Volume of distribution. This value is the ratio of the volume of the second compartment to the volume of the first compartment in a two-compartment model, and is dimensionless.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Fraction_Of_Adult_Dose</strong></td>
<td>DoseMap</td>
<td>0</td>
<td>1</td>
<td>NA</td>
<td>The fraction of an adult drug dose given to children below the corresponding age (in years).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Fractional_Dose_By_Upper_Age</strong></td>
<td>array</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>An array of doses that contain Upper_Age_In_Years and Fraction_Of_Adult_Dose values, in days.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Malaria_Drug_Params</strong></td>
<td>JSON object</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>This JSON structure defines the name of an anti-malaria drug and the parameters that define it.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
   <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
       <span class="nt">&quot;Artemether&quot;</span><span class="p">:</span> <span class="p">{</span>
           <span class="nt">&quot;Bodyweight_Exponent&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Cmax&quot;</span><span class="p">:</span> <span class="mi">114</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Decay_T1&quot;</span><span class="p">:</span> <span class="mf">0.12</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Decay_T2&quot;</span><span class="p">:</span> <span class="mf">0.12</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Vd&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
           <span class="nt">&quot;Drug_PKPD_C50&quot;</span><span class="p">:</span> <span class="mf">0.6</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Dose_Interval&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Fulltreatment_Doses&quot;</span><span class="p">:</span> <span class="mi">6</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.5</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span>
           <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mf">0.7</span><span class="p">,</span>
           <span class="nt">&quot;Drug_Hepatocyte_Killrate&quot;</span><span class="p">:</span> <span class="mf">0.0</span><span class="p">,</span>
           <span class="nt">&quot;Max_Drug_IRBC_Kill&quot;</span><span class="p">:</span> <span class="mf">8.9</span><span class="p">,</span>
           <span class="nt">&quot;Fractional_Dose_By_Upper_Age&quot;</span><span class="p">:</span> <span class="p">[</span>
               <span class="p">{</span>
                   <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span>
                   <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">3</span>
               <span class="p">},</span>
               <span class="p">{</span>
                   <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
                   <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">6</span>
               <span class="p">},</span>
               <span class="p">{</span> 
                   <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.75</span><span class="p">,</span>
                   <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">10</span>
               <span class="p">},</span>
            <span class="p">]</span>
       <span class="p">}</span>
   <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>&nbsp;</td>
<td>&nbsp;</td>
<td>&nbsp;</td>
<td>&nbsp;</td>
<td>&nbsp;</td>
<td>&nbsp;</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Max_Drug_IRBC_Kill</strong></td>
<td>float</td>
<td>5</td>
<td>100000</td>
<td>5</td>
<td>Maximum log reduction in IRBCs per day.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Upper_Age_In_Years</strong></td>
<td>DoseMap</td>
<td>0</td>
<td>125</td>
<td>NA</td>
<td>The age (in years) below which children are given the corresponding fraction of an adult drug dose.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="migration">
<h2><a class="toc-backref" href="#id7">Migration</a><a class="headerlink" href="#migration" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils" id="config-migration">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Air_Migration_Roundtrip_Duration</strong></td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent at the destination node during a round-trip migration by airplane.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Air_Migration_Roundtrip_Probability</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.8</td>
<td>Likelihood that an individual who flies to another cell will return to the cell of origin during the next migration. Relevant only when Enable_Air_Migration is selected.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Air_Migration</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to account for migration by airplane in and out of cities with airports. Set to 0 to assume by default that no migration occurs by air travel.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Local_Migration</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to enable local migration, the diffusion of people in and out of nearby nodes by foot travel. Set to 0 to disable local migration.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Migration_Heterogeneity</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Set to 1 to use migration rate distribution in the demographics file. Set to 0 to assume by default that the same migration rate applies for all individuals.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Regional_Migration</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to account for migration by road vehicle in and out of nodal cities in the road network. Set to 0 to assume by default that no migration occurs by road travel.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Sea_Migration</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to account for migration on ships in and out of coastal cities with seaports. Set to 0 to assume by default that no migration occurs by sea travel.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Local_Migration_Roundtrip_Duration</strong></td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent at the destination node during a round-trip migration by foot travel.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Local_Migration_Roundtrip_Probability</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.95</td>
<td>Likelihood that an individual who walks into a neighboring cell will return to the cell of origin during the next migration. Relevant only when Enable_Local_Migration is selected.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Migration_Model</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_MIGRATION</td>
<td>Model to use for migration. Possible values are: NO_MIGRATION, FIXED_RATE_MIGRATION, VARIABLE_RATE_MIGRATION and LEVY_FLIGHTS.  Note: VARIABLE_RATE_MIGRATION and LEVY FLIGHTS are currently not supported.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Migration_Pattern</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>RANDOM_WALK_DIFFUSION</td>
<td>The type of roundtrip. For example, a single roundtrip. Possible values are: RANDOM_WALK_DIFFUSION, SINGLE_ROUND_TRIPS and WAYPOINTS_HOME.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Regional_Migration_Roundtrip_Duration</strong></td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent at the destination node during a round-trip migration by road network.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Regional_Migration_Roundtrip_Probability</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>Likelihood that an individual who travels by vehicle to another cell will return to the cell of origin during the next migration. Relevant only when Enable_Regional_Migration is checked.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Roundtrip_Waypoints</strong></td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>10</td>
<td>The maximum number of points reached during a trip before steps are retraced on the return trip home.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Sea_Migration_Roundtrip_Duration</strong></td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent at the destination node during a round-trip migration by ship.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Sea_Migration_Roundtrip_Probability</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.25</td>
<td>Likelihood that an individual who travels by ship into a neighboring cell will return to the cell of origin during the next migration. Relevant only when Enable_Sea_Migration is checked.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="output-options">
<span id="config-outputoptions"></span><h2><a class="toc-backref" href="#id8">Output options</a><a class="headerlink" href="#output-options" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Enable_Default_Reporting</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Enables and disables the default InsetChart.json report.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Demographics_Reporting</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Outputs demographic summary data and age-binned reports to file.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Property_Output</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Use value 0 to disable Individual_Property reports (separate file). Use value 1 to enable.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Spatial_Output</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Use value 0 to disable all spatial output channels. Use value 1 to enable spatial output of all channels listed in the array parameter: Spatial_Output_Channels.  For full documentation of spatial output channels, please see the online documentation.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Report_Event_Recorder</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Enables or disables the ReportEventRecorder.csv report.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="sampling">
<span id="config-sampling"></span><h2><a class="toc-backref" href="#id9">Sampling</a><a class="headerlink" href="#sampling" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Base_Individual_Sample_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Base rate of sampling for individuals. This rate equals the fraction of individuals in each node being sampled.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Individual_Sampling_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>TRACK_ALL</td>
<td>Type of individual human sampling. Possible values are: TRACK_ALL, FIXED_SAMPLING, ADAPTED_SAMPLING_BY_POPULATION_SIZE, ADAPTED_SAMPLING_BY_AGE_GROUP, and ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Max_Node_Population_Samples</strong></td>
<td>float</td>
<td>1</td>
<td>3.40E+38</td>
<td>30</td>
<td>Number of individuals when the sampling rate starts dropping to the rate of adapted sampling by population size.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="scaling-factors">
<span id="config-scalingfactors"></span><h2><a class="toc-backref" href="#id10">Scaling factors</a><a class="headerlink" href="#scaling-factors" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>x_Air_Migration</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Multiplier for rate of migration by air.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>x_Birth</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Multiplier of birth rate from input demographics file.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>x_Local_Migration</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Multiplier for rate of migration by foot travel.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>x_Other_Mortality</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Multiplier for scaling mortality from causes other than disease being simulated.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>x_Regional_Migration</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Multiplier for rate of migration by road vehicle.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>x_Sea_Migration</strong></td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Multiplier for rate of migration by sea.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>x_Temporary_Larval_Habitat</strong></td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>1</td>
<td>Scales the habitat size for all mosquito populations.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="simulation-setup">
<span id="config-simulationsetup"></span><h2><a class="toc-backref" href="#id11">Simulation setup</a><a class="headerlink" href="#simulation-setup" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Config_Name</strong></td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>User-supplied title naming a configuration.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Interventions</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to load campaign interventions from file. Set to 0 to run a baseline simulation without campaigns.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Run_Number</strong></td>
<td>integer</td>
<td>0</td>
<td>2147480000</td>
<td>1</td>
<td>Sets the random number seed through a bit manipulation process for USE_PSEUDO_DES. When running a multi-core simulation, combines with processor rank to produce independent random number streams for each process.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Simulation_Duration</strong></td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>1</td>
<td>Elapsed time noted in days from the start to the end of a simulation.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Simulation_Timestep</strong></td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>1</td>
<td>Value indicating simulation time step in days.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Simulation_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>GENERIC_SIM</td>
<td>Type of disease being simulated. Supported values include GENERIC_SIM, VECTOR_SIM, MALARIA_SIM, TB_SIM, STI_SIM, HIV_SIM, and PY_SIM. To query the values supported for your specific build, use the &#8211;version option.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Start_Time</strong></td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>1</td>
<td>Time noted in days when the simulation begins. This time influences the point in the temporal input data, such as where in the climate input the DTK starts running the simulation. Note: The Start_Day of campaign events is in absolute time. So, time relative to the beginning of the simulation depends on this parameter.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="spatial-output-channels">
<span id="config-spatialoutput"></span><h2><a class="toc-backref" href="#id12">Spatial output channels</a><a class="headerlink" href="#spatial-output-channels" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Spatial_Output_Channels</strong></td>
<td>Fixed String Set</td>
<td>NA</td>
<td>NA</td>
<td>[]</td>
<td>An array of channel names for spatial output by Node and time step. The data from each channel will be written to a separate binary file. Channel names include Air_Temperature, Births, Campaign_Cost, Disease_Deaths, Human_Infectious_Reservoir, Infection_Rate, Land_Temperature, New_Infections, New_Reported_Infections, Population, Rainfall and Relative_Humidity.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="vector">
<span id="config-vector"></span><h2>Vector<a class="headerlink" href="#vector" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Age_Dependent_Biting_Risk_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>OFF</td>
<td>Type of functional form for age-dependent biting-risk.  Possible values are: OFF, LINEAR, SURFACE_AREA_DEPENDENT.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Egg_Hatch_Delay_Distribution</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_DELAY</td>
<td>Is there a delay in egg hatching from oviposition? If so, how is that delay distributed? Possible values are: NO_DELAY and EXPONENTIAL_DURATION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Egg_Saturation_At_Oviposition</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_SATURATION</td>
<td>Is there saturation in the number of viable eggs at oviposition depending on available larval habitat? Possible values are: NO_SATURATION and SATURATION_AT_OVIPOSITION.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Temperature_Dependent_Feeding_Cycle</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to have the duration between blood-feeds depend on the temperature.  Otherwise, if set to 0, days between feeds will be constant and specified by the parameter, Days_Between_Feeds, for each species.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Vector_Aging</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to simulate vector senescence. Set to 0 to assume that vector behavior does not change with age.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Enable_Vector_Migration</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to enable vector migration. Set to 0 to disable vector migration. Note: Only supported for individual-vector model with Vector_Sampling_Type value of TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Enable_Vector_Migration_Local</strong></td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to 1 to consider local vector migration by vectors to adjacent nodes. Set to 0 to not consider local vector migration.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>HEG_Fecundity_Limiting</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The fractional reduction in the number of eggs laid by a female mosquito who is homozygous in the homing endonuclease gene (HEG).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>HEG_Homing_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The fractional redistribution from heterozygous offspring to the homozygous homing-endonuclease-gene (HEG) behavior.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>HEG_Model</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>OFF</td>
<td>An enum governing the homing endonuclease gene (HEG) dynamics. Possible values are: OFF, GERMLINE_HOMING (early), EGG_HOMING (late).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Human_Feeding_Mortality</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>Fraction (dimensionless) of mosquitoes that die while feeding on a human. As of version 1.5, this is now a one-value-per-simulation quantity.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Larval_Density_Dependence</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>UNIFORM_WHEN_OVERPOPULATION</td>
<td>Set to UNIFORM_WHEN_OVERPOPULATION to apply mortality uniformly to all larvae when the population exceeds the specified carrying capacity for that habitat. Set to GRADUAL_INSTAR_SPECIFIC to apply mortality and delayed growth in an instar-specific fashion where the younger larvae are more susceptible to predation and competition from older larvae. The LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY and DENSITY_DELAYED_GROWTH_NOT_MORTALITY options configure only the age-dependent mortality component or only the delayed-growth component, respectively. For the NO_DENSITY_DEPENDENCE option, there is no additional larval density-dependent mortality factor. Possible values are: UNIFORM_WHEN_OVERPOPULATION, GRADUAL_INSTAR_SPECIFIC, LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY, DENSITY_DELAYED_GROWTH_NOT_MORTALITY, NO_DENSITY_DEPENDENCE.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Larval_Density_Mortality_Offset</strong></td>
<td>float</td>
<td>0.0001</td>
<td>1000</td>
<td>0.1</td>
<td>An offset factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Larval_Density_Mortality_Scalar</strong></td>
<td>float</td>
<td>0.01</td>
<td>1000</td>
<td>10</td>
<td>A scale factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Larval_Rainfall_Mortality_Threshold</strong></td>
<td>float</td>
<td>0.01</td>
<td>1000</td>
<td>100</td>
<td>The threshold value on daily rainfall in millimeters, above which larval mortality is applied when Vector_Larval_Rainfall_Mortality is set to either SIGMOID or SIGMOID_HABITAT_SHIFTING.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Mean_Egg_Hatch_Delay</strong></td>
<td>float</td>
<td>0</td>
<td>120</td>
<td>0</td>
<td>Mean delay in egg hatch time noted in days from the time of oviposition.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Mosquito_Weight</strong></td>
<td>integer</td>
<td>1</td>
<td>10000</td>
<td>1</td>
<td>Value indicating how many mosquitoes are represented by a sample mosquito.  This parameter is only used for Vector_Sampling_Type value SAMPLE_IND_VECTORS.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Rainfall_In_mm_To_Fill_Swamp</strong></td>
<td>float</td>
<td>1</td>
<td>10000</td>
<td>1000</td>
<td>Millimeters of rain to fill larval habitat to capacity. Only used for vector species with Habitat_Type set to BRACKISH_SWAMP.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Semipermanent_Habitat_Decay_Rate</strong></td>
<td>float</td>
<td>0.0001</td>
<td>100</td>
<td>0.01</td>
<td>Daily rate of larval habitat loss for semi-permanent habitats with Habitat_Type parameter of WATER_VEGETATION or BRACKISH_SWAMP.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Temporary_Habitat_Decay_Factor</strong></td>
<td>float</td>
<td>0.001</td>
<td>100</td>
<td>0.05</td>
<td>Factor to convert raw evaporation rate (ignoring boundary layer effects) to the daily rate of larval habitat loss for temporary habitats with Habitat_Type parameter of TEMPORARY_RAINFALL.  Units are (larval carrying capacity per day) / (kg per square meter per second).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Vector_Larval_Rainfall_Mortality</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONE</td>
<td>Type of vector larval mortality function due to rainfall.  Possible values are NONE, SIGMOID, where the mortality rate grows linearly from 0 at the threshold to 1 at twice the threshold value, and SIGMOID_HABITAT_SHIFTING, where the threshold value is reduced by a factor proportional to how full the larval habitat is.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Vector_Sampling_Type</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>TRACK_ALL_VECTORS</td>
<td>Type of vector module used in a simulation. Possible values are: TRACK_ALL_VECTORS, SAMPLE_IND_VECTORS, VECTOR_COMPARTMENTS_NUMBER and VECTOR_COMPARTMENTS_PERCENT.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Vector_Species_Names</strong></td>
<td>Dynamic String Set</td>
<td>NA</td>
<td>NA</td>
<td>[]</td>
<td>An array of vector species names for all species that are present in the simulation. For example, [“arabiensis”, “funestus”, “gambiae”]. Any vector species name may be used. However, you must specify the parameters as listed in Vector_Species_Params for each species. See Vector Species Parameters for a usage example.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Vector_Sugar_Feeding_Frequency</strong></td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>VECTOR_SUGAR_FEEDING_NONE</td>
<td>How often does a female mosquito take a sugar feed? Possible values are: VECTOR_SUGAR_FEEDING_NONE - No sugar feeding. VECTOR_SUGAR_FEEDING_ON_EMERGENCE_ONLY - Sugar feeding once at emergence. VECTOR_SUGAR_FEEDING_EVERY_FEED - Sugar feeding occurs once per blood meal. VECTOR_SUGAR_FEEDING_EVERY_DAY - Sugar feeding occurs everyday. Note: This parameter is used in conjunction with the SugarTrap and OvipositionTrap interventions. These features require Vector_Sampling_Type to be set to TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Wolbachia_Infection_Modification</strong></td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>1</td>
<td>The change in vector susceptibility to infection due to a Wolbachia infection.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Wolbachia_Mortality_Modification</strong></td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>1</td>
<td>The change in vector mortality due to a Wolbachia infection.</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="vector-species">
<span id="config-vectorspecies"></span><h2>Vector species<a class="headerlink" href="#vector-species" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>Acquire_Modifier</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Modifier of the probability of successful infection of a mosquito by a malaria-infected individual, given the individual&#8217;s infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span><span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><strong>Adult_Life_Expectancy</strong></td>
<td>float</td>
<td>0</td>
<td>730</td>
<td>10</td>
<td>Number of days an average mosquito survives. The daily adult mortality rate is 1 / (value of this parameter).</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Anthropophily</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Propensity (dimensionless) of the mosquito to feed on humans instead of animals, measured as a fraction of blood meals from human hosts. This propensity is important in differentiating the mosquito species.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Aquatic_Arrhenius_1</strong></td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8.42E+10</td>
<td>The duration of aquatic development is a decreasing function of temperature. The daily rate of fractional progression from egg-hatching to emergence is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a1, is a temperature-independent scale factor on the development rate.  Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Aquatic_Arrhenius_2</strong></td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8328</td>
<td>The duration of aquatic development is a decreasing function of temperature. The daily rate of fractional progression from egg-hatching to emergence is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a2, governs how quickly the rate changes with temperature. Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Aquatic_Mortality_Rate</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>Base aquatic mortality per day for the species before adjustment for effects of overpopulation and drying. The rate is locally adjusted for other considerations, including overpopulation and drying out of the aquatic habitat.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Days_Between_Feeds</strong></td>
<td>float</td>
<td>0</td>
<td>730</td>
<td>2</td>
<td>Factor (in days) used to calculate the rate at which female mosquitoes attempt to feed. The rate equals 1 / (value of this parameter). The Days_Between_Feeds parameter represents a 3-day feeding cycle at normal tropical temperatures (30 C). If it is colder, the cycle is 4 days at 24 C, and if the temperature is warmer, the cycle is shorter, increasing the fraction of 2-day cycles from 30 C to 48 C. Whole temperature-dependent distribution can be shifted by changing Days_Between_Feeds which is always the value at 30 C. See Enable_Temperature_Dependent_Feeding_Cycle for weather-dependent feeding.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Egg_Batch_Size</strong></td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>100</td>
<td>Size of egg batch laid by one female mosquito that has fed successfully.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Immature_Duration</strong></td>
<td>float</td>
<td>0</td>
<td>730</td>
<td>2</td>
<td>Factor (in days) used to calculate the rate at which immature organisms develop into adult mosquitoes, which equals 1 / (value of this parameter). Note that development from immature individual to adult is not dependent on temperature.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Indoor_Feeding_Fraction</strong></td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Fraction (dimensionless) of feeds on humans that occur indoors. Thus the fraction of feeds on humans that occur outdoors equals 1 - (value of this parameter).&#8221;</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Infected_Arrhenius_1</strong></td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>1.17E+11</td>
<td>The duration of sporogony is a decreasing function of temperature. The daily rate of fractional progression of infected mosquitoes to an infectious state is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a1, is a temperature-independent scale factor on the development rate.  Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Infected_Arrhenius_2</strong></td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8340</td>
<td>The duration of sporogony is a decreasing function of temperature. The daily rate of fractional progression of infected mosquitoes to an infectious state is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a2, governs how quickly the rate changes with temperature.   Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Infected_Egg_Batch_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>10</td>
<td>0.8</td>
<td>Factor (dimensionless) used to modify egg batch size to account for effects of infection in a female mosquito. Female mosquitoes undergoing sporogony have reduced fertility.&#8221;</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td><strong>Infectious_Human_Feed_Mortality_Factor</strong></td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1.5</td>
<td>Correction factor (dimensionless) used to modify the death rate of mosquitoes when feeding on humans, to account for the increase caused by infections in mosquitoes. Infected mosquitoes have a higher mortality rate while feeding on humans than uninfected mosquitoes. References: Wekesa J. W., Copeland R. S. , et al. (1992). Effect of Plasmodium Falciparum on Blood Feeding Behavior of Naturally Infected Anopheles Mosquitoes in Western Kenya. Am J Trop Med Hyg 47(4): 484-488.Anderson R. A., Knols B. G. J., et al. (2000). Plasmodium falciparum sporozoites increase feeding-associated mortality of their mosquito hosts Anopheles gambiae s.l. Parasitology 120(04): 329-333.</td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><strong>Larval_Habitat_Types</strong></td>
<td>nested JSON object</td>
<td>0</td>
<td>1.00E+15</td>
<td>1.25E+10</td>
<td><p class="first">A measure of the habitat type and scaling factors to use to estimate larval population. You can specify this with a dictionary specifying habitat type with a simple numeric scaling factor or, for LINEAR_SPLINE, with a more detailed configuration for scaling.</p>
<p>Possible habitat values are:</p>
<ul class="simple">
<li>&#8220;CONSTANT&#8221;</li>
<li>&#8220;TEMPORARY_RAINFALL&#8221;</li>
<li>&#8220;WATER_VEGETATION&#8221;</li>
<li>&#8220;HUMAN_POPULATION&#8221;</li>
<li>&#8220;BRACKISH_SWAMP&#8221;</li>
<li>&#8220;LINEAR_SPLINE&#8221;</li>
</ul>
<p>The numeric scaling value represents larval density with the number of larvae in a 1x1-degree area. The factor multiplicatively scales the resulting weather or population dependent functional form.</p>
<p class="last">The LINEAR_SPLINE configuration specifies the day of year, larval value, and larval capacity scaling number. The model linearly interpolates the values to estimate the habitat availability for each vector species without requiring climatological data.</p>
</td>
<td><p class="first">The following example shows how to specify the larval habitat using one of the climatological habitat types.</p>
<div class="highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span><span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span><span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">1000000</span><span class="p">}}</span>
</pre></div>
</div>
<p>The following example shows how to specify larval habitat using LINEAR_SPLINE.</p>
<div class="last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span>
    <span class="p">{</span>
        <span class="nt">&quot;LINEAR_SPLINE&quot;</span><span class="p">:</span>
        <span class="p">{</span>
            <span class="nt">&quot;Capacity_Distribution_Per_Year&quot;</span><span class="p">:</span>
            <span class="p">{</span>
                <span class="nt">&quot;Times&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mf">0.0</span><span class="p">,</span> <span class="mf">60.833</span><span class="p">,</span> <span class="mf">121.667</span><span class="p">,</span> <span class="mf">182.5</span><span class="p">,</span> <span class="mf">243.333</span><span class="p">,</span> <span class="mf">204.167</span><span class="p">],</span>
                <span class="nt">&quot;Values&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mf">0.0</span><span class="p">,</span> <span class="mf">0.0</span><span class="p">,</span> <span class="mf">0.2</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.0</span><span class="p">]</span>
            <span class="p">},</span>
            <span class="nt">&quot;Max_Larval_Capacity&quot;</span><span class="p">:</span> <span class="mi">10000000000</span>
        <span class="p">}</span>
    <span class="p">}</span>
  <span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td><strong>Vector_Species_Params</strong></td>
<td>JSON object</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>For each of the species specified in Vector_Species_Name, there is a set of required species-parameters. The code example, using the arabiensis parameter value, configures the vector <em>Anopheles arbiensis</em>.  The parameter name values for the other species are: farauti, fenestus and bambiae.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span> <span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
       <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
          <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span>
          <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span>
          <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span>
          <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span>
          <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span>
          <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span>
          <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span>
          <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span>
          <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
          <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span>
          <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span>
          <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span>
          <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span>
          <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
             <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mf">1e10</span><span class="p">,</span>
             <span class="nt">&quot;WATER_VEGETATION&quot;</span><span class="p">:</span> <span class="mf">1e6</span>

           <span class="p">},</span>
          <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
       <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
</div>


<div id="tipue_search_content"></div>

           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="parameter-campaign.html" class="btn btn-neutral float-right" title="Campaign parameters" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="parameter-demographics.html" class="btn btn-neutral" title="Demographics file structure and parameters" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Intellectual Ventures Management, LLC (IVM). All rights reserved.

    </p>
  </div>
<p>EMOD is licensed under the <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/legalcode">Creative Commons Attribution-Noncommercial-ShareAlike 4.0 License</a>.</p>
<p>Send documentation feedback to <a href="mailto:feedback@idmod.org" target="_top">feedback@idmod.org</a>.</p>


</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/tipuesearch/tipuesearch.js"></script>
      <script type="text/javascript" src="_static/tipuesearch/tipuesearch_content.js"></script>
      <script type="text/javascript" src="_static/tipuesearch/tipuesearch_set.js"></script>
      <script type="text/javascript" src="_static/runsearch.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
  
<!-- Google analytics for IDM docs -->
 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-87656833-3', 'auto');
  ga('send', 'pageview');

</script>



</body>
</html>