var tipuesearch_info = 'Generated by gti.py on 2017-06-22 16:30:51.547000';
var tipuesearch = {"pages": [{"url": "./genindex.html", "text": "Index A | B | C | D | E | F | G | H | I | J | K | L | M | N | O | P | R | S | T | V | W A agent-based model antigen B Boost boxcar function C campaign campaign event campaign file channel circumsporozoite protein (CSP) class factory Clausius-Clayperon relation cohort model coital dilution configuration file core CRISPR cytokine D demographics file deterministic diffusive migration disease-specific build dynamic link library (DLL) E EMODule entomological inoculation rate (EIR) Epidemiological MODeling software (EMOD) epitope Eradication.exe Euler method event coordinator exp( F flattened file G gametocyte gene drive mosquito gonotrophic H hepatocyte herd immunity Heterogeneous Intra-Node Transmission (HINT) high-performance computing (HPC) homing endonuclease gene (HEG) I incidence individual mosquito model indoor residual spraying (IRS) input data files insecticide-treated nets (ITN) intervention J JavaScript Object Notation (JSON) K Keyhole Markup Language (KML) Koppen-Geiger Climate Classification System L larval habitat larvicide Link-Time Code Generation (LTCG) loss to follow-up (LTFU) M mass drug administration (MDA) merozoite merozoite surface protein (MSP) merozoites Message Passing Interface (MPI) microsolver monolithic build Monte Carlo N node node-targeted intervention O ordinary differential equation (ODE) output report overlay file oviposition P Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1) pre-erythrocytic pre-erythrocytic vaccine (PEV) prevalence preview R regression test release reporter reproductive number routine immunization (RI) S scenario , [1] schema schizont SEIR model SEIRS model SI model simulation simulation type SIR model SIRS model SIS model solvers spatial repellent sporogony sporozoite Standard Template Library (STL) state transition event stochastic subpatent sugar-baited trap superinfection supplemental immunization activity (SIA) T time step transmission-blocking vaccine (TBV) tutorial V vaccine intervention types vector vector model vectorial capacity W WAIFW matrix Weibull distribution Wolbachia working directory", "tags": "", "title": "Index \u2014 Parameter Reference  documentation"}, {"url": "./glossary.html", "text": "Glossary \u00b6 The Epidemiological MODeling software (EMOD) glossary is divided into the following subsections that define terms related to software usage, general epidemiology, and the particular disease being modeled. Software terms \u00b6 The following terms are used to describe both general computing processes and concepts and the files, features, and functionality related to running simulations with Epidemiological MODeling software (EMOD). agent-based model A type of simulation that models the actions and interactions of autonomous agents (both individual and collective entities such as organizations or groups). Boost Free, peer-reviewed, portable C++ source libraries aimed at a wide range of uses including parallel processing applications (Boost.MPI). For more information, please see the Boost website, http://www.boost.org . boxcar function A mathematical function that is equal to zero over the entire real line except for a single interval where it is equal to a constant. campaign A collection of events that use interventions to modify a simulation . campaign event A JSON object that determines when and where an intervention is distributed during a campaign. campaign file A JavaScript Object Notation (JSON) formatted file that contains the parameters that specify the distribution instructions for all interventions used in a campaign, such as diagnostic tests, the target demographic, and the timing and cost of interventions. The location of this file is specified in the configuration file with the Campaign_Filename parameter. Typically, the file name is campaign.json. channel A property of the simulation (for example, \u201cParasite Prevalence\u201d) that is accumulated once per simulated time step and written to file, typically as an array of the accumulated values. class factory A function that instantiate objects at run-time and use information from JSON-based configuration information in the creation of these objects. configuration file A JavaScript Object Notation (JSON) formatted file that contains the parameters sufficient for initiating a simulation. It controls many different aspects of the simulation, such as population size, disease dynamics, and length of the simulation. Typically, the file name is config.json. core In computing, a core refers to an independent central processing unit (CPU) in the computer. Multi-core computers have more than one CPU. However, through technologies such as Hyper- Threading Technology (HTT or HT), a single physical core can actually act like two virtual or logical cores, and appear to the operating system as two processors. demographics file A JavaScript Object Notation (JSON) formatted file that contains the parameters that specify the demographics of a population, such as age distribution, risk, birthrate, and more. IDM provides demographics files for many geographic regions. This file is considered part of the input data files and is typically named <region>_demographics.json. disease-specific build A build of the EMOD executable (Eradication.exe) built using SCons without any dynamic link libraries (DLLs). dynamic link library (DLL) Microsoft\u2019s implementation of a shared library, separate from the EMOD executable (Eradication.exe), that can be dynamically loaded (and unloaded when unneeded) at runtime. This loading can be explicit or implicit. EMODule A modular component of EMOD that are consumed and used by the EMOD executable (Eradication.exe). Under Windows, a EMODule is implemented as a dynamic link library (DLL) and, under CentOS on Azure, EMODules are currently not supported. EMODules are primarily custom reporters. Epidemiological MODeling software (EMOD) The modeling software from the Institute for Disease Modeling (IDM) for disease researchers and developers to investigate disease dynamics, and to assist in combating a host of infectious diseases. You may see this referred to as Disease Transmission Kernel (DTK) in the source code. Eradication.exe Typical (default) name for the EMOD executable (Eradication.exe), whether built using monolithic build or modular (EMODule-enabled) build. event coordinator A JSON object that determines who will receive a particular intervention during a campaign. flattened file A single campaign or configuration file created by combining a default file with one or more overlay files. Multiple files must be flattened prior to running a simulation. Configuration files are flattened to a single-depth JSON file without nesting, the format required for consumption by the EMOD executable (Eradication.exe). Separating the parameters into multiple files is primarily used for testing and experimentation. Heterogeneous Intra-Node Transmission (HINT) A feature for modeling person-to-person transmission of diseases in heterogeneous population segments within a node for generic simulations. high-performance computing (HPC) The use of parallel processing for running advanced applications efficiently, reliably, and quickly. input data files Fixed information used as input to a simulation. Examples of such data are population, altitude, demographics, and transportation migrations. The configurable config.json and campaign.json files do not fall into this category of static fixed input. intervention An object aimed at reducing the spread of a disease that is distributed either to an individual; such as a vaccine, drug, or bednet; or to a node; such as a larvicide. Additionally, initial disease outbreaks and intermediate interventions that schedule another intervention are implemented as interventions in the campaign file . JavaScript Object Notation (JSON) A human-readable, open standard, text-based file format for data interchange. It is typically used to represent simple data structures and associative arrays, and is language- independent. Keyhole Markup Language (KML) A file format used to display geographic data in an Earth browser, for example, Google Maps. The format uses an XML-based structure (tag-based structure with nested elements and attributes). Tags are case-sensitive. Link-Time Code Generation (LTCG) A method for the linker to optimize code (for size and/or speed) after compilation has occurred. The compiled code is turned not into actual code, but instead into an intermediate language form (IL, but not to be confused with .NET IL which has a different purpose). The LTCG then, unlike the compiler, can see the whole body of code in all object files and be able to optimize the result more effectively. Message Passing Interface (MPI) An interface used to pass information between computing cores in parallel simulations. One example is the migration of individuals from one geographic location to another within EMOD simulations. microsolver A microsolver is a type of \u201cminiature model\u201d that operates within the framework of EMOD to compute a particular set of parameters. Each microsolver, in effect, is creating a microsimulation in order to accurately capture the dynamics of that particular aspect of the model. Monte Carlo Monte Carlo methods, or Monte Carlo experiments, are a class of computational algorithms utilizing repeated random sampling to obtain numerical results. Monte Carlo simulations create probability distributions for possible outcome values, which provides a more realistic way of describing uncertainty in variables. monolithic build A single EMOD executable (Eradication.exe) with no DLLs that includes all components as part of Eradication.exe itself. You can still use EMODules with the monolithic build; for example, a custom reporter is a common type of EMODule. View the documentation on EMODules and emodules_map.json for more information about creation and use of EMODules. node A grid size that is used for modeling geographies. Within EMOD, a node is a geographic region containing simulated individuals. Individuals migrate between nodes either temporarily or permanently using mobility patterns driven by local, regional, and long- distance transportation. node-targeted intervention An intervention that is distributed to a geographical node rather than to a single individual. One example is larvicides, which affect all mosquitoes living and feeding within a given node. output report A file that is the output from an EMOD simulation. Output reports are in JSON, CSV, or binary file format. You must pass the data from an output report to graphing software if you want to visualize the output of a simulation. overlay file An additional configuration, campaign, or demographic file that overrides the default parameter values in the primary file. Separating the parameters into multiple files is primarily used for testing a nd experimentation. In the case of configuration and campaign files, the files can use an arbitrary hierarchical structure to organize parameters into logical groups. Configuration and campaign files must be flattened into a single file before running a simulation. preview Software that undergoes a shorter testing cycle in order to make it available more quickly. Previews may contain software defects that could result in unexpected behavior. Use EMOD previews at your own discretion. regression test A test to verify that existing EMOD functionality works with new updates, typically used to refer to one of a set of scenarios included in the EMOD bundle and located in the Regression subdirectory of the bundle. Directory names of each subdirectory in Regression describe the main regression attributes, for example, \u201c1_Generic_Seattle_MultiNode\u201d. Also can refer to the process of regression testing of software. release Software that includes new functionality, scientific tutorials leveraging new or existing functionality, and/or bug fixes that have been thoroughly tested so that any defects have been fixed before release. EMOD releases undergo full regression testing. reporter Functionality that extracts simulation data, aggregates it, and saves it as an output report . EMOD provides several built-in reporters for outputting data from simulations and you also have the ability to create a custom reporter. scenario A collection of input, configuration, and campaign files that describes a real-world example of a disease outbreak and interventions. Many scenarios are included with both the Quick Start and EMOD source installations. The Quick Start documentation includes tutorials that describe how to run simulations using these scenarios to learn more about epidemiology and disease modeling. schema A text or JSON file that can be generated from the EMOD executable (Eradication.exe) that defines all configuration and campaign parameters. simulation An EMOD software execution performed by the kernel. Each simulation has an associated set of files that control the inputs, configuration, and campaign. simulation type The disease or disease class to model. EMOD supports the following simulation types for modeling a variety of diseases: Generic disease (GENERIC_SIM) Vector-borne diseases (VECTOR_SIM) Malaria (MALARIA_SIM) Tuberculosis (TB_SIM) Sexually transmitted infections (STI_SIM) HIV (HIV_SIM) solvers Solvers are used to find computational solutions to problems. In simulations, they can be used, for example, to determine the time of the next simulation step, or to compute the states of a model at particular time steps. Standard Template Library (STL) A library that contains a set of common C++ classes (including generic algorithms and data structures) that are independent of container and implemented as templates, which enables compile-time polymorphism (often more efficient than run-time polymorphism). For more information and discussion of STL, see Wikipedia . state transition event A change in state (e.g. healthy to infected, undiagnosed to positive diagnosis, or birth) that may trigger a subsequent action, often an intervention. \u201cCampaign events\u201d should not be confused with state transition events. time step A discrete number of hours or days in which the \u201csimulation states\u201d of all \u201csimulation objects\u201d (interventions, infections, immune systems, or individuals) are updated in a simulation. Each time step will complete processing before launching the next one. For example, a time step would process the migration data for populations moving between nodes via rail, airline, and road. The migration of individuals between nodes is the last step of the time step after updating states. tutorial A set of instructions in the Quick Start documentation to learn more about epidemiology and disease modeling. Tutorials are based on real-world scenarios and demonstrate the mechanics of the the model. Each tutorial consists of one or more scenarios. working directory The directory that contains the configuration and campaign files for a simulation. You must be in this directory when you invoke Eradication.exe at the command line to run a simulation. Epidemiology terms \u00b6 The following terms are used to describe general concepts and processes in the field of epidemiology and disease modeling. antigen A substance that is capable of inducing a specific immune response and that evokes the production of one or more antibodies. Clausius-Clayperon relation A way of characterizing a transition between two phases of matter; provides a method to find a relationship between temperature and pressure along phase boundaries. Frequently used in meteorology and climatology to describe the behavior of water vapor. See Wikipedia - Clausius-Clayperon relation for more information. deterministic Characterized by the output being fully determined by the parameter values and the initial conditions. Given the same inputs, a deterministic model will always produce the same output. diffusive migration The diffusion of people in and out of nearby nodes by foot travel. epitope The portion of an antigen that the immune system recognizes. An epitope is also called an antigenic determinant. Euler method Used in mathematics and computational science, this method is a first-order numerical procedure for solving ordinary differential equations with a given initial value. exp( The exponential function, , where is the number (approximately 2.718281828) such that the function is its own derivative. The exponential function is used to model a relationship in which a constant change in the independent variable gives the same proportional change (i.e. percentage increase or decrease) in the dependent variable. The function is often written as . The graph of is upward-sloping and increases faster as increases. herd immunity The resistance to the spread of a contagious disease within a population that results if a sufficiently high proportion of individuals are immune to the disease, especially through vaccination. incidence The rate of new cases of a disease during a specified time period. This is a measure of the risk of contracting a disease. Koppen-Geiger Climate Classification System A system based on the concept that native vegetation is a good expression of climate. Thus, climate zone boundaries have been selected with vegetation distribution in mind. It combines average annual and monthly temperatures and precipitation, and the seasonality of precipitation. The DTK has several options for configuring the climate, namely air temperature, rainfall, and humidity. One option utilizes Input Files that associate geographic nodes with Koppen climate indices. The modified Koppen classification uses three letters to divide the world into five major climate regions (A, B, C, D, and E) based on average annual precipitation, average monthly precipitation, and average monthly temperature. Each category is further divided into sub-categories based on temperature and precipitation. While the Koppen system does not take such things as temperature extremes, average cloud cover, number of days with sunshine, or wind into account, it is a good representation of our earth\u2019s climate. loss to follow-up (LTFU) Patients who at one point were actively participating in disease treatment or clinical research, but have become lost either by error or by becoming unreachable at the point of follow-up. ordinary differential equation (ODE) A differential equation containing one or more functions of one independent variable and its derivatives. prevalence The rate of all cases of a disease during a specified time period. This is a measure of how widespread a disease is. reproductive number In a fully susceptible population, the basic reproductive number R:sub:0is the number of new cases generated by one infectious case over the course of the infectious period. The effective reproductive number takes into account non-susceptible individuals. routine immunization (RI) The standard practice of vaccinating the majority of susceptible people in a population against vaccine-preventable diseases. scenario A real-world or hypothetical situation that demonstrates or represents disease dynamics. stochastic Characterized by having a random probability distribution that may be analyzed statistically but not predicted precisely. subpatent When an individual is infected but asymptomatic, so the infection is not readily detectable. SEIR model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, exposed, infectious, and recovered. SEIRS model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, exposed, infectious, recovered, and susceptible. SI model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible and infectious. SIR model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, infectious, and recovered. SIRS model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, infectious, recovered, and susceptible. SIS model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, infectious, and susceptible. supplemental immunization activity (SIA) In contrast to routine immunization (RI) , SIAs are large-scale operations with a goal of delivering vaccines to every household. WAIFW matrix A matrix of values that describes the rate of transmission between different population groups. WAIFW is an abbrevation for Who Acquires Infection From Whom. Weibull distribution A probability distribution often used in EMOD and that requires both a shape parameter and a scale parameter. The shape parameter governs the shape of the density function. When the shape parameter is equal to 1, it is an exponential distribution. For shape parameters above 1, it forms a unimodal (hump-shaped) density function. As the shape parameter becomes large, the function forms a sharp peak. The inverse of the shape parameter is sometimes referred to here as the \u201cheterogeneity\u201d of the distribution (heterogeneity = 1/shape), because it can be helpful to think about the degree of heterogeneity of draws from the distribution, especially for hump-shaped functions with heterogeneity values between 0 and 1 (i.e., shape parameters greater than 1). The scale parameter shifts the distribution from left to right. When heterogeneity is small (i.e., the shape parameter is large), the scale parameter sets the location of the sharp peak.", "tags": "", "title": "Glossary \u2014 Parameter Reference  documentation"}, {"url": "./index.html", "text": "Welcome to the complete IDM parameter reference \u00b6 The Institute for Disease Modeling (IDM) develops complex software modeling. The primary software, Epidemiological MODeling software (EMOD), helps determine the combination of health policies and intervention strategies that can lead to disease eradication. We share this modeling software with the research community to advance the understanding of disease dynamics. This documentation set contains an alphabetized list of all configuration, campaign, and demographic parameters available in the EMOD model. Parameter reference \u00b6 Demographics file structure and parameters Configuration parameters Campaign parameters Glossary", "tags": "", "title": "Welcome to the complete IDM parameter reference \u2014 Parameter Reference  documentation"}, {"url": "./parameter-campaign.html", "text": "Campaign parameters \u00b6 The parameters described in this reference section determine the interventions used as part of a campaign to control the spread of a disease and the outbreak of the disease itself. Interventions are configured in the campaign file . They are enabled in the configuration file by setting the parameter, Enable_Interventions , to 1. The configuration file also indicates the name of the campaign file. Warning The Outbreak must be the last event in the campaign file or none of the interventions will take place. Warning Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false. The tables below contain all campaign parameters available to use with the EMOD model. Collapse all parameter tables Contents Events array Campaign event Event coordinators NodeSet_Config Interventions Commonly used vector-control interventions AnimalFeedKill AntimalarialDrug ArtificialDiet ArtificialDietHousingModification BirthTriggeredIv BroadcastEvent DelayedIntervention HealthSeekingBehaviorUpdateable HumanHostSeekingTrap InsectKillingFence InsectKillingFenceHousingModification InterpolatedValueMap IRSHousingModification InputEIR IVCalendar Ivermectin Larvicides MalariaChallenge MosquitoRelease MultiInterventionDistributor NodeLevelHealthTriggeredIV Outbreak OutbreakIndividual OutdoorRestKill OvipositionTrap PMTCT PropertyValueChanger RTSSVaccine ScaleLarvalHabitat ScreeningHousingModification SimpleBednet SimpleDiagnostic SimpleHealthSeekingBehavior SimpleIndividualRepellent SimpleVaccine SpaceSpraying SpatialRepellent SpatialRepellentHousingModification Sugartrap Waning effect Events array \u00b6 The campaign file contains an Events array with one or more CampaignEvent elements and a Use_Defaults parameter. If Use_Defaults is set to 1, the simulation will use the default values for required parameters that are not configured in the campaign file. If it is set to 0, you must configure all required parameters. { \"Events\" : [{ \"Event_Name\" : \"Campaign Event 1\" , \"class\" : \"CampaignEvent\" }, { \"Event_Name\" : \"Campaign Event 2\" , \"class\" : \"CampaignEvent\" }], \"Use_Defaults\" : 1 } Campaign event \u00b6 To distribute an intervention, you must configure a campaign event, an event coordinator, and the intervention. The campaign event configures when the event occurs ( Start_Day ) and where it is distributed ( Nodeset_Config ). The event coordinator is a nested JSON object in Event_Coordinator_Config . { \"Event_Name\" : \"Campaign Event Example\" , \"class\" : \"CampaignEvent\" , \"Start_Day\" : 1 , \"Nodeset_Config\" : { }, \"Event_Coordinator_Config\" : { } } Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Event_Coordinator_Config EventCoordinator 0 3.40E+38 1 The configuration of the EventCoordinator to be instantiated. \u00a0 Nodeset_Config NodeSet 0 3.40E+38 1 The NodeSet describing the nodes covered by the event. \u00a0 Start_Day float 0 3.40E+38 1 Day of the simulation to activate the event\u2019s event coordinator. \u00a0 Event coordinators \u00b6 The Event_Coordinator_Config configures who can receive the intervention (for example, Target_Demographic and Demographic_Coverage ) and which intervention is distributed. The intervention is a nested JSON object in Intervention_Config . { \"Event_Coordinator_Config\" : { \"class\" : \"The event coordinator that will be used.\" , \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"Intervention_Config\" : {} } } Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Demographic_Coverage float 0 1 1 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Intervention_Config Intervention 0 3.40E+38 3.40E+38 The nested json of the actual intervention to be distributed by this event coordinator. \u00a0 Number_Repetitions integer -1 1000 -1 Number of times intervention given, used with tsteps_between_reps. \u00a0 Property_Restrictions Dynamic String Set NA NA [] Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Property_Restrictions_Within_Node PropertyRestrictions NA NA Everyone Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Target_Age_Max float 0 3.40E+38 3.40E+38 Upper end of age targeted for intervention, in years. \u00a0 Target_Age_Min float 0 3.40E+38 0 Lower end of age targeted for intervention, in years. \u00a0 Target_Demographic enum NA NA Everyone The demographic group (from a list of possibles) targeted by this intervention (e.g., Infants). \u00a0 Target_Gender enum NA NA All Specify the gender restriction for the intervention. Defaults to All. \u00a0 Target_Residents_Only boolean 0 1 0 If true, only distribute the intervention to those that claim the node as their residence - i.e. started in the node \u00a0 Timesteps_Between_Repetitions integer -1 10000 -1 Repetition interval. \u00a0 NodeSet_Config \u00b6 There are three configuration options: NodeSetAll , NodeSetNodeList and NodeSetPolygon . NodeSetAll \u00b6 If you are running a simulation that is in a single location (node) or you want the intervention to be distributed in all of the nodes, use NodeSetAll . There are no additional parameters. { \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" } } NodeSetNodeList \u00b6 The intervention is distributed to nodes in the intervention list. Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Node_List NodeListConfig NA NA SHAPE Comma-separated list of node IDs. \u00a0 NodeSetPolygon \u00b6 The intervention is distributed in nodes within the specified polygon. Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Polygon_Format enum NA NA SHAPE Right now just SHAPE. \u00a0 Vertices string NA NA SHAPE Comma-separated list of polygon points. \u00a0 Interventions \u00b6 Commonly used vector-control interventions \u00b6 Toggle parameter table Name Intervention level Target life stage Target biting preference Target biting location Effect Notes AnimalFeedKill node feeding cycle animal \u2013 killing \u00a0 ArtificialDiet node feeding cycle human all blocking Artificial_Diet_Target: AD_WithinVillage, AD_OutsideVillage HumanHostSeekingTrap individual feeding cycle human \u00a0 killing \u00a0 InsectKillingFence node feeding cycle all all killing Photonic fence. Two killing chances per feeding cycle: to and from oviposition site. InsectKillingFenceHousingModification individual \u00a0 \u00a0 \u00a0 killing \u00a0 IRSHousingModification individual feeding cycle human indoor killing, blocking \u00a0 Ivermectin individual feeding cycle human all killing \u00a0 Larvicides node larva all all killing, reduction Specify targeted habitat. MosquitoRelease node \u00a0 \u00a0 \u00a0 \u00a0 \u00a0 OutdoorRestKill node feeding cycle human outdoor killing \u00a0 OvipositionTrap node feeding cycle all all killing Specify targeted habitat. Requires individual vectors. ScaleLarvalHabitat node larva all all reduction Specify targeted habitat. ScreeningHousingModification individual feeding cycle human indoor killing, blocking \u00a0 SimpleBednet individual feeding cycle human indoor killing, blocking \u00a0 SimpleIndividualRepellent individual feeding cycle human all blocking \u00a0 SpaceSpraying ( SpaceSpray_Indoor ) node feeding cycle human indoor killing Node-based IRS for household. SpaceSpraying node feeding cycle human outdoor killing SpaceSpray_FemalesOnly, MalesOnly, FemalesAndMales SpatialRepellent node feeding cycle all outdoor blocking \u00a0 SpatialRepellentHousingModification individual feeding cycle human indoor killing, blocking \u00a0 SugarTrap node adults all all killing Requires individual vectors. Killing parameter specifies effective fraction of sugar feeds that occur on bait stations. AnimalFeedKill \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 43800 365 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 AntimalarialDrug \u00b6 When using antimalarial drugs in the model, the configuration file sets the main effects of the drug, while the intervention (configured in the campaign file) determines which drugs are used and to whom they are distributed. The following example provides the syntax to configure the AntiMalarialDrug intervention: { \"Intervention_Config\" : { \"class\" : \"AntimalarialDrug\" , \"Cost_To_Consumer\" : 3.75 , \"Drug_Type\" : \"Chloroquine\" , \"Dosing_Type\" : \"FullTreatmentParasiteDetect\" } } Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per drug (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Dosing_Type enum NA NA SingleDose Type of anti-malarial dosing to distribute in a drugs intervention. Possible values are: SingleDose, FullTreatmentCourse, Prophylaxis, SingleDoseWhenSymptom, FullTreatmentWhenSymptom, SingleDoseParasiteDetect, FullTreatmentParasiteDetect, SingleDoseNewDetectionTech, FullTreatmentNewDetectionTech. \u00a0 Drug_Type Constrained String 0 3.40E+38 3.40E+38 Type of anti-malarial drug to distribute in a drugs intervention. Possible values are: Artemisinin (AR), Chloroquine (CQ), Quinine (QN), Sulfadoxine-Pyramethamine (SP), Primaquine (PQ), Artemether_Lumefantrine (AR_LM), Generic Transmission Blocking (TBD), Generic Pre-erythrocytic (PED), Tafenoquine (TQ). \u00a0 Intervention_Name string 0 3.40E+38 3.40E+38 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 ArtificialDiet \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Artificial_Diet_Target enum NA NA AD_WithinVillage The targeted deployment of artificial diet. Possible values are: AD_WithinVillage, AD_OutsideVillage. \u00a0 Attraction_Config WaningEffect NA NA AD_WithinVillage Fraction of feeds attracted to the artificial diet. A parameter that is decayed. \u00a0 Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 ArtificialDietHousingModification \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 BirthTriggeredIv \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention 0 1 0 The configuration of the actual individual-based intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Demographic_Coverage float 0 1 1 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Duration float -1 3.40E+38 -1 The number of days to continue this intervention. \u00a0 Intervention_Name string -1 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Property_Restrictions Dynamic String Set NA NA [] Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Property_Restrictions_Within_Node PropertyRestrictions NA NA Everyone Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Target_Age_Max float 0 3.40E+38 3.40E+38 Upper end of age targeted for intervention, in years. \u00a0 Target_Age_Min float 0 3.40E+38 0 Lower end of age targeted for intervention, in years. \u00a0 Target_Demographic enum NA NA Everyone The demographic group (from a list of possibles) targeted by this intervention (e.g., Infants). \u00a0 Target_Gender enum NA NA All Specify the gender restriction for the intervention. Defaults to All. \u00a0 Target_Residents_Only boolean 0 1 0 If true, only distribute the intervention to those that claim the node as their residence - i.e. started in the node \u00a0 BroadcastEvent \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Broadcast_Event Constrained String 0 1 0 The event that should occur at the end of the delay period. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 DelayedIntervention \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Configs IndividualIntervention 0 3.40E+38 6 Array of nested interventions to be distributed at end of delay period to covered fraction. \u00a0 Coverage float 0 1 1 Fraction of individuals receiving delayed distribution of configured interventions. \u00a0 Delay_Distribution enum NA NA NOT_INITIALIZED Distribution of duration of delay period. Supported values are: FIXED_DURATION UNIFORM_DURATION GAUSSIAN_DURATION EXPONENTIAL_DURATION. \u00a0 Delay_Period float 0 3.40E+38 6 If a FIXED_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period to directly specify the time delay (in number of days). If EXPONENTIAL_DURATION is specified as the Delay_Distribution, Delay_Period, represents the exponential rate that describes the distribution of the time delay (in units of 1/days). \u00a0 Delay_Period_Max float 0.6 3.40E+38 0 If a UNIFORM_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Max to directly specify the maximum time delay (in number of days). \u00a0 Delay_Period_Mean float 0 3.40E+38 6 If a GAUSSIAN_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Mean to directly specify the mean time delay (in number of days). \u00a0 Delay_Period_Min float 0 3.40E+38 0 If a UNIFORM_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Min to directly specify the minimum time delay (in number of days). \u00a0 Delay_Period_Std_Dev float 0 3.40E+38 1 If a GAUSSIAN_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Std_Dev to specify the standard deviation describing the Gaussian distribution (in number of days). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 HealthSeekingBehaviorUpdateable \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention 0 1 1 The configuration of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Actual_IndividualIntervention_Event Constrained String 1900 2200 2000 The event of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Event_Or_Config enum NA NA Config Specifies whether the current intervention distributes a nested intervention (the Config option) or an event will be broadcast which may trigger other interventions in the campaign file (the Event option). See event list. \u00a0 Intervention_Name string 1900 2200 2000 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Single_Use boolean 0 1 1 One-time, or persistent? \u00a0 Tendency float 0 1 1 The probability of seeking healthcare. \u00a0 HumanHostSeekingTrap \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Attract_Config WaningEffect 0 999999 3.75 The configuration of killing efficacy and waning for human host-seeking trap. \u00a0 Cost_To_Consumer float 0 999999 3.75 Unit cost per trap (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 999999 10 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 3.75 The configuration of killing efficacy and waning for human host-seeking trap. \u00a0 InsectKillingFence \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 InsectKillingFenceHousingModification \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 InterpolatedValueMap \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Times float 0 99999 NA An array of years. \u00a0 Values float 0 3.40282e+38 NA An array of values to match the defined Times. \u00a0 IRSHousingModification \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 InputEIR \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Age_Dependence enum NA NA OFF Dependence of InputEIR on age. \u00a0 Intervention_Name string 0 3.40E+38 1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Monthly_EIR InputEIRConfig NA NA OFF An array of 12 elements that contain an entomological inoculation rate (EIR) for each month. \u00a0 IVCalendar \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Configs IndividualIntervention 0 1 0 An array of interventions distributed as specified in the calendar. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Calendar CalendarIV 0 1 0 An array of ages days and probability of receiving an intervention. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Dropout boolean 0 1 0 If an intervention in the series is missed, all subsequent interventions are also missed. If false (0), all calendar dates/doses are applied independently of each other. If true (1), a missed dose (by applying the probability) will result in no further doses, that is, a miss means you have dropped out altogether. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Ivermectin \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 8 Unit cost per Ivermectin dosing (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of drug-killing efficacy and waning. \u00a0 Larvicides \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect NA NA TEMPORARY_RAINFALL The configuration of larval habitat reduction and waning for targeted stage. \u00a0 Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Habitat_Target enum NA NA TEMPORARY_RAINFALL The larval habitat type targeted for larvicide. Possible values are: TEMPORARY_RAINFALL WATER_VEGETATION HUMAN_POPULATION CONSTANT BRACKISH_SWAMP ALL_HABITATS. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect NA NA TEMPORARY_RAINFALL The configuration of larval killing efficacy and waning for targeted stage. \u00a0 MalariaChallenge \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Challenge_Type enum NA NA InfectiousBites The type of malaria challenge. Possible values are: InfectiousBites Sporozoites. \u00a0 Coverage float 0 1 1 Fraction of individuals receiving distribution. \u00a0 Infectious_Bite_Count integer 0 1000 1 Number of infectious bites. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Sporozoite_Count integer 0 1000 1 Number of sporozoites. \u00a0 MosquitoRelease \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 0 Cost of each mosquito release. \u00a0 Intervention_Name string NA NA WOLBACHIA_FREE The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Mated_Genetics ResistanceHegGenetics 1 100000000 10000 The genetic properties of the mate if released mosquitoes have mated, e.g. HEG and pesticide resistance. \u00a0 Released_Gender enum NA NA VECTOR_FEMALE Gender of released mosquitoes. Possible values are: VECTOR_FEMALE VECTOR_MALE. \u00a0 Released_Genetics ResistanceHegGenetics 1 100000000 10000 The genetic properties of the released mosquito, e.g. HEG and pesticide resistance. \u00a0 Released_Number integer 1 100000000 10000 Number of mosquitos released in the intervention. \u00a0 Released_Species Constrained String 1 100000000 10000 Name of released mosquito species. \u00a0 Released_Sterility enum NA NA VECTOR_FERTILE Sterility of released mosquitoes. Possible values are: VECTOR_FERTILE VECTOR_STERILE. \u00a0 Released_Wolbachia enum NA NA WOLBACHIA_FREE Wolbachia type of released mosquitoes. Possible values are: WOLBACHIA_FREE VECTOR_WOLBACHIA_A VECTOR_WOLBACHIA_B VECTOR_WOLBACHIA_AB. \u00a0 MultiInterventionDistributor \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_List IndividualIntervention 0 1 0 The array of nested JSON parameters for the interventions to be distributed by this intervention. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 NodeLevelHealthTriggeredIV \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention -1 3.40E+38 -1 The configuration of the actual individual-based intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Actual_NodeIntervention_Config NodeIntervention NA NA [] The configuration of the actual node-level intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Blackout_Event_Trigger Constrained String 0 1 0 The event to broadcast if an intervention cannot be distributed due to the Blackout_Period. \u00a0 Blackout_Period float 0 3.40E+38 0 After the initial intervention distribution, the time, in days, to wait before distributing the intervention again. If it cannot distribute due to the blackout period, it will broadcast the user-defined Blackout_Event_Trigger. \u00a0 Demographic_Coverage float 0 1 1 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Duration float -1 3.40E+38 -1 The number of days to continue this intervention. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Property_Restrictions Dynamic String Set NA NA [] Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Property_Restrictions_Within_Node PropertyRestrictions 0 3.40E+38 3.40E+38 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Target_Age_Max float 0 3.40E+38 3.40E+38 Upper end of age targeted for intervention, in years. \u00a0 Target_Age_Min float 0 3.40E+38 0 Lower end of age targeted for intervention, in years. \u00a0 Target_Demographic enum NA NA Everyone The demographic group (from a list of possibles) targeted by this intervention (e.g., Infants). \u00a0 Target_Gender enum NA NA All Specify the gender restriction for the intervention. Defaults to All. \u00a0 Target_Residents_Only boolean 0 1 0 If true, only distribute the intervention to those that claim the node as their residence - i.e. started in the node \u00a0 Trigger_Condition Constrained String NA NA [] The condition for triggering a health seeking intervention. \u00a0 Trigger_Condition_List Vector String NA NA [] Included if Trigger_Condition is set to \u2018TriggerList\u2019. A list (JSON array) of triggers (strings) drawn from config.json \u2018Listed_Events\u2019 or from built-in events. \u00a0 Trigger_Condition_String Constrained String 0 1 0 Included if Trigger_Condition is set to \u2018TriggerString\u2019. Normally Trigger_Condition must specify an event from the built-in list of pre-defined events or triggers from the IndividualEventTriggerType enum. Trigger_Condition_String can be any string provided it is listed in the config.json\u2019s Listed_Events array. \u00a0 Outbreak \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Antigen integer 0 10 0 The antigenic ID of the outbreak infection. \u00a0 Genome integer -1 16777200 0 The genetic ID of the outbreak infection. \u00a0 Import_Age float 0 43800 365 Age (in days) of infected import cases \u00a0 Incubation_Period_Override boolean 0 1 0 0 = outbreak will bypass incubation, 1 = outbreak infection will start from beginning. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Number_Cases_Per_Node integer 0 2147480000 1 Number of new cases of Outbreak to import (per node). This will increase population. There is no control over demographics of these individuals. \u00a0 OutbreakIndividual \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Antigen integer 0 10 0 The antigenic ID of the outbreak infection. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Genome integer -1 16777200 0 The genetic ID of the outbreak infection. \u00a0 Incubation_Period_Override boolean 0 1 0 0 = outbreak will bypass incubation, 1 = outbreak infection will start from beginning. \u00a0 Intervention_Name string NA NA OFF The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 OutdoorRestKill \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 OvipositionTrap \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Habitat_Target enum NA NA TEMPORARY_RAINFALL The oviposition habitat type targeted by oviposition traps. Possible values are: TEMPORARY_RAINFALL WATER_VEGETATION HUMAN_POPULATION CONSTANT BRACKISH_SWAMP ALL_HABITATS. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect NA NA TEMPORARY_RAINFALL Killing of the targeted stage. A parameter that is decayed. \u00a0 PMTCT \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Efficacy float 0 1 0.5 Represents the efficacy of a PMTCT intervention, defined as the rate ratio of MTCT between women receiving the intervention and women not receiving the intervention. A setting of 1 is equivalent to 100% blocking efficacy, and 0 reverts to the default probability of transmission, configured through the config.json parameter Maternal_Transmission_Probability. \u00a0 Intervention_Name string 0 3.40E+38 6 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 PropertyValueChanger \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Daily_Probability float 0 1 1 The probability per day that an individual will move to the Target_Property_Value. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Maximum_Duration float -1 3.40E+38 3.40E+38 Maximum duration of \u2018delayed-release\u2019 property change, in days. \u00a0 Revert float 0 10000 0 Reversion rate, or the number of days until reversion, or a 2 if reversion is allowed. \u00a0 Target_Property_Key Constrained String 0 10000 0 The name of the EMOD-defined property of the group. \u00a0 Target_Property_Value Constrained String 0 10000 0 The value of the user-defined group where the individuals will be transitioned. \u00a0 RTSSVaccine \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Antibody_Type enum NA NA CSP Antibody type. Possible values are: CSP MSP1 PfEMP1_minor PfEMP1_major. \u00a0 Antibody_Variant integer 0 100000 0 Antibody variant index. \u00a0 Boosted_Antibody_Concentration float 0 3.40E+38 1 Boosted antibody concentration (where unity equals maximum from natural exposure). \u00a0 Cost_To_Consumer float 0 999999 3.75 Unit cost of RTS,S vaccination (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 ScaleLarvalHabitat \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Larval_Habitat_Multiplier LarvalHabitatMultiplier 0 1 0 The value by which to scale the larval habitat availability specified in Larval_Habitat_Types across all habitat types, for specific habitat types, or for specific mosquito species within each habitat type. \u00a0 ScreeningHousingModification \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string -1 3.40E+38 -1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 SimpleBednet \u00b6 The following example provides the syntax for configuring a SimpleBednet campaign. { \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.7 , \"Intervention_Config\" : { \"Bednet_Type\" : \"ITN\" , \"Cost_To_Consumer\" : 3.75 , \"Killing_Config\" : { \"Initial_Effect\" : 0.6 , \"Decay_Time_Constant\" : 1460 , \"class\" : \"WaningEffectExponential\" }, \"Blocking_Config\" : { \"Initial_Effect\" : 0.9 , \"Decay_Time_Constant\" : 730 , \"class\" : \"WaningEffectExponential\" }, \"class\" : \"SimpleBednet\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" } ], \"Use_Defaults\" : 1 } Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Bednet_Type enum NA NA Barrier Type of bednet. Possible values are: Barrier, ITN, LLIN, Retreatment. Note: This enum does not affect the code at all. \u00a0 Blocking_Config WaningEffect 0 999999 3.75 A fraction of feed attempts that have been blocked. The rate of blocking indoor feeds on individuals with an ITN. A parameter that is decayed. \u00a0 Cost_To_Consumer float 0 999999 3.75 Unit cost per bednet (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 3.75 A fraction of feed attempts that kill the vector. The rate at which the mosquito dies conditional on a successfully blocked feed. A parameter that is decayed. \u00a0 SimpleDiagnostic \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Base_Sensitivity float 0 1 1 The sensitivity of the diagnostic. This sets the proportion of the time that individuals with the condition being tested receive a positive diagnostic test. When set to 1, the diagnostic always accurately reflects the condition. When set to zero, then individuals who have the condition always receive a false-negative diagnostic test. \u00a0 Base_Specificity float 0 1 1 The specificity of the diagnostic. This sets the proportion of the time that individuals without the condition being tested receive a negative diagnostic test. When set to 1, the diagnostic always accurately reflects the lack of having the condition. When set to zero, then individuals who do not have the condition always receive a false-positive diagnostic test. \u00a0 Cost_To_Consumer float 0 3.40E+38 1 The unit \u2018cost\u2019 assigned to the intervention. Setting Cost_To_Consumer to zero for all other interventions, and to a non-zero amount for one intervention, provides a convenient way to track the number of times the intervention has been applied in a simulation. \u00a0 Days_To_Diagnosis float 0 3.40E+38 0 The number of days from test until diagnosis. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Event_Or_Config enum NA NA Config Specifies whether the current intervention distributes a nested intervention (the Config option) or an event will be broadcast which may trigger other interventions in the campaign file (the Event option). See event list. \u00a0 Intervention_Name string 0 1 1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Positive_Diagnosis_Config IndividualIntervention 0 1 0 The intervention distributed to individuals if they test positive. This is only valid if Event_Or_Config is set to \u2018Config\u2019. \u00a0 Positive_Diagnosis_Event Constrained String 0 3.40E+38 0 The next health care event that occurs for individuals who test negative. This is only valid if Event_Or_Config is set to \u2018Event\u2019. \u00a0 Treatment_Fraction float 0 1 1 The fraction of positive diagnoses that are treated. \u00a0 SimpleHealthSeekingBehavior \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention 0 1 1 The configuration of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Actual_IndividualIntervention_Event Constrained String 0 999999 10 The event of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Event_Or_Config enum NA NA Config Specifies whether the current intervention distributes a nested intervention (the Config option) or an event will be broadcast which may trigger other interventions in the campaign file (the Event option). See event list. \u00a0 Intervention_Name string 0 999999 10 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Single_Use boolean 0 1 1 One-time, or persistent? \u00a0 Tendency float 0 1 1 The probability of seeking healthcare. \u00a0 SimpleIndividualRepellent \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of efficacy and waning for individual repellent. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per repellent (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 SimpleVaccine \u00b6 The following example provides the syntax for configuring a SimpleVaccine campaign. { \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.8 , \"Intervention_Config\" : { \".Reduced_Acquire\" : 0.9 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"Cost_To_Consumer\" : 10.0 , \"class\" : \"SimpleVaccine\" , \"Waning_Config\" : { \"Initial_Effect\" : 0.9 , \"Box_Duration\" : 730 , \"class\" : \"WaningEffectBox\" } }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" } ], \"Use_Defaults\" : 1 } Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vaccine (unamortized). The unit \u2018cost\u2019 assigned to the intervention. Setting Cost_To_Consumer to zero for all other interventions, and to a non-zero amount for one intervention, provides a convenient way to track the number of times the intervention has been applied in a simulation. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Vaccine_Take float 0 1 1 This specifies the rate at which delivered vaccines will successfully stimulate an immune response and achieve the desired efficacy. If vaccine_take is set to 0.9, then with a 90 percent chance, the vaccine will start with the specified efficacy, and with a 10 percent chance it will have no efficacy at all. \u00a0 Vaccine_Type enum NA NA Generic Type of vaccine to distribute in a vaccine intervention. Possible values are: - Generic - The vaccine can have all of reduced_transmit, reduced_acquire, and reduced_mortality fields and the vaccine will have both effects. - TransmissionBlocking - Specifies the fraction by which the infectivity of the infected and vaccinated Individual is reduced (reduced_transmit). - AcquisitionBlocking - Specifies the fraction by which the force of infection experienced by the vaccinated Individual is reduced (reduced_acqu \u00a0 Waning_Config WaningEffect 0 1 1 The configuration of drug-killing efficacy and waning. \u00a0 SpaceSpraying \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Habitat_Target enum NA NA TEMPORARY_RAINFALL The larval habitat type targeted for habitat reduction. Possible values are: TEMPORARY_RAINFALL WATER_VEGETATION HUMAN_POPULATION CONSTANT BRACKISH_SWAMP ALL_HABITATS. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect NA NA NA The configuration of killing efficacy and waning for space spaying. \u00a0 Reduction_Config WaningEffect NA NA NA The configuration of larval habitat reduction and waning for space spraying. \u00a0 Spray_Kill_Target enum NA NA SpaceSpray_FemalesOnly The gender kill target of space spraying. Possible values are: SpaceSpray_FemalesOnly SpaceSpray_MalesOnly SpaceSpray_FemalesAndMales. \u00a0 SpatialRepellent \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Repellency_Config WaningEffect 0 999999 10 The configuration of efficacy and waning for spatial repellent. \u00a0 SpatialRepellentHousingModification \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 Sugartrap \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 100000 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 Waning effect \u00b6 Waning effect parameters are used with interventions such as vaccines, drugs, and bednets. It has six different classes. Each class specifies a different waning effect and uses a different combination of the waning effect parameters. The waning effect parameters are a nested JSON object and can be configured using several different parameters such as Killing_Config , Waning_Config , Blocking_Config , and Mortality_Config . The following example uses Killing_Config . See the specific intervention sections for the supported parameters. { \"Intervention_Config\" : { \"class\" : \"Ivermectin\" , \"Cost_To_Consumer\" : 1 , \"Killing_Config\" : { \"class\" : \"WaningEffectBox\" , \"Box_Duration\" : 3 , \"Initial_Effect\" : 1 } } } WaningEffectBox class \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Box_Duration float 0 100000 100 Box duration of effect in days. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectBoxExponential class \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Box_Duration float 0 100000 100 Box duration of effect in days. \u00a0 Decay_Time_Constant float 0 100000 100 Exponential decay length in days. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectConstant class \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectExponential class \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Decay_Time_Constant float 0 100000 100 Exponential decay length in days. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectLinear class \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Expire_At_Durability_Map_End boolean 0 1 0 Set to 1 to let the intervention expire when the end of the map is reached. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectPiecewise class \u00b6 Toggle parameter table Parameter Data type Minimum Maximum Default Description Example Expire_At_Durability_Map_End boolean 0 1 0 Set to 1 to let the intervention expire when the end of the map is reached. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect.", "tags": "", "title": "Campaign parameters \u2014 Parameter Reference  documentation"}, {"url": "./parameter-configuration.html", "text": "Configuration parameters \u00b6 The tables below contain an alphabetized list of all configuration parameters in the EMOD model. Warning Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false. The illustration below shows how the simulation types are built upon one another. All parameters available in the generic simulation type are inherited by the vector simulation type. The vector simulation type adds additional parameters specific to the biology of vector-borne diseases, which in turn are inherited by the malaria simulation type and so on. Therefore, depending on the simulation type you select, different parameters are available for you to use. In addition, simulation types for broader classes of disease can be extended to build your own disease-specific model. Simulation type inheritance Parameter Simulation type Data type Minimum Maximum Default Description Example Acquire_Modifier Vector float 0 1 1 Modifier of the probability of successful infection of a mosquito by a malaria-infected individual, given the individual\u2019s infectiousness. { \"Vector_Species_Params\" : { \"aegypti\" : { \"Acquire_Modifier\" : 1 } } } Acquisition_Blocking_Immunity_Decay_Rate Generic float 0 1000 0.001 The rate at which acquisition-blocking immunity decays after the initial period indicated by the base acquisition-blocking immunity offset. Only used when Enable_Immunity and Enable_Immune_Decay parameters are set to true (1). { \"Acquisition_Blocking_Immunity_Decay_Rate\" : 0.05 } Acquisition_Blocking_Immunity_Duration_Before_Decay Generic float 0 45000 0 The number of days after infection until acquisition-blocking immunity begins to decay. Enable_Immunity and Enable_Immune_Decay must be set to true (1). { \"Acquisition_Blocking_Immunity_Duration_Before_Decay\" : 10 } Acute_Duration_In_Months HIV float 0 5 2.9 The time since infection, in months, over which the Acute_Stage_Infectivity_Multiplier is applied to coital acts occurring in that time period. { \"Acute_Duration_In_Months\" : 2.9 } Acute_Stage_Infectivity_Multiplier HIV float 1 100 26 The multiplier acting on Base_Infectivity to determine the per-act transmission probability of an HIV+ individual during the acute stage. { \"Acute_Stage_Infectivity_Multiplier\" : 3 } Adult_Life_Expectancy Vector float 0 730 10 The number of days an adult mosquito survives. The daily adult mortality rate is 1 / (value of this parameter). { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Adult_Life_Expectancy\" : 5.9 } } } Age_Dependent_Biting_Risk_Type Vector enum NA NA OFF The type of functional form for age-dependent risk. Possible values are: OFF This is the default value. LINEAR The biting risk is 20% of the adult exposure rising linearly until age 20. SURFACE_AREA_DEPENDENT The biting risk rises linearly from 7% to 23% over the first two years of life and then rises with a shallower linear slope to the adult value at age 20. { \"Age_Dependent_Biting_Risk_Type\" : \"SURFACE_AREA_DEPENDENT\" } Age_Initialization_Distribution_Type Generic enum NA NA DISTRIBUTION_OFF The method for initializing the age distribution in the simulated population. Possible values are: DISTRIBUTION_OFF All individuals default to age 20 years old. DISTRIBUTION_SIMPLE Individual ages are drawn from a distribution whose functional form is specified in the demographics file using the IndividualAttributes simple Age distribution parameters. DISTRIBUTION_COMPLEX Individual ages are drawn from a piecewise linear distribution specified in the demographics file IndividualAttributes complex AgeDistribution distribution parameters. { \"Age_Initialization_Distribution_Type\" : \"DISTRIBUTION_SIMPLE\" } AIDS_Duration_In_Months HIV float 7 12 9 The length of time, in months, prior to an AIDS-related death over which the AIDS_Stage_Infectivity_Multiplier is applied. { \"AIDS_Duration_In_Months\" : 8 } AIDS_Stage_Infectivity_Multiplier HIV float 1 100 10 The multiplier acting on Base_Infectivity to determine the per-act transmission probability of an HIV+ individual during the AIDS stage. { \"AIDS_Stage_Infectivity_Multiplier\" : 8 } Air_Migration_Filename Generic string NA NA \u00a0 The path to the input data file that defines patterns of migration by airplane. Enable_Air_Migration must be set to true (1). The file must be in .bin format. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Air_Migration\" : 1 , \"Air_Migration_Filename\" : \"../Global_1degree_air_migration.bin\" } Air_Migration_Roundtrip_Duration Generic float 0 10000 1 The average time spent (in days) at the destination node during a round-trip migration by airplane. Enable_Air_Migration must be set to true (1). { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Air_Migration\" : 1 , \"Air_Migration_Roundtrip_Duration\" : 2 } Air_Migration_Roundtrip_Probability Generic float 0 1 0.8 The likelihood that an individual who flies to another node will return to the node of origin during the next migration. Enable_Air_Migration must be set to true (1). { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Air_Migration\" : 1 , \"Air_Migration_Roundtrip_Probability\" : 0.9 } Air_Temperature_Filename Generic string NA NA air_temp.json The path to the input data file that defines air temperature data measured two meters above ground. Climate_Model must be set to CLIMATE_BY_DATA. The file must be in .bin format. { \"Climate_Model\" : \"CLIMATE_BY_DATA\" , \"Air_Temperature_Filename\" : \"Namawala_single_node_air_temperature_daily.bin\" } Air_Temperature_Offset Generic float -20 20 0 The linear shift of air temperature in degrees Celsius. Climate_Model must be set to CLIMATE_BY_DATA. { \"Air_Temperature_Offset\" : 1 } Air_Temperature_Variance Generic float 0 5 2 The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily air temperature values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA. Enable_Climate_Stochasticity must be set to true (1). { \"Enable_Climate_Stochasticity\" : 1 , \"Air_Temperature_Variance\" : 2 , } Anemia_Mortality_Inverse_Width Malaria float 0.1 1000000 10 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for anemia. { \"Anemia_Mortality_Inverse_Width\" : 150 } Anemia_Mortality_Threshold Malaria double NA NA 3 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for hemoglobin count in grams per deciliter (g/dL) at which 50% of individuals die per day. { \"Anemia_Mortality_Threshold\" : 1.5 } Anemia_Severe_Inverse_Width Malaria float 0.1 1000000 10 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for anemia. { \"Anemia_Severe_Inverse_Width\" : 20 } Anemia_Severe_Threshold Malaria float 0 100 5 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the severe disease threshold level for anemia. Threshold units are in grams per deciliter (g/dL). { \"Anemia_Severe_Threshold\" : 3.0 } Animal_Reservoir_Type Generic enum NA NA NO_ZOONOSIS The type of animal reservoir and configuration for zoonosis. Use of the animal reservoir sets a low constant baseline of infectivity beyond what is present in the human population. It allows a more random introduction of cases in continuous time, which is more applicable for various situations such as zoonosis. Possible values are: NO_ZOONOSIS There is no animal reservoir. CONSTANT_ZOONOSIS The daily rate of zoonotic infection is configured by the parameter Zoonosis_Rate . ZOONOSIS_FROM_DEMOGRAPHICS The zoonosis rate is scaled first by the node-specific Zoonosis value in the NodeAttributes section of a demographics file. If the Zoonosis NodeAttribute is not configured, the simulation will then use the value for Zoonosis_Rate . { \"Animal_Reservoir_Type\" : \"CONSTANT_ZOONOSIS\" } Anthropophily Vector float 0 1 1 Mosquito preference for humans over animals, measured as a fraction of blood meals from human hosts. This dimensionless propensity is important in differentiating between mosquito species. { \"Vector_Species_Params\" : { \"aegypti\" : { \"Anthropophily\" : 0.95 } } } Antibody_Capacity_Growth_Rate Malaria float 0 1 0.1 The maximum daily rate of specific antibody capacity increase. Antibody production begins at a capacity of 0.3., and hyperimmunity results at 0.4. The actual growth rate tends to be lower and is a function of antigen concentrations. { \"Antibody_Capacity_Growth_Rate\" : 0.09 } Antibody_CSP_Decay_Days Malaria float 1 3.40E+38 90 Exponential decay time for the circumsporozoite protein (CSP) antibody concentration (in days) when boosted above natural maximum concentrations. { \"Antibody_CSP_Decay_Days\" : 80 } Antibody_CSP_Killing_Inverse_Width Malaria float 1.00E-0 1000000 1.5 The inverse width of the sigmoidal sporozoite killing function of circumsporozoite protein (CSP) antibody concentration. { \"Antibody_CSP_Killing_Inverse_Width\" : 1.7 } Antibody_CSP_Killing_Threshold Malaria float 1.00E-0 1000000 10 The threshold value on circumsporozoite protein (CSP) antibody concentration for sporozoite killing. { \"Antibody_CSP_Killing_Threshold\" : 20 } Antibody_IRBC_Kill_Rate Malaria double NA NA 2 The scale factor multiplied by antibody level to produce the rate of clearance of the infected red blood cell (IRBC) population. { \"Antibody_IRBC_Kill_Rate\" : 1.595 } Antibody_Memory_Level Malaria float 0 0.35 0.2 The limiting level of antibody capacity that remains after a prolonged absence of a specific antigen. The antibody capacity decays to this level gradually after infection is cleared. The decay rate in antibody capacity is set so that hyperimmunity is lost within 4 months, and capacity continues to decay to this level. The antibody memory level is relevant for year-scale dynamics, but not for long-term dynamics (10-20 years). { \"Antibody_Memory_Level\" : 0.3 } Antibody_Stimulation_C50 Malaria float 0.1 10000 10 The concentration of an antigen, measured in IRBC/ul at which growth in antibody capacity against the antigen increases at half the maximum rate specified by Antibody_Capacity_Growth_Rate . { \"Antibody_Stimulation_C50\" : 30 , \"Antibody_Capacity_Growth_Rate\" : 0.09 } Antigen_Switch_Rate Malaria double NA NA 2.00E-09 The antigenic switching rate per infected red blood cell per asexual cycle. See Parasite_Switch_Type for different switching patterns. { \"Antigen_Switch_Rate\" : 5e-09 } Aquatic_Arrhenius_1 Vector float 0 1.00E+15 8.42E+10 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the daily rate of fractional progression of mosquito aquatic development (egg-hatching through emergence). This duration is a decreasing function of temperature. The variable a1 is a temperature-independent scale factor on development rate. { \"Vector_Species_Params\" : { \"aegypti\" : { \"Aquatic_Arrhenius_1\" : 9752291.727 } } } Aquatic_Arrhenius_2 Vector float 0 1.00E+15 8328 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the daily rate of fractional progression of mosquito aquatic development (egg-hatching through emergence). This duration is a decreasing function of temperature. The variable a2 governs how quickly the rate changes with temperature. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Aquatic_Arrhenius_2\" : 8328 } } } Aquatic_Mortality_Rate Vector float 0 1 0.1 The base aquatic mortality per day for the species before adjusting for effects of overpopulation and drying out of aquatic habitat. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Aquatic_Mortality_Rate\" : 0.1 } } } ART_CD4_at_Initiation_Saturating_Reduction_in_Mortality HIV float 0 3.40E+38 350 The duration from ART enrollment to on-ART HIV-caused death increases with CD4 at ART initiation up to a threshold determined by this parameter value. { \"ART_CD4_at_Initiation_Saturating_Reduction_in_Mortality\" : 350 } ART_Viral_Suppression_Multiplier HIV float 0 1 0.08 Multiplier acting on Base_Infectivity to determine the per-act transmission probability of a virally suppressed HIV+ individual. { \"ART_Viral_Suppression_Multiplier\" : 0.3 } Base_Air_Temperature Generic float -55 45 22 The air temperature, in degrees Celsius, where Climate_Model is set to CLIMATE_CONSTANT. { \"Climate_Model\" : \"CLIMATE_CONSTANT\" , \"Base_Air_Temperature\" : 30 } Base_Gametocyte_Fraction_Male Malaria double NA NA 0.2 Fraction of gametocytes that are male. { \"Base_Gametocyte_Fraction_Male\" : 0.25 } Base_Gametocyte_Mosquito_Survival_Rate Malaria float 0 1 0.01 Average fraction of gametocytes in a blood meal that are successful in infecting the mosquito in the absence of other modulating effects, such as fever. { \"Base_Gametocyte_Mosquito_Survival_Rate\" : 0.015 } Base_Gametocyte_Production_Rate Malaria double NA NA 0.02 The fraction of infected red blood cells (IRBCs) producing gametocytes. { \"Base_Gametocyte_Mosquito_Survival_Rate\" : 0.01 } Base_Incubation_Period Generic float 0 3.40E+38 6 Average duration, in days, of the incubation period before infected individuals become infectious. Incubation_Period_Distribution must be set to either FIXED_DURATION or EXPONENTIAL_DURATION. { \"Incubation_Period_Distribution\" : \"EXPONENTIAL_DURATION\" , \"Base_Incubation_Period\" : 1 } Base_Individual_Sample_Rate Generic float 0 1 1 The base rate of sampling for individuals, equal to the fraction of individuals in each node being sampled. Reducing the sampling rate will reduce the time needed to run simulations. Individual_Sampling_Type must be set to FIXED_SAMPLING or ADAPTED_SAMPLING_BY_IMMUNE_STATE. { \"Base_Individual_Sample_Rate\" : 0.01 } Base_Infectious_Period Generic float 0 3.40E+38 6 Average duration, in days, of the infectious period before the infection is cleared. Infectious_Period_Distribution must be set to either FIXED_DURATION or EXPONENTIAL_DURATION. { \"Base_Infectious_Period\" : 4 } Base_Infectivity Generic float 0 1000 0.3 The base infectiousness of individuals before accounting for transmission-blocking effects of acquired immunity and/or campaign interventions. { \"Base_Infectivity\" : 0.5 } Base_Land_Temperature Generic float -55 60 26 The land temperature, in degrees Celsius, where Climate_Model is set to CLIMATE_CONSTANT. { \"Climate_Model\" : \"CLIMATE_CONSTANT\" , \"Base_Land_Temperature\" : 20 } Base_Mortality Generic float 0 1000 0.001 The base mortality of the infection before accounting for individual immune modification factors. Depending on the setting of Mortality_Time_Course , this is either the daily probability of the disease being fatal (DAILY_MORTALITY) or the probability of death at the end of the infection duration (MORTALITY_AFTER_INFECTIOUS). Enable_Vital_Dynamics must be set to true (1). { \"Enable_Vital_Dynamics\" : 1 , \"Mortality_Time_Course\" : \"DAILY_MORTALITY\" , \"Base_Mortality\" : 0.01 } Base_Population_Scale_Factor Generic float 0 3.40E+38 1 The scale factor for InitialPopulation in the demographics file. If Population_Scale_Type is set to FIXED_SCALING, the initial simulation population is uniformly scaled over the entire area to adjust for historical or future population density. { \"Base_Population_Scale_Factor\" : 0.0001 } Base_Rainfall Generic float 0 150 10 The value of rainfall per day in millimeters when Climate_Model is set to CLIMATE_CONSTANT. { \"Climate_Model\" : \"CLIMATE_CONSTANT\" , \"Base_Rainfall\" : 20 } Base_Relative_Humidity Generic float 0 1 0.75 The value of humidity where Climate_Model is set to CLIMATE_CONSTANT. { \"Base_Relative_Humidity\" : 0.1 } Base_Sporozoite_Survival_Fraction Malaria float 0 1 0.25 The fraction of sporozoites that survive to infect a hepatocyte in the absence of anti-CSP protection. { \"Base_Sporozoite_Survival_Fraction\" : 0.25 } Base_Year Generic float 1900 2200 2015 The absolute time in years when the simulation begins. This can be combined with CampaignEventByYear to trigger campaign events. Note: this parameter is only used in the STI sim type (which includes HIV), and for Typhoid. { \"Base_Year\" : 1960 } Birth_Rate_Boxcar_Forcing_Amplitude Generic float 0 3.40E+38 0 Fractional increase in birth rate during high birth season when Birth_Rate_Time_Dependence is set to ANNUAL_BOXCAR_FUNCTION. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Birth_Rate_Time_Dependence\" : \"ANNUAL_BOXCAR_FUNCTION\" , \"Birth_Rate_Boxcar_Forcing_Amplitude\" : 0.1 } Birth_Rate_Boxcar_Forcing_End_Time Generic float 0 365 0 Day of the year when the high birth rate season ends when Birth_Rate_Time_Dependence is set to ANNUAL_BOXCAR_FUNCTION. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Birth_Rate_Time_Dependence\" : \"ANNUAL_BOXCAR_FUNCTION\" , \"Birth_Rate_Boxcar_Forcing_End_Time\" : 220 } Birth_Rate_Boxcar_Forcing_Start_Time Generic float 0 365 0 Day of the year when the high birth rate season begins when Birth_Rate_Time_Dependence is set to ANNUAL_BOXCAR_FUNCTION. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Birth_Rate_Time_Dependence\" : \"ANNUAL_BOXCAR_FUNCTION\" , \"Birth_Rate_Boxcar_Forcing_Start_Time\" : 130 } Birth_Rate_Dependence Generic enum NA NA FIXED_BIRTH_RATE The method used to modify the value set in BirthRate in the demographics file. Possible values are: NONE Births are not allowed during the simulation, even if Enable_Birth is set to true (1). FIXED_BIRTH_RATE The absolute rate at which new individuals are born, as set by BirthRate . POPULATION_DEP_RATE Scales the node population to determine the birth rate. If BirthRate is greater than 0.005, a value of 2% per year (0.02/365) is used instead. DEMOGRAPHIC_DEP_RATE Scales the female population within fertility age ranges to determine the birth rate. If BirthRate is greater than 0.005, a value of 1 child every 8 years of fertility [1/8/365(~0.000342)] is used instead. INDIVIDUAL_PREGNANCIES Scales the female population within fertility age ranges to determine the birth rate, but pregnancies are assigned on an individual basis and result in a 40-week pregnancy for a specific individual with a birth at the end. INDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE Similar to INDIVIDUAL_PREGNANCIES, but determines the rate based on the FertilityDistribution (in IndividualAttributes ) and Urban (in NodeAttributes ), as well as the individual\u2019s age. INDIVIDUAL_PREGNANCIES_BY_AGE_AND_YEAR Similar to INDIVIDUAL_PREGNANCIES, but determines the rate based on the FertilityDistribution (in IndividualAttributes ), using the individual\u2019s age and the year of the simulation. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Birth_Rate_Dependence\" : \"POPULATION_DEP_RATE\" } Birth_Rate_Sinusoidal_Forcing_Amplitude Generic float 0 1 0 The amplitude of sinusoidal variations in birth rate when Birth_Rate_Time_Dependence is set to SINUSOIDAL_FUNCTION_OF_TIME. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Birth_Rate_Time_Dependence\" : \"SINUSOIDAL_FUNCTION_OF_TIME\" , \"Birth_Rate_Sinusoidal_Forcing_Amplitude\" : 0.1 } Birth_Rate_Sinusoidal_Forcing_Phase Generic float 0 365 0 The phase of sinusoidal variations in birth rate. Birth_Rate_Time_Dependence must be set to SINUSOIDAL_FUNCTION_OF_TIME. { \"Birth_Rate_Sinusoidal_Forcing_Phase\" : 20 } Birth_Rate_Time_Dependence Generic enum NA NA NONE A scale factor that allows the birth rate to be altered by time or season. Enable_Birth must be set to true (1). Possible values are: NONE Birth rate does not vary by time. SINUSOIDAL_FUNCTION_OF_TIME Allows birth rate to be time-dependent, following a sinusoidal shape. Set Birth_Rate_Sinusoidal_Forcing_Amplitude and Birth_Rate_Sinusoidal_Forcing_Phase . ANNUAL_BOXCAR_FUNCTION Allows birth rate to follow a boxcar function, such that it will be equal to zero for an entire time period (e.g. year) except for a single interval in which it is equal to a constant. Set Birth_Rate_Boxcar_Forcing_Amplitude , Birth_Rate_Boxcar_Forcing_End_Time , and Birth_Rate_Boxcar_Forcing_Start_Time . { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Birth_Rate_Time_Dependence\" : \"ANNUAL_BOXCAR_FUNCTION\" } Bodyweight_Exponent Malaria float 0 100000 0 The effect of bodyweight on maximum drug concentration in an individual patient. Drug_Cmax is divided by patient bodyweight raised to the power of Bodyweight_Exponent to account for the influence of body size on volume of distribution. { \"Malaria_Drug_Params\" : { \"Artemether_Lumefantrine\" : { \"Bodyweight_Exponent\" : 2 } } } Campaign_Filename Generic string NA NA \u00a0 The path to the campaign file. It is required when interventions are part of the simulation and Enable_Interventions is set to true (1). The file must be in .json format. { \"Enable_Interventions\" : 1 , \"Campaign_Filename\" : \"campaign.json\" } CD4_At_Death_LogLogistic_Heterogeneity HIV float 0 100 0 The inverse shape parameter of a Weibull distribution that represents the at-death CD4 cell count. { \"CD4_At_Death_LogLogistic_Heterogeneity\" : 0.7 } CD4_At_Death_LogLogistic_Scale HIV float 1 1000 31.63 The scale parameter of a Weibull distribution that represents the at-death CD4 cell count. { \"CD4_At_Death_LogLogistic_Scale\" : 2.96 } CD4_Post_Infection_Weibull_Heterogeneity HIV float 0 100 0.275642 The inverse shape parameter of a Weibull distribution that represents the post-acute-infection CD4 cell count. { \"CD4_Post_Infection_Weibull_Heterogeneity\" : 0 } CD4_Post_Infection_Weibull_Scale HIV float 1 1000 560.432 The scale parameter of a Weibull distribution that represents the post-acute-infection CD4 cell count. { \"CD4_Post_Infection_Weibull_Scale\" : 550 } CD4_Strata_Activation TBHIV array of floats 1 2000 1 An array of CD4 count thresholds by which to stratify the population. Increased risk of TB activation for each stratum is set using TB_CD4_Activation_Vector . { \"CD4_Strata_Activation\" : [ 0 , 350 , 500 , 1000 ], \"TB_CD4_Activation_Vector\" : [ 0.01 , 0.01 , 0.02 , 0.02 ] } Climate_Model Generic enum NA NA CLIMATE_OFF How and from what files the climate of a simulation is configured. The possible values are: CLIMATE_OFF No climate files used. CLIMATE_CONSTANT Uses the conditional parameters that give the fixed values of temperature or rain for land temperature, air temperature, rainfall, and humidity. CLIMATE_KOPPEN Uses an input data file that decodes Koppen codes by geographic location. CLIMATE_BY_DATA Reads everything out of several input data files with additional parameters that allow the addition of stochasticity or scale offsets. { \"Climate_Model\" : \"CLIMATE_CONSTANT\" } Climate_Update_Resolution Generic enum NA NA CLIMATE_UPDATE_YEAR The resolution of data in climate files. Climate_Model must be set to CLIMATE_CONSTANT, CLIMATE_BY_DATA, or CLIMATE_KOPPEN. Possible values are: CLIMATE_UPDATE_YEAR CLIMATE_UPDATE_MONTH CLIMATE_UPDATE_WEEK CLIMATE_UPDATE_DAY CLIMATE_UPDATE_HOUR { \"Climate_Update_Resolution\" : \"CLIMATE_UPDATE_DAY\" } Clinical_Fever_Threshold_High Malaria float 0 15 1 The fever threshold (in degrees Celsius above normal) to start a clinical incident. { \"Clinical_Fever_Threshold_High\" : 1.5 } Clinical_Fever_Threshold_Low Malaria float 0 5 1 The fever threshold (in degrees Celsius above normal) to end a clinical incident. { \"Clinical_Fever_Threshold_Low\" : 0.5 } Coital_Dilution_Factor_2_Partners STI float 1.19E-0 1 1 The multiplicative reduction in the coital act rate for all relationship types when an individual has exactly two current partners. Represents coital dilution. { \"Coital_Dilution_Factor_2_Partners\" 0.5 } Coital_Dilution_Factor_3_Partners STI float 1.19E-0 1 1 The multiplicative reduction in the coital act rate for all relationship types when an individual has exactly three current partners. Represents coital dilution. { \"Coital_Dilution_Factor_3_Partners\" : 0.33 } Coital_Dilution_Factor_4_Plus_Partners STI float 1.19E-0 1 1 The multiplicative reduction in the coital act rate for all relationship types when an individual has exactly three current partners. Represents coital dilution. { \"Coital_Dilution_Factor_4_Partners\" : 0.45 } Condom_Transmission_Blocking_Probability STI float 0 1 0.9 The per-act multiplier of the transmission probability when a condom is used. { \"Condom_Transmission_Blocking_Probability\" : 0.99 } Config_Name Generic string NA NA UNINITIALIZED STRING The optional, user-supplied title naming a configuration file. { \"Config_Name\" : \"My_First_Config\" } Custom_Reports_Filename Generic string NA NA UNINITIALIZED STRING The name of the file containing custom report configuration parameters. Set to NoCustomReports to prevent loading any of the custom report DLLs. The file must be in .json format. { \"Custom_Reports_Filename\" : \"custom_reports.json\" } Cycle_Arrhenius_1 Vector float 0 1.00E+15 4.09E+10 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the mosquito feeding cycle rate. This duration is a decreasing function of temperature. The variable a1 is a temperature-independent scale factor on feeding rate. Temperature_Dependent_Feeding_Cycle must be set to ARRHENIUS_DEPENDENCE. { \"Temperature_Dependent_Feeding_Cycle\" : \"ARRHENIUS_DEPENDENCE\" , \"Vector_Species_Params\" : { \"arabiensis\" : { \"Cycle_Arrhenius_1\" : 99 , \"Cycle_Arrhenius_2\" : 88 } } } Cycle_Arrhenius_2 Vector float 0 1.00E+15 7740 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the mosquito feeding cycle rate. This duration is a decreasing function of temperature. The variable a2 is a temperature-independent scale factor on feeding rate. Temperature_Dependent_Feeding_Cycle must be set to ARRHENIUS_DEPENDENCE. { \"Temperature_Dependent_Feeding_Cycle\" : \"ARRHENIUS_DEPENDENCE\" , \"Vector_Species_Params\" : { \"arabiensis\" : { \"Cycle_Arrhenius_1\" : 99 , \"Cycle_Arrhenius_2\" : 88 } } } Cycle_Arrhenius_Reduction_Factor Vector float 0 1 1 The scale factor applied to cycle duration (from oviposition to oviposition) to reduce the duration when primary follicles are at stage II rather than I in the case of newly emerged females. Temperature_Dependent_Feeding_Cycle must be set to ARRHENIUS_DEPENDENCE. { \"Temperature_Dependent_Feeding_Cycle\" : \"ARRHENIUS_DEPENDENCE\" , \"Vector_Species_Params\" : { \"funestus\" : { \"Cycle_Arrhenius_Reduction_Factor\" : 0.44 } } } Cytokine_Gametocyte_Inactivation Malaria float 0 1 0.02 The strength of inflammatory response inactivation of gametocytes. { \"Cytokine_Gametocyte_Inactivation\" : 0.0167 } Days_Between_Feeds Vector float 0 730 2 The number of days between female mosquito feeding attempts. In the cohort model, this value is converted to a feeding rate, equal to 1/(value of this parameter), and is used to determine the probability that one of each of the various feeding modes (animal host, human host, indoor, outdoor, etc) occurs. In the individual model, this parameter determines the number of days a vector waits between feeds. { \"Vector_Species_Params\" : { \"gambiae\" : { \"Days_Between_Feeds\" : 3 } } } Days_Between_Symptomatic_And_Death_Weibull_Heterogeneity HIV float 0.1 10 1 The time between the onset of AIDS symptoms and death is sampled from a Weibull distribution; this parameter governs the heterogeneity (inverse shape) of the Weibull. { \"Days_Between_Symptomatic_And_Death_Weibull_Heterogeneity\" : 0.5 } Days_Between_Symptomatic_And_Death_Weibull_Scale HIV float 1 3650 183 The time between the onset of AIDS symptoms and death is sampled from a Weibull distribution; this parameter governs the scale of the Weibull. { \"Days_Between_Symptomatic_And_Death_Weibull_Scale\" : 618.3 } Death_Rate_Dependence Generic enum NA NA NONDISEASE_MORTALITY_OFF Determines how likely individuals are to die from natural, non-disease causes. Possible values are: NONDISEASE_MORTALITY_OFF The daily mortality rate is 0, and no one dies from non-disease related causes. NONDISEASE_MORTALITY_BY_AGE_AND_GENDER The individual\u2019s age and gender are taken into account to determine the daily mortality rate. NONDISEASE_MORTALITY_BY_YEAR_AND_AGE_FOR_EACH_GENDER Gender, age, and year, are all taken into account to determine the daily mortality rate. Properties, rates, and bin sizes can be set for each gender in the demographics file. { \"Death_Rate_Dependence\" : \"NONDISEASE_MORTALITY_OFF\" } Default_Geography_Initial_Node_Population Generic integer 0 1000000 1000 When using the built-in demographics for default geography, the initial number of individuals in each node. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. Enable_Demographics_Builtin must be set to true (1). { \"Enable_Demographics_Builtin\" : 1 , \"Default_Geography_Initial_Node_Population\" : 1000 , \"Default_Geography_Torus_Size\" : 3 } Default_Geography_Torus_Size Generic integer 3 100 10 When using the built-in demographics for default geography, the square root of the number of nodes in the simulation. The simulation uses an N x N square grid of nodes with N specified by this parameter. If migration is enabled, the N x N nodes are assumed to be a torus and individuals can migrate from any node to all four adjacent nodes. To enable migration, set Migration_Model to FIXED_RATE_MIGRATION. Built-in migration is a form of \u201clocal\u201d migration where individuals only migrate to the adjacent nodes. You can use the x_Local_Migration parameter to control the rate of migration. The other migration parameters are ignored. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. Enable_Demographics_Builtin must be set to true (1). { \"Enable_Demographics_Builtin\" : 1 , \"Default_Geography_Initial_Node_Population\" : 1000 , \"Default_Geography_Torus_Size\" : 3 } Demographics_Filenames Generic array of strings NA NA \u00a0 An array of the paths to demographics files containing information on the identity and demographics of the region to simulate. The files must be in .json format. { \"Demographics_Filenames\" : [ \"Namawala_single_node_demographics.json\" , \"Namawala_demographics_overlay.json\" ] } Drought_Egg_Hatch_Delay Vector float 0 1 0.33 Proportion of regular egg hatching that still occurs when habitat dries up. Enable_Drought_Egg_Hatch_Delay must be set to 1. { \"Enable_Drought_Egg_Hatch_Delay\" : 1 , \"Drought_Egg_Hatch_Delay\" : 0.33 } Drug_Cmax Malaria float 0 100000 1000 The maximum drug concentration used. Note This parameter and the Drug_PKPD_C50 parameter must use the same units. { \"Malaria_Drug_Params\" : { \"Artemether_Lumefantrine\" : { \"Drug_Cmax\" : 1000 , \"Drug_PKPD_C50\" : 100 } } } Drug_Decay_T1 Malaria float 0 100000 1 The primary drug decay rate, in days. { \"Malaria_Drug_Params\" : { \"Artemisinin\" : { \"Drug_Decay_T1\" : 1 , \"Drug_Decay_T2\" : 1 } } } Drug_Decay_T2 Malaria float 0 100000 1 The secondary drug decay rate, in days. { \"Malaria_Drug_Params\" : { \"Artemisinin\" : { \"Drug_Decay_T1\" : 1 , \"Drug_Decay_T2\" : 1 } } } Drug_Dose_Interval Malaria float 0 100000 1 The interval between doses of drugs, in days. { \"Malaria_Drug_Params\" : { \"Artemisinin\" : { \"Drug_Dose_Interval\" : 1 } } } Drug_Fulltreatment_Doses Malaria float 1 100000 3 The number of doses for a full treatment of a drug. { \"Malaria_Drug_Params\" : { \"Chloroquine\" : { \"Drug_Fulltreatment_Doses\" : 3 } } } Drug_Gametocyte02_Killrate Malaria float 0 100000 0 The log reduction per day in early-stage gametocytes. { \"Malaria_Drug_Params\" : { \"Chloroquine\" : { \"Drug_Gametocyte02_Killrate\" : 0 , \"Drug_Gametocyte34_Killrate\" : 0 , \"Drug_GametocyteM_Killrate\" : 0 } } } Drug_Gametocyte34_Killrate Malaria float 0 100000 0 The log reduction per day in late-stage gametocytes. { \"Malaria_Drug_Params\" : { \"Chloroquine\" : { \"Drug_Gametocyte02_Killrate\" : 0 , \"Drug_Gametocyte34_Killrate\" : 0 , \"Drug_GametocyteM_Killrate\" : 0 } } } Drug_GametocyteM_Killrate Malaria float 0 100000 0 The log reduction per day in mature gametocyte numbers at saturated drug concentrations. { \"Malaria_Drug_Params\" : { \"Chloroquine\" : { \"Drug_Gametocyte02_Killrate\" : 0 , \"Drug_Gametocyte34_Killrate\" : 0 , \"Drug_GametocyteM_Killrate\" : 0 } } } Drug_Hepatocyte_Killrate Malaria float 0 100000 0 The log reduction in hepatocyte numbers per day. { \"Malaria_Drug_Params\" : { \"Chloroquine\" : { \"Drug_Hepatocyte_Killrate\" : 0 } } } Drug_PKPD_C50 Malaria float 0 100000 100 The concentration at which drug killing rates are half of the maximum. Note This parameter and the Drug_Cmax parameter must use the same units. { \"Malaria_Drug_Params\" : { \"Artemether_Lumefantrine\" : { \"Drug_Cmax\" : 1000 , \"Drug_PKPD_C50\" : 100 } } } Drug_Vd Malaria float 0 100000 10 The volume of drug distribution in a pharmacokinetic two compartment model. The first compartment comprises central organs and tissues and the second compartment comprises peripheral tissues. This value is the ratio of the volume of the second compartment to that of the first. { \"Malaria_Drug_Params\" : { \"Chloroquine\" : { \"Drug_Vd\" : 3 } } } Egg_Arrhenius1 Vector float 0 1.00E+10 6.16E+07 The Arrhenius equation, math: a_1^{-a_2/T} , with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature. The variable a1 is a temperature-independent scale factor on hatching rate. Enable_Temperature_Dependent_Egg_Hatching must be set to 1. { \"Enable_Temperature_Dependent_Egg_Hatching\" : 1 , \"Egg_Arrhenius1\" : 61599956 , \"Egg_Arrhenius2\" : 5754 } Egg_Arrhenius2 Vector float 0 1.00E+10 5754.03 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature. The variable a2 is a temperature-dependent scale factor on hatching rate. Enable_Temperature_Dependent_Egg_Hatching must be set to 1. { \"Enable_Temperature_Dependent_Egg_Hatching\" : 1 , \"Egg_Arrhenius1\" : 61599956 , \"Egg_Arrhenius2\" : 5754 } Egg_Batch_Size Vector float 0 10000 100 Number of eggs laid by one female mosquito that has fed successfully. The value of Egg_Batch_Size is used for both the number of female eggs and for the number of male eggs. For example, the default value of 100 means that there are 100 female eggs and 100 male eggs. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Egg_Batch_Size\" : 100 } } } Egg_Hatch_Delay_Distribution Vector enum NA NA NO_DELAY The distribution of the delay from oviposition to egg hatching. Possible values are: NO_DELAY No delay between oviposition and egg hatching. EXPONENTIAL_DURATION There is a delay that follows an exponential distribution. Set the mean delay in Mean_Egg_Hatch_Delay . { \"Egg_Hatch_Delay_Distribution\" : \"EXPONENTIAL_DURATION\" , \"Mean_Egg_Hatch_Delay\" : 2 } Egg_Hatch_Density_Dependence Vector enum NA NA NO_DENSITY_DEPENDENCE The effect of larval density on egg hatching rate. Possible values are: DENSITY_DEPENDENCE Egg hatching is reduced when the habitat is nearing its carrying capacity. NO_DENSITY_DEPENDENCE Egg hatching is not dependent upon larval density. { \"Egg_Hatch_Density_Dependence\" : \"NO_DENSITY_DEPENDENCE\" } Egg_Saturation_At_Oviposition Vector enum NA NA NO_SATURATION The means by which the number of viable eggs at oviposition become saturated in the available larval habitat. Possible values are: NO_SATURATION Egg number does not saturate. SATURATION_AT_OVIPOSITION Eggs are saturated at oviposition. SIGMOIDAL_SATURATION Eggs are saturated along a sigmoidal curve. { \"Egg_Saturation_At_Oviposition\" : \"SATURATION_AT_OVIPOSITION\" } Enable_Aging Generic boolean 0 1 1 Controls whether or not individuals in a population age during the simulation. Enable_Vital_Dynamics must be set to true (1). { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Aging\" : 1 } Enable_Air_Migration Generic boolean 0 1 0 Controls whether or not migration by air travel will occur. Migration_Model must be set to FIXED_RATE_MIGRATION. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Air_Migration\" : 1 , \"Air_Migration_Filename\" : \"../inputs/air_migration.bin\" } Enable_Birth Generic boolean 0 1 1 Controls whether or not individuals will be added to the simulation by birth. Enable_Vital_Dynamics must be set to true (1). If you want new individuals to have the same demographics coverage as existing individuals, you must add a BirthTriggeredIV to the campaign file. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 } Enable_Climate_Stochasticity Generic boolean 0 1 0 Controls whether or not the climate has stochasticity. Climate_Model must be set to CLIMATE_CONSTANT or CLIMATE_BY_DATA. Set the variance using the parameters Air_Temperature_Variance , Land_Temperature_Variance , Enable_Rainfall_Stochasticity , and Relative_Humidity_Variance . { \"Climate_Model\" : \"CLIMATE_BY_DATA\" , \"Enable_Climate_Stochasticity\" : 1 , \"Air_Temperature_Variance\" : 2 , \"Enable_Rainfall_Stochasticity\" : 1 , \"Land_Temperature_Variance\" : 2 , \"Relative_Humidity_Variance\" : 0.05 } Enable_Coital_Dilution STI boolean 0 1 1 Controls whether or not coital dilution will occur. { \"Enable_Coital_Dilution\" : 1 } Enable_Default_Reporting Generic boolean 0 1 1 Controls whether or not the default InsetChart.json report is created. { \"Enable_Default_Reporting\" : 1 } Enable_Demographics_Birth Generic boolean 0 1 0 Controls whether or not newborns have identical or heterogeneous characteristics. Set to false (0) to give all newborns identical characteristics; set to true (1) to allow for heterogeneity in traits such as sickle-cell status. Enable_Birth must be set to true (1). { \"Enable_Birth\" : 1 , \"Enable_Demographics_Birth\" : 1 } Enable_Demographics_Builtin Generic boolean 0 1 0 Controls whether or not built-in demographics for default geography will be used. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. Set to true (1), to define the initial population and number of nodes using Default_Geography_Initial_Node_Population and Default_Geography_Torus_Size . Set to false (1) to use demographics input files defined in Demographics_Filenames . { \"Enable_Demographics_Builtin\" : 1 , \"Default_Geography_Initial_Node_Population\" : 1000 , \"Default_Geography_Torus_Size\" : 3 } Enable_Demographics_Gender Generic boolean 0 1 1 Controls whether or not gender ratios are drawn from a Gaussian or 50/50 draw. Set to true (1) to create gender ratios drawn from a male/female ratio that is randomly smeared by a Gaussian of width 1%; set to false (0) to assign a gender ratio based on a 50/50 draw. { \"Enable_Demographics_Gender\" : 1 } Enable_Demographics_Other Generic boolean 0 1 0 Controls whether or not other demographic factors are included in the simulation, such as the fraction of individuals above poverty, urban/rural characteristics, heterogeneous initial immunity, or risk. These factors are set in the demographics file. { \"Enable_Demographics_Other\" : 1 } Enable_Demographics_Reporting Generic boolean 0 1 1 Controls whether or not demographic summary data and age-binned reports are outputted to file. { \"Enable_Demographics_Reporting\" : 1 } Enable_Disease_Mortality Generic boolean 0 1 1 Controls whether or not individuals die due to disease. { \"Enable_Disease_Mortality\" : 1 } Enable_Egg_Mortality Vector boolean 0 1 0 Controls whether or not to include a daily mortality rate on the egg population, which is independent of climatic factors. { \"Enable_Egg_Mortality\" : 1 , } Enable_Family_Migration Generic boolean 0 1 0 Controls whether or not all members of a household can migrate together. All residents must be home before they can leave on the trip. Migration_Model must be set to FIXED_RATE_MIGRATION. { \"Enable_Migration\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Family_Migration\" : 1 , \"Family_Migration_Filename\" : \"../inputs/family_migration.bin\" } Enable_Heterogeneous_Intranode_Transmission Generic boolean 0 1 0 Controls whether or not individuals experience heterogeneous disease transmission within a node. When set to true (1), individual property definitions and the matrix must be specified in the demographics file: each property set in IndividualProperties has a TransmissionMatrix that contains the values for the heterogeneous transmission between the groups defined for that property set. The values are multiplied with the 0 value configured by Base_Infectivity . This is used only in generic simulations, but must be set to false (0) for all other simulation types. Heterogeneous transmission for other diseases uses other mechanisms included with the simulation type. { \"Enable_Heterogeneous_Intranode_Transmission\" : 1 } Enable_Immune_Decay Generic boolean 0 1 1 Controls whether or not immunity decays after an infection clears. Set to true (1) if immunity decays; set to false (0) if recovery from the disease confers complete immunity for life. Enable_Immunity must be set to true (1). { \"Enable_Immunity\" : 1 , \"Enable_Immune_Decay\" : 0 } Enable_Immunity Generic boolean 0 1 1 Controls whether or not an individual has protective immunity after an infection clears. { \"Enable_Immunity\" : 1 } Enable_Interventions Generic boolean 0 1 0 Controls whether or not campaign interventions will be used in the simulation. Set Campaign_Filename to the path of the file that contains the campaign interventions. { \"Enable_Interventions\" : 1 , \"Campaign_Filename\" : \"campaign.json\" } Enable_Local_Migration Generic boolean 0 1 0 Controls whether or not local migration (the diffusion of people in and out of nearby nodes by foot travel) occurs. Migration_Model must be set to FIXED_RATE_MIGRATION. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Local_Migration\" : 1 , \"Local_Migration_Filename\" : \"../inputs/local_migration.bin\" } Enable_Maternal_Transmission Generic boolean 0 1 0 Controls whether or not infectious mothers infect infants at birth. Enable_Birth must be set to true (1). { \"Enable_Birth\" : 1 , \"Enable_Maternal_Transmission\" : 1 } Enable_Migration_Heterogeneity Generic boolean 0 1 1 Controls whether or not migration rate is heterogeneous among individuals. Set to true (1) to use a migration rate distribution in the demographics file; set to false (0) to use the same migration rate applied to all individuals. Migration_Model must be set to FIXED_RATE_MIGRATION. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Migration_Heterogeneity\" : 1 } Enable_Property_Output Generic boolean 0 1 0 Controls whether or not to create property output reports, which detail groups as defined in IndividualProperties in the demographics file. When there is more than one property type, the report will display the channel information for all combinations of the property type groups. { \"Enable_Property_Output\" : 1 } Enable_Rainfall_Stochasticity Generic boolean 0 1 1 Controls whether or not there is stochastic variation in rainfall; set to true (1) for stochastic variation of rainfall that is drawn from an exponential distribution (with a mean value as the daily rainfall from the Climate_Model values CLIMATE_CONSTANT or CLIMATE_BY_DATA), or set to false (0) to disable rainfall stochasticity. { \"Enable_Climate_Stochasticity\" : 1 , \"Air_Temperature_Variance\" : 2 , \"Enable_Rainfall_Stochasticity\" : 1 , \"Land_Temperature_Variance\" : 2 , \"Relative_Humidity_Variance\" : 0.05 } Enable_Regional_Migration Generic boolean 0 1 0 Controls whether or not there is migration by road vehicle into and out of nodes in the road network. Migration_Model must be set to FIXED_RATE_MIGRATION. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Regional_Migration\" : 1 , \"Regional_Migration_Filename\" : \"../inputs/regional_migration.bin\" } Enable_Sea_Migration Generic boolean 0 1 0 Controls whether or not there is migration on ships into and out of coastal cities with seaports. Migration_Model must be set to FIXED_RATE_MIGRATION. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Sea_Migration\" : 1 , \"Sea_Migration_Filename\" : \"../inputs/sea_migration.bin\" } Enable_Sexual_Combination Malaria boolean 0 1 0 Controls whether or not male and female gametocytes undergo sexual combination. Set to true (1) to allow for sexual combination; set to false (0) for transmission of each strain only. Note This parameter is currently not in use. { \"Enable_Sexual_Combination\" : 0 } Enable_Spatial_Output Generic boolean 0 1 0 Controls whether or not spatial output reports are created. If set to true (1), spatial output reports include all channels listed in the parameter Spatial_Output_Channels . Note Spatial output files require significant processing time and disk space. { \"Enable_Spatial_Output\" : 1 , \"Spatial_Output_Channels\" : [ \"Prevalence\" , \"New_Infections\" ] } Enable_Superinfection Generic boolean 0 1 0 Controls whether or not an individual can have multiple infections simultaneously. Set to true (1) to allow for multiple simultaneous infections; set to false (0) if multiple infections are not possible. Set the Max_Individual_Infections parameter. { \"Enable_Superinfection\" : 1 , \"Max_Individual_Infections\" : 2 } Enable_Temperature_Dependent_Egg_Hatching Vector boolean 0 1 0 Controls whether or not temperature has an effect on egg hatching, defined by Egg_Arrhenius_1 and Egg_Arrhenius_2 . { \"Enable_Temperature_Dependent_Egg_Hatching\" : 1 , \"Egg_Arrhenius1\" : 61599956.864 , \"Egg_Arrhenius2\" : 5754.033 } Enable_Vector_Aging Vector boolean 0 1 0 Controls whether or not vectors undergo senescence as they age. { \"Enable_Vector_Aging\" : 1 } Enable_Vector_Migration Vector boolean 0 1 0 Controls whether or not vectors can migrate. Vector_Sampling_Type must be set to TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. Specific migration types must be enabled or disabled. { \"Vector_Sampling_Type\" : \"TRACK_ALL_VECTORS\" , \"Enable_Vector_Migration\" : 1 , \"Enable_Vector_Migration_Local\" : 1 , \"Vector_Migration_Filename_Local\" : \"../inputs/vector_local.bin\" } Enable_Vector_Migration_Local Vector boolean 0 1 0 Controls whether or not vectors may migrate to adjacent nodes. Enable_Vector_Migration must be set to true (1). { \"Vector_Sampling_Type\" : \"TRACK_ALL_VECTORS\" , \"Enable_Vector_Migration\" : 1 , \"Enable_Vector_Migration_Local\" : 1 , \"Vector_Migration_Filename_Local\" : \"../inputs/vector_local.bin\" } Enable_Vector_Migration_Regional Vector boolean 0 1 0 Controls whether or not vectors can migration to non-adjacent nodes. Enable_Vector_Migration must be set to true (1). { \"Vector_Sampling_Type\" : \"TRACK_ALL_VECTORS\" , \"Enable_Vector_Migration\" : 1 , \"Enable_Vector_Migration_Regional\" : 1 , \"Vector_Migration_Filename_Regional\" : \"../inputs/vector_regional.bin\" } Enable_Vector_Mortality Vector boolean 0 1 1 Controls whether or not vectors can die. Vector_Sampling_Type must be set to TRACK_ALL_VECTORS. { \"Vector_Sampling_Type\" : \"TRACK_ALL_VECTORS\" , \"Enable_Vector_Mortality\" : 1 } Enable_Vital_Dynamics Generic boolean 0 1 1 Controls whether or not births and deaths occur in the simulation. Births and deaths must be individually enabled and set. { \"Enable_Vital_Dynamics\" : 1 , \"Enable_Birth\" : 1 , \"Death_Rate_Dependence\" : \"NONDISEASE_MORTALITY_OFF\" , \"Base_Mortality\" : 0.002 } Erythropoiesis_Anemia_Effect Malaria float 0 1000 3.5 The exponential rate of increased red-blood-cell production from reduced red-blood-cell availability. { \"Erythropoiesis_Anemia_Effect\" : 3 } Falciparum_MSP_Variants Malaria integer 0 1000 100 The number of distinct merozoite surface protein (MSP) variants for P. falciparum malaria in the overall parasite population in the simulation, not necessarily in an individual. { \"Falciparum_MSP_Variants\" : 4 } Falciparum_Nonspecific_Types Malaria integer 0 1000 20 The number of distinct non-specific types of P. falciparum malaria. { \"Falciparum_Nonspecific_Types\" : 92 } Falciparum_PfEMP1_Variants Malaria integer 0 100000 1000 The number of distinct Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1) variants for P. falciparum malaria in the overall parasite population in the simulation. { \"Falciparum_PfEMP1_Variants\" : 300 } Family_Migration_Filename Generic string NA NA \u00a0 The name of the binary file to use to configure family migration. Enable_Family_Migration must be set to true (1). The file must be in .bin format. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Enable_Family_Migration\" 1 , \"Family_Migration_Filename\" : \"../inputs/family_migration.bin\" } Family_Migration_Roundtrip_Duration Generic float 0 10000 1 The number of days to complete the trip and return to the original node. Migration_Pattern must be set to SINGLE_ROUND_TRIPS. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Migration_Pattern\" : \"SINGLE_ROUND_TRIPS\" , \"Family_Migration_Roundtrip_Duration\" : 100 } Fever_Detection_Threshold Malaria float 0.5 5 1 The level of body temperature above normal (defined as 37 C), in degrees Celsius, corresponding to detectable fever. { \"Fever_Detection_Threshold\" : 1 } Fever_IRBC_Kill_Rate Malaria float 0 1000 0.15 The factor multiplied by fever level to produce a decay rate for the infected red blood cell population due to inflammatory innate immune response. { \"Fever_IRBC_Kill_Rate\" : 1.4 } Fever_Mortality_Inverse_Width Malaria float 0.1 1000000 10 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for fever. { \"Fever_Mortality_Inverse_Width\" : 1000 } Fever_Mortality_Threshold Malaria double NA NA 3 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the fever mortality threshold, in units of degrees Celsius above normal body temperature (defined as 37 C). { \"Fever_Mortality_Threshold\" : 10 } Fever_Severe_Inverse_Width Malaria float 0.1 1000000 10 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for fever. { \"Fever_Severe_Inverse_Width\" : 18.7 } Fever_Severe_Threshold Malaria float 0 10 1.5 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the threshold (in degrees Celsius above normal) for fever indicating severe disease. { \"Fever_Severe_Threshold\" : 2.5 } Fractional_Dose_By_Upper_Age Malaria array of JSON objects NA NA NA An array to specify the fraction of the adult dose to use for children at various ages. Contain Upper_Age_In_Years and Fraction_Of_Adult_Dose values. { \"Malaria_Drug_Params\" : { \"Artemether\" : { \"Fractional_Dose_By_Upper_Age\" : [{ \"Fraction_Of_Adult_Dose\" : 0.25 , \"Upper_Age_In_Years\" : 3 }, { \"Fraction_Of_Adult_Dose\" : 0.5 , \"Upper_Age_In_Years\" : 6 }, { \"Fraction_Of_Adult_Dose\" : 0.75 , \"Upper_Age_In_Years\" : 10 }] } } } Fraction_Of_Adult_Dose Malaria float 0 1 NA The fraction of the adult drug dose given to children below the age defined in Upper_Age_In_Years . Set in the Fractional_Dose_By_Upper_Age array. { \"Malaria_Drug_Params\" : { \"Artemether\" : { \"Fractional_Dose_By_Upper_Age\" : [{ \"Fraction_Of_Adult_Dose\" : 0.25 , \"Upper_Age_In_Years\" : 3 }, { \"Fraction_Of_Adult_Dose\" : 0.5 , \"Upper_Age_In_Years\" : 6 }, { \"Fraction_Of_Adult_Dose\" : 0.75 , \"Upper_Age_In_Years\" : 10 }] } } } Gametocyte_Stage_Survival_Rate Malaria double 0 1 1 The rate of gametocyte survival from one development stage to the next in the absence of drugs or inflammatory immune response. { \"Gametocyte_Stage_Survival_Rate\" : 0.8 } HEG_Fecundity_Limiting Vector float 0 1 0 The fractional reduction in the number of eggs laid by a female mosquito who is homozygous in the homing endonuclease gene (HEG) . HEG_Model must be set to GERMLINE_HOMING, EGG_HOMING, DUAL_GERMLINE_HOMING, or DRIVING_Y. { \"HEG_Model\" : \"EGG_HOMING\" , \"HEG_Fecundity_Limiting\" : 0.7 } HEG_Homing_Rate Vector float 0 1 0 The fractional redistribution from heterozygous offspring to the homozygous homing endonuclease gene (HEG) behavior. HEG_Model\u201d must be set to GERMLINE_HOMING, EGG_HOMING, DUAL_GERMLINE_HOMING, or DRIVING_Y. { \"HEG_Model\" : \"EGG_HOMING\" , \"HEG_Homing_Rate\" : 0.4 } HEG_Model Vector enum NA NA OFF This parameter determines the structure of homing endonuclease gene (HEG) dynamics. Possible values are: OFF No HEG dynamics occur. GERMLINE_HOMING Early HEG dynamics. EGG_HOMING Late HEG dynamics. DUAL_GERMLINE_HOMING TBD DRIVING_Y TBD { \"HEG_Model\" : \"EGG_HOMING\" , \"HEG_Homing_Rate\" : 0.4 , \"HEG_Fecundity_Limiting\" : 0.9 } Heterogeneous_Infectiousness_LogNormal_Scale HIV float 0 10 0 Scale parameter of a log normal distribution that governs an infectiousness multiplier. The multiplier represents heterogeneity in infectivity and adjusts Base_Infectivity . { \"Heterogeneous_Infectiousness_LogNormal_Scale\" : 1 } HIV_Adult_Survival_Scale_Parameter_Intercept HIV float 0.001 1000 21.182 This parameter determines the intercept of the scale parameter, \u03bb, for the Weibull distribution used to determine HIV survival time. Survival time with untreated HIV infection depends on the age of the individual at the time of infection, and is drawn from a Weibull distribution with shape parameter (see HIV_Adult_Survival_Shape_Parameter ) and scale parameter, \u03bb. The scale parameter is allowed to vary linearly with age as follows: \u03bb = HIV_Adult_Survival_Scale_Parameter_Intercept + HIV_Adult_Survival_Scale_ Parameter_Slope * Age (in years). { \"HIV_Adult_Survival_Scale_Parameter_Intercept\" : 21.182 , \"HIV_Adult_Survival_Scale_Parameter_Slope\" : -0.2717 , \"HIV_Adult_Survival_Shape_Parameter\" : 2.0 , \"HIV_Age_Max_for_Adult_Age_Dependent_Survival\" : 50.0 } HIV_Adult_Survival_Scale_Parameter_Slope HIV float -1000 1000 -0.2717 This parameter determines the slope of the scale parameter, \u03bb, for the Weibull distribution used to determine HIV survival time. Survival time with untreated HIV infection depends on the age of the individual at the time of infection, and is drawn from a Weibull distribution with shape parameter (see HIV_Adult_Survival_Shape_Parameter ) and scale parameter, \u03bb. The scale parameter is allowed to vary linearly with age as follows: \u03bb = HIV_Adult_Survival_Scale_Parameter_Intercept + HIV_Adult_Survival_Scale_ Parameter_Slope * Age (in years). Because survival time with HIV becomes shorter with increasing age, HIV_Adult_Survival_Scale_ Parameter_Slope should be set to a negative number. { \"HIV_Adult_Survival_Scale_Parameter_Intercept\" : 21.182 , \"HIV_Adult_Survival_Scale_Parameter_Slope\" : -0.2717 , \"HIV_Adult_Survival_Shape_Parameter\" : 2.0 , \"HIV_Age_Max_for_Adult_Age_Dependent_Survival\" : 50.0 } HIV_Adult_Survival_Shape_Parameter HIV float 0.001 1000 2 This parameter determines the shape of the Weibull distribution used to determine age-dependent survival time for individuals infected with HIV. { \"HIV_Adult_Survival_Scale_Parameter_Intercept\" : 21.182 , \"HIV_Adult_Survival_Scale_Parameter_Slope\" : -0.2717 , \"HIV_Adult_Survival_Shape_Parameter\" : 2.0 , \"HIV_Age_Max_for_Adult_Age_Dependent_Survival\" : 50.0 } HIV_Age_Max_for_Adult_Age_Dependent_Survival HIV float 0 75 70 Survival time with untreated HIV infection depends on the age of the individual at the time of infection, and is drawn from a Weibull distribution with shape parameter and scale parameter, \u03bb. See HIV_Adult_Survival_Scale_Parameter_Intercept , HIV_Adult_Survival_Scale_ Parameter_Slope , and HIV_Adult_Survival_Shape_Parameter . Although the scale parameter for survival time declines with age, it cannot become negative. To avoid negative survival times at older ages, this parameter, HIV_Age_Max_for_Adult_Age_Dependent_Survival , determines the age beyond which HIV survival is no longer affected by further aging. { \"HIV_Adult_Survival_Scale_Parameter_Intercept\" : 21.182 , \"HIV_Adult_Survival_Scale_Parameter_Slope\" : -0.2717 , \"HIV_Adult_Survival_Shape_Parameter\" : 2.0 , \"HIV_Age_Max_for_Adult_Age_Dependent_Survival\" : 50.0 } HIV_Age_Max_for_Child_Survival_Function HIV float 0 75 15 The maximum age at which an individual\u2019s survival will be fit to the child survival function. If the value of this parameter falls between zero and the age of sexual debut, model results are not sensitive to this parameter as there is no mechanism for children to become infected between infancy and sexual debut. { \"HIV_Adult_Survival_Scale_Parameter_Intercept\" : 21.182 , \"HIV_Adult_Survival_Scale_Parameter_Slope\" : -0.2717 , \"HIV_Adult_Survival_Shape_Parameter\" : 2.0 , \"HIV_Age_Max_for_Adult_Age_Dependent_Survival\" : 50.0 , \"HIV_Age_Max_for_Child_Survival_Function\" : 15.0 } HIV_Child_Survival_Rapid_Progressor_Fraction HIV float 0 1 0.57 The proportion of HIV-infected children who are rapid HIV progressors. { \"HIV_Child_Survival_Rapid_Progressor_Fraction\" : 0.57 , \"HIV_Child_Survival_Rapid_Progressor_Rate\" : 1.52 } HIV_Child_Survival_Rapid_Progressor_Rate HIV float 0 1000 1.52 The exponential decay rate, in years, describing the distribution of HIV survival for children who are rapid progressors. { \"HIV_Child_Survival_Rapid_Progressor_Fraction\" : 0.57 , \"HIV_Child_Survival_Rapid_Progressor_Rate\" : 1.52 } HIV_Child_Survival_Slow_Progressor_Scale HIV float 0.001 1000 16 The Weibull scale parameter describing the distribution of HIV survival for children who are slower progressors. { \"HIV_Child_Survival_Slow_Progressor_Scale\" : 16.0 , \"HIV_Child_Survival_Slow_Progressor_Shape\" : 2.7 } HIV_Child_Survival_Slow_Progressor_Shape HIV float 0.001 1000 2.7 The Weibull shape parameter describing the distribution of HIV survival for children who are slower progressors. { \"HIV_Child_Survival_Slow_Progressor_Scale\" : 16.0 , \"HIV_Child_Survival_Slow_Progressor_Shape\" : 2.7 } Human_Feeding_Mortality Vector float 0 1 0.1 The fraction of mosquitoes, for all species, that die while feeding on a human. { \"Human_Feeding_Mortality\" : 0.15 } Immature_Duration Vector float 0 730 2 The number of days for larvae to develop into adult mosquitoes. The value is used to calculate mosquito development rate, which equals 1 / (value of this parameter). Development from immature to adult is not dependent on temperature. { \"Vector_Species_Params\" : { \"funestus\" : { \"Immature_Duration\" : 4 } } } Immune_Threshold_For_Downsampling Generic float 0 1 0 Threshold on acquisition immunity at which to apply immunity dependent downsampling. Individual_Sampling_Type must set to ADAPTED_SAMPLING_BY_IMMUNE_STATE. { \"Individual_Sampling_Type\" : \"ADAPTED_SAMPLING_BY_IMMUNE_STATE\" , \"Immune_Threshold_For_Downsampling\" : 0.5 } Immunity_Acquisition_Factor Generic float 0 1000 0 The multiplicative reduction in the probability of reacquiring disease. Only used when Enable_Immunity and Enable_Immune_Decay are set to 1. { \"Enable_Immunity\" : 1 , \"Enable_Immune_Decay\" : 1 , \"Immunity_Acquisition_Factor\" : 0.9 } Immunity_Initialization_Distribution_Type Generic enum NA NA DISTRIBUTION_OFF The method for initializing the immunity distribution in the simulated population. Enable_Immunity must be set to true (1). Possible values are: DISTRIBUTION_OFF All individuals default to no immunity. DISTRIBUTION_SIMPLE Individual immunities are drawn from a distribution whose functional form and parameters are specified in the demographics file in IndividualAttributes using ImmunityDistributionFlag , ImmunityDistribution1 , and ImmunityDistribution2 . DISTRIBUTION_COMPLEX Individual immnunities are drawn from an age-dependent piecewise l for each specific antibody in the demographics file. { \"Immunity_Initialization_Distribution_Type\" : \"DISTRIBUTION_COMPLEX\" } Immunity_Mortality_Factor Generic float 0 1000 0 The multiplicative reduction in the probability of dying from infection after getting re-infected. Enable_Immunity and Enable_Immune_Decay must be set to true (1). { \"Enable_Immunity\" : 1 , \"Enable_Immune_Decay\" : 1 , \"Immunity_Mortality_Factor\" : 0.5 } Immunity_Transmission_Factor Generic float 0 1000 0 The multiplicative reduction in the probability of transmitting infection after getting re-infected. Only used when Enable_Immunity and Enable_Immune_Decay are set to 1. { \"Enable_Immunity\" : 1 , \"Enable_Immunity_Decay\" : 1 , \"Immunity_Transmission_Factor\" : 0.9 } Incubation_Period_Distribution Generic enum NA NA NOT_INITIALIZED The distribution for the duration of the incubation period. Possible values are: NOT_INITIALIZED No distribution set. FIXED_DURATION Base_Incubation_Period is the constant duration incubation period. UNIFORM_DURATION Incubation_Period_Min and Incubation_Period_Max define the ranges of a uniform random draw for the duration of the incubation period. GAUSSIAN_DURATION Incubation_Period_Mean and Incubation_Period_Std_Dev define the mean and standard deviation of the Gaussian from which the incubation period is drawn. Negative values are truncated at zero. EXPONENTIAL_DURATION Base_Incubation_Period is the mean of the exponential random draw. POISSON_DURATION Incubation_Period_Mean is the mean of the random Poisson draw. LOG_NORMAL_DURATION Incubation_Period_Log_Mean is the mean of log normal and Incubation_Period_Log_Width is the log width of log normal. BIMODAL_DURATION The distribution is bimodal, the duration is a fraction of Base_Incubation_Period for a specified period of time and equal to Base_Incubation_Period otherwise. PIECEWISE_CONSTANT The distribution is specified with a list of years and a matching list of values. The duration at a given year is that specified for the nearest previous year. PIECEWISE_LINEAR The distribution is specified with a list of years and matching list of values. The duration at a given year is a linear interpolation of the specified values. WEIBULL_DURATION The duration is a Weibull distribution with a given scale and shape. DUAL_TIMESCALE_DURATION The duration is two exponential distributions with given means. { \"Base_Incubation_Period\" : 5 , \"Incubation_Period_Distribution\" : \"EXPONENTIAL_DURATION\" } Incubation_Period_Log_Mean Generic float 0 3.40E+38 6 The mean of log normal for the incubation period distribution. Incubation_Period_Distribution must be set to LOG_NORMAL_DURATION. { \"Incubation_Period_Distribution\" : \"LOG_NORMAL_DURATION\" , \"Incubation_Period_Log_Mean\" : 5.758 , \"Incubation_Period_Log_Width\" : 0.27 } Incubation_Period_Log_Width Generic float 0 3.40E+38 1 The log width of log normal for the incubation period distribution. Incubation_Period_Distribution must be set to LOG_NORMAL_DURATION. { \"Incubation_Period_Distribution\" : \"LOG_NORMAL_DURATION\" , \"Incubation_Period_Log_Mean\" : 5.758 , \"Incubation_Period_Log_Width\" : 0.27 } Incubation_Period_Max Generic float 0.6 3.40E+38 0 The maximum length of the incubation period. Incubation_Period_Distribution must be set to UNIFORM_DURATION. { \"Incubation_Period_Distribution\" : \"UNIFORM_DURATION\" , \"Incubation_Period_Min\" : 2 , \"Incubation_Period_Max\" : 6 } Incubation_Period_Mean Generic float 0 3.40E+38 6 The mean of the incubation period. Incubation_Period_Distribution must be set to either GAUSSIAN_DURATION or POISSON_DURATION. { \"Incubation_Period_Distribution\" : \"GAUSSIAN_DURATION\" , \"Incubation_Period_Mean\" : 7 , \"Infectious_Period_Std_Dev\" : 2 } Incubation_Period_Min Generic float 0 3.40E+38 0 The minimum length of the incubation period. Incubation_Period_Distribution must be set to UNIFORM_DURATION. { \"Incubation_Period_Distribution\" : \"UNIFORM_DURATION\" , \"Incubation_Period_Min\" : 2 , \"Incubation_Period_Max\" : 6 } Incubation_Period_Std_Dev Generic float 0 3.40E+38 1 The standard deviation incubation period. Incubation_Period_Distribution must be set to GAUSSIAN_DURATION. { \"Incubation_Period_Distribution\" : \"GAUSSIAN_DURATION\" , \"Incubation_Period_Mean\" : 7 , \"Infectious_Period_Std_Dev\" : 2 } Individual_Sampling_Type Generic enum NA NA TRACK_ALL The type of individual human sampling. Possible values are: TRACK_ALL Each person in the population is tracked as a single Individual object with a sampling weight of 1. FIXED_SAMPLING Members of the population are added to the simulation with a probability of Base_Individual_Sample_Rate and sampling weight of 1 / Base_Individual_Sample_Rate . This allows a smaller set of Individual objects to represent the overall population. ADAPTED_SAMPLING_BY_POPULATION_SIZE For each node, a maximum number of Individual objects is tracked (set in Max_Node_Population_Samples ). If the population is smaller than this number, each person is tracked with a sampling rate of 1; if the population is larger, members of the population are added to the simulation with a probability of Max_Node_Population_Samples / population and sampling weight of population / Max_Node_Population_Samples . This setting is useful for simulations over large geographic areas with large heterogeneities in population density. ADAPTED_SAMPLING_BY_AGE_GROUP The population is binned by age and each age bin is sampled at a different rate as defined by Sample_Rate parameters. The setting is useful for diseases in which only infants or children are relevant by allowing full sampling of these age groups while using fewer objects to represent the rest of the population. ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE Each node has a maximum number of Individual objects as described in ADAPTED_SAMPLING_BY_POPULATION_SIZE, but when the population grows beyond that threshold, sampling is not done at a fixed rate, but varies by age as described in ADAPTED_SAMPLING_BY_AGE_GROUP. ADAPTED_SAMPLING_BY_IMMUNE_STATE Individuals who have acquired immunity are sampled at a lower rate as defined by Relative_Sample_Rate_Immune and Immune_Threshold_For_Downsampling parameters. The following example shows how to sampling immune individuals at a lower rate. { \"Individual_Sampling_Type\" : \"ADAPTED_SAMPLING_BY_IMMUNE_STATE\" , \"Immune_Threshold_For_Downsampling\" : 0.5 , \"Relative_Sample_Rate_Immune\" : 0.1 } The following example shows how to sampling older individuals at a lower rate. { \"Individual_Sampling_Type\" : \"ADAPTED_SAMPLING_BY_AGE_GROUP\" , \"Sample_Rate_0_18mo\" : 1 , \"Sample_Rate_10_14\" : 0.5 , \"Sample_Rate_15_19\" : 0.3 , \"Sample_Rate_18mo_4yr\" : 1 , \"Sample_Rate_20_Plus\" : 0.2 , \"Sample_Rate_5_9\" : 1 , \"Sample_Rate_Birth\" : 1 } Indoor_Feeding_Fraction Vector float 0 1 1 The fraction (dimensionless) of feeds in which mosquitoes feed on humans indoors; the fraction of feeds on humans that occur outdoors equals 1 \u2013 (value of this parameter). { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Cycle_Arrhenius_1\" : 0 , \"Cycle_Arrhenius_2\" : 0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 }, \"Nighttime_Feeding_Fraction\" : 1 , \"Transmission_Rate\" : 0.5 } } } Infected_Arrhenius_1 Vector float 0 1.00E+15 1.17E+11 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the daily rate of fractional progression of infected mosquitoes to an infectious state. The duration of sporogony is a decreasing function of temperature. The variable a1 is a temperature-independent scale factor on the progression rate to infectiousness. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Cycle_Arrhenius_1\" : 0 , \"Cycle_Arrhenius_2\" : 0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 }, \"Nighttime_Feeding_Fraction\" : 1 , \"Transmission_Rate\" : 0.5 } } } Infected_Arrhenius_2 Vector float 0 1.00E+15 8340 The Arrhenius equation, , with T in degrees Kelvin, parameterizes the daily rate of fractional progression of infected mosquitoes to an infectious state. The duration of sporogony is a decreasing function of temperature. The variable a2 is a temperature-dependent scale factor on the progression rate to infectiousness. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Cycle_Arrhenius_1\" : 0 , \"Cycle_Arrhenius_2\" : 0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 }, \"Nighttime_Feeding_Fraction\" : 1 , \"Transmission_Rate\" : 0.5 } } } Infected_Egg_Batch_Factor Vector float 0 10 0.8 The dimensionless factor used to modify mosquito egg batch size in order to account for reduced fertility effects arising due to infection (e.g. when females undergo sporogony). { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Cycle_Arrhenius_1\" : 0 , \"Cycle_Arrhenius_2\" : 0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 }, \"Nighttime_Feeding_Fraction\" : 1 , \"Transmission_Rate\" : 0.5 } } } Infection_Updates_Per_Timestep Generic integer 0 144 1 The number of infection updates executed during each timestep; note that a timestep defaults to one day. { \"Infection_Updates_Per_Timestep\" : 1 } Infectious_Human_Feed_Mortality_Factor Vector float 0 1000 1.5 The (dimensionless) factor used to modify the death rate of mosquitoes when feeding on humans, to account for the higher mortality rate infected mosquitoes experience during human feeds versus uninfected mosquitoes. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Cycle_Arrhenius_1\" : 0 , \"Cycle_Arrhenius_2\" : 0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 }, \"Nighttime_Feeding_Fraction\" : 1 , \"Transmission_Rate\" : 0.5 } } } Infectious_Period_Distribution Generic enum NA NA NOT_INITIALIZED The distribution of the duration of the infectious period. Possible values are: FIXED_DURATION Base_Infectious_Period is the constant-duration infectious period. UNIFORM_DURATION Infectious_Period_Min and Infectious_Period_Max define the ranges of a uniform random draw for the duration of the infectious period. GAUSSIAN_DURATION Infectious_Period_Mean and Infectious_Period_Std_Dev define the mean and standard deviation of the Gaussian from which the infectious period is drawn. Negative values are truncated at zero. EXPONENTIAL_DURATION Base_Infectious_Period is the mean of the exponential random draw. POISSON_DURATION Infectious_Period_Mean is the mean of the random Poisson draw. NOT_INITIALIZED No distribution set. LOG_NORMAL_DURATION The duration is a log normal distribution defined by a specified mean and log width. BIMODAL_DURATION The distribution is bimodal, the duration is a fraction of Base_Incubation_Period for a specified period of time and equal to Base_Incubation_Period otherwise. PIECEWISE_CONSTANT The distribution is specified with a list of years and a matching list of values. The duration at a given year is that specified for the nearest previous year. PIECEWISE_LINEAR The distribution is specified with a list of years and matching list of values. The duration at a given year is a linear interpolation of the specified values. WEIBULL_DURATION The duration is a Weibull distribution with a given scale and shape. DUAL_TIMESCALE_DURATION The duration is two exponential distributions with given means. { \"Infectious_Period_Distribution\" : \"EXPONENTIAL_DURATION\" } Infectious_Period_Max Generic float 0.6 3.40E+38 0 The maximum length of the infectious period; used when Infectious_Period_Distribution is set to UNIFORM_DURATION. { \"Infectious_Period_Distribution\" : \"UNIFORM_DURATION\" , \"Infectious_Period_Max\" : 15 , \"Infectious_Period_Min\" : 5 } Infectious_Period_Mean Generic float 0 3.40E+38 6 The mean of the infectious period; used when Infectious_Period_Distribution is set to either GAUSSIAN_DURATION or POISSON_DURATION. { \"Infectious_Period_Distribution\" : \"GAUSSIAN_DURATION\" , \"Infectious_Period_Mean\" : 12 , \"Infectious_Period_Std_Dev\" : 10 } Infectious_Period_Min Generic float 0 3.40E+38 0 The minimum length of the infectious period; used when Infectious_Period_Distribution is set to UNIFORM_DURATION. { \"Infectious_Period_Distribution\" : \"UNIFORM_DURATION\" , \"Infectious_Period_Max\" : 15 , \"Infectious_Period_Min\" : 5 } Infectious_Period_Std_Dev Generic float 0 3.40E+38 1 The standard deviation of the infectious period; used when Infectious_Period_Distribution is set to GAUSSIAN_DURATION. { \"Infectious_Period_Distribution\" : \"GAUSSIAN_DURATION\" , \"Infectious_Period_Mean\" : 12 , \"Infectious_Period_Std_Dev\" : 10 } Infectivity_Boxcar_Forcing_Amplitude Generic float 0 3.40E+38 0 The fractional increase in R0 during the high-infectivity season when Infectivity_Scale_Type is equal to ANNUAL_BOXCAR_FUNCTION. { \"Infectivity_Boxcar_Forcing_Amplitude\" : 0.25 , \"Infectivity_Boxcar_Forcing_End_Time\" : 270 , \"Infectivity_Boxcar_Forcing_Start_Time\" : 90 , \"Infectivity_Scale_Type\" : \"ANNUAL_BOXCAR_FUNCTION\" } Infectivity_Boxcar_Forcing_End_Time Generic float 0 365 0 The end of the high-infectivity season when Infectivity_Scale_Type is equal to ANNUAL_BOXCAR_FUNCTION. { \"Infectivity_Boxcar_Forcing_Amplitude\" : 0.25 , \"Infectivity_Boxcar_Forcing_End_Time\" : 270 , \"Infectivity_Boxcar_Forcing_Start_Time\" : 90 , \"Infectivity_Scale_Type\" : \"ANNUAL_BOXCAR_FUNCTION\" } Infectivity_Boxcar_Forcing_Start_Time Generic float 0 365 0 The beginning of the high-infectivity season, in days, when Infectivity_Scale_Type is equal to ANNUAL_BOXCAR_FUNCTION. { \"Infectivity_Boxcar_Forcing_Amplitude\" : 0.25 , \"Infectivity_Boxcar_Forcing_End_Time\" : 270 , \"Infectivity_Boxcar_Forcing_Start_Time\" : 90 , \"Infectivity_Scale_Type\" : \"ANNUAL_BOXCAR_FUNCTION\" } Infectivity_Exponential_Baseline Generic float 0 1 0 The scale factor applied to Base_Infectivity at the beginning of a simulation, before the infectivity begins to grow exponentially. Infectivity_Scale_Type must be set to EXPONENTIAL_FUNCTION_OF_TIME. { \"Infectivity_Exponential_Baseline\" : 0.1 , \"Infectivity_Exponential_Delay\" : 90 , \"Infectivity_Exponential_Rate\" : 45 , \"Infectivity_Scale_Type\" : \"EXPONENTIAL_FUNCTION_OF_TIME\" } Infectivity_Exponential_Delay Generic float 0 3.40E+38 0 The number of days before infectivity begins to ramp up exponentially. Infectivity_Scale_Type must be set to EXPONENTIAL_FUNCTION_OF_TIME. { \"Infectivity_Exponential_Baseline\" : 0.1 , \"Infectivity_Exponential_Delay\" : 90 , \"Infectivity_Exponential_Rate\" : 45 , \"Infectivity_Scale_Type\" : \"EXPONENTIAL_FUNCTION_OF_TIME\" } Infectivity_Exponential_Rate Generic float 0 3.40E+38 0 The daily rate of exponential growth to approach to full infectivity after the delay set by Infectivity_Exponential_Delay has passed. Infectivity_Scale_Type must be set to EXPONENTIAL_FUNCTION_OF_TIME. { \"Infectivity_Exponential_Rate\" : 45 } Infectivity_Scale_Type Generic enum NA NA CONSTANT_INFECTIVITY A scale factor that allows infectivity to be altered by time or season. Possible values are: CONSTANT_INFECTIVITY No infectivity correction is applied. FUNCTION_OF_TIME_AND_LATITUDE Infectivity is corrected for approximate seasonal forcing. The use of a seasonal infectivity correction is a proxy for the effects of varying climate. From October through March, infectivity increases in the Northern Hemisphere and decreases in the Southern Hemisphere. From April through September, the trend reverses: regions closer to the equator have reduced forcing compared to temperate regions. FUNCTION_OF_CLIMATE Allows infectivity to be modulated by weather directly, for example, relative humidity in airborne simulations or rainfall in waterborne simulations. There is no default climate dependence enabled for generic simulations. EXPONENTIAL_FUNCTION_OF_TIME To facilitate certain burn-in scenarios, infectivity ramps up from zero at the beginning of the simulation according to the functional form, 1-exp(-rate*time), where the rate is specified by the parameter Infectivity_Scaling_Rate . SINUSOIDAL_FUNCTION_OF_TIME Allows infectivity to be time-dependent, following a sinusoidal shape. ANNUAL_BOXCAR_FUNCTION Allows infectivity to follow a boxcar function, such that it will be equal to zero for an entire time period (e.g. year) except for a single interval in which it is equal to a constant. { \"Infectivity_Scale_Type\" : \"FUNCTION_OF_CLIMATE\" } Infectivity_Sinusoidal_Forcing_Amplitude Generic float 0 1 0 Sets the amplitude of sinusoidal variations in Base_Infectivity . Only used when Infectivity_Scale_Type is set to SINUSOIDAL_FUNCTION_OF_TIME. { \"Infectivity_Scale_Type\" : \"SINUSOIDAL_FUNCTION_OF_TIME\" , \"Infectivity_Sinusoidal_Forcing_Amplitude\" : 0.1 , \"Infectivity_Sinusoidal_Forcing_Phase\" : 0 } Infectivity_Sinusoidal_Forcing_Phase Generic float 0 365 0 Sets the phase of sinusoidal variations in Base_Infectivity . Only used when Infectivity_Scale_Type is set to SINUSOIDAL_FUNCTION_OF_TIME. { \"Infectivity_Scale_Type\" : \"SINUSOIDAL_FUNCTION_OF_TIME\" , \"Infectivity_Sinusoidal_Forcing_Amplitude\" : 0.1 , \"Infectivity_Sinusoidal_Forcing_Phase\" : 0 } Innate_Immune_Variation_Type Malaria enum NA NA NONE The type of innate immune variation in an individual. Possible values are: NONE Default value; no variation in innate immunity between individuals. PYROGENIC_THRESHOLD There is a threshold for parasite density which triggers cytokine production (and results in fever). PYROGENIC_THRESHOLD_VS_AGE Cytokine production (and fever) is age-dependent. CYTOKINE_KILLING Individuals differ in the effectiveness of cytokines in killing infected red-blood cells. { \"Innate_Immune_Variation_Type\" : \"PYROGENIC_THRESHOLD\" } Koppen_Filename Generic string NA NA UNINITIALIZED STRING The path to the input file used to specify Koppen climate classifications; only used when Climate_Model is set to CLIMATE_KOPPEN. The file must be in .dat format. { \"Koppen_Filename\" : \"Mad_2_5arcminute_koppen.dat\" } Land_Temperature_Filename Generic string NA NA land_temp.json The path of the input file defining temperature data measured at land surface; used only when Climate_Model is set to CLIMATE_BY_DATA. The file must be in .bin format. { \"Land_Temperature_Filename\" : \"Namawala_single_node_land_temperature_daily.bin\" } Land_Temperature_Offset Generic float -20 20 0 The linear shift of land surface temperature in degrees Celsius; only used when Climate_Model is set to CLIMATE_BY_DATA. { \"Land_Temperature_Offset\" : 0 } Land_Temperature_Variance Generic float 0 7 2 The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily land temperature values when Climate_Model is configured to CLIMATE_CONSTANT or CLIMATE_BY_DATA; only used if the Enable_Climate_Stochasticity is set to true (1). { \"Land_Temperature_Variance\" : 1.5 } Larval_Density_Dependence Vector enum NA NA UNIFORM_WHEN_OVERPOPULATION The functional form of mortality and growth delay for mosquito larvae based on population density. Possible values are: UNIFORM_WHEN_OVERPOPULATION Mortality is uniformly applied to all larvae when the population exceeds the specified carrying capacity for that habitat. GRADUAL_INSTAR_SPECIFIC Mortality and delayed growth are instar-specific, where the younger larvae are more susceptible to predation and competition from older larvae. LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY Mortality is based only on larval age. DENSITY_DELAYED_GROWTH_NOT_MORTALITY There is no mortality, only delayed growth in larvae. NO_DENSITY_DEPENDENCE There is no additional larval density-dependent mortality factor. { \"Larval_Density_Dependence\" : \"GRADUAL_INSTAR_SPECIFIC\" } Larval_Density_Mortality_Offset Vector float 0.0001 1000 0.1 An offset factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models. { \"Larval_Density_Mortality_Offset\" : 0.001 } Larval_Density_Mortality_Scalar Vector float 0.01 1000 10 A scale factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models. { \"Larval_Density_Mortality_Scalar\" : 1.0 } Larval_Habitat_Types Vector JSON object NA NA NA A measure of the habitat type and scale factors used to estimate larval population. This parameter is a dictionary that specifies habitat type with a simple numeric scale factor or, for LINEAR_SPLINE, with a more detailed configuration for scaling. The numeric scaling value represents larval density with the number of larvae in a 1x1-degree area. The factor multiplicatively scales the resulting weather or population-dependent functional form. Possible habitat values are: CONSTANT Larval carrying capacity is constant throughout the year and does not depend on weather. However, mosquito abundance will exhibit a seasonal signal due to the effects of temperature on aquatic developmental rates. TEMPORARY_RAINFALL This habitat type corresponds to temporary puddles which are replenished by rainfall and drained through evaporation and infiltration. Habitat availability is proportional to temperature and humidity, and decays over time as configured by the parameter Temporary_Habitat_Decay_Factor . WATER_VEGATATION This habitat type corresponds to semi-permanent habitats, such as developing vegetation on the edges of small water sources such as swamps or rice cultivation settings. Development of habitat lags behind rainfall, and typically peaks near the end of the rainy season. Habitat decays daily, as specified by the parameter Semipermanent_Habitat_Decay_Rate . HUMAN_POPULATION This habitat type scales with correlates of human development, such as available water in pots in urban areas. It is configured by multiplying the number of people in a node\u2019s population times the capacity value set for this parameter, and is not climate-dependent. BRACKISH_SWAMP This habitat type deals with the dynamics of how rain fills brackish swamps, how this habitat availability decays, and includes a rainfall-driven mortality threshold. Habitat decay rate is specified by the parameter Semipermanent_Habitat_Decay_Rate . LINEAR_SPLINE The LINEAR_SPLINE configuration specifies the day of year, larval value, and larval capacity scaling number. The model linearly interpolates the values to estimate the habitat availability for each vector species without requiring climatological data. The values set here can be scaled per node using LarvalHabitatMultiplier in the demographics file (see NodeAttributes) or per intervention using Larval_Habitat_Multiplier in the campaign file (see ScaleLarvalHabitat). The following example shows how to specify larval habitat using LINEAR_SPLINE. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Larval_Habitat_Types\" : { \"LINEAR_SPLINE\" : { \"Capacity_Distribution_Per_Year\" : { \"Times\" : [ 0.0 , 60.833 , 121.667 , 182.5 , 243.333 , 204.167 ], \"Values\" : [ 0.0 , 0.0 , 0.2 , 1.0 , 0.5 , 0.0 ] }, \"Max_Larval_Capacity\" : 10000000000.0 } } } } } The following example shows how to specify the larval habitat using climatological habitat types (you may use one or more for each vector species). { \"arabiensis\" : { \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 , \"WATER_VEGETATION\" : 6000000000 , \"BRACKISH_SWAMP\" : 30000000 } } } Larval_Rainfall_Mortality_Threshold Vector float 0.01 1000 100 The threshold value on daily rainfall in millimeters, above which larval mortality is applied when Vector_Larval_Rainfall_Mortality is set to either SIGMOID or SIGMOID_HABITAT_SHIFTING. { \"Larval_Rainfall_Mortality_Threshold\" : 30.0 } Listed_Events Generic array of strings NA NA [] The list of valid, user-defined events that will be included in the campaign. Any event used in the campaign must either be one of the built-in events or in this list. { \"Listed_Events\" : [ \"Vaccinated\" , \"VaccineExpired\" ] } Load_Balance_Filename Generic string NA NA UNINITIALIZED STRING The path to the input file used when a static load balancing scheme is selected. The file must be in .json format. { \"Load_Balance_Filename\" : \"GitHub_426_LoadBalance.json\" } Local_Migration_Filename Generic string NA NA \u00a0 The path of the input file which defines patterns of migration to adjacent nodes by foot travel. The file must be in .bin format. { \"Local_Migration_Filename\" : \"Local_Migration.bin\" } Local_Migration_Roundtrip_Duration Generic float 0 10000 1 The average time spent (in days) at the destination node during a round-trip migration by foot travel. Only used if Enable_Local_Migration is set to true (1). { \"Local_Migration_Roundtrip_Duration\" : 1.0 } Local_Migration_Roundtrip_Probability Generic float 0 1 0.95 The likelihood that an individual who walks into a neighboring cell will return to the cell of origin during the next migration. Only used when Enable_Local_Migration is set to true (1). { \"Local_Migration_Roundtrip_Probability\" : 1.0 } Malaria_Drug_Params Malaria JSON object NA NA NA This JSON structure contains the names of anti-malarial drugs and the parameters that define them. { \"Malaria_Drug_Params\" : { \"Artemether_Lumefantrine\" : { \"Bodyweight_Exponent\" : 0 , \"Drug_Cmax\" : 1000 , \"Drug_Decay_T1\" : 1 , \"Drug_Decay_T2\" : 1 , \"Drug_Dose_Interval\" : 1 , \"Drug_Fulltreatment_Doses\" : 3 , \"Drug_Gametocyte02_Killrate\" : 2.3 , \"Drug_Gametocyte34_Killrate\" : 2.3 , \"Drug_GametocyteM_Killrate\" : 0 , \"Drug_Hepatocyte_Killrate\" : 0 , \"Drug_PKPD_C50\" : 100 , \"Drug_Vd\" : 10 , \"Max_Drug_IRBC_Kill\" : 4.6 }, \"Artemisinin\" : { \"Bodyweight_Exponent\" : 0 , \"Drug_Cmax\" : 1000 , \"Drug_Decay_T1\" : 1 , \"Drug_Decay_T2\" : 1 , \"Drug_Dose_Interval\" : 1 , \"Drug_Fulltreatment_Doses\" : 3 , \"Drug_Gametocyte02_Killrate\" : 2.3 , \"Drug_Gametocyte34_Killrate\" : 2.3 , \"Drug_GametocyteM_Killrate\" : 0 , \"Drug_Hepatocyte_Killrate\" : 0 , \"Drug_PKPD_C50\" : 100 , \"Drug_Vd\" : 10 , \"Max_Drug_IRBC_Kill\" : 4.61 } } } Malaria_Model Malaria enum NA NA MALARIA_MECHANISTIC_MODEL The type of malaria model used in a simulation. Possible values are: MALARIA_FIXED_DURATION MALARIA_EXPONENTIAL_DURATION MALARIA_REDUCEDSTATE_MODEL MALARIA_MECHANISTIC_MODEL Note All malaria simulations should use MALARIA_MECHANISTIC_MODEL. The others are placeholder values. { \"Malaria_Model\" : \"MALARIA_MECHANISTIC_MODEL\" } Malaria_Strain_Model Malaria enum NA NA FALCIPARUM_NONRANDOM_STRAIN The generator that is used to construct the antigenic repertoire of a malaria infection. To create parasite diversity, various antigenic strains are created by conducting draws for merozoite surface protein (MSP) variants, Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1) variants, and minor surface epitopes, out of available populations. Each possible value for this parameter utilizes different settings for the available population draws. Possible values are: FALCIPARUM_NONRANDOM_STRAIN The strain created uses the exact same MSP variant, non-specific PfEMP1 variant, and ordered repertoire of PfEMP1 major epitopes. Minor surface protein epitopes are randomly drawn from a set of 5. FALCIPARUM_RANDOM50_STRAIN Strain creation extends the logic used for FALCIPARUM_NONRANDOM_STRAIN, but increases variation by randomly drawing PfEMP1 variants from a population of 50, and allowing the variants to have a random switching order. Random draws are done with replacement. FALCIPARUM_RANDOM_STRAIN The MSP variant is drawn from the population set by the parameter Falciparum_MSP_Variants , a repertoire of 50 PfEMP1 variants is drawn from the population set by the parameter Falciparum_PfEMP1_Variants , and the set of five minor surface protein epitope variants is drawn from the population set by the parameter Falciparum_Nonspecific_Types . FALCIPARUM_STRAIN_GENERATOR This is an exploratory effort created to provide a pseudo-random generation of strains similar to FALCIPARUM_RANDOM_STRAIN, but with variant indices deterministically assigned according to the strain ID. { \"Malaria_Strain_Model\" : \"FALCIPARUM_STRAIN_GENERATOR\" } Male_To_Female_Relative_Infectivity_Ages STI array of floats NA NA 0 The vector of ages governing the susceptibility of females relative to males, by age. Used with Male_To_Female_Relative_Infectivity_Multipliers . { \"Male_To_Female_Relative_Infectivity_Ages\" : [ 15 , 25 , 35 ] } Male_To_Female_Relative_Infectivity_Multipliers STI array of floats NA NA 1 An array of scale factors governing the susceptibility of females relative to males, by age. Used with Male_To_Female_Relative_Infectivity_Ages . Scaling is linearly interpolated between ages. The first value is used for individuals younger than the first age in Male_To_Female_Relative_Infectivity_Ages and the last value is used for individuals older than the last age. { \"Male_To_Female_Relative_Infectivity_Multipliers\" : [ 5 , 1 , 0.5 ] } Maternal_Antibodies_Type Malaria enum NA NA OFF The type of maternal antibody protection. Possible values are: OFF No maternal antibody protection provided. SIMPLE_WANING The protection from maternal antibodies is a function of the configured maximum fraction protection ( Maternal_Antibody_Protection ) and the age of the child (through Maternal_Antibody_Decay_Rate ), but also of the transmission-intensity in the simulation. In particular, the protection is modified by a factor corresponding to the fraction of PfEMP1 antigenic variants to which possible mothers (14 to 45-year-old females) have been exposed in their lifetimes. CONSTANT_INITIAL_IMMUNITY Protection is independent of acquired immunity in possible mothers. { \"Maternal_Antibodies_Type\" : \"CONSTANT_INITIAL_IMMUNITY\" } Maternal_Antibody_Decay_Rate Malaria float 0 3.40E+38 0.01 The decay rate per day in protection from maternal antibodies. Maternal_Antibodies_Type must be set to SIMPLE_WANING or CONSTANT_INITIAL_IMMUNITY. { \"Maternal_Antibodies_Type\" : \"SIMPLE_WANING\" , \"Maternal_Antibody_Decay_Rate\" : 0.01 , \"Maternal_Antibody_Protection\" : 0.1327 } Maternal_Antibody_Protection Malaria float 0 1 0.1 The strength of protection from maternal antibodies as a multiple of full antibody killing effect. Maternal_Antibodies_Type must be set to SIMPLE_WANING or CONSTANT_INITIAL_IMMUNITY. { \"Maternal_Antibodies_Type\" : \"SIMPLE_WANING\" , \"Maternal_Antibody_Decay_Rate\" : 0.01 , \"Maternal_Antibody_Protection\" : 0.1327 } Maternal_Transmission_ART_Multiplier HIV float 0 1 0.1 The maternal transmission multiplier for on-ART mothers. { \"Maternal_Transmission_ART_Multiplier\" : 0.03 } Maternal_Transmission_Probability Generic float 0 1 0 The probability of transmission of infection from mother to infant at birth. Enable_Maternal_Transmission must be set to 1. Note: For malaria and vector simulations, set this to 0. Instead, use the Maternal_Antibody_Protection , Maternal_Antibody_Decay_Rate , and Maternal_Antibodies_Type parameters. { \"Maternal_Transmission_Probability\" : 0.3 } Max_Drug_IRBC_Kill Malaria float 5 100000 5 The maximum log reduction in IRBCs per day due to treatment. { \"Malaria_Drug_Params\" : { \"Artemether_Lumefantrine\" : { \"Bodyweight_Exponent\" : 0 , \"Drug_Cmax\" : 1000 , \"Drug_Decay_T1\" : 1 , \"Drug_Decay_T2\" : 1 , \"Drug_Dose_Interval\" : 1 , \"Drug_Fulltreatment_Doses\" : 3 , \"Drug_Gametocyte02_Killrate\" : 2.3 , \"Drug_Gametocyte34_Killrate\" : 2.3 , \"Drug_GametocyteM_Killrate\" : 0 , \"Drug_Hepatocyte_Killrate\" : 0 , \"Drug_PKPD_C50\" : 100 , \"Drug_Vd\" : 10 , \"Max_Drug_IRBC_Kill\" : 4.61 } } } Max_Individual_Infections Generic integer 0 1000 1 The limit on the number of infections that an individual can have simultaneously. Enable_Superinfection must be set to 1. { \"Max_Individual_Infections\" : 5 } Max_MSP1_Antibody_Growthrate Malaria float 0 1 0.02 The maximum increase in MSP1 antibody capacity during each asexual cycle. The higher this value, the sooner early clearances are observed and the earlier the parasite density envelope decreases. { \"Max_MSP1_Antibody_Growthrate\" : 0.045 } Max_Node_Population_Samples Generic float 1 3.40E+38 30 The maximum number of individuals to track in a node. When the population exceeds this number, the sampling rate will drop according to the value set in Individual_Sampling_Type . { \"Individual_Sampling_Type\" : \"ADAPTED_SAMPLING_BY_POPULATION_SIZE\" , \"Max_Node_Population_Samples\" : 100000 } Mean_Egg_Hatch_Delay Vector float 0 120 0 The mean delay in egg hatch time, in days, from the time of oviposition. Set the delay distribution with Egg_Hatch_Delay_Distribution . { \"Egg_Hatch_Delay_Distribution\" : \"EXPONENTIAL_DURATION\" , \"Mean_Egg_Hatch_Delay\" : 2 } Mean_Sporozoites_Per_Bite Malaria float 0 1000 11 The mean number of sporozoites per infectious mosquito bite. { \"Mean_Sporozoites_Per_Bite\" : 11 } Memory_Usage_Halting_Threshold_Working_Set_MB Generic integer 0 1.00E+06 8000 The maximum size (in MB) of working set memory before the system throws an exception and halts. { \"Memory_Usage_Halting_Threshold_Working_Set_MB\" : 4000 } Memory_Usage_Warning_Threshold_Working_Set_MB Generic integer 0 1.00E+06 7000 The maximum size (in MB) of working set memory before memory usage statistics are written to the log regardless of log level. { \"Memory_Usage_Warning_Threshold_Working_Set_MB\" : 3500 } Merozoites_Per_Hepatocyte Malaria double NA NA 15000 The number of IRBCs caused by a single infected hepatocyte at the start of infection. { \"Merozoites_Per_Hepatocyte\" : 15000 } Merozoites_Per_Schizont Malaria double NA NA 16 The number of merozoites released by a single infected schizont after each asexual cycle. The number of resulting IRBC\u2019s depends on the RBC availability and merozoite-specific immunity. { \"Merozoites_Per_Schizont\" : 16 } Migration_Model Generic enum NA NA NO_MIGRATION Model to use for migration. Possible values are: NO_MIGRATION Migration into and out of nodes will not occur. FIXED_RATE_MIGRATION Migration into and out of nodes will occur at a fixed rate as defined in the migration files. At the beginning of the simulation or whenever an individual has just moved, they pick their next destination and the time and type of the migration. If an individual is on an outbound leg of their journey, they will query the node\u2019s MigrationInfo object and, through probability, pick a new destination; if the individual is inbound, they will travel back to their previous location. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Local_Migration_Filename\" : \"../inputs/local_migration.bin\" , \"Enable_Local_Migration\" : 1 } Migration_Pattern Generic enum NA NA RANDOM_WALK_DIFFUSION Describes the type of roundtrip used during migration. Migration_Model must be set to FIXED_RATE_MIGRATION. Possible values are: RANDOM_WALK_DIFFUSION Individuals retain no memory of where they came from; every move is to a new destination with no thought of returning home. SINGLE_ROUND_TRIPS There is a certain probability that an individual\u2019s move will be a roundtrip (determined Local_Migration_Roundtrip_Probability , Air_Migration_Roundtrip_Probability , etc.). Otherwise, the departure point is forgotten and the individual does not return to their original location. WAYPOINTS_HOME Individuals go on a multi-step journey along several waypoints and then retrace their steps back along their path once they have reached a maximum number of waypoints from their home. { \"Migration_Model\" : \"FIXED_RATE_MIGRATION\" , \"Migration_Pattern\" : \"SINGLE_ROUND_TRIPS\" } Minimum_Adult_Age_Years Generic float 0 3.40E+38 15 The age, in years, after which an individual is considered an adult. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP. { \"Minimum_Adult_Age_Years\" : 17 } Min_Adapted_Response Malaria float 0 1 0.02 The minimum level of antibody stimulation to a novel antigen. The value sets the low-range asymptote for antibody capacity growth, which is calculated from Antibody_Capacity_Growth_Rate and antigen density, in the presence of any nonzero antigen level. { \"Min_Adapted_Response\" : 0.01 } Min_Days_Between_Adding_Relationships STI float 0 365 60 The minimum number of days between adding two consecutive relationships as a means to control concurrency. The constraint does not apply if an individual breaks an existing relationship. { \"Min_Days_Between_Adding_Relationships\" : 1 } Min_Days_Between_Clinical_Incidents Malaria float 0 1000000 3 The number of days with fever below the low-threshold before a new incident can start, when fever again exceeds the high-threshold. { \"Min_Days_Between_Clinical_Incidents\" : 14 } Mortality_Blocking_Immunity_Decay_Rate Generic float 0 1000 0.001 The rate at which mortality-blocking immunity decays after the mortality-blocking immunity offset period. Enable_Immune_Decay must be set to 1. { \"Mortality_Blocking_Immunity_Decay_Rate\" : 0.1 } Mortality_Blocking_Immunity_Duration_Before_Decay Generic float 0 45000 0 The number of days after infection until mortality-blocking immunity begins to decay. Enable_Immunity and Enable_Immune_Decay must be set to 1. { \"Mortality_Blocking_Immunity_Duration_Before_Decay\" : 270 } Mortality_Time_Course Generic enum NA NA DAILY_MORTALITY The method used to calculate disease deaths. Possible values are: DAILY_MORTALITY Calculated at every time step. MORTALITY_AFTER_INFECTIOUS Calculated once at the end of the disease duration. { \"Mortality_Time_Course\" : \"MORTALITY_AFTER_INFECTIOUS\" } Mosquito_Weight Vector integer 1 10000 1 The value indicating how many mosquitoes are represented by a sample mosquito in the simulation. Vector_Sampling_Type must be set to SAMPLE_IND_VECTORS. { \"Mosquito_Weight\" : 10 } MSP1_Merozoite_Kill_Fraction Malaria double NA NA 0.5 The fraction of merozoites inhibited from invading new erythrocytes when MSP1-specific antibody level is 1. { \"MSP1_Merozoite_Kill_Fraction\" : 0.4364623975644405 } Newborn_Biting_Risk_Multiplier Vector float 0 1 0.2 The scale factor that defines the y-intercept of the linear portion of the biting risk curve when Age_Dependent_Biting_Risk_Type is set to LINEAR. { \"Newborn_Biting_Risk_Multiplier\" : 0.2 } New_Diagnostic_Sensitivity Malaria float 0.0001 100000 0.01 The number of microliters of blood tested to find single parasites in a new diagnostic (corresponds to inverse parasites/microliters sensitivity). { \"New_Diagnostic_Sensitivity\" : 0.025 } Nighttime_Feeding_Fraction Vector float 0 1 1 The fraction of feeds on humans (for a specific mosquito species) that occur during the nighttime. Thus the fraction of feeds on humans that occur during the day equals 1 - (value of this parameter). { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Cycle_Arrhenius_1\" : 0 , \"Cycle_Arrhenius_2\" : 0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 }, \"Nighttime_Feeding_Fraction\" : 1 , \"Transmission_Rate\" : 0.5 } } } Node_Grid_Size Generic float 0.00416 90 0.004167 The spatial resolution indicating the node grid size for a simulation in degrees. { \"Node_Grid_Size\" : 0.042 } Nonspecific_Antibody_Growth_Rate_Factor Malaria float 0 1000 0.5 The factor that adjusts Antibody_Capacity_Growth_Rate for less immunogenic surface proteins, called minor epitopes. { \"Nonspecific_Antibody_Growth_Rate_Factor\" : 0.5 } Nonspecific_Antigenicity_Factor Malaria double NA NA 0.2 The nonspecific antigenicity factor that adjusts antibody IRBC kill rate to account for IRBCs caused by antibody responses to antigenically weak surface proteins. { \"Nonspecific_Antigenicity_Factor\" : 0.39192559432597257 } Number_Basestrains Generic integer 1 10 1 The number of base strains in the simulation, such as antigenic variants. { \"Number_Basestrains\" : 1 } Number_Of_Asexual_Cycles_Without_Gametocytes Malaria integer 0 500 1 The number of asexual reproduction cycles that do not produce gametocytes. All later cycles will produce gametocytes according to Base_Gametocyte_Production_Rate . { \"Number_Of_Asexual_Cycles_Without_Gametocytes\" : 1 } Number_Substrains Generic integer 1 16777200 256 The number of disease substrains for each base strain, such as genetic variants. { \"Number_Substrains\" : 16777216 } Num_Cores Generic integer NA NA NA The number of cores used to run a simulation. { \"Num_Cores\" : 4 } Parasite_Mortality_Inverse_Width Malaria float 0.1 1000000 10 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ). This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for parasite density. { \"Parasite_Mortality_Inverse_Width\" : 100 } Parasite_Mortality_Threshold Malaria double NA NA 2000000 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the parasite-density mortality threshold for a simulation. { \"Parasite_Mortality_Threshold\" : 3000000 } Parasite_Severe_Inverse_Width Malaria float 0.1 1000000 10 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for parasite density. { \"Parasite_Severe_Inverse_Width\" : 48.66825295 } Parasite_Severe_Threshold Malaria float 0 3.40E+38 1000000 The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function: probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ) This parameter configures parasite density threshold level that results in severe disease. { \"Parasite_Severe_Threshold\" : 253871 } Parasite_Smear_Sensitivity Malaria float 0.0001 100 0.1 The number of microliters of blood tested to find single parasites in a traditional smear (corresponds to inverse parasites/microliters sensitivity). { \"Parasite_Smear_Sensitivity\" : 0.1 } Parasite_Switch_Type Malaria enum NA NA CONSTANT_SWITCH_RATE_2VARS The parasite switch type for erythrocyte surface antigens. Possible values are: CONSTANT_SWITCH_RATE_2VARS Introduction of new variants has a fixed probability which does not depend on the level of parasitemia. RATE_PER_PARASITE_7VARS Each IRBC has a fixed rate of switching to each of seven possible variants. RATE_PER_PARASITE_5VARS_DECAYING Each IRBC has a rate of switching to several possible variants, and the rate decreases with distance from the original variant. { \"Parasite_Switch_Type\" : \"RATE_PER_PARASITE_7VARS\" } PFA_Burnin_Duration_In_Days STI float 1 3.40E+38 365000 The number of days to continue tuning the pair formation rates. After this duration, the rates will remain at the last rate value calculated. { \"PFA_Burnin_Duration_In_Days\" : 5475 } PFA_Cum_Prob_Selection_Threshold STI float 0 1 0.2 This parameter serves to minimize the extent to which relationships with unlikely age gaps are formed. These unlikely relationships could be generated due to lack of diversity in the PFA. Within the algorithm, males pick from amongst the available females by age bin, weighted by the conditional of the joint_probability matrix given the male age bin. If the sum of the probabilities in the female age bins that are not empty is below this threshold, the male will wait till the next update. Setting this parameter to a value near 1 dramatically increases the delay that individuals will experience in seeking relationships. Setting this parameter to 0 disables the feature. { \"PFA_Cum_Prob_Selection_Threshold\" : 0.4 } PKPD_Model Malaria enum NA NA FIXED_DURATION_CONSTANT_EFFECT Determines which pharmacokinetic pharmacodynamic module to use. Possible values are: FIXED_DURATION_CONSTANT_EFFECT This is the simplified model of drug action. Each dose of the drug has a constant effect for a fixed duration of time, after which the effect is zero. CONCENTRATION_VERSUS_TIME This model of drug action uses continuous pharmacokinetics with two-compartment decay. Effects of each drug depend on the concentration at that point in time. { \"PKPD_Model\" : \"FIXED_DURATION_CONSTANT_EFFECT\" } Population_Density_C50 Generic float 0 3.40E+38 10 The population density at which R0 for a 2.5-arc minute square reaches half of its initial value. Population_Density_Infectivity_Correction must be set to SATURATING_FUNCTION_OF_DENSITY. { \"Population_Density_C50\" : 30 } Population_Density_Infectivity_Correction Generic enum NA NA CONSTANT_INFECTIVITY Correction to alter infectivity by population density set in the Population_Density_C50 parameter. Measured in people per square kilometer. Possible values are: CONSTANT_INFECTIVITY SATURATING_FUNCTION_OF_DENSITY Note Sparsely populated areas have a lower infectivity, while densely populated areas have a higher infectivity, which rises to saturate at the Base_Infectivity value. { \"Population_Density_Infectivity_Correction\" : \"SATURATING_FUNCTION_OF_DENSITY\" } Population_Scale_Type Generic enum NA NA USE_INPUT_FILE The method to use for scaling the initial population specified in the demographics input file. Possible values are: USE_INPUT_FILE Turns off population scaling and uses InitialPopulation in the demographics file. FIXED_SCALING Enables Base_Population_Scale_Factor . { \"Population_Scale_Type\" : \"FIXED_SCALING\" } Pyrogenic_Threshold Malaria float 0.1 20000 1000 The level of bloodstream infection, measured in IRBC per microliter, at which stimulation of the innate inflammatory immune response is half its maximum value. { \"Pyrogenic_Threshold\" : 15000 } Rainfall_Filename Generic string NA NA rainfall.json The path of the input file which defines rainfall data. Climate_Model must be set to CLIMATE_BY_DATA. The file must be in .bin format. { \"Rainfall_Filename\" : \"Namawala_single_node_rainfall_daily.bin\" } Rainfall_In_mm_To_Fill_Swamp Vector float 1 10000 1000 Millimeters of rain to fill larval habitat to capacity. This is only used for vector species with Larval_Habitat_Types set to BRACKISH_SWAMP. { \"Rainfall_In_mm_To_Fill_Swamp\" : 1000.0 } Rainfall_Scale_Factor Generic float 0.1 10 1 The scale factor used in multiplying rainfall value(s). Climate_Model must be set to CLIMATE_BY_DATA. { \"Rainfall_Scale_Factor\" : 1 } RBC_Destruction_Multiplier Malaria double NA NA 9.5 The number of total red blood cells destroyed per infected rupturing schizont . { \"RBC_Destruction_Multiplier\" : 3.291048711 } Regional_Migration_Filename Generic string NA NA \u00a0 The path of the input file which defines patterns of migration by vehicle via road or rail network. If the node is not on a road or rail network, regional migration focuses on the closest hub city in the network. The file must be in .bin format. { \"Regional_Migration_Filename\" : \"Regional_Migration.bin\" } Regional_Migration_Roundtrip_Duration Generic float 0 10000 1 The average time spent (in days) at the destination node during a round-trip migration by road network. Enable_Regional_Migration must be set to true (1). { \"Regional_Migration_Roundtrip_Duration\" : 1.0 } Regional_Migration_Roundtrip_Probability Generic float 0 1 0.1 The likelihood that an individual who travels by vehicle to another cell will return to the cell of origin during the next migration. Migration_Pattern must be set to SINGLE_ROUND_TRIPS. { \"Regional_Migration_Roundtrip_Probability\" : 1.0 } Relative_Humidity_Filename Generic string NA NA rel_hum.json The path of the input file which defines relative humidity data measured 2 meters above ground. Climate_Model must be set to CLIMATE_BY_DATA. The file must be in .bin format. { \"Relative_Humidity_Filename\" : \"Namawala_single_node_relative_humidity_daily.bin\" } Relative_Humidity_Scale_Factor Generic float 0.1 10 1 The scale factor used in multiplying relative humidity values. Climate_Model must be set to CLIMATE_BY_DATA. { \"Relative_Humidity_Scale_Factor\" : 1 } Relative_Humidity_Variance Generic float 0 0.12 0.05 The standard deviation (in percentage) for normally distributed noise applied to the daily relative humidity values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA. Enable_Climate_Stochasticity must be set to true (1). { \"Relative_Humidity_Variance\" : 0.05 } Relative_Sample_Rate_Immune Generic float 0.001 1 0.1 The relative sampling rate for people who have acquired immunity through recovery or vaccination. { \"Relative_Sample_Rate_Immune\" : 0.1 } Report_Coital_Acts STI boolean 0 1 0 Set to true (1) to enable or to false (0) to disable the RelationshipConsummatedReport.csv output report. { \"Report_Coital_Acts\" : 1 } Report_Event_Recorder Generic boolean 0 1 0 Set to true (1) to enable or to false (0) to disable the ReportEventRecorder.csv output report. { \"Report_Event_Recorder\" : 1 , \"Report_Event_Recorder_Events\" : [ \"VaccinatedA\" , \"VaccineExpiredA\" , \"VaccinatedB\" , \"VaccineExpiredB\" ], \"Report_Event_Recorder_Ignore_Events_In_List\" : 0 } Report_Event_Recorder_Events Generic array NA NA \u00a0 The list of events to include or exclude in the ReportEventRecorder.csv output report, based on how Report_Event_Recorder_Ignore_Events_In_List is set. { \"Report_Event_Recorder\" : 1 , \"Report_Event_Recorder_Events\" : [ \"VaccinatedA\" , \"VaccineExpiredA\" , \"VaccinatedB\" , \"VaccineExpiredB\" ], \"Report_Event_Recorder_Ignore_Events_In_List\" : 0 } Report_Event_Recorder_Ignore_Events_In_List Generic boolean 0 1 0 If set to false (0), only the events listed in the Report_Event_Recorder_Events array will be included in the ReportEventRecorder.csv output report. If set to true (1), only the events listed in the array will be excluded, and all other events will be included. If you want to return all events from the simulation, leave the events array empty. Value Events Array Output File 0 No events No events 0 One or more events Only the listed events 1 No events All events occurring in the simulation 1 One or more events All simulation events occurring in the simulation, except for those listed { \"Report_Event_Recorder\" : 1 , \"Report_Event_Recorder_Events\" : [ \"VaccinatedA\" , \"VaccineExpiredA\" , \"VaccinatedB\" , \"VaccineExpiredB\" ], \"Report_Event_Recorder_Ignore_Events_In_List\" : 0 } Report_Event_Recorder_Individual_Properties Generic array of strings NA NA [] Specifies an array of events that will be excluded from the property output report; all events NOT listed in the array will be included in the report. To report all events from the simulation, leave the events array empty. This example demonstrates reporting all individual property events: { \"Report_Event_Recorder_Individual_Properties\" : [] } The following example demonstrates the syntax for excluding particular properties from the report: { \"Report_Event_Recorder_Individual_Properties\" : [ \"Accessibility\" , \"Risk\" ] } Report_HIV_ART HIV boolean 0 1 0 Set to true (1) to enable or to false (0) to disable the ReportHIVART.csv output report. { \"Report_HIV_ART\" : 0 } Report_HIV_ByAgeAndGender HIV boolean 0 1 0 Set to true (1) to enable or to false (0) to disable the ReportHIVByAgeAndGener.csv output report. { \"Report_HIV_ByAgeAndGender\" : 1 } Report_HIV_ByAgeAndGender_Add_Relationships HIV boolean 0 1 0 Sets whether or not the ReportHIVByAgeAndGender.csv output file will contain data by relationship type on population currently in a relationship and ever in a relationship. A sum of those in two or more partnerships and a sum of the lifetime number of relationships in each bin will be included. { \"Report_HIV_ByAgeAndGender_Add_Relationships\" : 1 } Report_HIV_ByAgeAndGender_Add_Transmitters HIV boolean 0 1 0 When Set to to true (1), the \u201ctransmitters\u201d column is included in the output report. For a given row, \u201cTransmitters\u201d indicates how many people that have transmitted the disease meet the specifications of that row. { \"Report_HIV_ByAgeAndGender_Add_Transmitters\" : 1 } Report_HIV_ByAgeAndGender_Collect_Age_Bins_Data HIV array of floats -3.40282e+38 3.40282e+38 1 This array of floats allows the user to define the age bins used to stratify the report by age. Each value defines the minimum value of that bin, while the next value defines the maximum value of the bin. The maximum number of age bins is 100. For example, if you had: \u201cReport_HIV_ByAgeAndGender_Collect_Age_Bins_Data\u201d : [ 0, 10, 20, 50, 100 ] The report would have the following age bins: From 0 up to (but not including) 10, from 10 up to (but not including) 20, from 20 up to (but not including) 50, from 50 up to (but not including) 100, and 100 and over. If no values are specified in the array, then the output report will have no age binning. { \"Report_HIV_ByAgeAndGender_Collect_Age_Bins_Data\" : [ 0 , 1 , 2 , 3 , 4 , 5 , 6 , 7 , 8 , 9 , 10 , 11 , 12 , 13 , 14 , 15 , 16 , 17 , 18 , 19 , 20 , 21 , 22 , 23 , 24 , 25 , 26 , 27 , 28 , 29 , 30 , 31 , 32 , 33 , 34 , 35 , 36 , 37 , 38 , 39 , 40 , 41 , 42 , 43 , 44 , 45 , 46 , 47 , 48 , 49 , 50 , 51 , 52 , 53 , 54 , 55 , 56 , 57 , 58 , 59 , 60 , 61 , 62 , 63 , 64 , 65 , 66 , 67 , 68 , 69 , 70 , 71 , 72 , 73 , 74 , 75 , 76 , 77 , 78 , 79 , 80 , 81 , 82 , 83 , 84 , 85 , 86 , 87 , 88 , 89 , 90 , 91 , 92 , 93 , 94 , 95 , 96 , 97 , 98 , 99 ] } Report_HIV_ByAgeAndGender_Collect_Circumcision_Data HIV boolean 0 1 0 When set to 1, the ReportHIVByAgeAndGender.csv output report will have separate rows for those who do or do not have the MaleCircumcision intervention and an additional column with 0 and 1 indicating whether the row corresponds to those who are or are not circumcised. Setting this to 1 doubles the number of rows in ReportHIVByAgeAndGender.csv. { \"Report_HIV_ByAgeAndGender_Collect_Circumcision_Data\" : 0 } Report_HIV_ByAgeAndGender_Collect_Gender_Data HIV boolean 0 1 0 Controls whether or not the report is stratified by gender; to enable gender stratification, set to true (1). { \"Report_HIV_ByAgeAndGender_Collect_Gender_Data\" : 1 } Report_HIV_ByAgeAndGender_Collect_HIV_Data HIV boolean 0 1 0 When set to 1, the ReportHIVByAgeAndGender.csv output report will have separate rows for those who do or do not have the ART intervention and an additional column with 0 and 1 indicating whether the row corresponds to those who are or are not on ART. Setting this to 1 doubles the number of rows in ReportHIVByAgeAndGender.csv. { \"Report_HIV_ByAgeAndGender_Collect_HIV_Data\" : 1 } Report_HIV_ByAgeAndGender_Collect_Intervention_Data HIV array of strings NA NA NA Stratifies the population by adding a column of 0s and 1s depending on whether or not the person has the indicated intervention. This only works for interventions that remain with a person for a period of time, such as ART, VMMC, vaccine/PrEP, or a delay state in the cascade of care. You can name the intervention by adding a parameter Intervention_Name in the campaign file, and then give this parameter the same Intervention_Name . This allows you to have multiple types of vaccines, VMMCs, etc., but to only stratify on the type you want. { \"Report_HIV_ByAgeAndGender_Collect_Intervention_Data\" : [ \"ART_Intervention\" , \"PrEP_Intervention\" ] } Report_HIV_ByAgeAndGender_Collect_IP_Data HIV boolean 0 1 0 When set to 1, the ReportHIVByAgeAndGender.csvoutput report will have separate rows for those with different IndividualProperties values and an additional column for each property indicating which row corresponds to which value of that property. Setting this to 1 typically increases by severalfold the number of rows in ReportHIVByAgeAndGender.csv. { \"Report_HIV_ByAgeAndGender_Collect_IP_Data\" : 0 } Report_HIV_ByAgeAndGender_Event_Counter_List HIV array of strings NA NA NA A list of columns to add to the ReportHIVByAgeAndGender.csv output files counting the number of times an intervention with the corresponding Distributed_Event_Trigger has been distributed. Note that the interventions will be specified in the campaign file. { \"Report_HIV_ByAgeAndGender_Event_Counter_List\" : [ \"Reduced_Acquire_Lowest\" , \"Reduced_Acquire_Medium\" , \"Reduced_Acquire_Low\" , \"Reduced_Acquire_Highest_Not_Duplicated\" ] } Report_HIV_ByAgeAndGender_Start_Year HIV float 1900 2200 1900 The beginning calendar year that will be collected by the ReportHIVByAgeAndGender.csv report. { \"Report_HIV_ByAgeAndGender_Start_Year\" : 1962 , \"Report_HIV_ByAgeAndGender_Stop_Year\" : 1978 } Report_HIV_ByAgeAndGender_Stop_Year HIV float 1900 2200 2200 The ending calendar year that will be collected by the HIVByAgeAndGender.csv report. { \"Report_HIV_ByAgeAndGender_Start_Year\" : 1962 , \"Report_HIV_ByAgeAndGender_Stop_Year\" : 1978 } Report_HIV_ByAgeAndGender_Stratify_Infected_By_CD4 HIV boolean 0 1 0 When set to 1, the ReportHIVByAgeAndGender.csv output file will separate the count of the number of infected individuals by CD4 stratum. When set to 0, the number of infected individuals are aggregated into one column regardless of CD4 count. { \"Report_HIV_ByAgeAndGender_Stratify_Infected_By_CD4\" : 0 } Report_HIV_Event_Channels_List HIV array of strings NA NA NA This is the list of events included in the InsetChart report. If events are specified with this parameter, the InsetChart will include a channel for each event listed. If no events are listed, a \u201cNumber of Events\u201d channel will display the total number of all events that occurred during the simulation. { \"Report_HIV_Event_Channels_List\" : [ \"NewInfectionEvent\" , \"HIVNeedsHIVTest\" , \"HIVPositiveHIVTest\" , \"HIVNegativeHIVTest\" , \"HIVDiagnosedEligibleForART\" , \"HIVSymptomatic\" , \"DiseaseDeaths\" , \"NonDiseaseDeaths\" , \"Births\" ], } Report_HIV_Infection HIV boolean 0 1 0 Enables or disables the ReportHIVInfection.csv output report. { \"Report_HIV_Infection\" : 0 } Report_HIV_Infection_Start_Year HIV float 1900 2200 1900 The beginning calendar year that will be collected by the ReportHIVInfection.csv output report. Report_HIV_Infection must be set to 1. { \"Report_HIV_Infection_Start_Year\" : 1900 } Report_HIV_Infection_Stop_Year HIV float 1900 2200 2200 The ending calendar year that will be collected by the ReportHIVInfection.csv output report. Report_HIV_Infection must be set to 1. { \"Report_HIV_Infection_Stop_Year\" : 2100 } Report_HIV_Mortality HIV boolean 0 1 0 Enables or disables the HIVMortality.csv (disease and non-disease deaths) output report. { \"Report_HIV_Mortality\" : 0 } Report_HIV_Period HIV float 30 36500 730 The number of days between records in the HIV_By_Age_And_Gender output report. Output data will only be recorded during a time step, so if Report_HIV_Period is set to a value less than the value set for Simulation_Timestep , more than one period may occur before the next time step. When Report_HIV_Period is greater than the value set for Simulation_Timestep , a record may not be written during each time step. Note that the number of days between records is half the time specified by this parameter. For example, if Report_HIV_Period is set to 40, the actual time between records is 20 days. For best results, use integers for this value. { \"Report_HIV_Period\" : 365 } Report_Relationship_End STI boolean 0 1 0 Enables or disables the RelationshipEnd.csv output report. For HIV simulations, there will be no additional columns. { \"Report_Relationship_End\" : 0 } Report_Relationship_Start STI boolean 0 1 0 Enables or disables the RelationshipStart.csv output report. For HIV simulations, there will be some additional columns. { \"Report_Relationship_Start\" : 0 } Report_Transmission STI boolean 0 1 0 Enables or disables the TransmissionReport.csv output report. For HIV simulations, there will be some additional columns. { \"Report_Transmission\" : 0 } Report_Typhoid_ByAgeAndGender_Start_Year Typhoid float 0 3.40E+38 0 The year to start the typhoid output report. { \"Report_Typhoid_ByAgeAndGender_Start_Year\" : 1970 } Report_Typhoid_ByAgeAndGender_Stop_Year Typhoid float 0 3.40E+38 3.40E+38 The year to stop the typhoid output report. { \"Report_Typhoid_ByAgeAndGender_Stop_Year\" : 2000 } Roundtrip_Waypoints Generic integer 0 1000 10 The maximum number of points reached during a trip before steps are retraced on the return trip home. Migration_Pattern must be set to WAYPOINTS_HOME. { \"Roundtrip_Waypoints\" : 5 } Run_Number Generic integer 0 2147480000 1 Sets the random number seed through a bit manipulation process. When running a multi-core simulation, combines with processor rank to produce independent random number streams for each process. { \"Run_Number\" : 1 } Sample_Rate_0_18mo Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for infants age 0 to 18 months. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_0_18mo\" : 1 } Sample_Rate_10_14 Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for children age 10 to 14 years. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_10_14\" : 1 } Sample_Rate_15_19 Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for adolescents age 15 to 19 years. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_15_19\" : 1 } Sample_Rate_18mo_4yr Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for children age 18 months to 4 years. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_18mo_4yr\" : 1 } Sample_Rate_20_plus Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for adults age 20 and older. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_20_plus\" : 1 } Sample_Rate_5_9 Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for children age 5 to 9 years. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_5_9\" : 1 } Sample_Rate_Birth Generic float 0 1000 1 For age-adapted sampling, the relative sampling rate for births. Individual_Sampling_Type must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. { \"Sample_Rate_Birth\" : 1 } Sea_Migration_Filename Generic string NA NA \u00a0 The path of the input file which defines patterns of migration by ship. Only used when Enable_Sea_Migration is set to true (1). The file must be in .bin format. { \"Sea_Migration_Filename\" : \"5x5_Households_Work_Migration.bin\" } Sea_Migration_Roundtrip_Duration Generic float 0 10000 1 The average time spent at the destination node during a round-trip migration by ship. Used only when Enable_Sea_Migration is set to true (1). { \"Sea_Migration_Roundtrip_Duration\" : 10000 } Sea_Migration_Roundtrip_Probability Generic float 0 1 0.25 The likelihood that an individual who travels by ship into a neighboring cell will return to the cell of origin during the next migration. Used only when Enable_Sea_Migration is set to true (1). { \"Sea_Migration_Roundtrip_Probability\" : 0 } Semipermanent_Habitat_Decay_Rate Vector float 0.0001 100 0.01 Daily rate of larval habitat loss for semi-permanent habitats with Larval_Habitat_Types parameter value of WATER_VEGETATION or BRACKISH_SWAMP. { \"Semipermanent_Habitat_Decay_Rate\" : 0.01 } Serialization_Time_Steps Generic array of integers 0 2.15E+09 \u00a0 The list of time steps after which to save the serialized state. 0 (zero) indicates the initial state before simulation, n indicates after the nth time step. By default, no serialized state is saved. { \"Serialization_Time_Steps\" : [ 0 , 10 ] } Serialized_Population_Filenames Generic array of strings NA NA NA Array of filenames with serialized population data. The number of filenames must match the number of cores used for the simulation. The file must be in .dtk format. { \"Serialized_Population_Filenames\" : [ \"state-00010.dtk\" ] } Serialized_Population_Path Generic string NA NA . The root path for the serialized population files. { \"Serialized_Population_Path\" : \"../00_Generic_Version_1_save/output\" } Sexual_Debut_Age_Female_Weibull_Heterogeneity STI float 0 50 20 The inverse shape of the Weibull distribution for female debut age. { \"Sexual_Debut_Age_Female_Weibull_Heterogeneity\" : 0.05 } Sexual_Debut_Age_Female_Weibull_Scale STI float 0 50 16 The scale term of the Weibull distribution for female debut age. { \"Sexual_Debut_Age_Female_Weibull_Scale\" : 15.919654846191 } Sexual_Debut_Age_Male_Weibull_Heterogeneity STI float 0 50 20 The inverse shape of the Weibull distribution for male debut age. { \"Sexual_Debut_Age_Male_Weibull_Heterogeneity\" : 0.05 } Sexual_Debut_Age_Male_Weibull_Scale STI float 0 50 16 The scale term of the Weibull distribution for male debut age. { \"Sexual_Debut_Age_Male_Weibull_Scale\" : 16.946729660034 } Sexual_Debut_Age_Min STI float 0 3.40E+38 13 The minimum age at which individuals become eligible to form sexual relationships. { \"Sexual_Debut_Age_Min\" : 13 } Simulation_Duration Generic float 0 1000000 1 The elapsed time (in days) from the start to the end of a simulation. { \"Simulation_Duration\" : 7300 } Simulation_Timestep Generic float 0 1000000 1 The duration of a simulation time step, in days. { \"Simulation_Timestep\" : 1 } Simulation_Type Generic enum NA NA GENERIC_SIM The type of disease being simulated. Possible IDM-supported values are: GENERIC_SIM VECTOR_SIM MALARIA_SIM TB_SIM STI_SIM HIV_SIM { \"Simulation_Type\" : \"STI_SIM\" } Spatial_Output_Channels Generic array of strings NA NA [] An array of channel names for spatial output by node and time step. The data from each channel will be written to a separate binary file. Enable_Spatial_Output must be set to true (1). Possible values are: Air_Temperature Data related to air temperature. Births Data related to the number of births. Campaign_Cost Data related to the costs of a campaign. Disease_Deaths Data related to the number of deaths due to disease. Human_Infectious_Reservoir Data related to the total infectiousness of the population. Infection_Rate Data related to infection rates. Land_Temperature Data related to the average land temperature over all nodes. New_Infections Data related to the presence of new infections. New_Reported_Infections Data related to the presence of reported new infections. Population Data related to the total population in the simulation. Prevalence Data related to the fraction of the population that is infected. Rainfall Data related to the presence of rainfall. Relative_Humidity Data related to the presence of relative humidity. { \"Spatial_Output_Channels\" : [ \"Prevalence\" , \"New_Infections\" ] } Start_Time Generic float 0 1000000 1 The time, in days, when the simulation begins. This time is used to identify the starting values of the temporal input data, such as specifying which day\u2019s climate values should be used for the first day of the simulation. Note: The Start_Day of campaign events is in absolute time, so time relative to the beginning of the simulation depends on this parameter. { \"Start_Time\" : 365 } Susceptibility_Scale_Type Generic enum NA NA CONSTANT_SUSCEPTIBILITY The effect of time or season on infectivity. Possible values are: CONSTANT_SUSCEPTIBILITY LOG_LINEAR_FUNCTION_OF_TIME LINEAR_FUNCTION_OF_AGE LOG_LINEAR_FUNCTION_OF_AGE { \"Susceptibility_Scale_Type\" : \"CONSTANT_SUSCEPTIBILITY\" } TB_Active_Cure_Rate TB float 0 1 0 The mean rate at which active symptomatic individuals progress to a cured (uninfected) state. In the absence of treatment, this rate is added to TB_Inactivation_Rate and TB_Active_Mortality_Rate to determine the total rate of the active smear-positive state, then the ratio of TB_Active_Cure_Rate to total rate determines the probability that an actively infected individual will go to a cured state after a mean duration of 1 / total rate. { \"TB_Active_Cure_Rate\" : 1e-09 } TB_Active_Mortality_Rate TB float 0 1 0.0009589 The mean rate at which active smear-positive individuals die from TB disease. In the absence of treatment, this rate is added to TB_Active_Cure_Rate and TB_Inactivation_Rate to determine the total rate of the active smear-positive state, then the ratio of TB_Active_Mortality_Rate to total rate determines the probability that an active smear-positive individual will die from TB disease after a mean duration of 1 / total rate. { \"TB_Active_Mortality_Rate\" : 5e-08 } TB_Active_Period_Distribution TB enum NA NA NOT_INITIALIZED Distribution for duration of active disease. For all possible values, the mean is defined by total rate, which is the sum of TB_Inactivation_Rate and TB_Active_Mortality_Rate corrected for smear status. Possible values are: NOT_INITIALIZED No distribution set. FIXED_DURATION A constant duration incubation period. UNIFORM_DURATION A uniform random draw for the duration of the incubation period. GAUSSIAN_DURATION TB_Active_Period_Std_Dev defines the standard deviation. Negative values are truncated at zero. GAUSSIAN_DURATION Duration of the active period is defined by the mean and standard deviation of the Gaussian distribution. Negative values are truncated at zero. EXPONENTIAL_DURATION The active period is the mean of the exponential random draw. POISSON_DURATION The active period is the mean of the random Poisson draw. LOG_NORMAL_DURATION The active period is a log normal distribution defined by a mean and log width. BIMODAL_DURATION The distribution is bimodal, the duration is a fraction of the active period for a specified period of time and equal to the active period otherwise. PIECEWISE_CONSTANT The distribution is specified with a list of years and a matching list of values. The duration at a given year is that specified for the nearest previous year. PIECEWISE_LINEAR The distribution is specified with a list of years and matching list of values. The duration at a given year is a linear interpolation of the specified values. WEIBULL_DURATION The duration is a Weibull distribution with a given scale and shape. DUAL_TIMESCALE_DURATION The duration is two exponential distributions with given means. { \"TB_Active_Period_Distribution\" : \"EXPONENTIAL_DURATION\" } TB_Active_Period_Std_Dev TB float 0 3.40E+38 1 The standard deviation for the distribution of the duration of active TB. Used when TB_Active_Period_Distribution is set to GAUSSIAN_DURATION. { \"TB_Active_Period_Std_Dev\" : 200 } TB_Active_Presymptomatic_Infectivity_Multiplier TB float 0 1 0.0274 The scale factor for Base_Infectivity to give the infectiousness of active presymptomatic individuals. { \"TB_Active_Presymptomatic_Infectivity_Multiplier\" : 0.19999999999999998 } TB_CD4_Activation_Vector TBHIV array of floats 0 3.40E+38 1 The increased risk per time step of TB activation by CD4 strata defined in CD4_Strata_Activation . { \"CD4_Strata_Activation\" : [ 0 , 350 , 500 , 1000 ], \"TB_CD4_Activation_Vector\" : [ 0.01 , 0.01 , 0.02 , 0.02 ] } TB_Drug_Clearance_Rate TB float 0 1 1 The daily rate at which treatment with an anti-TB drug causes disease clearance. This parameter, TB_Drug_Inactivation_Rate , TB_Drug_Mortality_Rate , and TB_Drug_Relapse_Rate specify the daily probability of a change in the TB disease state during the course of treatment. The sum of these rates and the total duration of treatment specify the probability that any event occurs. If no event occurs, the individual fails drug treatment. { \"TB_Drug_Params\" : { \"BurninDrug\" : { \"TB_Drug_Clearance_Rate\" : 0.0111 , \"TB_Drug_Inactivation_Rate\" : 0 , \"TB_Drug_Mortality_Rate\" : 0.00162 , \"TB_Drug_Relapse_Rate\" : 0.00223 , \"TB_Drug_Resistance_Rate\" : 0 } } } TB_Drug_Efficacy_Multiplier_Failed TB float 0 1 1 The scale factor for drug efficacy parameters ( TB_Drug_Inactivation_Rate and TB_Drug_Clearance_Rate ) to give the efficacy of first-line drugs when given to individuals who have failed treatment. { \"TB_Drug_Efficacy_Multiplier_Failed\" : 1.0 } TB_Drug_Efficacy_Multiplier_MDR TB float 0 1 1 The scale factor for drug efficacy parameters ( TB_Drug_Inactivation_Rate and TB_Drug_Clearance_Rate ) to give the efficacy of first-line drugs when given to individuals with MDR-TB. { \"TB_Drug_Efficacy_Multiplier_MDR\" : 1.0 } TB_Drug_Efficacy_Multiplier_Relapsed TB float 0 1 1 The scale factor for drug efficacy parameters ( TB_Drug_Inactivation_Rate and TB_Drug_Clearance_Rate ) to give the efficacy of first-line drugs when given to individuals who have relapsed after treatment. { \"TB_Drug_Efficacy_Multiplier_Relapsed\" : 1.0 } TB_Drug_Inactivation_Rate TB float 0 1 1 The daily rate at which treatment with an anti-TB drug causes inactivation. This parameter, TB_Drug_Clearance_Rate , TB_Drug_Mortality_Rate , and TB_Drug_Relapse_Rate specify the daily probability of a change in the TB disease state during the course of treatment. The sum of these rates and the total duration of treatment specify the probability that any event occurs. If no event occurs, the individual fails drug treatment. { \"TB_Drug_Params\" : { \"BurninDrug\" : { \"TB_Drug_Clearance_Rate\" : 0.0111 , \"TB_Drug_Inactivation_Rate\" : 0 , \"TB_Drug_Mortality_Rate\" : 0.00162 , \"TB_Drug_Relapse_Rate\" : 0.00223 , \"TB_Drug_Resistance_Rate\" : 0 } } } TB_Drug_Mortality_Rate TB float 0 1 0 The daily rate at which treatment with an anti-TB drug causes death. This parameter, TB_Drug_Inactivation_Rate , TB_Drug_Relapse_Rate , and TB_Drug_Clearance_Rate specify the daily probability of a change in the TB disease state during the course of treatment. The sum of these rates and the total duration of treatment specify the probability that any event occurs. If no event occurs, the individual fails drug treatment. { \"TB_Drug_Params\" : { \"BurninDrug\" : { \"TB_Drug_Clearance_Rate\" : 0.0111 , \"TB_Drug_Inactivation_Rate\" : 0 , \"TB_Drug_Mortality_Rate\" : 0.00162 , \"TB_Drug_Relapse_Rate\" : 0.00223 , \"TB_Drug_Resistance_Rate\" : 0 } } } TB_Drug_Params TB JSON object NA NA NA This JSON structure contains the names of anti-tuberculosis drugs and the parameters that define them. { \"TB_Drug_Params\" :{ \"CDCDrug\" :{ \"TB_Drug_Clearance_Rate\" : 0.1 , \"TB_Drug_Inactivation_Rate\" : 1e-09 , \"TB_Drug_Mortality_Rate\" : 1e-09 , \"TB_Drug_Primary_Decay_Time_Constant\" : 180.0 , \"TB_Drug_Relapse_Rate\" : 1e-09 , \"TB_Drug_Resistance_Rate\" : 1e-09 }, \"CDCDrugMDR\" :{ \"TB_Drug_Clearance_Rate\" : 1e-09 , \"TB_Drug_Inactivation_Rate\" : 1e-09 , \"TB_Drug_Mortality_Rate\" : 1e-09 , \"TB_Drug_Primary_Decay_Time_Constant\" : 180.0 , \"TB_Drug_Relapse_Rate\" : 1e-09 , \"TB_Drug_Resistance_Rate\" : 1e-09 } } } TB_Drug_Primary_Decay_Time_Constant TB float 0 100000 0 The number of days before the efficacy of a specific anti-TB drug begins to decay. { \"TB_Drug_Params\" : { \"CDCDrug\" : { \"TB_Drug_Clearance_Rate\" : 0.1 , \"TB_Drug_Inactivation_Rate\" : 1e-09 , \"TB_Drug_Mortality_Rate\" : 1e-09 , \"TB_Drug_Primary_Decay_Time_Constant\" : 180.0 , \"TB_Drug_Relapse_Rate\" : 1e-09 , \"TB_Drug_Resistance_Rate\" : 1e-09 } } } TB_Drug_Relapse_Rate TB float 0 1 0 The daily rate at which treatment with an anti-TB drug causes disease relapse. This parameter, TB_Drug_Inactivation_Rate , TB_Drug_Mortality_Rate , and TB_Drug_Clearance_Rate specify the daily probability of a change in the TB disease state during the course of treatment. The sum of these rates and the total duration of treatment specify the probability that any event occurs. If no event occurs, the individual fails drug treatment. { \"TB_Drug_Params\" : { \"CDCDrug\" : { \"TB_Drug_Clearance_Rate\" : 0.1 , \"TB_Drug_Inactivation_Rate\" : 1e-09 , \"TB_Drug_Mortality_Rate\" : 1e-09 , \"TB_Drug_Primary_Decay_Time_Constant\" : 180.0 , \"TB_Drug_Relapse_Rate\" : 1e-09 , \"TB_Drug_Resistance_Rate\" : 1e-09 } } } TB_Drug_Resistance_Rate TB float 0 1 0 The daily probability that an individual with drug-sensitive TB will acquire MDR-TB. Only individuals who return to the latent state or fail can acquire MDR-TB. { \"TB_Drug_Params\" : { \"CDCDrug\" : { \"TB_Drug_Clearance_Rate\" : 0.1 , \"TB_Drug_Inactivation_Rate\" : 1e-09 , \"TB_Drug_Mortality_Rate\" : 1e-09 , \"TB_Drug_Primary_Decay_Time_Constant\" : 180.0 , \"TB_Drug_Relapse_Rate\" : 1e-09 , \"TB_Drug_Resistance_Rate\" : 1e-09 } } } TB_Drug_Types_For_This_Sim TB array of strings NA NA [] The list of drug types that will be used with the AntiTBPropDep intervention. The values for each drug type are defined by TB_Drug_Params . { \"TB_Drug_Types_For_This_Sim\" : [ \"CDCDrug\" , \"CDCDrugRetx\" , \"CDCDrugMDR\" ] } TB_Extrapulmonary_Fraction_Adult TB float 0 1 0 The fraction of active cases in adults (age 15 or over) that are extrapulmonary. The sum of this and TB_Smear_Positive_Fraction_Adult must be less than or equal to 1. { \"TB_Extrapulmonary_Fraction_Adult\" : 0.33 } TB_Extrapulmonary_Fraction_Child TB float 0 1 0 The fraction of active cases in children (under age 15) that are extrapulmonary. The sum of this and TB_Smear_Positive_Fraction_Child must be less than or equal to 1. { \"TB_Extrapulmonary_Fraction_Child\" : 0.33 } TB_Extrapulmonary_Mortality_Multiplier TB float 0 1 0.4 The scale factor for TB_Active_Mortality_Rate to give the mortality rate of active extrapulmonary individuals. In the absence of treatment, the product is added to TB_Active_Cure_Rate and TB_Inactivation_Rate to determine the total rate of the active extrapulmonary state, then the ratio of the aforementioned product to total rate determines the probability that an active extrapulmonary individual will die from TB disease after mean duration of 1 / total rate. { \"TB_Extrapulmonary_Mortality_Multiplier\" : 1 } TB_Fast_Progressor_Fraction_Adult TB float 0 1 1 The fraction of adults (age 15 or over) that are fast progressors. { \"TB_Fast_Progressor_Fraction_Adult\" : 0.5 } TB_Fast_Progressor_Fraction_Child TB float 0 1 1 The fraction of children (under age 15) that are fast progressors. { \"TB_Fast_Progressor_Fraction_Child\" : 0.5 } TB_Fast_Progressor_Rate TB float 0 1 4.11E-05 The mean rate at which individuals who are fast progressors progress from a latently infected state to an active presymptomatic state. This rate is added to the TB_Latent_Cure_Rate to determine the total rate of the latently infected state, then the ratio of TB_Fast_Progressor_Rate to the total rate is calculated. This ratio determines the probability that a latently infected fast progressor will go to an active presymptomatic state after a mean duration of 1 / total rate. { \"TB_Fast_Progressor_Rate\" : 0.02 } TB_Immune_Loss_Fraction TB float 0 1 0.5 The fraction of individuals that lose immunity with time. { \"TB_Immune_Loss_Fraction\" : 0 } TB_Inactivation_Rate TB float 0 1 0.00041096 The mean rate at which active symptomatic (smear-positive, smear-negative, and extrapulmonary) individuals progress to an inactivated (latently infected) state. In the absence of treatment, this rate is added to TB_Active_Cure_Rate and TB_Active_Mortality_Rate to determine the total rate of the active symptomatic state, then the ratio of TB_Active_Cure_Rate to total rate determines the probability that an active syptomatic individual will go to a latently infected state after a mean duration of 1 / total rate. { \"TB_Inactivation_Rate\" : 0.05 } TB_Latent_Cure_Rate TB float 0 1 0.0005479 The mean rate at which individuals progress from a latently infected state to a cured (uninfected) state. This rate is added to either the TB_Fast_Progressor_Rate or the TB_Slow_Progressor_Rate to determine the total rate of the latently infected state (depending on whether the individual is a fast progressor or slow progressor, respectively), then the ratio of TB_Latent_Cure_Rate to total rate is calculated. This ratio determines the probability that a latently infected individual will go to a cured state. { \"TB_Latent_Cure_Rate\" : 0.1 } TB_MDR_Fitness_Multiplier TB float 0 1 1 The scale factor for Base_Infectivity to give the infectiousness of MDR-TB strains. { \"TB_MDR_Fitness_Multiplier\" : 0.65 } TB_Presymptomatic_Cure_Rate TB float 0 1 0.0274 The mean rate at which active presymptomatic individuals progress to a cured (uninfected) state. This rate is added to TB_Presymptomatic_Rate to determine the total rate of the active presymptomatic state, then the ratio of TB_Presymptomatic_Cure_Rate to total rate is calculated. This ratio determines the probability that an active presymptomatic individual will go to a cured state after a mean duration of 1 / total rate. { \"TB_Presymptomatic_Cure_Rate\" : 0.05 } TB_Presymptomatic_Rate TB float 0 1 0.0274 The mean rate at which active presymptomatic individuals progress to an active symptomatic state. This rate is added to TB_Presymptomatic_Cure_Rate to determine the total rate of the active presymptomatic state, then the ratio of TB_Presymptomatic_Rate to total rate is calculated. This ratio determines the probability that an active presymptomatic individual will go to active symptomatic disease after a mean duration of 1 / total rate. { \"TB_Presymptomatic_Rate\" : 0.0333 } TB_Relapsed_to_Active_Rate TB float 0 1 4.11E-05 The mean rate at which relapsed individuals progress to an active symptomatic state. { \"TB_Relapsed_to_Active_Rate\" : 0.01 } TB_Slow_Progressor_Rate TB float 0 1 2.05E-06 The mean rate at which individuals who are slow progressors progress from a latently infected state to an active presymptomatic state. This rate is added to the TB_Latent_Cure_Rate to determine the total rate of the latently infected state, then the ratio of TB_Slow_Progressor_Rate to the total rate is calculated. This ratio determines the probability that a latently infected fast progressor will go to an active presymptomatic state after a mean duration of 1 / total rate. { \"TB_Slow_Progressor_Rate\" : 0.02 } TB_Smear_Negative_Infectivity_Multiplier TB float 0 1 0.25 The scale factor for Base_Infectivity to give the infectiousness of active smear-negative individuals. { \"TB_Smear_Negative_Infectivity_Multiplier\" : 1 } TB_Smear_Negative_Mortality_Multiplier TB float 0 1 0.4 The scale factor to increase the mortality rate set by TB_Active_Mortality_Rate for individuals with smear-negative TB. In the absence of treatment, the product is added to TB_Active_Cure_Rate and TB_Inactivation_Rate to determine the total rate of the active smear-negative state, then the ratio of the aforementioned product to total rate determines the probability that an active smear-negative individual will die from TB disease after mean duration of 1 / total rate. { \"TB_Smear_Negative_Mortality_Multiplier\" : 1 } TB_Smear_Positive_Fraction_Adult TB float 0 1 1 The fraction of active cases in adults (age 15 and over) that are smear-positive. The sum of TB_Smear_Positive_Fraction_Adult and TB_Extrapulmonary_Fraction_Adult must be less than or equal to 1. This sum plus the fraction of smear-negative adults is equal to 1. { \"TB_Smear_Positive_Fraction_Adult\" : 1 } TB_Smear_Positive_Fraction_Child TB float 0 1 1 The fraction of active cases in children (under age 15) that are smear-positive. The sum of TB_Smear_Positive_Fraction_Child and TB_Extrapulmonary_Fraction_Child must be less than or equal to 1. This sum plus the fraction of smear-negative children is equal to 1. { \"TB_Smear_Positive_Fraction_Child\" : 1 } Temperature_Dependent_Feeding_Cycle Vector enum NA NA NO_TEMPERATURE_DEPENDENCE The effect of temperature on the duration between blood feeds. Possible values are: NO_TEMPERATURE_DEPENDENCE No relation to temperature; days between feeds will be constant and specified by Days_Between_Feeds for each species. ARRHENIUS_DEPENDENCE Use the Arrhenius equation with parameters Cycle_Arrhenius_1 and Cycle_Arrhenius_2 . BOUNDED_DEPENDENCE Use an equation bounded at 10 days at 15C and use Days_Between_Feeds to set the duration at 30C. { \"Temperature_Dependent_Feeding_Cycle\" : \"BOUNDED_DEPENDENCE\" } Temporary_Habitat_Decay_Factor Vector float 0.001 100 0.05 The factor to convert raw evaporation rate (ignoring boundary layer effects) to the daily rate of larval habitat loss for temporary habitats ( Larval_Habitat_Types set to TEMPORARY_RAINFALL). Units are (larval carrying capacity per day) / (kg per square meter per second). { \"Temporary_Habitat_Decay_Factor\" : 0.05 } Transmission_Blocking_Immunity_Decay_Rate Generic float 0 1000 0.001 The rate at which transmission-blocking immunity decays after the base transmission-blocking immunity offset period. Used only when Enable_Immunity and Enable_Immune_Decay parameters are set to true (1). { \"Transmission_Blocking_Immunity_Decay_Rate\" : 0.01 } Transmission_Blocking_Immunity_Duration_Before_Decay Generic float 0 45000 0 The number of days after infection until transmission-blocking immunity begins to decay. Only used when Enable_Immunity and Enable_Immune_Decay parameters are set to true (1). { \"Transmission_Blocking_Immunity_Duration_Before_Decay\" : 90 } Transmission_Rate Vector float 0 1 0.5 The probability that the bite of an infected mosquito establishes a new infection in an immunologically naive and uninfected individual, or the modifier of the probability of success for an individual with pre-erythrocytic immunity. Note that each mosquito species will have their own Transmission_Rate parameter. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Cycle_Arrhenius_1\" : 99.0 , \"Cycle_Arrhenius_2\" : 88.0 , \"Cycle_Arrhenius_Reduction_Factor\" : 0.77 , \"Nighttime_Feeding_Fraction\" : 1.0 , \"Acquire_Modifier\" : 0.2 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 11250000000 , \"BRACKISH_SWAMP\" : 10000000000 }, \"Transmission_Rate\" : 0.5 } } } Upper_Age_In_Years Malaria float 0 125 NA The age, in years, below which children are given a fraction of the adult drug dose, as defined in Fraction_Of_Adult_Dose . Note that this parameter can be specified for each drug included in the configuration file, and different fractional doses may be used for different age groups. { \"Malaria_Drug_Params\" : { \"Artemether\" : { \"Fractional_Dose_By_Upper_Age\" : [{ \"Fraction_Of_Adult_Dose\" : 0.25 , \"Upper_Age_In_Years\" : 3 }, { \"Fraction_Of_Adult_Dose\" : 0.5 , \"Upper_Age_In_Years\" : 6 }, { \"Fraction_Of_Adult_Dose\" : 0.75 , \"Upper_Age_In_Years\" : 10 }] } } } Vector_Larval_Rainfall_Mortality Vector enum NA NA NONE The type of vector larval mortality function due to rainfall. Possible values are: NONE No larval mortality due to rainfall. SIGMOID The mortality rate grows linearly from 0 at the threshold to 1 at twice the threshold value. SIGMOID_HABITAT_SHIFTING The threshold value is reduced by a factor proportional to how full the larval habitat is. { \"Vector_Larval_Rainfall_Mortality\" : \"SIGMOID\" } Vector_Migration_Filename_Local Vector string NA NA UNSPECIFIED_FILE The path to the vector migration file which defines patterns of vector migration to adjacent nodes. Enable_Vector_Migration must be set to 1. The file must be in .bin format. { \"Vector_Migration_Filename_Local\" : \"5x5_Households_Local_Vector_Migration.bin\" } Vector_Migration_Filename_Regional Vector string NA NA UNSPECIFIED_FILE The path to the vector migration file which defines patterns of vector migration to non-adjacent nodes. Enable_Vector_Migration must be set to 1. The file must be in .bin format. { \"Vector_Migration_Filename_Regional\" : \"5x5_Households_Regional_Vector_Migration.bin\" } Vector_Migration_Food_Modifier Vector float 0 3.40E+38 0 The preference of a vector to migrate toward a node currently occupied by humans, independent of the number of humans in the node. Used only when Vector_Sampling_Type is set to TRACK_ALL_VECTORS. Enable_Vector_Migration must be set to 1. { \"Vector_Migration_Food_Modifier\" : 1.0 } Vector_Migration_Habitat_Modifier Vector float 0 3.40E+38 0 The preference of a vector to migrate toward a node with more habitat. Only used when Vector_Sampling_Type is set to TRACK_ALL_VECTORS. Enable_Vector_Migration must be set to 1. { \"Vector_Migration_Habitat_Modifier\" : 1.0 } Vector_Migration_Modifier_Equation Vector enum NA NA LINEAR The functional form of vector migration modifiers. Enable_Vector_Migration must be set to 1. Possible values are: LINEAR and EXPONENTIAL. { \"Vector_Migration_Modifier_Equation\" : \"EXPONENTIAL\" } Vector_Migration_Stay_Put_Modifier Vector float 0 3.40E+38 0 The preference of a vector to remain in its current node rather than migrate to another node. Used only when Vector_Sampling_Type is set to TRACK_ALL_VECTORS. Enable_Vector_Migration must be set to 1. { \"Vector_Migration_Stay_Put_Modifier\" : 1.0 } Vector_Sampling_Type Vector enum NA NA TRACK_ALL_VECTORS The type of vector sampling used. Possible values are: TRACK_ALL_VECTORS Each vector at a location is part of the simulation as a separate object. SAMPLE_IND_VECTORS A subset of the vectors at a location are part of the simulation and represent the overall vector population. VECTOR_COMPARTMENTS_NUMBER The default option. This does not track every vector individually, but it provides a compartment for every possible entry in state space and tracks the number of vectors in each compartment. This successfully accounts for each vector, but does not distinguish between vectors that have identical states, just counts how many identical vectors of each state are present at a location. VECTOR_COMPARTMENTS_PERCENT Similar to VECTOR_COMPARTMENTS_NUMBER, but it only tracks the percentage of the population in each state. Less useful, but included for comparison to other models. { \"Vector_Sampling_Type\" : \"TRACK_ALL_VECTORS\" } Vector_Species_Names Vector array of strings NA NA [] An array of vector species names for all species that are present in the simulation. You may provide any species names you desire, but the same names must be used in Vector_Species_Params . { \"Vector_Species_Names\" : [ \"arabiensis\" , \"funestus\" , \"gambiae\" ] } Vector_Species_Params Vector JSON object NA NA NA A JSON object that lists each vector species, as defined in Vector_Species_Names , and the parameters that define them. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 1 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 1e10 , \"WATER_VEGETATION\" : 1e6 }, \"Transmission_Rate\" : 0.5 } } Vector_Sugar_Feeding_Frequency Vector enum NA NA VECTOR_SUGAR_FEEDING_NONE The frequency of sugar feeding by a female mosquito. Used is used in conjunction with the SugarTrap and OvipositionTrap interventions. Vector_Sampling_Type must be set to TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. Possible values are: VECTOR_SUGAR_FEEDING_NONE No sugar feeding. VECTOR_SUGAR_FEEDING_ON_EMERGENCE_ONLY Sugar feeding once at emergence. VECTOR_SUGAR_FEEDING_EVERY_FEED Sugar feeding occurs once per blood meal. VECTOR_SUGAR_FEEDING_EVERY_DAY Sugar feeding occurs every day. { \"Vector_Sugar_Feeding_Frequency\" : \"VECTOR_SUGAR_FEEDING_NONE\" } Wolbachia_Infection_Modification Vector float 0 100 1 The change in vector susceptibility to infection due to a Wolbachia infection. { \"Wolbachia_Infection_Modification\" : 1.0 } Wolbachia_Mortality_Modification Vector float 0 100 1 The change in vector mortality due to a Wolbachia infection. { \"Wolbachia_Infection_Modification\" : 0.0 } x_Air_Migration Generic float 0 3.40E+38 1 Scale factor for the rate of migration by air, as provided by the migration file. Enable_Air_Migration must be set to 1. { \"x_Air_Migration\" : 1 } x_Birth Generic float 0 3.40E+38 1 Scale factor for birth rate, as provided by the demographics file. Enable_Birth must be set to 1. { \"x_Birth\" : 1 } x_Family_Migration Generic float 0 3.40E+38 1 Scale factor for the rate of migration by families, as provided by the migration file. Enable_Family_Migration must be set to true (1). { \"x_Family_Migration\" : 1 } x_Local_Migration Generic float 0 3.40E+38 1 Scale factor for rate of migration by foot travel, as provided by the migration file. Enable_Local_Migration must be set to 1. { \"x_Local_Migration\" : 1 } x_Other_Mortality Generic float 0 3.40E+38 1 Scale factor for mortality from causes other than the disease being simulated, as provided by the demographics file. Enable_Vital_Dynamics must be set to 1. { \"x_Other_Mortality\" : 1 } x_Regional_Migration Generic float 0 3.40E+38 1 Scale factor for the rate of migration by road vehicle, as provided by the migration file. Enable_Regional_Migration must be set to 1. { \"x_Regional_Migration\" : 1 } x_Sea_Migration Generic float 0 3.40E+38 1 Scale factor for the rate of migration by sea, as provided by the migration file. Enable_Sea_Migration must be set to 1. { \"x_Sea_Migration\" : 1 } x_Temporary_Larval_Habitat Vector double NA NA 1 Scale factor for the habitat size for all mosquito populations. { \"x_Temporary_Larval_Habitat\" : 1 } x_Vector_Migration_Local Vector float 0 3.40E+38 1 Scale factor for the rate of vector migration to adjacent nodes, as provided by the vector migration file. Enable_Vector_Migration must be set to 1. { \"x_Vector_Migration_Local\" : 1.0 } x_Vector_Migration_Regional Vector float 0 3.40E+38 1 Scale factor for the rate of vector migration to non-adjacent nodes, as provided by the vector migration file. Enable_Vector_Migration must be set to 1. { \"x_Vector_Migration_Regional\" : 1.0 } Zoonosis_Rate Generic float 0 1 0 The daily rate of zoonotic infection per individual. Animal_Reservoir_Type must be set to CONSTANT_ZOONOSIS or ZOONOSIS_FROM_DEMOGRAPHICS. If Animal_Reservoir_Type is set to ZOONOSIS_FROM_DEMOGRAPHICS, the value for the Zoonosis NodeAttribute in the demographics file will override the value set for Zoonosis_Rate . { \"Zoonosis_Rate\" : 0.005 }", "tags": "", "title": "Configuration parameters \u2014 Parameter Reference  documentation"}, {"url": "./parameter-demographics.html", "text": "Demographics file structure and parameters \u00b6 The parameters described in this reference section can be added to the JavaScript Object Notation (JSON) formatted demographics file to determine the demographics of the population within each geographic node in a simulation. For example, the number of individuals and the distribution for age, gender, immunity, risk, and mortality. This is in contrast to the configuration parameters related to demographics that are simulation-wide and generally control whether certain events, such as births or deaths, are enabled in a simulation. At least one demographics file is required for every simulation unless you set the parameter Enable_Demographics_Builtin to 0 (zero) in the configuration file . This is generally only used for testing and validating code pathways by using a standard testing sandbox instead of actual demographics information. If Migration_Model is set to \u201cFIXED_RATE_MIGRATION\u201d, built-in demographics will use a form of local migration where individuals migrate only to adjacent nodes. Demographics files are usually the only input data file you will modify. Demographics files are named using the name of the region, appended with \u201c_demographics.json\u201d. Additionally, you can use more than one demographics file, with one serving as the base layer and the one or more others acting as overlays that override the values in the base layer. Collapse all parameter tables Contents Demographic file structure Metadata Defaults Nodes Parameters NodeAttributes IndividualAttributes IndividualProperties Example Demographic file structure \u00b6 Demographics files are organized into three sections: Metadata Information such as data source, created date or region name. Metadata is used to provide provenance information. Defaults Default parameter values applied to all nodes. Nodes Parameter values specific to individual geographic nodes, which override the default values. There is one entry for each node in the simulation. The following example shows the skeletal format of a demographics file. { \"Metadata\" : { \"DateCreated\" : \"dateTime\" , \"Tool\" : \"scriptUsedToGenerate\" , \"Author\" : \"author\" , \"IdReference\" : \"ID reference\" , \"NodeCount\" : 2 }, \"Defaults\" : { \"NodeAttributes\" : {}, \"IndividualAttributes\" : {}, \"IndividualProperties\" : {} }, \"Nodes\" : [{ \"NodeID\" : 1 , \"NodeAttributes\" : {}, \"IndividualAttributes\" : {}, \"IndividualProperties\" : {} }, { \"NodeID\" : 2 , \"NodeAttributes\" : {}, \"IndividualAttributes\" : {}, \"IndividualProperties\" : {} }] } Metadata \u00b6 Most of the items in the Metadata section informational only and are not used by EMOD. This includes information such as author, date created, and tool used to create the file. However, the following two values are used by EMOD: NodeCount The number of nodes to expect in the demographics file. IdReference The unique string that indicates the method used for generating the NodeID , the identifier used for each node in the simulation. Defaults \u00b6 Parameter values in the Defaults section is applied to all nodes. However, any node-level information provided in the Nodes section will take precedence for that node. The Defaults section is optional and can be completely omitted, as is often the case of single- node simulations. The Defaults section can contain any of the parameters listed below. Nodes \u00b6 Parameter values in the Nodes section is applied to specific nodes. The intent of the Nodes section is to have only the information that is unique to each node, such as identifier and location (longitude and latitude). The section contains an array with each element in the array representing a node that is identified by its NodeID . The NodeID is a unique integer value. If a parameter appears in both the Defaults and Nodes sections, the value in the Nodes section will take precedence. The Nodes section can contain any of the parameters listed below, except for the Society parameters that can only be applied in the Defaults section. Parameters \u00b6 The demographic parameters are divided into the following broad categories. The parameters will be contained in a nested JSON object of the same name. Warning Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false. NodeAttributes \u00b6 NodeAttributes are a JSON object that contains parameters that add or modify information regarding the location, migration, habitat, and population of a simulation. Some NodeAttributes depend on values set in the configuration parameters. Toggle parameter table Parameter Data type Minimum Maximum Default Description Example AbovePoverty float 0 1 1 The proportion of the population that is living above the poverty line. { \"NodeAttributes\" : { \"AbovePoverty\" : 0.5 } } Airport bool 0 1 0 Set to 1 to indicate that the node is connected by air migration. { \"NodeAttributes\" : { \"Airport\" : 0 } } Altitude float -3.40282e+038 3.40E+38 0 Required. Only used when Climate_Model is set to \u201cCLIMATE_KOPPEN\u201d. The unit of measurement is in meters. { \"NodeAttributes\" : { \"Altitude\" : 250 } } BirthRate float 0 1 0.25 The birth rate. In the configuration file, Enable_Birth must be set to 1 and Birth_Rate_Dependence will affect how this rate is used. { \"NodeAttributes\" : { \"BirthRate\" : 0.0001 } } InitialPopulation integer 0 2.15E+09 \u00a0 The number of people that will be populated into the node at the beginning of the simulation. You can scale this number using Base_Population_Scale_Factor in the configuration file. { \"NodeAttributes\" : { \"InitialPopulation\" : 1000 } } Latitude float -3.40282e+038 3.40E+38 \u00a0 Latitude of the node in decimal degrees. This can be used for several things including determining infectiousness by latitude and defining the size of grid cells. { \"NodeAttributes\" : { \"Latitude\" : 12.4 } } Longitude float -3.40282e+038 3.40E+38 \u00a0 Longitude of the node in decimal degrees. This can be used for several things including defining the size of grid cells. { \"NodeAttributes\" : { \"Longitude\" : 9.1 } } Region bool 0 1 0 Set to 1 to indicate that the node is connected by regional migration. { \"NodeAttributes\" : { \"Region\" : 1 } } Seaport bool 0 1 0 Set to 1 to indicate that the node is connected by sea migration. { \"NodeAttributes\" : { \"Seaport\" : 1 } } Urban bool 0 1 0 Set to 1 to enable urban settings. In the configuration file, Enable_Demographics_Other must be set to 1 and Birth_Rate_Dependence must be set to \u201cINDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE\u201d. { \"NodeAttributes\" : { \"Urban\" : 1 } } Zoonosis float 0 1 0 The daily rate of zoonotic infection per individual. If Animal_Reservoir_Type is set to \u201cZOONOSIS_FROM_DEMOGRAPHICS\u201d, the value for Zoonosis overrides the value set for Zoonosis_Rate in the configuration file. { \"NodeAttributes\" : { \"Zoonosis\" : 0 } } Toggle parameter table Parameter Data type Minimum Maximum Default Description Example AbovePoverty float 0 1 1 The proportion of the population that is living above the poverty line. { \"AbovePoverty\" : 0.5 } Airport bool 0 1 0 Set to 1 to indicate that the node is connected by air migration. { \"Airport\" : 0 } Altitude float -3.40282e+038 3.40E+38 0 Required. Only used when Climate_Model is set to \u201cCLIMATE_KOPPEN\u201d. The unit of measurement is in meters. { \"Altitude\" : 250 } BirthRate float 0 1 0.25 The birth rate. In the configuration file, Enable_Birth must be set to 1 and Birth_Rate_Dependence will affect how this rate is used. { \"BirthRate\" : 0.0001 } InitialPopulation integer 0 2.15E+09 \u00a0 The number of people that will be populated into the node at the beginning of the simulation. You can scale this number using Base_Population_Scale_Factor in the configuration file. { \"InitialPopulation\" : 1000 } LarvalHabitatMultiplier float or nested JSON object \u00a0 \u00a0 \u00a0 The value by which to scale the larval habitat availability specified in the configuration file with Larval_Habitat_Types across all habitat types, for specific habitat types, or for specific mosquito species within each habitat type. The following example scales the larval habitat equally across all habitat types and mosquito species. { \"LarvalHabitatMultiplier\" : 2.0 } The following example scales the larval habitat only in the temporary rainfall habitat for all mosquito species. { \"LarvalHabitatMultiplier\" : { \"TEMPORARY_RAINFALL\" : 2.0 } } The following example scales the larval habitat independently for A. gambiae in the temporary rainfall habitat and for A. arabiensis in the brackish swamp habitat. { \"LarvalHabitatMultiplier\" : { \"TEMPORARY_RAINFALL\" : { \"gambiae\" : 2.0 }, \"BRACKISH_SWAMP\" : { \"arabiensis\" : 2.5 } } } Latitude float -3.40282e+038 3.40E+38 \u00a0 Latitude of the node in decimal degrees. This can be used for several things including determining infectiousness by latitude and defining the size of grid cells. { \"Latitude\" : 12.4 } Longitude float -3.40282e+038 3.40E+38 \u00a0 Longitude of the node in decimal degrees. This can be used for several things including defining the size of grid cells. { \"Longitude\" : 9.1 } Region bool 0 1 0 Set to 1 to indicate that the node is connected by regional migration. { \"Region\" : 1 } Seaport bool 0 1 0 Set to 1 to indicate that the node is connected by sea migration. { \"Seaport\" : 1 } Urban bool 0 1 0 Set to 1 to enable urban settings. In the configuration file, Enable_Demographics_Other must be set to 1 and Birth_Rate_Dependence must be set to \u201cINDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE\u201d. { \"Urban\" : 1 } Zoonosis float 0 1 0 The daily rate of zoonotic infection per individual. If Animal_Reservoir_Type is set to \u201cZOONOSIS_FROM_DEMOGRAPHICS\u201d, the value for Zoonosis overrides the value set for Zoonosis_Rate in the configuration file. { \"Zoonosis\" : 0 } IndividualAttributes \u00b6 IndividualAttributes are a JSON object that contains parameters that add or modify the distribution of attributes across individuals in a simulation. For example, the age or immunity distribution. An initial value for an individual is a randomly selected value from a distribution. For example, if you use a uniform distribution to initialize age, the initial ages of individuals in the simulation will be evenly distributed between some minimum and maximum value. There are two types of distributions: simple and complex. Simple distributions are defined by three parameters where one is the distribution type and the other two are used to further define the distribution. You can use complex distributions to define a distribution that does not fit some standard. For example, a complex distribution is useful when trying to represent real-world data. The table below shows the simple distributions and their associated parameters. Toggle parameter table Distribution Demographic parameters Related configuration parameters Age AgeDistributionFlag AgeDistribution1 AgeDistribution2 Age_Initialization_Distribution_Type must be set to \u201cDISTRIBUTION_SIMPLE\u201d. Prevalence PrevalenceDistributionFlag PrevalenceDistribution1 PrevalenceDistribution2 Enable_Demographics_Other must be set to 1. Immunity ImmunityDistributionFlag ImmunityDistribution1 ImmunityDistribution2 \u00a0 Risk RiskDistributionFlag RiskDistribution1 RiskDistribution2 Enable_Demographics_Other must be set to 1. Migration heterogeneity MigrationHeterogeneityDistributionFlag MigrationHeterogeneityDistribution1 MigrationHeterogeneityDistribution2 Enable_Migration_Heterogeneity must be set to 1. The table below shows the available probability distribution types to use with simple distributions and how to set the other two demographic parameters to initialize the distribution. Distribution type Flag value Distribution1 value Distribution2 value Constant 0 Value N/A (set to 0) Uniform 1 Minimum Maximum Gaussian 2 Mean Standard deviation Exponential 3 Mean N/A (set to 0) Poisson 4 Mean N/A (set to 0) Log normal 5 Mean Log (standard deviation) Bimodal 6 Proportion of time to return the value specified by Distribution2. Value must be between 0 and 1. A positive value to be returned. If the value is not positive, a value of 1 is returned. Weibull 7 Scale parameter Shape parameter EMOD also supports the following complex distributions: AgeDistribution FertilityDistribution MortalityDistribution MortalityDistributionMale MortalityDistributionFemale HIVCoinfectionDistribution HIVMortalityDistribution MSP_mean_antibody_distribution nonspec_mean_antibody_distribution PfEMP1_mean_antibody_distribution MSP_variance_antibody_distribution nonspec_variance_antibody_distribution Malaria PfEMP1_variance_antibody_distribution IndividualProperties \u00b6 IndividualProperties are a JSON array that contains parameters that add properties to individuals in a simulation as a means of setting up groups. For example, you can define values for accessibility, age, geography, risk, and other properties and assign individuals to different groups based on those property values. Toggle parameter table Parameter Data type Description Example Property string The property type for which you will assign values to create groups. Accepted values are: Accessibility Geographic Place Risk QualityofCare HasActiveTB Age_Bin { \"IndividualProperties\" : [{ \"Property\" : \"Risk\" }] } Values array An array of the user-defined values that can be assigned to individuals for this property. You can have up to 125 values for the Geographic property type and up to 5 values for all other types. For \u201cAge_Bin\u201d property types, omit and use Age_Bin_In_Years instead. { \"IndividualProperties\" : [{ \"Values\" : [ \"Low\" , \"Medium\" , \"High\" ] }] } Initial_Distribution array An array of numbers that define the proportion of individuals to assign to each group at the beginning of the simulation. Their total must equal 1 and the number of members in this array must match the number of members in Values . For \u201cAge_Bin\u201d property types, omit this parameter as the demographics file controls the age distribution. { \"IndividualProperties\" : [{ \"Initial_Distribution\" : [ 0.2 , 0.4 , 0.4 ] }] } Age_Bin_In_Years array An array of integers that represents the ages, in years, at which to demarcate the age groups. Used only with the \u201cAge_Bin\u201d property type. The first number must be 0, the last must be -1, and they must be listed in ascending order. The following example creates three age groups: 0 to 5, older than 5 to 13, and older than 13 to the maximum age. { \"IndividualProperties\" : [{ \"Age_Bin_In_Years\" : [ 0 , 5 , 13 , -1 ] }] } Transitions array An array contains multiple JSON objects that each define how an individual transitions from one property group to another. See the following table for information about the parameters to include in the object. For \u201cAge_Bin\u201d property types, set to an empty array, as individuals will transition to the next age bin based on the passing of time. { \"IndividualProperties\" : [{ \"Transitions\" : [] }] } TransmissionMatrix array An object that contains Route and Matrix parameters that define how to scale the base infectivity from individuals in one group to individuals in another. Enable_Heterogeneous_Intranode_Transmission in the configuration file must be set to 1 or this parameter is ignored. The HINT feature is used only in the generic simulation type. { \"IndividualProperties\" : [{ \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 10 , 0.1 ], [ 0.1 , 1 ] ] } }] } The following table contains the parameters that are available to use in the Transitions array. Toggle parameter table Parameter Data type Description Example From string Which group an individual will transition from. { \"Transitions\" : [{ \"From\" : \"Low\" }] } To string Which group an individual will transition to. { \"Transitions\" : [{ \"To\" : \"High\" }] } Type string The type of condition that starts transitioning individuals. Set to \u201cAt_Age\u201d or \u201cAt_Timestep\u201d. The other parameters you must set depend on the condition type set here. { \"Transitions\" : [{ \"Type\" : \"At_Age\" }] } Age_In_Years_Restriction nested JSON object The age when an individual is eligible to transition. Min is optional and Max is required. Do not use this with \u201cAt_Age\u201d types because it will conflict with other age parameters in Age_In_Years . This is required for \u201cAt_Timestep\u201d types, though it can be empty for no age restriction. { \"Transitions\" : [{ \"Type\" : \"At_Timestep\" , \"Age_In_Years_Restriction\" : { \"Min\" : 5 , \"Max\" : 40 } }] } Timestep_Restriction nested JSON object The time step when transitioning starts and stops. Required for both \u201cAt_Age\u201d and \u201cAt_Timestep\u201d types. Start is required and End is optional. { \"Transitions\" : [{ \"Timestep_Restriction\" : { \"Start\" : 20 } }] } Coverage float A value between 0 and 1 that indicates the proportion of the population that EMOD will attempt to transition. The calculation used with Probability_Per_Timeset only uses the proportion of the population specified in Cover to calculate the probability of an individual transitioning at a given time step. Coverage has no effect when the value is set to 1. Required for both \u201cAt_Age\u201d and \u201cAt_Timestep\u201d types. { \"Transitions\" : [{ \"Coverage\" : 0.5 }] } Probability_Per_Timestep float A value between 0 and 1 that is used to calculate the probability of an individual transitioning at the current time step given the number of individuals remaining in the group. Required for both \u201cAt_Age\u201d and \u201cAt_Timestep\u201d types. { \"Transitions\" : [{ \"Probability_Per_Timestep\" : 0.25 }] } Age_In_Years integer The age at which individuals are eligible to transition. Do not use for \u201cAt_Timestep\u201d types. Required for \u201cAt_Age\u201d types. { \"Transitions\" : [{ \"Type\" : \"At_Age\" , \"Age_In_Years\" : 10 }] } Timesteps_Until_Reversion integer The number of time steps after the start of transitioning when individuals revert back to their original group. The start of transitioning is specified by the Start parameter in Timestep_Restriction . { \"Transitions\" : [{ \"Timestep_Restriction\" : { \"Start\" : 20 }, \"Timesteps_Until_Reversion\" : 100 }] } Example \u00b6 The example below shows a complete multi-node demographics file that uses a gridded IdReference type. { \"Metadata\" : { \"DateCreated\" : \"03/03/2016\" , \"Tool\" : \"DemographicsWorker\" , \"Author\" : \"Institute for Disease Modeling\" , \"NodeType\" : \"Grid\" , \"ProjectName\" : \"IDM-Madagascar\" , \"RegionName\" : \"Madagascar\" , \"IdReference\" : \"Gridded world grump2.5arcmin\" , \"NodeCount\" : 15 , \"DataProvenance\" : \"Generated by internally-developed demographic tools\" , \"Resolution\" : 150 , \"UpperLatitude\" : -10.6666666666667 , \"LeftLongitude\" : 41.6666666666667 , \"BottomLatitude\" : -27.3333333333333 , \"RightLongitude\" : 51.3333333333333 }, \"Defaults\" : { \"NodeAttributes\" : { \"Altitude\" : 0 , \"Airport\" : 0 , \"Region\" : 1 , \"Seaport\" : 0 , \"BirthRate\" : 0.000102 , \"AbovePoverty\" : 0 }, \"IndividualAttributes\" : { \"AgeDistributionFlag\" : 3 , \"AgeDistribution1\" : 0.00015 , \"AgeDistribution2\" : 0 , \"PrevalenceDistributionFlag\" : 1 , \"PrevalenceDistribution1\" : 0.1 , \"PrevalenceDistribution2\" : 0.2 , \"ImmunityDistributionFlag\" : 0 , \"ImmunityDistribution1\" : 1 , \"ImmunityDistribution2\" : 0 , \"RiskDistributionFlag\" : 0 , \"RiskDistribution1\" : 1 , \"RiskDistribution2\" : 0 , \"MigrationHeterogeneityDistributionFlag\" : 0 , \"MigrationHeterogeneityDistribution1\" : 1 , \"MigrationHeterogeneityDistribution2\" : 0 , \"MortalityDistribution\" : { \"NumDistributionAxes\" : 2 , \"AxisNames\" : [ \"gender\" , \"age\" ], \"AxisUnits\" : [ \"male=0,female=1\" , \"years\" ], \"AxisScaleFactors\" : [ 1 , 365 ], \"NumPopulationGroups\" : [ 2 , 3 ], \"PopulationGroups\" : [ [ 0 , 1 ], [ 0 , 100 , 2000 ] ], \"ResultUnits\" : \"annual deaths per 1000 individuals\" , \"ResultScaleFactor\" : 0.00000273972602739726027397260273973 , \"ResultValues\" : [ [ 0 , 20.0000035 , 400.00007 ], [ 0 , 20.0000035 , 400.00007 ] ] } } }, \"Nodes\" : [{ \"NodeID\" : 358876977 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.1875 , \"Altitude\" : 34 , \"Area\" : 25 , \"InitialPopulation\" : 2088 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358876978 , \"NodeAttributes\" : { \"Latitude\" : -13.2708330154419 , \"Longitude\" : 48.1875 , \"Altitude\" : 19 , \"Area\" : 25 , \"InitialPopulation\" : 1682 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358942512 , \"NodeAttributes\" : { \"Latitude\" : -13.3541669845581 , \"Longitude\" : 48.2291679382324 , \"Altitude\" : 127 , \"Area\" : 25 , \"InitialPopulation\" : 5160 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358942513 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.2291679382324 , \"Altitude\" : 128 , \"Area\" : 25 , \"InitialPopulation\" : 2860 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358942514 , \"NodeAttributes\" : { \"Latitude\" : -13.2708330154419 , \"Longitude\" : 48.2291679382324 , \"Altitude\" : 26 , \"Area\" : 25 , \"InitialPopulation\" : 790 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008047 , \"NodeAttributes\" : { \"Latitude\" : -13.3958330154419 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 48 , \"Area\" : 25 , \"InitialPopulation\" : 6436 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008048 , \"NodeAttributes\" : { \"Latitude\" : -13.3541669845581 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 111 , \"Area\" : 25 , \"InitialPopulation\" : 4983 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008049 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 124 , \"Area\" : 25 , \"InitialPopulation\" : 1245 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008050 , \"NodeAttributes\" : { \"Latitude\" : -13.2708330154419 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 57 , \"Area\" : 25 , \"InitialPopulation\" : 1387 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008051 , \"NodeAttributes\" : { \"Latitude\" : -13.2291669845581 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 66 , \"Area\" : 25 , \"InitialPopulation\" : 1516 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073583 , \"NodeAttributes\" : { \"Latitude\" : -13.3958330154419 , \"Longitude\" : 48.3125 , \"Altitude\" : 132 , \"Area\" : 25 , \"InitialPopulation\" : 5957 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073584 , \"NodeAttributes\" : { \"Latitude\" : -13.3541669845581 , \"Longitude\" : 48.3125 , \"Altitude\" : 44 , \"Area\" : 25 , \"InitialPopulation\" : 643 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073585 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.3125 , \"Altitude\" : 55 , \"Area\" : 25 , \"InitialPopulation\" : 1171 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073587 , \"NodeAttributes\" : { \"Latitude\" : -13.2291669845581 , \"Longitude\" : 48.3125 , \"Altitude\" : 36 , \"Area\" : 25 , \"InitialPopulation\" : 378 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359139117 , \"NodeAttributes\" : { \"Latitude\" : -13.4791669845581 , \"Longitude\" : 48.3541679382324 , \"Altitude\" : 282 , \"Area\" : 25 , \"InitialPopulation\" : 674 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }] }", "tags": "", "title": "Demographics file structure and parameters \u2014 Parameter Reference  documentation"}, {"url": "./search-local.html", "text": "Search results \u00b6", "tags": "", "title": "Search results \u2014 Parameter Reference  documentation"}, {"url": "./search.html", "text": "Please activate JavaScript to enable the search functionality.", "tags": "", "title": "Search \u2014 Parameter Reference  documentation"}]};
