var tipuesearch_info = 'Generated by gti.py on 2017-05-01 15:41:00.751000';
var tipuesearch = {"pages": [{"url": "./generic-simulation-tutorial-overview.html", "text": "Generic simulation tutorials \u00b6 The generic simulation tutorials consist of a group of scenarios that guide you through simulations that will help you understand how to use a configuration file , campaign file , and input data files to run EMOD simulations. Each scenario has a corresponding directory and a demographics file in the ScenariosInputFiles directory so you can run the simulation on your computer. All of these configuration files (config.json) have Simulation_Type set to GENERIC_SIM. There are two ways to use the tutorials. You can read the tutorials information and graphs to learn about the model; installing the Quick Start is not required. Or, for a more hands-on approach, you can also run the simulations that are associated with the tutorials and make the suggested changes to the configuration files to see the impact on the output graphs. The EMOD executable and all of the files needed to run the simulations are included in the Quick Start installation. You can refer to EMOD parameter reference for more information regarding individual parameters that you are interested in modifying. Note These scenarios are for learning purposes only and are not considered scientifically valid. Generic 1: SIR - Simple single-node outbreak Generic 2: SIR - Targeted vaccination campaigns Generic 3: SIER - Incubation periods Generic 4: SEIR - Vital dynamics (birth and death) Generic 5: SIRS - Waning immunity Generic 6: SEIRS - Disease-specific mortality and endemics Generic 7: SI - Lifelong infection Generic 8: SIS - Reoccurring infections Generic 9: Density scaling of infectivity Generic 10: Animal reservoir zoonosis Heterogeneous Intra-Node Transmission (HINT) tutorials HINT 1: Risk groups HINT 2: School closure HINT 3: Age and accessibility HINT 4: Washington state pertussis HINT 5: Seattle commuting", "tags": "", "title": "Generic simulation tutorials \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial-10Zoonosis.html", "text": "Generic 10: Animal reservoir zoonosis \u00b6 Zoonosis represents a list of diseases that transmit from animals to humans. They are of particular interest because typically a zoonosis disease has previously not been in the population, and the whole population is susceptible. This tutorial demonstrates an animal reservoir, independently of human infections, reintroducing zoonotic cases at a specified rate. This tutorial is important because for many zoonotic diseases, there is a background animal reservoir that continuously exposes humans to infection. In these situations, the interest is not only in a single large outbreak which can be simulated by an Outbreak event, but also in the population-level dynamics where there is a continuous background rate of introducing a new zoonotic infection to the human population which can be configured by parameter Zoonosis_Rate . In this tutorial, the rate of infection from the animal reservoir can be adjusted with respect to the human-to-human infectiousness to vary along a spectrum, from a low rate of primary zoonotic infection with substantial secondary human transmission to a high zoonosis rate with low human transmission. This shifts the fraction of cases that are primary zoonotic infections. By varying the parameter Zoonosis_Rate , you can observe the daily rate of zoonotic infection per individual and the impact of the parameter changes on the outbreak. Input files \u00b6 The input files for this tutorial differ from the previous tutorials. The previous tutorials were simple single-node simulations without migration. This tutorial uses Seattle as an example of a multi-node simulation with migration rates. The input files include not a only a human population demographics file but also a zoonosis demographics file and local and regional migration files. Seattle_30arcsec_demographics_zoonosis Seattle_Seattle_30arcsec_local_migration Seattle_Seattle_30arcsec_regional_migration You can view these files at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. For more information on data input files, see Input data files . Key configuration parameters \u00b6 This tutorial uses the config.json file in the <path_to_directory>ScenariosGeneric10_Zoonosis directory. Input file parameters \u00b6 This tutorial uses two demographics files. The first demographics file provides information about the people in the region while the second provides information on zoonosis. Multiple demographics file are listed under one parameter. When there is more than one demographics file, the file names are separated by a semi-colon. This is a multi-node simulation that also supports migration between the nodes. This simulation supports only local and regional migration. The input files for the different types of migration are specified with different parameters. The following parameters specify the demographic and migration input files. For more information, see Input files parameters. Set the following parameters: Set Demographics_Filename to \u201cSeattle_30arcsec_demographics.json;Seattle_30arcsec_demographics_zoonosis.json.\u201d Set Local_Migration__Filename to \u201cSeattle_30arcsec_local_migration.bin.\u201d Set Regional_Migration_Filename to \u201cSeattle_30arcsec_regional_migration.bin.\u201d Demographics parameters \u00b6 The following parameters enable vital dynamics for birth but disable non-disease related death and disease related deaths. For more information, see Demographics parameters. Set the following parameters: Set Enable_Vital_Dynamics to 1. Set Enable_Birth to 1. Set Death_Rate_Dependence to \u201cNONDISEASE_MORTALITY_BY_AGE_AND_GENDER.\u201d Set Enable_Disease_Mortality to 0. Migration input file parameters \u00b6 The following parameters enable or disable the specific type of migration. For more information, see Migration parameters. Set the following parameters: Set Enable_Air_Migration to 0. Set Enable_Local_Migration to 1. Set Enable_Regional_Migration to 1. Set Enable_Sea_Migration to 0. SEIRS model setup parameters \u00b6 This tutorial uses an SEIRS model . The following parameters configure the EMOD executable to run an SEIRS-like disease model with person-to-person direct transmission. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 3. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Enable_Immunity to 1. Set Enable_Immune_Decay to 1. Migration parameters \u00b6 In a multi-node simulation with migration enabled, individuals have a probably of \u201cmigrating\u201d to another node at every time step unless they have already migrated. When an individual has migrated to another node, they can shed or acquire infection in that node. The migration pattern for this simulation is configured as \u201cWAYPOINTS_HOME\u201d with Roundtrip_Waypoints set to 1. This means, individuals \u201cmigrate\u201d one day and return home the next day. There are no intermediate nodes between the home and the destination node. For more information, see Migration parameters. Set the following parameters: Set Migration_Model to \u201cFIXED_RATE_MIGRATION.\u201d Set Migration_Pattern to \u201cWAYPOINTS_HOME.\u201d Set Roundtrip_Waypoints to 1. Disease parameters \u00b6 These parameters configure the EMOD executable to run an SEIRS-like disease model with person- to- person direct transmission. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 0.25. Set Base_Infectious_Period to 7. Set Infectious_Period_Distribution to \u201cFIXED_DURATION.\u201d Zoonosis parameters \u00b6 The following parameters configure the animal zoonosis. The Zoonosis_Rate is multiplied with the Zoonosis rate in the demographic file. For more information, see General disease parameters. Set the following parameters: Set Animal_Reservoir_Type to \u201cZOONOSIS_FROM_DEMOGRAPHICS.\u201d Set Zoonosis_Rate to 0.005. Immunity parameters \u00b6 At the time of recovery when the infection is cleared, the values of immunity against acquisition, transmission, and mortality are multiplied by Immunity_Acquisition_Factor , Immunity_Transmission_Factor , and Immunity_Mortality_Factor parameters. The default values are set to 0 in the files, which means full and complete protection against each. For more information, see General disease parameters. In this scenario, people\u2019s immunity decays with an average half-life of 33 days and a 400-day offset. This assumption applies to transmission blocking, acquisition blocking, and mortality blocking immunity. Set the following parameters: Set Acquisition_Blocking_Immunity_Duration_Before_Decay to 400. Set Acquisition_Blocking_Immunity_Decay_Rate to 0.03. Set Transmission_Blocking_Immunity_Duration_Before_Decay to 400. Set Transmission_Blocking_Immunity_Decay_Rate to 0.03. Set Immunity_Acquisition_Factor to 0. Set Immunity_Mortality_Factor to 0. Set Immunity_Transmission_Factor to 0. Set Mortality_Blocking_Immunity_Duration_Before_Decay to 400. Set Mortality_Blocking_Immunity_Decay_Rate to 0.03. Intervention \u00b6 This simulation uses an Outbreak event for repeated infection seeding. This simulation uses the campaign.json in the <path_to_directory>ScenariosGeneric10_Zoonosis directory. For more information, see Outbreak parameters. { \"Use_Defaults\" : 1 , \"Campaign_Name\" : \"Generic Seattle Regression Campaign\" , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"ImportCases\" , \"class\" : \"Outbreak\" }, \"Number_Distributions\" : 10 , \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"Polygon_Format\" : \"SHAPE\" , \"Vertices\" : \"-122.320726936496,47.6597902588541 -122.320406460197,47.6520988276782 -122.308388598985,47.6471314450438 -122.301658596706,47.6541819236217 -122.315118601264,47.6644371651896 -122.320726936496,47.6597902588541\" , \"class\" : \"NodeSetPolygon\" }, \"Start_Day\" : 40 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 After the initial outbreak, there are no additional imported cases set in the campaign file. However, because of the non-zero zoonosis rate, there are several introductions of infections in the background, as shown in the Log Prevalence chart in Figure 1. When enough susceptible individuals are accumulated due to immunity waning, one of the seeds starts a disease outbreak later due to zoonosis (see Figure 1 and Figure 2). The seeding and occasional outbreaks after the seeding is shown in the following graph plotting infections over time. Figure 1: Animal reservoir zoonosis charts Figure 2: Re-seeding of infections cased by zoonosis and waning immunity Exploring the model \u00b6 Change the Zoonosis_Rate in the configuration file to see the changing frequency of outbreaks.", "tags": "", "title": "Generic 10: Animal reservoir zoonosis \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial1SIR.html", "text": "Generic 1: SIR - Simple single-node outbreak \u00b6 This first tutorial is a simple SIR epidemic in a single- node , village-scale location. The tutorial describes the differences between the stochastic EMOD model and the deterministic SIR model and demonstrates how the Epidemiological MODeling software (EMOD) is configured to simulate an SIR epidemic. After running the base simulation of the SIR epidemic, you can vary the parameters for disease infectivity and infection duration to see the effect on the height of the epidemic peak, the time to reach the peak, and the cumulative number of disease cases. The SIR model \u00b6 The simplest way to model epidemic spread in populations is to classify people into different population groups or compartments. One of the classic but still widely used models divides people into three groups: S - susceptible, I - infected/infectious, and R - recovered/immune. This model was first used by Kermack and McKendrick in 1927 and has subsequently been applied to a variety of diseases, especially airborne childhood diseases with life-long immunity upon recovery, such as measles, mumps, rubella, and pertussis. S, I and R represent the number of susceptible,infected, and recovered individuals, and N = S + I + R is the total population. The infectious rate \u03b2 controls the rate of spread which represents the probability of transmitting disease between a susceptible and an infectious individual. Recovery rate, \u03b3 = 1/D, is determined by the average duration, D, of infection. If the course of the infection is short (emergent outbreak) compared with the life time of an individual, and the disease is non-fatal, vital dynamics (birth and death) can be ignored. In the deterministic form, the SIR model can be written as follows: It is challenging to derive exact analytical solutions of the previous equations because of the non- linear dynamics. However, the key metrics that control the spread can be derived. At the initial seeding of the infection, the following condition needs to be satisfied for a disease to spread: If the number of infections at the initial stage is small and the initial R = 0 then S is close to N and the condition becomes: where \u03b2/\u03b3 is named the reproductive number (R 0 ). R 0 is the average number of secondary cases generated by an index case in a fully susceptible population. The disease will spread in the population when R 0 > 1 and will die out if R 0 < 1. SIR and EMOD model comparison \u00b6 The EMOD model is a discrete and stochastic version of the SIR model with state changes occurring at fixed time steps and an exponentially distributed duration of infection. The discrete form of the previous equation at each time step from to can be written as: The infection and recovery process can be represented as probabilistic binomial draws where: The total rate of contagion shedding from all infected individuals (the force of infection): The probability of infection: Probability of recovery: Therefore, Because the EMOD model is stochastic and discrete, the accuracy of the result will depend on \u03b4 which is set with the parameter Simulation_Timestep . By default, \u03b4 = 1 which is a day but you can choose a smaller time step in order to get more accurate results. One of the key the differences between stochastic and deterministic systems is the value of R 0 . A ordinary differential equation (ODE) model will never predict an outbreak when R 0 < 1, especially when R 0 is close to 1. In stochastic simulations it is possible to see outbreaks. For more information on the mathematics of the EMOD model, see Heterogeneity and transmission scaling in the generic documentation set. Demographic inputs \u00b6 The demographic file generic_scenarios_demographics, defines a single node with 10,000 individuals and has a fixed birth and death rate at 20 per 1000 individual per year (0.0000548 per day). This demographics file is used by this tutorial and several of the other Generic simulation tutorials. { \"NodeID\" : 1 , \"NodeAttributes\" : { \"InitialPopulation\" : 10000 , \"BirthRate\" : 0.0000548 }, \"IndividualAttributes\" : { \"MortalityDistribution\" : { \"NumDistributionAxes\" : 2 , \"AxisNames\" : [ \"gender\" , \"age\" ], \"AxisUnits\" : [ \"male=0,female=1\" , \"years\" ], \"AxisScaleFactors\" : [ 1 , 365 ], \"NumPopulationGroups\" : [ 2 , 1 ], \"PopulationGroups\" : [ [ 0 , 1 ], [ 0 ] ], \"ResultUnits\" : \"deaths per day\" , \"ResultScaleFactor\" : 1 , \"ResultValues\" : [ [ 0.0000548 ], [ 0.0000548 ] ] } } } For some of the Generic tutorials that use generic_scenarios_demographics , including this tutorial, the birth and death rate needs to be 0. In this case, the birth and death process is disabled in the configuration file (config.json) by setting the parameter Enable_Vital_Dynamics to 0. You can view the demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where either the Quick Start was installed. For more information on demographics files, see Input data files . Key configuration parameters \u00b6 You can view the complete config.json at <path_to_directory>ScenariosGeneric Scenarios01_SIR directory. Demographics parameters \u00b6 Set Enable_Vital_Dyanamics to 0. This will disable vital dynamics (birth and death) in the simulation. SIR model setup parameters \u00b6 The EMOD generic simulation uses an SEIR-like disease model by default. You can modify the default SEIR model to an SIR model by turning off the incubation period and immune decay in the config.json file. For more information, see General disease parameters. Set the following parameters: Set Siumulation_Type to \u201cGENERIC SIM.\u201d Set Base_Incubation_Period to 0. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Enable_Immune_Decay to 0. Disease parameters \u00b6 As demonstrated above, the spread of disease in the SIR model depends on the basic reproductive number R 0 =\u03b2/\u03b3 . In this simulation, the average infection duration is 4 days, and R 0 =14 (measles-like). An exponentially distributed period is used to match the behavior of ODE. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 3.5. Set Base_Infectious_Period to 4. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Simulation duration parameters \u00b6 The simulation is run for 3 months with a 1-day time step. For more information, see Simulation setup parameters. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 90. Interventions \u00b6 You can define interventions and events for a simulation through a JavaScript Object Notation (JSON) formatted campaign file, typically called campaign.json. Interventions can have either a positive effect on a simulation (for example, a vaccination campaign) or a negative impact (for example, an outbreak). For this simulation, the intervention is an outbreak. Initial outbreak \u00b6 This simulation uses an OutbreakIndividual event at day 1 as the initial seeding of the infection. For more information, see OutbreakIndividual parameters. { \"Use_Defaults\" : 1 , \"Campaign_Name\" : \"Initial Seeding\" , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" }] } You can view the complete campaign.json in <path_to_directory>ScenariosGeneric Scenarios01_SIR directory. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 Figure 1: growth of infection and depletion of the susceptible population is a graph of the complete course of the outbreak. After the initial fast growth, the epidemic depletes the susceptible population. Eventually the virus cannot find enough new susceptible people and dies out. Figure 1: Growth of infection and depletion of the susceptible population Figure 2: All inset charts Exploring the model \u00b6 The EMOD model is individual-based, which allows the implementation of a flexible distribution. In an ODE model the time constants are exponentially distributed, however, this is not case for some diseases. You can test the effects of different distributions in the EMOD executable by changing Infectious_Period_Distribution in the configuration file. Set the value of Infectious_Period_Distribution to \u201cFIXED_DURATION\u201d and let all other parameters remain the same. The following output shows a much faster spread of disease when using a fixed duration for the infection period. Figure 3: Infectious_Period_Distribution is FIXED_DURATION You can also try the same SIR model with different basic reproductive number values. For example, run the simulation with R 0 equal to 0.2 and 2.0. Simulation with R 0 =0.2 : Keep Base_Infectious_Period equal to 4 and change Base_Infectivity to 0.05. Simulation with R 0 =2.0 : Keep Base_Infectious_Period equal to 4 and change Base_Infectivity to 0.5. The following images shows the simulation results with R 0 equal to 0.2 and 2.0. Figure 4: R 0 =0.2 - No Outbreak ( Base_Infectivity = 0.05) Figure 5: R 0 =2.0 - Outbreak ( Base_Infectivity = 0.5)", "tags": "", "title": "Generic 1: SIR - Simple single-node outbreak \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial2SIR.html", "text": "Generic 2: SIR - Targeted vaccination campaigns \u00b6 This tutorial assumes that you have read Generic 1: SIR - Simple single-node outbreak . This tutorial implements vaccination strategies to control an SIR epidemic in a single-node, village-scale location. The tutorial demonstrates how to configure vaccination interventions and the effect of those vaccine interventions on herd immunity. The tutorial also demonstrates how to target vaccines to heterogeneous groups within the node. By default, the EMOD executable assumes every individual has the same probability of being selected for vaccination. However, it is possible to model certain groups with low or no vaccination coverage through the use of IndividualProperties . IndividualProperties is added to the demographics file and configured to create the groups. For more information see IndividualProperties. This tutorial consists of three scenarios: Baseline outbreak Vaccination campaigns all individuals Vaccination campaigns targeted groups Vaccination and herd immunity \u00b6 In Generic 1: SIR - Simple single-node outbreak , the SIR model assumed all individuals are susceptible before the importation of an outbreak. However, vaccination is the one of the most effective ways to prevent or contain a disease outbreak. This scenario demonstrates the effect of vaccinations. Starting from the SIR equations discussed in the previous tutorials: For an outbreak to start, the following condition needs to be satisfied: If a vaccination campaign with coverage p [0, 1] is performed before the outbreak, only a fraction of susceptible people move to the recovered compartment. If the vaccine take is e [0, 1], representing the probability of being protected after receiving a vaccine dose, the fraction of immune people due to vaccination is ep . Therefore, the previous condition can be reduced to: R eff is also called the effective reproductive number. Vaccination can reduce the disease\u2019s ability to spread, and the outbreak can be prevented or stopped with less than 100% coverage. When R eff = 1, there exists a minimum vaccination coverage that can prevent a disease outbreak. This minimum coverage required to prevent an outbreak is usually called herd immunity (represented as P herd ). Therefore, the analytical form of P herd can be derived based on the previous condition: Multiple rounds of vaccinations \u00b6 If multiple vaccine interventions are performed in the same area, people are selected on a binomial basis and all individuals have the same probability of being included. If the vaccine coverage is p, every individual has a probability p of being selected in a given round. After the first round, the fraction of non-vaccinated is 1- p , and after n rounds this fraction is (1 - p) n . Therefore, the fraction of people getting at least 1 dose is: For example, the fraction of people getting at least one vaccination with a 50% campaign coverage is shown in the following table. After a few rounds, the coverage increases significantly with the number of campaigns. Scenario B demonstrates this example. Number of Campaigns Covered Fraction (>=1 dose) 1 50% 2 75% 3 87.5% 4 93.75% 5 96.875% However, this is usually not the case. In recent polio eradication campaigns, the number of supplemental immunization activity (SIA) campaigns planned in certain high-risk districts is greater than 6 per year, but poliovirus still persists in the area due to certain groups that are chronically missed. This can be caused by low accessibility, exclusion from SIA microplans, or vaccine refusal. R eff has not been driven below 1 because these susceptible people in the chronically missed groups are still in contact with the rest of population. In some modeling simulations, this assumption has to be included. In this second example, the same 50% campaign coverage is repeated so that the same amount of vaccine is used. However, 30% of the population is not accessible to any vaccine campaigns. Although the number of vaccine doses used is the same as the previous example, the overall coverage is much less. The scenario C demonstrates this example. Number of Campaigns Covered Fraction of Total Population Covered Fraction of Groups with Access Covered Fraction of Groups without Access 1 50% 71.43% 0% 2 29% 91.84% 0% 3 68.37% 97.67% 0% 4 69.53% 99.33% 0% 5 69.87% 99.81% 0% Demographics inputs \u00b6 This simulation uses the demographics file \u201cgeneric_scenarios_individualproperties_demographics\u201d. Except for an IndividualProperties structure, the demographics file is the same as \u201cgeneric_scenarios_demographics\u201d used in Generic 1: SIR - Simple single-node outbreak . Individual Properties \u00b6 An IndividualProperties structure is added to Defaults in the demographics file. Its purpose for this scenario is to divide individuals into two groups so the individuals in different groups can be configured differently. In the third scenario, a vaccine campaign will be targeted at only one of the groups but not the other group. This models the situation where individuals in the general population have a chance of receiving a vaccine while the other individuals are in the chronically missed group. { \"Metadata\" : { }, \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Accessibility\" , \"Values\" : [ \"Easy\" , \"Hard\" ], \"Initial_Distribution\" : [ 0.7 , 0.3 ], \"Transitions\" : [] }] }, \"Nodes\" : [{ }] } Property and values \u00b6 The IndividualProperties structure creates groups using the Property and Values parameters. The Property parameter specifies the property type. This tutorial uses the property type Accessibility . The Values array defines the groups. Each element in the array is a group. In this tutorial, the groups are \u201cEasy\u201d and \u201cHard\u201d. They represent the group that is easy to access during a vaccine campaign (Easy), and the chronically missed group (Hard). Note Property and Values creates the groups. However, the different behavior of the two groups must be configured with parameters in the campaign files. Initial_Distribution \u00b6 At the beginning of the simulation individuals are assigned a group using the probabilities in the Initial_Distribution array. The values in the Initial_Distribution array must be in the same order as the groups in the Values array. At the beginning of the simulation, each individual has a 70% of belonging to the \u201cEasy\u201d accessibility group and a 30% chance of belonging to the \u201cHard\u201d accessibility group as specified by the Initial_Distribution array. { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Accessibility\" , \"Values\" : [ \"Easy\" , \"Hard\" ], \"Initial_Distribution\" : [ 0.7 , 0.3 ], \"Transitions\" : [] }] } } Note Transitions is not used in this tutorial but it must be include as an empty array. When the simulation is run, it will create an empty Transitions output file which can be ignored. Note that this tutorial only uses IndividualProperties to divide the population into groups. In more complex simulations, IndividualProperties can be used to differentiate the disease transmission between multiple groups within a node. This is described in Heterogeneous Intra-Node Transmission (HINT) tutorials . Key configuration parameters \u00b6 All three of the scenarios in this tutorial use the same simulation set up. Please note each subdirectory has a copy of the config.json file. You can view the complete config.json in the following directories: <path_to_directory>ScenariosGeneric Scenarios02_SIR_VaccinationsA_BaselineOutbreak <path_to_directory>ScenariosGeneric Scenarios02_SIR_VaccinationsB_VaccinationAll <path_to_directory>ScenariosGeneric Scenarios02_SIR_VaccinationsC_TargetedVaccination Demographics parameters \u00b6 Set Enable_Vital_Dyanamics to 0. This will disable vital dynamics (birth and death) in the simulation. For more information, see Demographics parameters. SIR model setup parameters \u00b6 The generic simulation uses an SEIR-like disease model by default. You can modify the default SEIR model to an SIR model by turning off the incubation period and immune decay in the config.json file. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 0. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Enable_Immune_Decay to 0. Disease parameters \u00b6 In the following example, we assume the average infection duration is 7 days, and R 0 = 4.9. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 7. Set Base_Infectious_Period to 0.7. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DISTRIBUTION.\u201d Timestep and duration parameters \u00b6 For more information, see Simulation setup parameters. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 120. Interventions \u00b6 The three scenarios have different campaign.json files. A. Baseline outbreak \u00b6 This scenario uses an OutbreakIndividual event at day 30 as the initial seeding of the infection. For more information, see OutbreakIndividual parameters. { \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 30 , \"class\" : \"CampaignEvent\" } You can view the complete campaign.json at <path_to_directory>ScenariosGeneric Scenarios02_SIR_VaccinationsA_BaselineOutbreak. B. Vaccinations for all individuals \u00b6 The following campaign event has three vaccination campaigns that are 7 days apart. The campaigns have a 100% vaccine take rate and a 50% per-campaign coverage. The vaccine is acquisition-blocking and full immunity lasts for 10 years. For more information, see SimpleVaccine parameters. Set the following parameters: Set Number_Repetitions to 3. Set Timesteps_Between_Repetitions to 7. Set Vaccine_Take to 1.0. Set Demographic_Coverage to 3. Set Vaccine_Type to \u201cAcquistionBlocking\u201d. Set Simulation_Duration to 120. Set Primar_Decay_Time_Constant to 3650. You can also view campaign.json at <path_to_directory>ScenariosGeneric Scenarios02_SIR_VaccinationsB_Vaccinations. { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.5 , \"Intervention_Config\" : { \"Cost_To_Consumer\" : 10.0 , \"Durability_Time_Profile\" : \"BOXDURABILITY\" , \"Primary_Decay_Time_Constant\" : 3650 , \"Reduced_Acquire\" : 1 , \"Reduced_Transmit\" : 0 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"class\" : \"SimpleVaccine\" }, \"Number_Repetitions\" : 3 , \"Timesteps_Between_Repetitions\" : 7 , \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" } C. Vaccination campaigns for targeted groups \u00b6 The following campaign event is the same as the campaign event in scenario B, but it restricts the vaccine campaign to the \u201cEasy\u201d accessibility group by adding a Property_Restrictions parameter and setting it to \u201cAccessibility\u201d: \u201cEasy\u201d. As only 70% of the population is in the \u201cEasy\u201d accessibility group, vaccine coverage is also reduced. { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.5 , \"Intervention_Config\" : { \"Cost_To_Consumer\" : 10.0 , \"Durability_Time_Profile\" : \"BOXDURABILITY\" , \"Primary_Decay_Time_Constant\" : 3650 , \"Reduced_Acquire\" : 1 , \"Reduced_Transmit\" : 0 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"class\" : \"SimpleVaccine\" }, \"Number_Repetitions\" : 3 , \"Timesteps_Between_Repetitions\" : 7 , \"Target_Demographic\" : \"Everyone\" , \"Property_Restrictions\" : [ \"Accessibility:Easy\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" } You can view the complete campaign.json at <path_to_directory>ScenariosGeneric Scenarios02_SIR_VaccinationsC_TargetedVaccinations. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. A. Baseline outbreak \u00b6 Figure 1: Outbreak without vaccination campaigns shows the complete trajectory of the outbreak without vaccination campaigns. Figure 1: Outbreak without vaccination campaigns Figure 2: Outbreak without vaccination campaigns - property report B. Vaccinations for all individuals \u00b6 With three campaigns randomly administered, the fraction of immune people is above the herd immunity (in this case, close to 80%), therefore, an outbreak will not start. Figure 3: Outbreak with vaccination campaigns Figure 4: Outbreak with vaccination campaigns - property report C. Vaccination campaigns for targeted groups \u00b6 With three campaigns administered only to the easily accessible group, the fraction of immune people is never above the group size (70%). If there is an outbreak, the group will not reach the herd immunity no matter how many vaccine campaigns are done. Figure 5: Outbreak with targeted vaccination campaigns Figure 6: Outbreak with targeted vaccination campaigns - property report Exploring the model \u00b6 Starting with the campaign.json file in scenario B, change one of the single vaccination intervention by varying the vaccination coverage and observe the herd immunity.", "tags": "", "title": "Generic 2: SIR - Targeted vaccination campaigns \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial3SEIR.html", "text": "Generic 3: SIER - Incubation periods \u00b6 This tutorial assumes that you have read Generic 1: SIR - Simple single-node outbreak and Generic 2: SIR - Targeted vaccination campaigns . The SIR model in Generic 1: SIR - Simple single-node outbreak has several limitations that might not be true for some diseases. Many diseases have a latent phase during which the individual is infected but not yet infectious. This delay between the acquisition of infection and the infectious state can be incorporated within the existing model by adding a latent/exposed population, E, and letting infected (but not yet infectious) individuals move from S to E and from E to I. If is the rate of latent individuals becoming infectious (average duration of latency is 1/ ), the SEIR model becomes: with Since the latency delays the start of the individual\u2019s infectious period, the secondary spread from an infected individual will occur at a later time compared with an SIR model which has no latency. Therefore, including a longer latency period will result in slower initial growth of the outbreak. However, since the model does not include mortality, the basic reproductive number , R 0 = , does not change. Demographics inputs \u00b6 This tutorial uses generic_scenarios_demographics file for demographics input, as do all of the generic simulation scenarios except for Generic 2: SIR - Targeted vaccination campaigns , and Generic 10: Animal reservoir zoonosis . You can view the complete demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where you installed the Quick Start. For more information on demographics files, see Demographics file structure and parameters . Key configuration parameters \u00b6 This tutorial uses the config.json in <path_to_directory>ScenariosGeneric Scenarios03_SEIR directory. Demographics parameters \u00b6 Set Enable_Vital_Dyanamics to 0. This will disable vital dynamics (birth and death) in the simulation. For more information, see Demographics parameters. SEIR model setup parameters \u00b6 The incubation period, infection period, and immunity must be enabled for an SEIR model . Immunity waning must be disabled. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 8. Set Enable_Immunity to 1. Set Enable_Immune_Decay to 0. Set Incubation_Period_Distribution to \u201cEXPONENTIAL_DISTRIBUTION.\u201d Disease parameters \u00b6 In the following example, we still use the same parameters as in the SIR model: the average infectious duration is 4 days; and R 0 = 14 (measles-like). However, this time we will add a latent period of 8 days. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 3.5. Set Base_Infectious_Period to 4. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Time step and duration parameters \u00b6 For more information, see Simulation setup parameters. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 90. Interventions \u00b6 This tutorial uses the campaign.json file in the <path_to_directory>ScenariosGeneric Scenarios03_SEIR directory. Initial outbreak \u00b6 This simulation uses an OutbreakIndividual event at day 1 to create the initial seeding. For more information, see OutbreakIndividual parameters. { \"Use_Defaults\" : 1 , \"Campaign_Name\" : \"Initial Seeding\" , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 The complete course of outbreak is observed. After the initial fast growth, the epidemic depletes the susceptible population. Eventually the virus cannot find enough new susceptible people and dies out. Although R 0 and the cumulative number of infected individuals does not change, including latency slows down the growth of the epidemic. Figure 1: Depletion of the susceptible population Exploring the model \u00b6 Change the latent period to observe the initial growth. The following figure shows the results of changing to a shorter latent period (2 days). Figure 2: Impact of the length of the latent period", "tags": "", "title": "Generic 3: SIER - Incubation periods \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial4SEIR.html", "text": "Generic 4: SEIR - Vital dynamics (birth and death) \u00b6 This tutorial assumes that you have read Generic 3: SIER - Incubation periods . This tutorial explores the effect of birth and death on a simulation. In a closed population with no vital dynamics, an epidemic will eventually die out due to an insufficient number of susceptible individuals to sustain the disease. Infected individuals who are added later will not start another epidemic due to the life-long immunity of the existing population. If there are vital dynamics, new births can provide more susceptible individuals to the population. In this case, when infected individuals are added later, another epidemic may start. Demographics inputs \u00b6 In the demographics input file, the node has 10,000 individuals. The birth rate and death rate are equal to 20 per 1000 individuals per year (0.0000548 per day). The birth rate and initial population values are set in NodeAttributes using BirthRate and InitialPopulation , respectively. The death rate is set in IndividualAttributes using the MortalityDistribution . Both the male and female death rate is equal to 0.0000548 per day. { \"Metadata\" : { }, \"Nodes\" : [{ \"NodeID\" : 1 , \"NodeAttributes\" : { \"InitialPopulation\" : 10000 , \"BirthRate\" : 0.001 }, \"IndividualAttributes\" : { \"MortalityDistribution\" : { \"NumDistributionAxes\" : 2 , \"AxisNames\" : [ \"gender\" , \"age\" ], \"AxisUnits\" : [ \"male=0,female=1\" , \"years\" ], \"AxisScaleFactors\" : [ 1 , 365 ], \"NumPopulationGroups\" : [ 2 , 1 ], \"PopulationGroups\" : [ [ 0 , 1 ], [ 0 ] ], \"ResultUnits\" : \"deaths per day\" , \"ResultScaleFactor\" : 1 , \"ResultValues\" : [ [ 0.0000548 ], [ 0.0000548 ] ] } } }] } To view the complete demographics file, see generic_scenarios_demographics at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. For more information on demographics files, see Demographics file structure and parameters . Key simulation parameters \u00b6 This tutorial simulates the person-person disease transmission as an SEIR model in a hypothetical community over a longer period. You can view the complete config.json in <path_to_directory>ScenariosGeneric04_VitalDynamics directory. Demographics parameters \u00b6 The following parameters enable vital dynamics for birth and non-disease related deaths, and disable disease-related mortality. For more information, see Demographics parameters. Set the following parameters: Set Enable_Vital_Dyanamics to 1. Set Enable_Birth to 1. Set Death_Rate_Dependence to \u201cNONDISEASE_MORTALITY_BY_AGE_AND_GENDER.\u201d Set Enable_Disease_Mortality to 0. SEIR model setup parameters \u00b6 The incubation period, infection period, and immunity must be enabled for an SEIR model. Immunity waning must be disabled. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 3. Set Enable_Immunity to 1. Set Enable_Immune_Decay to 0. Set Incubation_Period_Distribution to \u201cEXPONENTIAL_DISTRIBUTION.\u201d Disease parameters \u00b6 These parameters configure an SEIR-like disease model with person-to-person direct transmission. The basic reproductive number is 17.5. The birth and death rates are calculated based on the base birth and mortality rates in the demographics file, as well as the multiplier in the configuration file. The birth and death (mortality) rates are increased by a factor of 10 using multipliers, x_Birth and x_Other_Mortality , to show the effect of the vital process. In this tutorial, the birth rate equals the mortality rate. For more information, see General disease and Scaling factors parameters. Set the following parameters: Set Base_Infectivity to 2.5. Set Base_Infectious_Period to 7. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Set x_Birth to 10. Set x_Other_Mortality to 10. Simulation duration parameters \u00b6 The simulation runs a 10 year simulation with a 1-day time step. For more information, see Simulation setup parameters. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 3650. Interventions \u00b6 Prior immunity \u00b6 This tutorial assumes that 80% of individuals are immune at the beginning of the simulation. This is simulated using a vaccination campaign with 80% coverage and 100% take at day 1. { \"Campaign_Name\" : \"Pre-outbreak Immunity\" , \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.8 , \"Intervention_Config\" : { \"Cost_To_Consumer\" : 10.0 , \"Durability_Time_Profile\" : \"BOXDURABILITY\" , \"Primary_Decay_Time_Constant\" : 3650 , \"Reduced_Acquire\" : 1 , \"Reduced_Transmit\" : 0 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"class\" : \"SimpleVaccine\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" } Outbreaks \u00b6 This simulation uses an OutbreakIndividual event for an infrequent infection seeding. The Timesteps_Between_Repetitions and Number_Repetitions parameters are set to reintroduce the outbreak once every two years for 10 years. With larger population sizes, the endemic can sustain itself. For more information, see OutbreakIndividual parameters. { \"Campaign_Name\" : \"Outbreak\" , \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.001 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Event_Name\" : \"Outbreak\" , \"Genome\" : 0 , \"Outbreak_Source\" : \"ImportCases\" , \"class\" : \"OutbreakIndividual\" }, \"Number_Repetitions\" : 10 , \"Target_Demographic\" : \"Everyone\" , \"Timesteps_Between_Repetitions\" : 730 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 5 , \"class\" : \"CampaignEvent\" } You can view the complete campaign.json in <path_to_directory>ScenariosGeneric04_VitalDynamics directory. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 The output shows the periodic outbreaks when an infection is reseeded because of the accumulation of new susceptibles due to the vital process (new births). Since periodic endemic behavior cannot be maintained due to stochastic fadeout in a small population, OutbreakIndividual was used to reintroduce the outbreak every two years for 10 years. Figure 1: Periodic outbreaks on reintroduction Figure 2: Periodic outbreaks on reintroduction - all channels Exploring the model \u00b6 Shut down the vital dynamics by setting Enable_Vital_Dynamics to 0 in the ScenariosGeneric04_SEIR_VitalDynamicsconfig.json file. Eventually, the number of susceptible individuals will be exhausted and a reintroduction of the outbreak will not cause a new outbreak. Experiment with increasing the birth rate and population size in the generic_scenarios_demographics file. At a certain point, the influx of population might be enough to sustain the dynamics without reintroducing an outbreak.", "tags": "", "title": "Generic 4: SEIR - Vital dynamics (birth and death) \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial5SIRS.html", "text": "Generic 5: SIRS - Waning immunity \u00b6 This tutorial assumes that you have read Generic 4: SEIR - Vital dynamics (birth and death) . The SIR model assumes people carry lifelong immunity to a disease upon recovery; and this is the case for a variety of diseases. For another class of airborne diseases, for example seasonal influenza, an individual\u2019s immunity may wane over time. In this case, the SIRS model is used allow recovered individuals return to a susceptible state. Specifically, is the rate which recovered individuals return to the susceptible statue due to loss of immunity. If there is sufficient influx to the susceptible population, at equilibrium the dynamics will be in an endemic state with damped oscillation. The EMOD executable (Eradication.exe) simulates waning immunity by a delayed exponential distribution. Individuals stay immune for a certain period of time then immunity wanes following an exponential distribution. Because the shedding and acquisition process is separate, two kinds of immunity can be waned: transmission and acquisition immunity. Demographic inputs \u00b6 This tutorial uses generic_scenarios_demographics file for demographics input. You can view the demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where either the Quick Start was installed. For more information on demographics files, see Demographics file structure and parameters . Key configuration parameters \u00b6 You can view the complete config.json in the ScenariosGeneric05_SIRS directory. Demographics parameters \u00b6 Set Enable_Vital_Dyanamics to 0. This will disable vital dynamics (birth and death) in the simulation. For more information, see Demographics parameters. SIRS model setup parameters \u00b6 The model setup is similar to an SIR model. The immunity decay is enabled so that recovered individuals can return to susceptible at a later time. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 0. Set Enable_Immunity to 1. Set Enable_Immune_Decay to 1. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Disease parameters \u00b6 In the following example, the average infection duration is 4 days, and R 0 = 1.5 which is similar to influenza. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 0.25. Set Base_Infectious_Period to 7. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Immunity parameters \u00b6 There are three immunity factor parameters: Immunity_Acquisition_Factor , Immunity_Transmission_Factor , and Immunity_Mortality_Factor . At the time of recovery when the infection is cleared, the values of immunity against acquisition, transmission, and mortality are multiplied by (1- the value of the corresponding immunity factor parameter). For example, the multiplier for Acquisition_Blocking_Immunity_Decay_Rate is Immunity_Acquisition_Factor and the multiplicative reduction in the probability of reacquiring the disease is: Acquisition_Blocking_Immunity_Decay_Rate x (1 - Immunity_Acquisition_Factor ) In this simulation, individual\u2019s immunity decays with an average half-life of 180 days and a 90-day offset. This applies to both transmission blocking and acquisition blocking immunity. This tutorial does not use the Mortality_Blocking_Immunity_Decay_Rate , however, the next tutorial, Generic 6: SEIRS - Disease-specific mortality and endemics , does. For more information, see General disease parameters. Set the following parameters: Set Acquisition_Blocking_Immunity_Duration_Before_Decay to 90. Set Transmission_Blocking_Immunity_Duration_Before_Decay to 90. Set Acquisition_Blocking_Immunity_Decay_Rate to 0.0056. Set Transmission_Blocking_Immunity_Decay_Rate to 0.0056. Set Immunity_Acquisition_Factor to 0. Set Immunity_Mortality_Factor to 0. Set Immunity_Transmission_Factor to 0. Simulation duration parameters \u00b6 The simulation is run for 10 years with a 1-day time step. For more information, see Simulation setup parameters. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 3650. Interventions \u00b6 This simulation uses the campaign.json file in the ScenariosGeneric05_SIRS directory. Outbreak with re-seeding \u00b6 This simulation uses an OutbreakIndividual event for an infrequent infection seeding. The Timesteps_Between_Repetitions and Number_Repetitions parameters are set to reintroduce the outbreak once every 180 days for 10 years, to allow reintroduction of disease. For more information, see OutbreakIndividual parameters. { \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"Number_Repetitions\" : 20 , \"Timesteps_Between_Repetitions\" : 180 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 Figure 1 shows damped oscillation due to people losing immunity and becoming susceptible again. Note Individuals who are susceptible people due to waning immunity are not classified as susceptible in the simulation. They are reported under the \u201cWane Population\u201d channel. Figure 1: Simulation output - damped oscillation Figure 2: Simulation output - all channels Exploring the model \u00b6 Change the waning immunity distribution by modifying the number of days until immunity starts to decay and the immunity decay rate. For example, to represent a step-like function: Set the Acquisition_Blocking_Immunity_Duration_Before_Decay and Transmission_Blocking_Immunity_Duration_Before_Decay to 180. Set Acquisition_Blocking_Immunity_Duration_Before_Decay and Transmission_Blocking_Immunity_Duration_Before_Decay to 1.", "tags": "", "title": "Generic 5: SIRS - Waning immunity \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial6SEIRS.html", "text": "Generic 6: SEIRS - Disease-specific mortality and endemics \u00b6 This tutorial assumes that you have read Generic 5: SIRS - Waning immunity . This tutorial is a simple SEIRS model in a single-node, village-scale location. Similar to the SIRS model, recovered individuals transition to susceptible individuals by enabling waning immunity. As in a previous tutorial, Generic 4: SEIR - Vital dynamics (birth and death) , the vital dynamics are enabled. This scenario also enables the disease-specific mortality rate. By varying the disease infectivity and infection durations, you will observe the effect of these parameters on the height of the epidemic peak, the time to reach the peak, and the cumulative number of disease cases. Demographics inputs \u00b6 This scenario uses generic_scenarios_demographics file for demographics input. You can view the complete demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where you installed the Quick Start. For more information on demographics files, see Demographics file structure and parameters . Key configuration parameters \u00b6 You can view the complete config.json in the <path_to_directory>ScenariosGeneric06_SEIRS directory. Demographics parameters \u00b6 The following parameters enable vital dynamics for birth , non-disease related deaths, and disease related deaths. For more information, see Demographics parameters. Set the following parameters: Set Enable_Vital_Dynamics to 1. Set Enable_Birth to 1. Set Death_Rate_Dependence to \u201cNONDISEASE_MORTALITY_BY_AGE_AND_GENDER.\u201d Set Enable_Disease_Mortality to 1. SEIRS model setup parameters \u00b6 The incubation period, infection period, immunity, and immunity waning must be enabled for an SEIRS model. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 3. Set Enable_Immunity to 1. Set Enable_Immune_Decay to 1. Set Incubation_Period_Distribution to \u201cEXPONENTIAL_DISTRIBUTION.\u201d Disease parameters \u00b6 These parameters set the EMOD executable to run as an SEIRS-like disease model with person-to- person direct contact. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 0.25. Set Base_Infectious_Period to 6. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Mortality rate \u00b6 The following parameters set the mortality for the disease. These parameters determine whether disease mortality is enabled by the simulation when the mortality is applied, and the rate of mortality. For more information, see General disease parameters. Set the following parameters: Set Base_Mortality to 0.01. Set Mortality_Time_Course to \u201cDAILY_MORTALITY.\u201d Immunity parameters \u00b6 For this scenario the Mortality_Blocking_Immunity_Duration_Before_Decay and Mortality_Blocking_Immunity_Decay_Rate are added. On recovery, people\u2019s immunity decays with an average half-life of 10 days and a 100-day offset. This applies to transmission blocking, acquisition blocking and mortality blocking immunity. Set the following parameters: Set Acquisition_Blocking_Immunity_Duration_Before_Decay to 100. Set Acquisition_Blocking_Immunity_Decay_Rate to 0.1. Set Transmission_Blocking_Immunity_Duration_Before_Decay to 100. Set Transmission_Blocking_Immunity_Decay_Rate to 0.1. Set Immunity_Acquisition_Factor to 0. Set Immunity_Mortality_Factor to 0. Set Immunity_Transmission_Factor to 0. Set Mortality_Blocking_Immunity_Duration_Before_Decay to 100. Set Mortality_Blocking_Immunity_Decay_Rate to 0.1. Simulation duration parameters \u00b6 The simulation is run for 2 years with a 1-day time step. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 730. Interventions \u00b6 You can view the complete campaign.json in the SamplesGeneric06_SEIRS directory. Baseline outbreak \u00b6 This scenario uses an OutbreakIndividual event at day 1 as the initial seeding of the infection. The value of Demographic_Coverage is 0.001 which is about 10 infected individuals in a population of 10,000. For more information, see OutbreakIndividual parameters. { \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.001 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" } Vaccination campaigns \u00b6 The following campaign event adds a vaccination campaign at day 500, with 100% vaccine take rate ( Vaccine_Take ) and 80% coverage ( Demographic_Coverage ) for all individuals. The vaccine is acquisition-blocking and full immunity lasts for 10 years (3650 days). For more information, see SimpleVaccine parameters. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.80 , \"Intervention_Config\" : { \"Cost_To_Consumer\" : 10.0 , \"Durability_Time_Profile\" : \"BOXDURABILITY\" , \"Primary_Decay_Time_Constant\" : 3650 , \"Reduced_Acquire\" : 1 , \"Reduced_Transmit\" : 0 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"class\" : \"SimpleVaccine\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 5060 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 The following figure shows the complete trajectory of the outbreak: the disease endemicity due to vital process and waning immunity; and the effect of vaccination campaigns that eradicates the outbreak after day 500. Figure 1: Trajectory of the outbreak The following figure shows the number of disease-specific deaths, and cumulative number of infections for all of the channels in the simulation output. Figure 2: Disease-specific deaths and cumulative number of infections Exploring the model \u00b6 In the this simulation, the Base_Infectivity parameter is set to 0.25 and with the incubation period, the basic reproductive numbers is 1.5. Change the Base_Infectivity parameter to 0.5. The base reproductive number should now be 3. Due to this larger value, the disease did not persist in the simulation because the rate of consuming susceptible individuals was larger than the inflow of new susceptible individuals. The infection could not find enough new susceptible individuals to sustain the transmission. Keep the base reproductive number at 3, and try shortening the immunity duration by changing Acquisition_Blocking_Immunity_Duration_Before_Decay and Transmission_Blocking_Immunity_Duration_Before_Decay from 100 to 50. Run the simulation again, and observe the persistence of the outbreak again due to increased influx of susceptible individuals.", "tags": "", "title": "Generic 6: SEIRS - Disease-specific mortality and endemics \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial7SI.html", "text": "Generic 7: SI - Lifelong infection \u00b6 This tutorial assumes that you have read Generic 6: SEIRS - Disease-specific mortality and endemics . This tutorial demonstrates how to configure an SI model and vary EMOD input parameters to observe the effect on the disease. The SI model is the simplest form of all disease models. Individuals are born into the simulation with no immunity (susceptible). Once infected and with no treatment, individuals stay infected and infectious throughout their life, and remain in contact with the susceptible population. This model matches the behavior of diseases like cytomegalovirus (CMV) or herpes. If S and I represent the number of susceptible and infected individuals, and is the total population. The infectious rate controls the spread. The SI model can be written as follows: The dynamics of I are also known as logistic growth. If there are no vital processes (birth and death), every susceptible will eventually become infected. Let and represent the birth and death rates, respectively, for the model. To maintain a constant population, assume that . Therefore, the model becomes: In steady state and given that , the final proportion of infected people is related to both the vital dynamics and . Demographics inputs \u00b6 This tutorial uses generic_scenarios_demographics file for demographics input. You can view the complete demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where you installed either the Quick Start. For more information on demographics files, see Demographics file structure and parameters . Key configuration parameters \u00b6 This tutorial simulates the person-to-person disease transmission of an SI model in a hypothetical community for more than a 100 year period. You can view the complete config.json at <path_to_directory>ScenariosGeneric07_SI directory. Demographics parameters \u00b6 The following parameter enables vital dynamics (birth and death) in the simulation. Set the following parameters: Set Enable_Vital_Dynamics to 1. Set Enable_Birth to 1. Set Death_Rate_Dependence to \u201cNONDISEASE_MORTALITY_BY_AGE_AND_GENDER.\u201d SI model setup parameters \u00b6 The EMOD generic simulation uses an SEIR model by default. However, it can be modified to an SI model with the following configurations. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 0. Set Base_Infectious_Period to 50000. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Infectious_Period_Distribution to \u201cFIXED_DURATION.\u201d Disease parameters \u00b6 As demonstrated previously, the spread of disease in the SI model depends on \u03b2, the base infectivity, and, for simplicity, assume the infectivity stays constant over time. Assume that the anticipated final epidemic size, , is 85%, and the birth/death rate equals 0.0000548 per day (2% per year). can be calculated by looking at the steady state: and =0.00003653. Set the following parameters: Set Base_Infectivity to 0.00003653. Simulation time parameters \u00b6 The simulation runs for 100 years. Because the disease dynamics are slow, the larger time step speeds up the simulation. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 36500. Interventions \u00b6 Initial outbreak \u00b6 This simulation uses an OutbreakIndividual event at day 30 for the initial seeding. For more information, see OutbreakIndividual parameters. { \"Use_Defaults\" : 1 , \"Campaign_Name\" : \"Initial Seeding\" , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.05 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 30 , \"class\" : \"CampaignEvent\" }] } You can view the complete campaign.json at <path_to_directory>ScenariosGeneric07_SI directory. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 Figure 1: Outbreak reaching steady-state shows the growth of an outbreak over time and eventually approaching steady state. Notice that the fraction of infected people at steady state approaches the anticipated value of 85%. Figure 1: Outbreak reaching steady-state Exploring the model \u00b6 You can change the Base_Infectivity in the config.json file and the birth and death rates in the demographics file to have a different fraction of infected value at steady state. If you want to change vital dynamics without updating the demographics file, disable vital dynamics in the configuration file ( Enable_Vital_Dynamics = 0). There will be no births in the simulation and eventually everyone becomes infected. The following output shows the result without vital dynamics. Figure 2: No vital dynamics", "tags": "", "title": "Generic 7: SI - Lifelong infection \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial8SIS.html", "text": "Generic 8: SIS - Reoccurring infections \u00b6 This tutorial assumes that you have read Generic 7: SI - Lifelong infection and Generic 1: SIR - Simple single-node outbreak . Similar to the SIRS model , the infected individuals return to the susceptible state. This model is appropriate for diseases that commonly have repeat infections, for example, the common cold (rhinoviruses), or sexually transmitted diseases like gonorrhea or chlamydia. For this model, we can analytically solve the ordinary differential equation (ODE) to understand the dynamics. For simplicity, the following ODE model ignores the vital process (birth and death): At equilibrium, solving: There are two equilibrium states for the SIS model, the first is (disease free state), and the second is: For disease to spread, we need to have . Therefore, similar to the previously described concept of the basic reproductive number , when , the disease will spread and approach the second steady state; otherwise, it will eventually reach the disease-free state. Demographics inputs \u00b6 This tutorial uses generic_scenarios_demographics file for demographics input. You can view the complete demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where you installed the Quick Start. For more information on demographics files, see Demographics file structure and parameters . Key configuration parameters \u00b6 The tutorial simulates a sexually-transmitted disease using the SIS model in a hypothetical closed community over a 10-year period. The infectious duration is 25 days, and the basic reproductive number is 1.2. You can view the complete config.json in the <path_to_directory>ScenariosGeneric08_SIS directory where <path_to_directory> is the location where you installed the Quick Start. Demographics parameter \u00b6 Set Enable_Vital_Dyanamics to 0. This will disable vital dynamics (birth and death) in the simulation. For more information, see Demographics parameters. SIS model parameters \u00b6 The generic simulation type uses an SEIR model by default. However, it can be modified to an SIS model with the following configurations: no incubation period and no immunity. For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 0. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Enable_Immunity to 0. Disease parameters \u00b6 As demonstrated previously, the spread of disease in the SIS model depends on (set by base infectivity) and which are set by duration of infection. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 0.048. Set Base_Infectious_Period to 25. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Simulation duration parameters \u00b6 The simulation is run for 3 months with a 1-day time step. For more information, see Simulation setup parameters. Set the following parameters: Set Simulation_Timestep to 1. Set Simulation_Duration to 3650. Interventions \u00b6 You can view the complete campaign.json in the <path_to_directory>ScenariosGeneric08_SIS directory where <path_to_directory> is the location where you installed the Quick Start. Initial outbreak \u00b6 This simulation uses an OutbreakIndividual event at day 1 to provide the initial seeding. For more information, see OutbreakIndividual parameters. { \"Use_Defaults\" : 1 , \"Campaign_Name\" : \"Initial Seeding\" , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Event_Name\" : \"Outbreak\" , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 The following figure shows the growth of an outbreak over time that eventually approaches steady state. You can compare the fraction of infected people with the anticipated value based on the previous calculation. If we have a reproductive number of 1.2, the infected fraction at equilibrium will be 1 - (1/1.2) ~ 17%. Figure 1: Outbreak approaching steady-state Exploring the model \u00b6 You can change the Base_Infectivity value to alter the basic reproductive number (R 0 >1 and R 0 <1) to observe the endemic state behavior.", "tags": "", "title": "Generic 8: SIS - Reoccurring infections \u2014 Training Tutorials  documentation"}, {"url": "./generic-tutorial9Density.html", "text": "Generic 9: Density scaling of infectivity \u00b6 This tutorial explores the effect of population density on the transmissibility of disease. By default, the EMOD executable (Eradication.exe) uses frequency-dependent transmission and the overall transmissibility does not change with population size or density. An infected individual will infect the same number of people either in a small village or in a metropolitan area. However, it is possible when population size is low, the transmissibility might be less and follow density- dependent behavior instead of frequency-dependent behavior. Based on existing evidence, it is possible that neither a frequency-dependent nor a density- dependent mechanism may adequately describe the correct scaling over the entire population density spectrum. The EMOD executable uses the spatial contact model with constant population density and limited activity range to estimate contact rates as discussed in the article The scaling of contact rates with population density for the infectious disease models , by Hu et al ., 2013 Journal of Mathematical Bioscience . 244(2):125-134. The scaling function can be specified for \u03b2, in the form of: \u03b2 = \u03b2 0 (1- e \u03c1/\u03c1 c50 ) where \u03c1 c50 is a constant determining the speed of increase of infectivity. For this tutorial, the simulation is run three times with a different node size for each run. The change of node grid size will result in the change of population density for the simulations. Demographics inputs \u00b6 This tutorial uses generic_scenarios_demographics file for demographics input. You can view the complete demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where you installed the Quick Start. For more information on demographics files, see Demographics file structure and parameters . Key configuration parameters \u00b6 The tutorial simulates the person-person disease transmission of an influenza-like-illness model with waning immunity (SEIRS) over a short period, in a hypothetical community over a short period. After infection, all individuals will stay immune for a certain period of time before they become susceptible again. The saturated R 0 is set to 1.5 and R 0 initially increases with population density You can view the complete config.json at <path_to_directory>ScenariosGeneric Scenarios09_DensityScaling directory. SEIRS model setup parameters \u00b6 The incubation period, infection period, immunity, and immunity waning, must be enabled for an SEIRS model . For more information, see General disease parameters. Set the following parameters: Set Base_Incubation_Period to 3. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Enable_Immunity to 1. Set Enable_Immune_Decay to 1. Disease parameters \u00b6 These parameters set Eradication.exe to run an SEIRS-like disease model with person-to-person direct transmission. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 0.375. Set Base_Infectious_Period to 14. Set Infectious_Period_Distribution to \u201cFIXED_DURATION.\u201d Immunity parameters \u00b6 The following parameters configure a 90 day immunity. After 90 days, the immunity is lost immediately. For more information, see General disease parameters. Set the following parameters: Set Acquisition_Blocking_Immunity_Duration_Before_Decay to 90. Set Transmission_Blocking_Immunity_Duration_Before_Decay to 90. Set Acquisition_Blocking_Immunity_Decay_Rate to 1. Set Transmission_Blocking_Immunity_Decay_Rate to 1. Density scaling parameters \u00b6 These parameters are used by the scaling function to calculate \u03b2. You can change Node_Grid_Size to change population density for different simulations. For more information, see General disease parameters. Set the following parameters: Set Population_Density_Infectivity_Correction to \u201cSATURATING_FUNCTION_OF_DENSITY.\u201d Set Population_Density_C50 to 30. Set Node_Grid_Size to 0.1, 0.15, and 0.3 respectively for the three different simulations. Interventions \u00b6 This simulation uses an OutbreakIndividual event for repeated infection seeding. The campaign.json file is in the ScenariosGeneric09_DensityScaling directory. For more information, see OutbreakIndividual parameters. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Campaign_Name\" : \"Infections\" , \"Event_Coordinator_Config\" : { \"Coverage\" : 0.001 , \"Demographic_Coverage\" : 0.001 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Event_Name\" : \"Outbreak\" , \"Genome\" : 0 , \"Outbreak_Source\" : \"ImportCases\" , \"class\" : \"OutbreakIndividual\" }, \"Number_Repetitions\" : 10 , \"Target_Age_Max\" : 100 , \"Target_Age_Min\" : 0 , \"Target_Demographic\" : \"Everyone\" , \"Timesteps_Between_Repetitions\" : 120 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 0 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Simulation output graphs \u00b6 The simulation was run three times with the Node_Grid_Size set to three different values: 0.1, 0.15, and 0.3 degrees. The node was assumed to be near the equator (both latitude and longitude were set to 0) and the population was 10,000. Using the three different Node_Grid_Size values, the node area for the three simulations were 124, 278, and 1112 km 2 , respectively. With a population of 10,000, the population density was 80, 36, and 9 people/km 2 , respectively. Using the data described in the previous paragraph, the three densities correspond to different R:sub:0values: close to 1.5, close to 1, and well below 1, respectively, as shown in the following graph from Hu *et al*., 2013. *Journal of Mathematical Bioscience*. 244(2):125-134. . Figure 1: Effect of population density on transmissibility The following table shows the values for the three simulations: Simulation Population Latitude, Longitude Node_Grid_Size Node Area Density R 0 1 10,000 0, 0 0.1 124 80 R 0 > 1 2 10,000 0, 0 0.15 278 36 R 0 ~ 1 3 10,000 0, 0 0.3 1112 9 R 0 < 1 The following graphs show the effect of population density on transmissibility, in terms of maintaining endemic status. When population density is large enough, it is easy to maintain an endemic status while in lower density is difficult to do so. Simulation 1: Node_Grid_Size = 0.1, population density 80/km 2 and R 0 > 1 Simulation 2: Node_Grid_Size = 0.15, population density 36/km 2 and R 0 ~ 1 Simulation 3: Node_Grid_Size = 0.3, population density 9/km 2 and R 0 < 1 Exploring the model \u00b6 Change the Node_Grid_Size to other values. Explore a range of population densities and run the simulation multiple times to get a probability of fade-out under different densities. You can download the sample Excel file, calculate_cell_area_pop_density, to calculate the node area (cell area) and population density. There are four inputs that you can modify: latitude, longitude, grid size (in degrees) and population. Click calculate_cell_area_pop_density to download the file.", "tags": "", "title": "Generic 9: Density scaling of infectivity \u2014 Training Tutorials  documentation"}, {"url": "./genindex.html", "text": "Index A | B | C | D | E | F | G | H | I | J | K | L | M | N | O | P | R | S | T | V | W A agent-based model antigen B Boost C campaign campaign event campaign file channel circumsporozoite protein (CSP) class factory Clausius-Clayperon relation cohort model configuration file core CRISPR cytokine D demographics file deterministic diffusive migration disease-specific build dynamic link library (DLL) E EMODule entomological inoculation rate (EIR) Epidemiological MODeling software (EMOD) epitope Eradication.exe Euler method event coordinator exp( F flattened file G gametocyte gene drive mosquito gonotrophic H hepatocyte herd immunity Heterogeneous Intra-Node Transmission (HINT) high-performance computing (HPC) homing endonuclease genes (HEGs) I incidence individual mosquito model indoor residual spraying (IRS) input data files insecticide-treated nets (ITN) intervention J JavaScript Object Notation (JSON) K Keyhole Markup Language (KML) Koppen-Geiger Climate Classification System L larval habitat larvicide Link-Time Code Generation (LTCG) loss to follow-up (LTFU) M mass drug administration (MDA) merozoite merozoite surface protein (MSP) merozoites Message Passing Interface (MPI) microsolver monolithic build Monte Carlo N node node-targeted intervention O ordinary differential equation (ODE) output report overlay file oviposition P Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1) pre-erythrocytic pre-erythrocytic vaccine (PEV) prevalence preview R regression test release reporter reproductive number routine immunization (RI) S scenario , [1] schema schizont SEIR model SEIRS model SI model simulation simulation type SIR model SIRS model SIS model solvers spatial repellent sporogony sporozoite Standard Template Library (STL) state transition event stochastic subpatent sugar-baited trap superinfection supplemental immunization activity (SIA) T time step transmission-blocking vaccine (TBV) tutorial V vaccine intervention types vector vector model vectorial capacity W WAIFW matrix Wolbachia working directory", "tags": "", "title": "Index \u2014 Training Tutorials  documentation"}, {"url": "./glossary.html", "text": "Glossary \u00b6 The Epidemiological MODeling software (EMOD) glossary is divided into the following subsections that define terms related to software usage, general epidemiology, and the particular disease being modeled. Contents Software terms Epidemiology terms Malaria and vector terms Software terms \u00b6 The following terms are used to describe both general computing processes and concepts and the files, features, and functionality related to running simulations with Epidemiological MODeling software (EMOD). agent-based model A type of simulation that models the actions and interactions of autonomous agents (both individual and collective entities such as organizations or groups). Boost Free, peer-reviewed, portable C++ source libraries aimed at a wide range of uses including parallel processing applications (Boost.MPI). For more information, please see the Boost website, http://www.boost.org . campaign A collection of events that use interventions to modify a simulation . campaign event A JSON object that determines when and where an intervention is distributed during a campaign. campaign file A JavaScript Object Notation (JSON) formatted file that contains the parameters that specify the distribution instructions for all interventions used in a campaign, such as diagnostic tests, the target demographic, and the timing and cost of interventions. The location of this file is specified in the configuration file with the Campaign_Filename parameter. Typically, the file name is campaign.json. channel A property of the simulation (for example, \u201cParasite Prevalence\u201d) that is accumulated once per simulated time step and written to file, typically as an array of the accumulated values. class factory A function that instantiate objects at run-time and use information from JSON-based configuration information in the creation of these objects. configuration file A JavaScript Object Notation (JSON) formatted file that contains the parameters sufficient for initiating a simulation. It controls many different aspects of the simulation, such as population size, disease dynamics, and length of the simulation. Typically, the file name is config.json. core In computing, a core refers to an independent central processing unit (CPU) in the computer. Multi-core computers have more than one CPU. However, through technologies such as Hyper- Threading Technology (HTT or HT), a single physical core can actually act like two virtual or logical cores, and appear to the operating system as two processors. demographics file A JavaScript Object Notation (JSON) formatted file that contains the parameters that specify the demographics of a population, such as age distribution, risk, birthrate, and more. IDM provides demographics files for many geographic regions. This file is considered part of the input data files and is typically named <region>_demographics.json. disease-specific build A build of the EMOD executable (Eradication.exe) built using SCons without any dynamic link libraries (DLLs). dynamic link library (DLL) Microsoft\u2019s implementation of a shared library, separate from the EMOD executable (Eradication.exe), that can be dynamically loaded (and unloaded when unneeded) at runtime. This loading can be explicit or implicit. EMODule A modular component of EMOD that are consumed and used by the EMOD executable (Eradication.exe). Under Windows, a EMODule is implemented as a dynamic link library (DLL) and, under CentOS on Azure, EMODules are currently not supported. EMODules are primarily custom reporters. Epidemiological MODeling software (EMOD) The modeling software from the Institute for Disease Modeling (IDM) for disease researchers and developers to investigate disease dynamics, and to assist in combating a host of infectious diseases. You may see this referred to as Disease Transmission Kernel (DTK) in the source code. Eradication.exe Typical (default) name for the EMOD executable (Eradication.exe), whether built using monolithic build or modular (EMODule-enabled) build. event coordinator A JSON object that determines who will receive a particular intervention during a campaign. flattened file A single campaign or configuration file created by combining a default file with one or more overlay files. Multiple files must be flattened prior to running a simulation. Configuration files are flattened to a single-depth JSON file without nesting, the format required for consumption by the EMOD executable (Eradication.exe). Separating the parameters into multiple files is primarily used for testing and experimentation. Heterogeneous Intra-Node Transmission (HINT) A feature for modeling person-to-person transmission of diseases in heterogeneous population segments within a node for generic simulations. high-performance computing (HPC) The use of parallel processing for running advanced applications efficiently, reliably, and quickly. input data files Fixed information used as input to a simulation. Examples of such data are population, altitude, demographics, and transportation migrations. The configurable config.json and campaign.json files do not fall into this category of static fixed input. intervention An object aimed at reducing the spread of a disease that is distributed either to an individual; such as a vaccine, drug, or bednet; or to a node; such as a larvicide. Additionally, initial disease outbreaks and intermediate interventions that schedule another intervention are implemented as interventions in the campaign file . JavaScript Object Notation (JSON) A human-readable, open standard, text-based file format for data interchange. It is typically used to represent simple data structures and associative arrays, and is language- independent. Keyhole Markup Language (KML) A file format used to display geographic data in an Earth browser, for example, Google Maps. The format uses an XML-based structure (tag-based structure with nested elements and attributes). Tags are case-sensitive. Link-Time Code Generation (LTCG) A method for the linker to optimize code (for size and/or speed) after compilation has occurred. The compiled code is turned not into actual code, but instead into an intermediate language form (IL, but not to be confused with .NET IL which has a different purpose). The LTCG then, unlike the compiler, can see the whole body of code in all object files and be able to optimize the result more effectively. Message Passing Interface (MPI) An interface used to pass information between computing cores in parallel simulations. One example is the migration of individuals from one geographic location to another within EMOD simulations. microsolver A microsolver is a type of \u201cminiature model\u201d that operates within the framework of EMOD to compute a particular set of parameters. Each microsolver, in effect, is creating a microsimulation in order to accurately capture the dynamics of that particular aspect of the model. Monte Carlo Monte Carlo methods, or Monte Carlo experiments, are a class of computational algorithms utilizing repeated random sampling to obtain numerical results. Monte Carlo simulations create probability distributions for possible outcome values, which provides a more realistic way of describing uncertainty in variables. monolithic build A single EMOD executable (Eradication.exe) with no DLLs that includes all components as part of Eradication.exe itself. You can still use EMODules with the monolithic build; for example, a custom reporter is a common type of EMODule. View the documentation on EMODules and emodules_map.json for more information about creation and use of EMODules. node A grid size that is used for modeling geographies. Within EMOD, a node is a geographic region containing simulated individuals. Individuals migrate between nodes either temporarily or permanently using mobility patterns driven by local, regional, and long- distance transportation. node-targeted intervention An intervention that is distributed to a geographical node rather than to a single individual. One example is larvicides, which affect all mosquitoes living and feeding within a given node. output report A file that is the output from an EMOD simulation. Output reports are in JSON, CSV, or binary file format. You must pass the data from an output report to graphing software if you want to visualize the output of a simulation. overlay file An additional configuration, campaign, or demographic file that overrides the default parameter values in the primary file. Separating the parameters into multiple files is primarily used for testing a nd experimentation. In the case of configuration and campaign files, the files can use an arbitrary hierarchical structure to organize parameters into logical groups. Configuration and campaign files must be flattened into a single file before running a simulation. preview Software that undergoes a shorter testing cycle in order to make it available more quickly. Previews may contain software defects that could result in unexpected behavior. Use EMOD previews at your own discretion. regression test A test to verify that existing EMOD functionality works with new updates, typically used to refer to one of a set of scenarios included in the EMOD bundle and located in the Regression subdirectory of the bundle. Directory names of each subdirectory in Regression describe the main regression attributes, for example, \u201c1_Generic_Seattle_MultiNode\u201d. Also can refer to the process of regression testing of software. release Software that includes new functionality, scientific tutorials leveraging new or existing functionality, and/or bug fixes that have been thoroughly tested so that any defects have been fixed before release. EMOD releases undergo full regression testing. reporter Functionality that extracts simulation data, aggregates it, and saves it as an output report . EMOD provides several built-in reporters for outputting data from simulations and you also have the ability to create a custom reporter. scenario A collection of input, configuration, and campaign files that describes a real-world example of a disease outbreak and interventions. Many scenarios are included with both the Quick Start and EMOD source installations. The Quick Start documentation includes tutorials that describe how to run simulations using these scenarios to learn more about epidemiology and disease modeling. schema A text or JSON file that can be generated from the EMOD executable (Eradication.exe) that defines all configuration and campaign parameters. simulation An EMOD software execution performed by the kernel. Each simulation has an associated set of files that control the inputs, configuration, and campaign. simulation type The disease or disease class to model. EMOD supports the following simulation types for modeling a variety of diseases: Generic disease (GENERIC_SIM) Vector-borne diseases (VECTOR_SIM) Malaria (MALARIA_SIM) Tuberculosis (TB_SIM) Sexually transmitted infections (STI_SIM) HIV (HIV_SIM) solvers Solvers are used to find computational solutions to problems. In simulations, they can be used, for example, to determine the time of the next simulation step, or to compute the states of a model at particular time steps. Standard Template Library (STL) A library that contains a set of common C++ classes (including generic algorithms and data structures) that are independent of container and implemented as templates, which enables compile-time polymorphism (often more efficient than run-time polymorphism). For more information and discussion of STL, see Wikipedia . state transition event A change in state (e.g. healthy to infected, undiagnosed to positive diagnosis, or birth) that may trigger a subsequent action, often an intervention. \u201cCampaign events\u201d should not be confused with state transition events. time step A discrete number of hours or days in which the \u201csimulation states\u201d of all \u201csimulation objects\u201d (interventions, infections, immune systems, or individuals) are updated in a simulation. Each time step will complete processing before launching the next one. For example, a time step would process the migration data for populations moving between nodes via rail, airline, and road. The migration of individuals between nodes is the last step of the time step after updating states. tutorial A set of instructions in the Quick Start documentation to learn more about epidemiology and disease modeling. Tutorials are based on real-world scenarios and demonstrate the mechanics of the the model. Each tutorial consists of one or more scenarios. working directory The directory that contains the configuration and campaign files for a simulation. You must be in this directory when you invoke Eradication.exe at the command line to run a simulation. Epidemiology terms \u00b6 The following terms are used to describe general concepts and processes in the field of epidemiology and disease modeling. antigen A substance that is capable of inducing a specific immune response and that evokes the production of one or more antibodies. Clausius-Clayperon relation A way of characterizing a transition between two phases of matter; provides a method to find a relationship between temperature and pressure along phase boundaries. Frequently used in meteorology and climatology to describe the behavior of water vapor. See Wikipedia - Clausius-Clayperon relation for more information. deterministic Characterized by the output being fully determined by the parameter values and the initial conditions. Given the same inputs, a deterministic model will always produce the same output. diffusive migration The diffusion of people in and out of nearby nodes by foot travel. epitope The portion of an antigen that the immune system recognizes. An epitope is also called an antigenic determinant. Euler method Used in mathematics and computational science, this method is a first-order numerical procedure for solving ordinary differential equations with a given initial value. exp( The exponential function, , where is the number (approximately 2.718281828) such that the function is its own derivative. The exponential function is used to model a relationship in which a constant change in the independent variable gives the same proportional change (i.e. percentage increase or decrease) in the dependent variable. The function is often written as . The graph of is upward-sloping and increases faster as increases. herd immunity The resistance to the spread of a contagious disease within a population that results if a sufficiently high proportion of individuals are immune to the disease, especially through vaccination. incidence The rate of new cases of a disease during a specified time period. This is a measure of the risk of contracting a disease. Koppen-Geiger Climate Classification System A system based on the concept that native vegetation is a good expression of climate. Thus, climate zone boundaries have been selected with vegetation distribution in mind. It combines average annual and monthly temperatures and precipitation, and the seasonality of precipitation. The DTK has several options for configuring the climate, namely air temperature, rainfall, and humidity. One option utilizes Input Files that associate geographic nodes with Koppen climate indices. The modified Koppen classification uses three letters to divide the world into five major climate regions (A, B, C, D, and E) based on average annual precipitation, average monthly precipitation, and average monthly temperature. Each category is further divided into sub-categories based on temperature and precipitation. While the Koppen system does not take such things as temperature extremes, average cloud cover, number of days with sunshine, or wind into account, it is a good representation of our earth\u2019s climate. loss to follow-up (LTFU) Patients who at one point were actively participating in disease treatment or clinical research, but have become lost either by error or by becoming unreachable at the point of follow-up. ordinary differential equation (ODE) A differential equation containing one or more functions of one independent variable and its derivatives. prevalence The rate of all cases of a disease during a specified time period. This is a measure of how widespread a disease is. reproductive number In a fully susceptible population, the basic reproductive number R:sub:0is the number of new cases generated by one infectious case over the course of the infectious period. The effective reproductive number takes into account non-susceptible individuals. routine immunization (RI) The standard practice of vaccinating the majority of susceptible people in a population against vaccine-preventable diseases. scenario A real-world or hypothetical situation that demonstrates or represents disease dynamics. stochastic Characterized by having a random probability distribution that may be analyzed statistically but not predicted precisely. subpatent When an individual is infected but asymptomatic, so the infection is not readily detectable. SEIR model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, exposed, infectious, and recovered. SEIRS model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, exposed, infectious, recovered, and susceptible. SI model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible and infectious. SIR model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, infectious, and recovered. SIRS model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, infectious, recovered, and susceptible. SIS model A generic epidemiological model that provides a simplified means of describing the transmission of an infectious disease through individuals where those individuals can pass through the following five states: susceptible, infectious, and susceptible. supplemental immunization activity (SIA) In contrast to routine immunization (RI) , SIAs are large-scale operations with a goal of delivering vaccines to every household. WAIFW matrix A matrix of values that describes the rate of transmission between different population groups. WAIFW is an abbrevation for Who Acquires Infection From Whom. Malaria and vector terms \u00b6 The following terms are used to describe the biological processes involved in malaria and other vector-borne diseases. circumsporozoite protein (CSP) One of two proteins (the other being thrombospondin-related adhesive protein) involved in sporozoite recognition of host cells in malaria. cohort model A vector model that tracks properties of mosquito \u201ccohorts\u201d, for example, population, age, gender, fertility, etc. cytokine Cytokines are small secreted proteins that are released by cells, and have specific effects on interactions and communications between cells. They are especially important in the immune system, and can function in inflammatory (and anti-inflammatory) response. CRISPR A genome-editing technique that utilizes the CRISPR/Cas9 system (a prokaryotic immune system) that can be used to splice DNA sequences in very specific target locations, enabling the precise insertion or removal of genes of interest. entomological inoculation rate (EIR) The commonly-used measure of the intensity of malaria transmission. EIR is equal to the number of mosquito bites per night times the proportion of those bites that test positive for sporozoites. gametocyte The male and female sexual forms of the blood-stage malaria parasite. During blood-feeding, gametocytes are taken up into the mosquito mid-gut where they commence the mosquito-phase of the parasite life cycle. gene drive mosquito Gene drive is a genetic technique that promotes the inheritance of particular genes or specific genetic elements. In mosquitoes, approaches are being tested that work to suppress the spread of malaria by creating mosquito lines through fertility suppression, driving-Y chromosomes, or express genes that limit malaria transmission. gonotrophic A female mosquito\u2019s feeding and egg-laying cycle, where the duration is defined as the average number of days that gravid mosquitoes take to oviposit after taking a blood meal. hepatocyte The cell type that comprises liver tissue. homing endonuclease genes (HEGs) \u201cSelfish\u201d genetic elements that can spread through the mosquito population even at a cost to the host. The HEG allele status is a tracked property of mosquitoes. Large numbers of HEG- positive mosquitoes may be introduced into the simulation by a MosquitoRelease intervention. individual mosquito model A vector model simulation where every individual mosquito in the population is modeled, as well as a simulation of a sampled subset of mosquitoes to represent the population as the whole. indoor residual spraying (IRS) The process of spraying insecticide on the interior surfaces of dwellings in order to repel or kill mosquitoes. insecticide-treated nets (ITN) Bednets hung over sleeping areas that are treated with insecticide to repel mosquitoes and kill those that land on them. larval habitat Mosquito larval ecology includes several habitat types including brackish swamp, constant, human population, temporary rainfall, and semi-permanent water vegetation. larvicide A node-targeted intervention that may be configured to have a direct killing effect on the larval population and a temporary reduction on available larval habitat. mass drug administration (MDA) The treatment of an entire population in a geographic area with a curative dose of an antimalarial drug without first testing for infection, and regardless of the presence of symptoms. merozoite The malaria parasite stage that invades red blood cells. After each cycle of asexual reproduction within the infected red blood cell, merozoites are released into the bloodstream to invade new red blood cells. merozoites see \u201cmerozoite\u201d merozoite surface protein (MSP) A type of protein integral to red blood cell invasion by malaria parasites that is an important target of the human immune response. oviposition to deposit or lay eggs. pre-erythrocytic Stages of a malaria parasite that occur before the red blood cells are invaded. This includes the period starting from inoculation with sporozoites, through the liver-stage multiplication, up to infected hepatocyte rupture, and the first invasion of erythrocytes by merozoites. pre-erythrocytic vaccine (PEV) A vaccine targeted at the earliest stage of malaria parasite development, namely before the blood-stage infection. Other vaccines target the asexual and sexual stages of the parasite. Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1) A protein encoded by the malaria parasite and expressed on the surface of infected red blood cells. PfEMP1 adheres to various human receptors, allowing sequestration of parasitized red blood cells and avoidance of clearance in the spleen. PfEMP1 is an important antigenic target of the human adapted immune response. It exhibits a high degree of antigenic variation via a genetic switching mechanism evolved to escape immune pressure. schizont Cells formed during the asexual stage of the life cycle of sporozoan protozoans, such as the malaria parasite. spatial repellent An intervention that may be individually-distributed (protective indoors and outdoors), or node-distributed at the community-level, that repels mosquitoes from biting. sporogony The asexual process of spore formation in parasitic sporozoans. sporozoite a motile, spore-like stage of the malaria parasite that is the infectious agent introduced into the host. sugar-baited trap Bait stations, often placed inside houses, that have an attractive and toxic sugar used to attract and kill sugar-feeding male and female mosquitoes. superinfection The simultaneous infection with multiple strains of the same pathogen. transmission-blocking vaccine (TBV) A vaccine that blocks gametocytes from infected humans by producing viable sporozoites in mosquitoes. Also referred to as a \u201csexual-stage\u201d vaccine, since vaccines in this class block gametocytes from infected humans by producing viable sporozoites in mosquitoes. vaccine intervention types In EMOD, vaccine intervention types are either \u201cAcquisitionBlocking\u201d, \u201cTransmissionBlocking\u201d or \u201cMortalityBlocking\u201d. For example, \u201cmode-of-action\u201d, \u201ctargeted parasite stages\u201d, (such as pre-erythrocytic, asexual, or sexual), etc. vector Insect or other living carrier that transmits an infectious agent. The vector life cycle and feeding cycle in the model are described in the article Eckhoff, Malaria Journal 2011, 10:303 . vectorial capacity The daily rate of all mosquitoes that would be infected after biting a single infectious host. This measure can be used to describe the transmission intensity of malaria, and serves as a rate at which future infections arise from a currently infected host. vector model The vector model includes both a cohort model and an individual mosquito model . The model is a closed-loop feeding cycle where successful animal feeds (both indoors and outdoors) produce eggs, larva, vector mating, and then, adult mosquitoes. The vector model inherits the same human- infection model structure from the generic simulation type: uninfected, latent incubation, infectious, multiple immune variables, and superinfection. However, the transmission of infections is not between individual humans, but rather via the human-to-vector and vector- to-human pathways. Wolbachia This is a genus of bacteria that infects many arthopod species (especially insects). It has been discovered that insects infected with Wolbachia have increased resistance to viral or other parasitic infections. Work is underway to determine if strains of modified Wolbachia can be used to control malaria, by effectively preventing mosquitoes from becoming infected with Plasmodium parasites.", "tags": "", "title": "Glossary \u2014 Training Tutorials  documentation"}, {"url": "./hint-tutorial-overview.html", "text": "Heterogeneous Intra-Node Transmission (HINT) tutorials \u00b6 The Heterogeneous Intra-Node Transmission (HINT) tutorials consist of a group of scenarios that guide you through simulations that will help you understand the HINT feature, which allows you to configure heterogeneous disease transmission between different groups defined with IndividualProperties . Each tutorial has a corresponding Scenarios directory and a demographics file in the ScenariosInputFiles directory. All of these tutorials have an IndividualProperties attribute in their demographics files. An IndividualProperties attribute with a TransmissionMatrix is a requirement for a HINT simulation. Additionally, Enable_Heterogeneous_Intranode_Transmission must be set to 1 (true) in the simulation configuration file (config.json). There are two ways to use the tutorials. You can read the tutorials information and graphs to learn about the model; installing the Quick Start is not required. Or, for a more hands-on approach, you can also run the simulations that are associated with the tutorials and make the suggested changes to the configuration files to see the impact on the output graphs. The EMOD executable (Eradication.exe) and all of the files needed to run the simulations are included in the Quick Start installation. Refer to EMOD parameter reference for more information regarding individual parameters that you are interested in modifying. For more information about the mathematics behind HINT, see Heterogeneity and transmission scaling in the generic disease documentation. HINT 1: Risk groups HINT 2: School closure HINT 3: Age and accessibility HINT 4: Washington state pertussis HINT 5: Seattle commuting", "tags": "", "title": "Heterogeneous Intra-Node Transmission (HINT) tutorials \u2014 Training Tutorials  documentation"}, {"url": "./hint-tutorial-scenario1.html", "text": "HINT 1: Risk groups \u00b6 This tutorial is a simple simulation using risk groups to investigate the disease spread between heterogeneous groups. The tutorial uses EMOD\u2019s HINT feature and sets up 2 risk groups (Low, High) with the level of risk associated with disease transmission. By default, the disease transmission for a EMOD generic simulation is frequency-dependent. The expected number of secondary infections from an infected person does not change with population size. HINT uses a \u201cwho-acquires-infection-from-whom\u201d (WAIFW) matrix with values that define the force of infection between the risk groups. Contents Demographics inputs IndividualProperties Property and values Initial_Distribution Transitions TransmissionMatrix Key configuration parameters Interventions Run the simulation Expected simulation output Exploring the model Demographics inputs \u00b6 In the demographics input file, the node has 10,000 individuals. There are no births and deaths. The birth rate and initial population values are set by BirthRate and InitialPopulation , respectively, in NodeAttributes . The death rate is set in IndividualAttributes using the MortalityDistribution attribute. The ResultValues matrix in MortalityDistribution contains the death rates for males and females. Both the male and female death rate is equal to 0. { \"Metadata\" : {}, \"Defaults\" : { \"IndividualProperties\" : {} }, \"Nodes\" : [{ \"NodeID\" : 340461476 , \"NodeAttributes\" : { \"InitialPopulation\" : 10000 , \"BirthRate\" : 0.001 }, \"IndividualAttributes\" : { \"MortalityDistribution\" : { \"NumDistributionAxes\" : 2 , \"AxisNames\" : [ \"gender\" , \"age\" ], \"AxisUnits\" : [ \"male=0,female=1\" , \"years\" ], \"AxisScaleFactors\" : [ 1 , 365 ], \"NumPopulationGroups\" : [ 2 , 1 ], \"PopulationGroups\" : [ [ 0 , 1 ], [ 0 ] ], \"ResultUnits\" : \"annual deaths per day\" , \"ResultScaleFactor\" : 1 , \"ResultValues\" : [ [ 0.001 ], [ 0.001 ] ] } } }] } You can view the complete hint_riskgroups_demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. IndividualProperties \u00b6 For HINT, IndividualProperties must be added to Defaults in the demographics file and contain a TransmissionMatrix . The TransmissionMatrix is configured with the values of the WAIFW matrix that are needed for HINT\u2018s mathematical model. Property and values \u00b6 The Property and Values parameters of IndividualProperties define two risk groups: \u201cHigh\u201d and \u201cLow\u201d. Initial_Distribution \u00b6 At the beginning of the simulation, 20% of the population is in the high-risk population and 80% are in the low-risk population as specified by the Initial_Distribution array. Transitions \u00b6 Individuals will not move from one risk group to the other so Transitions is empty. TransmissionMatrix \u00b6 In the TransmissionMatrix structure, the Matrix member is a Values x Values-sized WAIFW matrix with values that scale the base infectivity. For this simulation, the risk groups are defined in the following order: \u201cHigh\u201d and \u201cLow.\u201d The following matrix represents the direction of the disease transmission between the groups: This simulation assumes that the disease transmission within the same risk group is higher than between risk groups, and people in the high-risk group have higher transmissibility. The following values of the WAIFW matrix are used to specify the disease transmission between the risk groups: { \"Metadata\" : {}, \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Risk\" , \"Values\" : [ \"High\" , \"Low\" ], \"Initial_Distribution\" : [ 0.2 , 0.8 ], \"Transitions\" : [], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 10 , 0.1 ], [ 0.1 , 1 ] ] } }] }, \"Nodes\" : [] } For more information, see IndividualProperties parameters. Key configuration parameters \u00b6 The tutorial simulates the person-person disease transmission of influenza-like-illness model (SEIR) in a hypothetical community over a short period. You can view the config.json file in the ScenariosHINT01_RiskGroups directory. Set the following parameters: Set Base_Infectivity to 1. Set Base_Infectious_Period to 1.8. Set Base_Incubation_Period to 1. Set Enable_Heterogeneous_Intranode_Transmission to 1. Set Incubation_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Interventions \u00b6 An OutbreakIndividual event seeds the initial infection on day 1. Property_Restrictions targets the outbreak at the low risk group. For more information on targeting an intervention , see Target interventions to nodes or groups in the general documentation set. This simulation uses the campaign.json file in the ScenariosHINT01_RiskGroups directory. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Start_Day\" : 1 , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Event_Coordinator_Config\" : { \"Number_Distributions\" : -1 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Demographic_Coverage\" : 0.001 , \"Target_Demographic\" : \"Everyone\" , \"Property_Restrictions\" : [ \"Risk:Low\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Expected simulation output \u00b6 The output shows the different spreading behavior of the two risk groups. The following graphs show the daily number of infected people for each group in the simulation Although the disease starts in the low risk group, the initial spread in the high-risk group is much faster because of the high transmission rates in the high-risk group, and is the main driver of the overall epidemic. Figure 1: Daily number of infected individuals by risk group Exploring the model \u00b6 Compare the HINT results with homogeneous mixing where people mix equally with each other. To do this, disable the HINT feature by setting Enable_Heterogeneous_Intranode_Transmission to 0 in the config.json file in the ScenariosHINT01_RiskGroups directory. All of the IndividualProperties will be used except for the TransmissionMatrix . Re-run the simulation and compare the following results with the previous results. You can use the sample batch file, plotPropertyReport, in the ScenariosHINT01_RiskGroups directory to generate the following graphs from the PropertyReport output file. Change the transmission rate values in the WAIFW matrix, re-run the model and see the dynamics. Make sure that the HINT feature is enabled by checking that Enable_Heterogeneous_Intranode_Transmission equals 1 in the config.json. Note It is possible that the high-low risk groups in this example are orthogonal to other factors related to transmission, for example, age or urban/rural status. If you would like to explore other factors related to heterogeneous transmission, a secondary axis can be added to IndividualProperties . See HINT 3: Age and accessibility for more information.", "tags": "", "title": "HINT 1: Risk groups \u2014 Training Tutorials  documentation"}, {"url": "./hint-tutorial-scenario2.html", "text": "HINT 2: School closure \u00b6 This tutorial replicates disease transmission dynamics through the different activities of individuals. In this tutorial, a group is defined by the place where individuals go for similar activities but it is not tied to one specific geographic location. For example, \u201cschool\u201d represents all of the students that are going to school instead of having a group for every school that the students attend. As another example, \u201ccommunity\u201d represents places that people meet for varying activities outside of school and work, such as shopping, getting gas, or going to a movie. This tutorial also shows the effect of disease-control policies that modify group mixing during a disease outbreak (for example, the effect of closing schools during a hypothetical pandemic). Students are transitioned from the school group to another group during school closure, and reverted back to the school group when the school is reopened. There are two distinct groups that represent the different activities of school-age children during a school closure: Children who mostly stay at home and the contact rates were largely reduced. Children who spend more time outside of home and have more contacts with other groups Contents Demographics inputs IndividualProperties Key configuration parameters Interventions Run the simulation Expected simulation output Exploring the model Demographics inputs \u00b6 You can view the complete \u201chint_schoolclosure_demographics\u201d file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. IndividualProperties \u00b6 Property and values \u00b6 The Property and Values members of IndividualProperties define five groups. There are three groups that represent the places where individuals normally go when schools are open. There are two groups that represent the place where students go during a school closure. Normal activity groups: School Students who are in school. Work People who are working outside of the home. Community General activities for people who neither work nor go to school. School closure groups: SchoolBreak_Interaction Students who do activities outside of the home during a school break. SchoolBreak_NonInteraction Students who mostly stay at home during a school break. Initial_Distribution \u00b6 At the beginning of the simulation, every individual has a 10% chance of being assigned to \u201cCommunity\u201d, 30% chance to \u201cSchool\u201d, and 60% chance to \u201cWork\u201d as specified by the Initial_Distribution array. Individuals are not assigned to SchoolBreak_Interaction and SchoolBreak_NonInteraction at the beginning of the simulation. Transitions \u00b6 Transitions is used to move individuals from one property value (group) to another. All of the individuals in \u201cSchool\u201d are moved to SchoolBreak_NonInteraction on day 20 and moved back to \u201cSchool\u201d 20 days later (day 40). SchoolBreak_Interaction is not used in this tutorial. It is added to the demographics file so that you can easily run some variations of this simulation. TransmissionMatrix \u00b6 TransmissionMatrix contains the values of the WAIFW matrix . The transmission Route is \u201cContact\u201d. The values are added to Matrix . The first three rows and columns of Matrix demonstrate a hypothetical transmission rate between the three groups: \u201cCommunity\u201d, \u201cSchool\u201d, and \u201cWork\u201d. Students are a main driver of disease transmission. The last two columns and rows demonstrate the different behaviors when students are not in school. Note that the fourth column and row for SchoolBreak_Interaction have a much higher transition rate than SchoolBreak_NonInteraction in the fifth row and column. { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Place\" , \"Values\" : [ \"Community\" , \"School\" , \"Work\" , \"SchoolBreak_Interaction\" , \"SchoolBreak_NonInteraction\" ], \"Initial_Distribution\" : [ 0.1 , 0.3 , 0.6 , 0.0 , 0.0 ], \"Transitions\" : [{ \"From\" : \"School\" , \"To\" : \"SchoolBreak_NonInteraction\" , \"Type\" : \"At_Timestep\" , \"Coverage\" : 1 , \"Timestep_Restriction\" : { \"Start\" : 20 }, \"Age_In_Years_Restriction\" : {}, \"Probability_Per_Timestep\" : 1 , \"Timesteps_Until_Reversion\" : 20 }], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 0.5 , 0.5 , 0.5 , 1 , 0.1 ], [ 0.5 , 3 , 1.5 , 0.0 , 0.0 ], [ 0.5 , 1.5 , 1 , 1 , 0.1 ], [ 1 , 0.0 , 1 , 1 , 0 ], [ 0.1 , 0.0 , 0.1 , 0 , 0.1 ] ] } }] } } Key configuration parameters \u00b6 The simulation simulates the person-person disease transmission of influenza-like-illness model (SEIR) in a hypothetical community over a short period. For more information, see General disease and Demographics parameters. You can view the config.json file in the ScenariosHINT02_SchoolClosure directory. Set the following parameters: Set Base_Infectivity to 1. Set Base_Infectious_Period to 1.8. Set Base_Incubation_Period to 1. Set Enable_Heterogeneous_Intranode_Transmission to 1. Set Incubation_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Interventions \u00b6 An OutbreakIndividual event seeds the initial infection on day 1 of the simulation. For more information, see OutbreakIndividual parameters. You can view the campaign.json file in the in the ScenariosHINT02_SchoolClosure directory. { \"Use_Defaults\" : 1 , \"Events\" : [ { \"Start_Day\" : 1 , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Event_Coordinator_Config\" : { \"Number_Distributions\" : -1 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"outbreak_source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Demographic_Coverage\" : 0.001 , \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"class\" : \"CampaignEvent\" } ] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Expected simulation output \u00b6 The output shows the effect of school break under the assumption that students will reduce their contact during the break ( SchoolBreak_NonInteraction ). The following graph shows the daily number of infected people for each group in the simulation. It is clear that since students are the major driver of transmission, and the fact that students do not interact much with other groups during the break, the school break was able to delay the epidemic for 20 days. Figure 1: School closure - SchoolBreak_NonInteraction Exploring the model \u00b6 Change the transition in intervention to SchoolBreak_Interaction . In this case, students will interact more with other groups, therefore, it is expected that a school break does not necessarily slow down the epidemic.", "tags": "", "title": "HINT 2: School closure \u2014 Training Tutorials  documentation"}, {"url": "./hint-tutorial-scenario3.html", "text": "HINT 3: Age and accessibility \u00b6 This tutorial illustrates the effect of heterogeneity in both mixing between population groups and in access to interventions, specifically vaccination. The scenarios simulate a measles outbreak over a single year in an initially fully susceptible population, highlighting differences in age and accessibility group-specific disease dynamics. It consists of three scenarios that illustrate the measles outbreak and targeted vaccination campaigns: A measles outbreak with no interventions. The supporting configuration and campaign files are in ScenariosHINT03_AgeAndAccessA_BaselineOutbreak. An age-targeted outreach vaccination campaign for 6 to 10 year olds. The outreach is designed to reach populations who are normally inaccessible. The supporting configuration and campaign files are in ScenariosHINT03_AgeAndAccessB_AgeTargetedVaccine. A mass measles vaccination campaign of the entire population. This is a standard campaign, not an outreach campaign, as it targets individuals who are in the accessible group of the population and easily reached by the vaccination campaign. The supporting configuration and campaign files are in ScenariosHINT03_AgeAndAccessC_AccessTargetedVaccine. Note The fully susceptible population in this tutorial is used to illustrate the model features in the most parsimonious manner. For calibration to real data, you would need to initialize a distribution of immunity in the population at the beginning of the epidemic to represent past history of infection and vaccination. This could be done through matching serology data and/or matching vaccination dose history. This tutorial assumes that you have read the previous HINT tutorial, HINT 2: School closure . Contents Demographics inputs IndividualProperties Key configuration parameters Demographics parameters Pathogenesis and immunity parameters Interventions A: Baseline outbreak B: Age-targeted outreach vaccination campaign C: Standard vaccination campaign Exploring the model Demographics inputs \u00b6 You can view the complete hint_ageandaccess_demographics and hint_ageandaccess_overlay files at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. For more information, see Demographics file structure and parameters . IndividualProperties \u00b6 IndividualProperties defines two Property types: Accessibility and Age_Bin . The supporting software for Accessibility is the same as for the property types, Risk and Place , that were used in the previous tutorials. However, the Age_Bin property uses data from the demographics files for initial age distribution, and the membership in an Age_Bin group is updated as an individual ages. Therefore, the Age_Bin attribute has different structures than those of the other property types. For more information, see IndividualProperties parameters. Accessibility: Property and Values \u00b6 Property and Values define two Accessibility groups: \u201cEasy\u201d and \u201cHard\u201d. These two groups will be used to initialize the population with differential vaccination coverage. Individuals are randomly assigned one of these two values at birth and do not transition afterwards. Accessibility: Initial_Distribution \u00b6 At the beginning of the simulation, individuals have an equal chance of being assigned to either the \u201cEasy\u201d or \u201cHard\u201d group as specified in the Initial_Distribution array . Accessibility: Transitions \u00b6 Individuals will not transition from one Accessibility group to the other therefore Transitions is empty. Accessibility: TransmissionMatrix \u00b6 TransmissionMatrix contains the values for the WAIFW matrix that scales the base infectivity. The transmission Route is \u201cContact\u201d. The values are added to Matrix . Matrix specifies the interaction between the \u201cEasy\u201d and \u201cHard\u201d groups. Individuals in the \u201cEasy\u201d group are nine times as likely to be in contact with individuals of their own group. Similarly, individuals in the \u201cHard\u201d accessibility group are five times as likely to interact with their own group. The interpretation for this tutorial is that individuals in the \u201cHard\u201d accessibility group have only 60% of the contacts of those in the \u201cEasy\u201d accessibility group. As an example, \u201cHard\u201d accessibility could be caused by geographic isolation. Age_Bin: Property and Age_Bin_Edges_In_Years \u00b6 The second Property is Age_Bin . Instead of a Values array there is an Age_Bin_Edges_In_Years array that specifies the upper bounds of the age groups. All of the middle values in the Age_Bins_In_Years array specify the upper bound of Age_Bin group that is neither the last nor the first group. This scenario has the following age groups: Birth to age 6 Older than 6 to age 10 Older than 10 to age 20 Older than 20 Age_Bin: Initial_Distribution \u00b6 The Age_Bin property does not support Initial_Distribution . As an age-dependent property, the distribution of the individuals in age bins is determined by the demographics file. Age_Bin: Transitions \u00b6 The Age_Bin property does not use Transitions in IndividualProperties . As an age-dependent property, the membership in an Age_Bin group is updated for individuals as they age. Transitions must be added to the IndividualProperties but it should be empty. Age_Bin: TransmissionMatrix \u00b6 TransmissionMatrix contains the values that scale the base infectivity. The transmission Route is \u201cContact\u201d. The values are added to Matrix . The assumption for this scenario is that individuals interact more frequently with their own age group, and those groups closer in age. Therefore, Matrix has higher values on the diagonal entries. IndividualProperties for Accessibility and Age_Bin \u00b6 { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Accessibility\" , \"Values\" : [ \"Easy\" , \"Hard\" ], \"Initial_Distribution\" : [ 0.5 , 0.5 ], \"Transitions\" : [], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 0.9 , 0.1 ], [ 0.1 , 0.5 ] ] } }, { \"Property\" : \"Age_Bin\" , \"Age_Bin_Edges_In_Years\" : [ 0 , 6 , 10 , 20 , -1 ], \"Transitions\" : [], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 0.4 , 0.2 , 0.1 , 0.3 ], [ 0.2 , 0.4 , 0.1 , 0.3 ], [ 0.1 , 0.2 , 0.6 , 0.1 ], [ 0.1 , 0.1 , 0.1 , 0.7 ] ] } }] } } Key configuration parameters \u00b6 The following sections describe the parameters that define measles pathogenesis and immunity of an individual for all scenarios in this tutorial. Each scenario has a directory (A_BaselineOutbreak, B_AgeTargetedVaccine and C_AccessTargetedVaccine) that contains a simulation configuration files (config.json) file. The three directories are located in ScenariosHINT03_AgeAndAccess. The configuration files are the same in each directory but are copied in each directory to make it more convenient to run simulations. Demographics parameters \u00b6 The following parameters disable vital dynamics and disease mortality in the simulations. HINT is enabled. For more information, see Demographics parameters. Set the following parameters: Set Enable_Vital_Dynamics to 0. Set Death_Rate_Dependence to \u201cNONDISEASE_MORTALITY_BY_AGE_AND_GENDER\u201d. Set Enable_Heterogeneous_Intranode_Transmission to 1. Pathogenesis and immunity parameters \u00b6 Measles is a highly infectious virus that is typically transmitted by large respiratory and aerosolized droplets. Some transmission also occurs via fomites, however this is believed to be a less significant route for transmission as the virus only survives a short time on dry surfaces. The pathogen is characterized by a high basic reproductive number , often estimated at 16-20 combined with short incubation and infectious periods on the order of days. Immunity conferred is typically lifelong and vaccination has high efficacy (~95%). For more information, see General disease parameters. The following parameters define measles pathogenesis and immunity of an individual. Set the following parameters: Set Enable_Immunity to 1. Set Enable_Immune_Decay to 0. Set Base_Infectivity to 4. Set Base_Incubation_Period to 5. Set Base_Infectious_Period to 5. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Interventions \u00b6 Run the simulation and generate graphs of the simulation output for each of the following three scenarios. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. A: Baseline outbreak \u00b6 The simulation seeds an initial measles outbreak with a prevalence of 1% in the age group of 10 to 20 years old who are in the \u201cEasy\u201d accessibility group. Age groups do not use the PropertyRestrictions parameter. It uses the Target_Demographic parameter with the value set to \u201cExplicitAgeRanges\u201d, which will use the values in Target_Age_Min and Target_Age_Max . For more information, see Event coordinators parameters. The campaign file (campaign.json) is in the ScenariosHINT03_AgeAndAccessA_BaselineOutbreak directory. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Start_Day\" : 2 , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.01 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"outbreak_source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Target_Demographic\" : \"ExplicitAgeRanges\" , \"Target_Age_Min\" : 10 , \"Target_Age_Max\" : 20 , \"Property_Restrictions\" : [ \"Accessibility:Easy\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"class\" : \"CampaignEvent\" }] } Expected simulation output \u00b6 In the fixed population of 1000 susceptible individuals, a total of 700 individuals become infected with a peak prevalence of 14% roughly 100 days into the epidemic. The number of individuals in the \u201cEasy\u201d and \u201cHard\u201d groups is equal for each age group as specified in the initial distribution. Looking at the number of infected individuals by age and accessibility groups, notice that the numbers are significantly higher (peak prevalence) in the \u201cEasy\u201d accessibility group compared to \u201cHard\u201d. This is due to the decreased transmission rate specified in the accessibility group matrix. Figure 1: Baseline outbreak B: Age-targeted outreach vaccination campaign \u00b6 This simulation adds an age targeted vaccination on day 1 followed by the same outbreak seeding specified in the first scenario on day 2. The campaign configuration file (campaign.json) is in the ScenariosHINT03_AgeAndAccessB_AgeTargetedVaccine directory. For more information, see SimpleVaccine parameters. { \"Use_Defaults\" : 1 , \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.95 , \"Intervention_Config\" : { \"Cost_To_Consumer\" : 10.0 , \"Durability_Time_Profile\" : \"BOXDURABILITY\" , \"Primary_Decay_Time_Constant\" : 3650 , \"Reduced_Acquire\" : 1 , \"Reduced_Transmit\" : 0 , \"Vaccine_Take\" : 0.95 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"class\" : \"SimpleVaccine\" }, \"Target_Demographic\" : \"ExplicitAgeRanges\" , \"Target_Age_Min\" : 6 , \"Target_Age_Max\" : 10 , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" } ] } Expected simulation output \u00b6 Compare the number of infected-by-age in this simulation with the baseline outbreak scenario. Notice the dramatic decrease (an order of magnitude) of the infection in 6 to 10 year olds due to the vaccination of this group. Compare infections in other ages, noting the overall decrease in infection across age groups despite the fact that only 6 to 10 year olds were vaccinated. Figure 2: Age-targeted outreach vaccination campaign C: Standard vaccination campaign \u00b6 On day 1, a non-targeted vaccination campaign begins which reaches all those individuals in the \u201cEasy\u201d accessibility group but does not reach those in the \u201cHard\u201d accessibility group. This is followed by an outbreak that originates in the \u201cHard\u201d accessibility group in the 0.5 to 6 year olds. In the first two scenarios, the outbreak originated in the \u201cEasy\u201d accessibility group among older children. This outbreak begins in the \u201cHard\u201d accessibility group. For more information, see OutbreakIndividual and SimpleVaccine parameters. The campaign configuration file (campaign.json) is in the ScenariosHINT03_AgeAndAccessC_AccessTargetedVaccine directory. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Start_Day\" : 2 , \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Event_Coordinator_Config\" : { \"Number_Distributions\" : -1 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"outbreak_source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Demographic_Coverage\" : 0.01 , \"Target_Demographic\" : \"ExplicitAgeRanges\" , \"Target_Age_Min\" : 0.5 , \"Target_Age_Max\" : 6.0 , \"Property_Restrictions\" : [ \"Accessibility:Hard\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"class\" : \"CampaignEvent\" }, { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.95 , \"Intervention_Config\" : { \"Cost_To_Consumer\" : 10.0 , \"Durability_Time_Profile\" : \"BOXDURABILITY\" , \"Primary_Decay_Time_Constant\" : 36500 , \"Reduced_Acquire\" : 1 , \"Reduced_Transmit\" : 1 , \"Vaccine_Take\" : 0.95 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"class\" : \"SimpleVaccine\" }, \"Target_Demographic\" : \"Everyone\" , \"Property_Restrictions\" : [ \"Accessibility:Easy\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1 , \"class\" : \"CampaignEvent\" }] } Expected simulation output \u00b6 Notice the dramatic decrease in cumulative incidence (~300 vs. ~700) and peak prevalence from the first scenario because of the vaccination campaign that targeted the \u201cEasy\u201d accessibility group. Figure 3: Easy accessibility targeted vaccine\u2013Accessibility easy group Notice the almost complete elimination of infection in the \u201cEasy\u201d accessibility group due to 95% vaccination coverage with a 95% take. There is only a slight reduction in infection in the \u201cHard\u201d accessibility group which was not reachable by vaccination in this campaign. Recall that the epidemic was seeded in the \u201cHard\u201d accessibility group. Figure 4: Easy accessibility targeted vaccine\u2013Accessibility hard group Exploring the model \u00b6 Adjust both the targeted age and accessibility groups of the initial outbreak. In IndividualProperties , decrease the value of the off-diagonal entries in the TransmissionMatrix . Investigate how this slows down the spread of measles to the other age and accessibility groups in terms of both the speed of the disease spread and total incidence of the disease. For one age and/or accessibility group, set all interactions with other groups to zero. To do this for the n th population group, set all values of the n th row to 0 except for the diagonal entry (n th column and n th row), which should remain unchanged. Use the campaign cost output, which is proportional to the number of doses, to compare the cost- effectiveness of vaccination campaigns targeted at different age and accessibility groups. Also adjust coverage in different groups. For example, the following Matrix values configure the interactions with other group for the third age group (older than 10 to 20 years). { \"Matrix\" : [ [ 0.4 , 0.2 , 0.10 , 0.3 ], [ 0.2 , 0.4 , 0.10 , 0.3 ], [ 0.0 , 0.0 , 0.6 , 0.0 ], [ 0.1 , 0.1 , 0.10 , 0.7 ] ] } Verify that if the epidemic started in a different group, it cannot spread to this group. How do results of the third scenario change if the infection is seeded in the \u201cEasy\u201d accessibility group? See that the infection fails to spread to the \u201cHard\u201d accessibility group. Lower the vaccination coverage of the \u201cEasy\u201d accessibility group to find the vaccination level where the spread to the Hard accessibility group occurs.", "tags": "", "title": "HINT 3: Age and accessibility \u2014 Training Tutorials  documentation"}, {"url": "./hint-tutorial-scenario4.html", "text": "HINT 4: Washington state pertussis \u00b6 The Washington state pertussis tutorial highlights several important features of the 2012 pertussis outbreak in Washington state. These features include the timing and spatial pattern of the spreading outbreak, the relationship between vaccine-derived immunity levels in different population groups and disease incidence over the course of the outbreak, and the effects of within-age-group mixing and the switch from whole-cell to acellular vaccines on incidence in different age groups. The Washington state pertussis tutorial consists of two scenarios: The baseline outbreak scenario initializes the historical vaccination of vaccine-accepting individuals before introducing a pertussis outbreak in Skagit county and letting it spread throughout the state. The supporting configuration and campaign files are in ScenariosHINT04_WAPertussisA_BaselineOutbreak. The target vaccine refusers scenario uses the baseline outbreak setup with the addition of a targeted vaccination before the outbreak of the previously vaccine-refusing populations in Snohomish, King, and Pierce counties. The supporting configuration and campaign files are in ScenariosHINT04_WAPertussisB_TargetVaccineRefusers. This tutorial assumes that you have read the previous HINT tutorial, HINT 3: Age and accessibility . Contents Demographics inputs Geography and population IndividualAttributes IndividualProperties Migration Key configuration parameters Pertussis disease parameters Adapted population sampling Migration Interventions Initialization of population immunity Targeted outbreak Run the simulation Spatial spread Incidence in different groups Incidence by age Exploring the model Demographics inputs \u00b6 You can see the complete hint_wapertussis_demographics file at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. For more information, see Demographics file structure and parameters . Geography and population \u00b6 Each of Washington state counties is specified as a geographical node with population estimates by county from 2013 WA State Office of Financial Management and geographic location (latitude and longitude) centered on county seats. The following code example is for the first county (node) listed in the demographics file. { \"Nodes\" : [{ \"County\" : \"Adams\" , \"NodeAttributes\" : { \"Latitude\" : 47.1274 , \"InitialPopulation\" : 19200 , \"Longitude\" : -118.38 }, \"NodeID\" : 1 }] } IndividualAttributes \u00b6 The initial age distribution is a uniform distribution of individuals between the ages of 0 and 60 years old. { \"Defaults\" : { \"NodeAttributes\" : {}, \"IndividualAttributes\" : { \"AgeDistribution_DESCRIPTION\" : \"Box between age 0 and 60 years: Age_Initialization_Distribution_Type=DISTRIBUTION_SIMPLE\" , \"AgeDistributionFlag\" : 1 , \"AgeDistribution1\" : 0 , \"AgeDistribution2\" : 21900 } } } Age-specific mortality is taken from Washington statewide numbers from CDC 1999-2010 in annual deaths per 1000 individuals. { \"Defaults\" : { \"NodeAttributes\" : {}, \"IndividualAttributes\" : { \"MortalityDistribution_DESCRIPTION\" : \"WA state (1999-2010). Source: wonder.cdc.gov\" , \"MortalityDistribution\" : { \"NumDistributionAxes\" : 2 , \"AxisNames\" : [ \"gender\" , \"age\" ], \"AxisUnits\" : [ \"male=0,female=1\" , \"years\" ], \"AxisScaleFactors\" : [ 1 , 365 ], \"NumPopulationGroups\" : [ 2 , 15 ], \"PopulationGroups\" : [ [ 0 , 1 ], [ 0 , 1 , 2.5 , 7.5 , 12.5 , 17.5 , 22.5 , 30 , 40 , 50 , 60 , 70 , 80 , 90 , 120 ] ], \"ResultUnits\" : \"annual deaths per 1000 individuals\" , \"ResultScaleFactor\" : 0.00000273972602739726027397260273973 , \"ResultValues\" : [ [ 4.8 , 0.2 , 0.1 , 0.2 , 0.5 , 1.1 , 1.1 , 1.7 , 4.1 , 9.2 , 19.8 , 53.7 , 154.2 , 1000 ], [ 4.2 , 0.2 , 0.2 , 0.2 , 0.2 , 0.4 , 0.5 , 1.1 , 2.6 , 5.5 , 14.3 , 40.1 , 129.3 , 1000 ] ] } } } } The birth rate used in the simulation is the value corresponding to stable population replacement. { \"Defaults\" : { \"NodeAttributes\" : { \"BirthRate_DESCRIPTION\" : \"Replacement of stable age distribution: Birth_Rate_Dependence=DEMOGRAPHIC_DEP_RATE (i.e. 14-45 year-old PossibleMothers)\" , \"BirthRate\" : 0.00017675 }, \"IndividualAttributes\" : {} } } IndividualProperties \u00b6 In this simulation IndividualProperties defines two Property types: Accessibility and Age_Bin . The IndividualProperties configuration is in the hint_wapertussis_demographics_overlay file. Accessibility: Property and Values \u00b6 The Property and Values parameters define two Accessibility groups: VaccineTake and VaccineRefuse . Accessibility will be used to initialize the population with differential vaccination coverage. Individuals are randomly assigned one of these two values at birth and do not transition afterwards. { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Accessibility\" , \"Values\" : [ \"VaccineTake\" , \"VaccineRefuse\" ], \"Initial_Distribution\" : [ 0.85 , 0.15 ], \"Transitions\" : [] }] } } Accessibility: Initial_Distribution \u00b6 At the beginning of the simulation, individuals have an 85% chance of being assigned to VaccineTake and a 15% chance to VaccineRefuse as specified by the Initial_Distribution array. Accessibility: Transitions \u00b6 In this simulation, individuals will not transition from one Accessibility group to the other therefore the Transitions structure is empty. Accessibility: TransmissionMatrix \u00b6 The specification of enhanced within-group interaction is specified by the WAIFW matrix between the two groups: VaccineTake and VaccineRefuse . For example, a VaccineRefuse -type individual is 5 times more likely (compared to homogeneous mixing) to be exposed to infections from another individual within their group and 0.3 times as likely to be exposed to an individual outside their group. The transmission Route is \u201cContact\u201d. The values of the WAIFW matrix for the heterogeneous mixing between the Accessibility groups is defined in Matrix . { \"Defaults\" : { \"IndividualProperties\" : [{ \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 1.1 , 0.3 ], [ 0.3 , 5.0 ] ] } }] } } Age_Bin: Property and Age_Bin_Edges_In_Years \u00b6 The second Property is Age_Bin . Three age bins are defined by the Age_Bins_In_Years array with heterogeneous mixing defined by the following TransmissionMatrix. The beginning 0 and ending -1 are mandatory elements in the Age_Bins_In_Years array. The 0 is the lower bound for the first Age_Bin group and specifies birth. The -1 is the upper bound of the last Age_Bin group and specifies the maximum age of the simulation. All of the middle values in the Age_Bins_In_Years array specify the upper bound of Age_Bin group that is neither the last nor the first group. This simulation has the following age groups: Birth to age 5 Older than 5 to age 13 Older than 13 { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Age_Bin\" , \"Age_Bin_Edges_In_Years\" : [ 0 , 5 , 13 , -1 ], \"Transitions\" : [] }] } } Age_Bin: Initial_Distribution \u00b6 The Age_Bin Property type does not support Initial_Distribution . As an age-dependent property, the distribution of the individuals in age bins is determined by the demographics file. Age_Bin: Transitions \u00b6 The Age_Bin property does not use Transitions . As an age-dependent property, the membership in an Age_Bin group is updated for each individual as he or she ages. Transitions must be added to the IndividualProperties but it should be empty. Age_Bin: TransmissionMatrix \u00b6 TransmissionMatrix contains the values of the WAIFW matrix that scale the base infectivity. The transmission Route is \u201cContact\u201d. The heterogeneous mixing between the age groups is defined in Matrix . { \"Defaults\" : { \"IndividualProperties\" : [{ \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 1.4 , 1.0 , 1.0 ], [ 1.0 , 2.5 , 0.7 ], [ 1.0 , 0.7 , 1.0 ] ] } }] } } IndividualProperties for Accessibility and Age_Bin \u00b6 { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Accessibility\" , \"Values\" : [ \"VaccineTake\" , \"VaccineRefuse\" ], \"Initial_Distribution\" : [ 0.85 , 0.15 ], \"Transitions\" : [], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 1.1 , 0.3 ], [ 0.3 , 5.0 ] ] } }, { \"Property\" : \"Age_Bin\" , \"Age_Bin_Edges_In_Years\" : [ 0 , 5 , 13 , -1 ], \"Transitions\" : [], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 1.4 , 1.0 , 1.0 ], [ 1.0 , 2.5 , 0.7 ], [ 1.0 , 0.7 , 1.0 ] ] } }] } } For more information on IndividualProperties , see Heterogeneous Intra-Node Transmission (HINT). Migration \u00b6 The daily migration rates between the various counties are taken from daily commuter data and are packed into the binary file in hint_wapertussis_regional_migration.bin. These rates are most likely a lower bound on actual migration, especially between more distant counties, as they consider only daily commutes and not other longer-range modes. Key configuration parameters \u00b6 Enable_Heterogeneous_Intranode_Transmission must be set to 1 to enable the TransmissionMatrix structure in IndividualProperties . For more information, see Demographics parameters. Each scenario has a directory (A_BaselineOutbreak and B_TargetVaccineRefusers) that contains a configuration files (config.json) file. The two directories are located in ScenariosHINT04_WAPertussis. The configuration files are the same but are copied in each directory to make it more convenient to run simulations. Pertussis disease parameters \u00b6 The generic-disease-model parameters are configured to approximate the dynamics of pertussis. There is a 7-day fixed incubation period followed by an infectious period drawn from an exponential with a mean duration of 22 days. The product of the daily infectiousness (0.6) and the mean infectious duration corresponds to an R 0 of about 14. For more information, see General disease parameters. Set the following parameters: Set Base_Infectivity to 0.6. Set Base_Infectious_Period to 22. Set Base_Incubation_Period to 7. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Upon clearing natural pertussis infections, individuals are configured to have complete protection against re-acquisition for a duration of 4 years. After 4 years, the protection wanes exponentially with a time constant of 16 years. The reduced transmission potential of immune individuals is configured to follow a similar waning pattern. Set the following parameters: Set Immunity_Acquisition_Factor to 0. Set Acquisition_Blocking_Immunity_Duration_Before_Decay to 1460. Set Acquisition_Blocking_Immunity_Decay_Rate to 0.00017. Adapted population sampling \u00b6 The simulation is configured to use adapted sampling rates in the various nodes as a function of population size. This improves the computing time of the simulation while minimizing the effect of over-sampling, (i.e. a compromised representation of the small-number stochastic effects). The maximum number of sampled individuals in any given node is configured to be 1000. For example, a county with 2 million individuals would have 1000 samples each with weight=2000, while a county with 10,000 individuals would have 1000 samples each with weight=10. For more information, see Sampling parameters. Set the following parameters: Set Individual_Sampling_Type to \u201cADAPTED_SAMPLING_BY_POPULATION_SIZE.\u201d Set Max_Node_Population_Samples to 1000. Migration \u00b6 The regional migration between nodes has been configured in a commute-driven mode with all migration events consisting of single round-trips where the migrating individual draws a duration of his or her stay from an exponential with a mean of one day before returning with probability = 1 to the home node. Without regenerating the binary migration file, it is possible to scale all regional migration rates up and down with the x_Regional_Migration multiplicative parameter. For more information, see:ref: Migration <config-migration> parameters. Set the following parameters: Set Migration_Pattern to \u201cSINGLE_ROUND_TRIPS.\u201d Set Regional_Migration_Roundtrip_Duration to 1. Set Regional_Migration_Roundtrip_Probability to 1. Interventions \u00b6 You can see the complete campaign.json in ScenariosHINT04_WAPertussis. Initialization of population immunity \u00b6 A ten-year burn-in allows for the age-structure of the vaccination schedule (including waning) to be initialized. Four vaccine-distribution events are specified in the campaign file, each targeted only at individuals with the \u201cVaccineTake\u201d value for the \u201cAccessibility\u201d property in the demographics file. For more information, see SimpleVaccine . A vaccination calendar to all children under 12 years old, so that the rest of their childhood vaccinations are distributed as they age: 2 month, 4 month, 6 month, 17 month, 5 year, 11.5 year with 90% coverage at the fourth dose and 95% on each of the last two doses. { \"Calendar\" : [{ \"Age\" : 60 , \"Probability\" : 1 }, { \"Age\" : 120 , \"Probability\" : 1 }, { \"Age\" : 180 , \"Probability\" : 1 }, { \"Age\" : 510 , \"Probability\" : 0.9 }, { \"Age\" : 1825 , \"Probability\" : 0.95 }, { \"Age\" : 4200 , \"Probability\" : 0.95 }], \"Dropout\" : 0 , \"class\" : \"IVCalendar\" } A birth-triggered distribution of the same calendar to newborns as they are born throughout the simulation. A health-seeking behavior for people over 12 years old to seek booster doses with a daily tendency (0.000548) corresponding to a 5-year mean waiting time. A dose to over-12s at the beginning of the simulation to prime their immune systems to account for their historic childhood vaccination. The acellular pertussis vaccine is configured such that 95% of vaccinated individuals, for whom the vaccine takes, will have an initial 90% reduction in the probability of acquiring infection per exposure and a 60% reduction in their transmission potential if infected. The protection wanes exponentially (DECAYDURABILITY) with a time constant of approximately 7 years. { \"Cost_To_Consumer\" : 20.0 , \"Durability_Time_Profile\" : \"DECAYDURABILITY\" , \"Primary_Decay_Time_Constant\" : 2525 , \"Reduced_Acquire\" : 0.9 , \"Reduced_Transmit\" : 0.6 , \"Reduced_Mortality\" : 1 , \"Vaccine_Take\" : 0.95 , \"Vaccine_Type\" : \"Generic\" , \"class\" : \"SimpleVaccine\" } The initial immunity given to the over-12 population is intended to capture the slightly more immunogenic whole-cell vaccine, which is configured with 90% initial protection against both acquisition and transmission as well as a longer 20-year decay time. Targeted outbreak \u00b6 As a demonstration of the ability to target an outbreak at both a geographical region (or set of nodes) and at a target population (based on IndividualProperties ), this simulation has two OutbreakIndividual interventions. The first (shown below) is distributed 10 years into the simulation and infects 1% of the \u201cVaccineTake\u201d population in the specified list of nodes (in this case only Skagit county with NodeID = 57). The other intervention, distributed 100 days later, infects 10% of the \u201cVaccineRefuse\u201d population in the same county. { \"Campaign_Name\" : \"Washington State Pertussis Outbreak\" , \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Antigen\" : 0 , \"Genome\" : 0 , \"Outbreak_Source\" : \"PrevalenceIncrease\" , \"class\" : \"OutbreakIndividual\" }, \"Demographic_Coverage\" : 0.01 , \"Number_Repetitions\" : 1 , \"Target_Demographic\" : \"Everyone\" , \"Property_Restrictions\" : [ \"Accessibility:VaccineTake\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"Node_List\" : [ 57 ], \"County_DESCRIPTION\" : \"Skagit\" , \"class\" : \"NodeSetNodeList\" }, \"Start_Day\" : 3650 , \"class\" : \"CampaignEvent\" } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Spatial spread \u00b6 The Plot_Infections_By_Node.py script unpacks the SpatialOutput binary files, which are written out for each channel listed in the configuration parameter, Spatial_Output_Channels , provided that Enable_Spatial_Output is set to 1. For more information, see Output options and Spatial output channels parameters. The following plot shows the daily incidence of new infections for a representative sampling of counties in the simulation. From top to bottom, one can see the characteristic evolution of each local outbreak, starting in Skagit county and progressing southward through Snohomish and King counties down to Clark county on the Oregon border. Similarly, there is a delay before the outbreak makes it over Cascades into the Tri-Cities (Benton and Franklin counties) and Spokane areas. Figure 1: Daily incidence of new infections in representative sample of counties Incidence in different groups \u00b6 The Plot_Infections_by_Property.py script parses the output of the PropertyReport.json file (provided that Enable_Property_Output is set to 1) and plots the output channels grouped according to the property values specified within the script. In the following example, the total number of cases and the population-normalized incidence is compared between the \u201cVaccineTake\u201d and \u201cVaccineRefuse\u201d groups. Figure 2: Population-normalized incidence of new infection by accessibility groups Incidence by age \u00b6 Finally, the Plot_Age_Distributions.py script parses the BinnedReport.json output (provided that Enable_Demographics_Reporting is set to 1) and plots the population in each age bin through time and also the cumulative incidence of infection in age bins as shown below. Figure 3: Population of age bins and cumulated incidence Exploring the model \u00b6 There are a number of different areas that one could explore within this tutorial: Modifying the vaccination coverage, efficacy, and waning values to see the impact on the size and make-up of the effected populations. Modifying the migration rates and county-level variation in vaccine-derived immunity to see the impact on the spatio-temporal pattern of the spreading outbreak. Modifying the heterogeneous mixing within age groups and the relative protective efficacy of the whole-cell and acellular vaccines to see the impact on the age distribution of the effected population. Introducing targeted interventions before the outbreak to minimize its impact. As an example, the last category is provided in the directory ScenariosHINT04_WAPertussisB_TargetVaccineRefusers and consists of a targeted vaccination 100 days before the outbreak that reaches 95% of the previously unvaccinated VaccineRefuse population in the three most-populous counties: King, Snohomish and Pierce. Figure 4: Impact of targeted vaccines in King, Snohomish and Pierce counties", "tags": "", "title": "HINT 4: Washington state pertussis \u2014 Training Tutorials  documentation"}, {"url": "./hint-tutorial-scenario5.html", "text": "HINT 5: Seattle commuting \u00b6 This tutorial simulates the person-person disease transmission of influenza in Seattle over the course of one year. The tutorial assumes that individuals have round-trip travels, or commutes, between census tracts in less than one day. The return time is less than or equal to 8 hours and movement is in equilibrium. Different from homogeneous mixing, the heterogeneous movement links and rates between census tracts will create a timing difference of epidemic trajectories upon initial seeding. The tutorial represents Seattle as a single node. There are 125 groups based on Seattle census tract data. The heterogeneous infectivity between individuals and the 125 groups is represented by a WAIFW matrix with values. The values, which represent the heterogeneous force of infection at each census tract, are based on the movement rate between census tracts and the return rate. Individuals interact mostly with other individuals in their own census tract but also shed into or acquire infection from the census tracts that they commuted to during the day. Contents Demographics inputs Generating Seattle as a single node IndividualProperties Key configuration parameters Interventions Run the simulation Expected simulation output Exploring the model Updating a demographics file Demographics inputs \u00b6 To view the complete demographics file, see hint_seattlecommuting_demographics at <path_to_directory>ScenariosInputFiles where <path_to_directory> is the location where the Quick Start was installed. For more information, see Demographics file structure and parameters . Generating Seattle as a single node \u00b6 The hint_seattlecommuting_demographics file was generated from the multi-node Seattle_30arcsec_demographics file, the Seattle_30arcsec_local_migration file, and the Seattle_30arcsec_regional_migration file using the Python script ConvertMultipleNodeDemographicsToSingleNodePools. The script also added the IndividualProperties attribute that is needed for this tutorial. The ConvertMultipleNodeDemographicsToSingleNodePools Python file, Seattle_Template, which is a copy of the hint_seattlecommuting_demographics file, and the three multi-node input files are in the ScenariosHINT05_SeattleCommutingscript directory. IndividualProperties \u00b6 For more information, see IndividualProperties parameters. Property and Values \u00b6 The Property and Values parameters of IndividualProperties define 125 groups based on Seattle census tracts. The Property type must be \u201cGeographic\u201d as it is the only property that supports 125 Values . The Values are the 3-digit ID for the 125 Seattle census tracts for Seattle. Initial_Distribution \u00b6 Initial_Distribution is an array with 125 elements. See the hint_seattlecommuting_demographics file for those values. Transitions \u00b6 Individuals will not move from one Seattle census track to the other so Transitions is empty. TransmissionMatrix \u00b6 TransmissionMatrix contains the values that scale the base infectivity. The transmission Route is \u201cContact\u201d. The values are added to Matrix . The Matrix was created using a script with a specified return rate. The script loads the existing multi-node demographics and migration input files, and converts them to a single-node demographics file. Each component in the TransmissionMatrix is calculated based on the spatial coupling of round- trip movement flow detailed in the article: Balcan, D., Colizza, V., Gon\u00e7alves, B., et al. (2009). Multiscale mobility networks and the spatial spreading of infectious diseases. PNAS 106(51), 21484-21489. doi: 10.1073/pnas.0906910106. Since the average hours to stay at a destination is 8 hours (1/3 of a day) and the return rate is the inverse of the average hours, the return rate is 3. That is, . There are 125 Seattle census tracts so the Matrix is a 125 x 125 matrix. See the hint_seattlecommuting_demographics file in the ScenarioInputFiles directory to see the complete IndividualProperties for this simulation. { \"Defaults\" : { \"IndividualProperties\" : [{ \"Property\" : \"Geographic\" , \"Values\" : [ \"001\" , \"002\" , \"003\" , \"124\" , \"125\" ], \"Initial_Distribution\" : [ 125 ], \"Transitions\" : [], \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [] } }] } } Key configuration parameters \u00b6 The generic disease model parameters are configured to approximate the dynamics of influenza-like- illness disease. The key influenza disease parameters are listed below. For more information, see General disease parameters. You can see the complete config.json file in the ScenariosHINT05_SeattleCommuting directory. Set the following parameters: Set Base_Infectivity to 0.3. Set Base_Infectious_Period to 7. Set Base_Incubation_Period to 3. Set Enable_Heterogeneous_Intranode_Transmission to 1. Set Incubation_Period_Distribution to \u201cFIXED_DURATION.\u201d Set Infectious_Period_Distribution to \u201cFIXED_DURATION.\u201d Interventions \u00b6 This simulation uses an OutbreakIndividual event to demonstrate the ability to target an outbreak at a geographical region represented as a heterogeneous group. The OutbreakIndividual targets the infection at 1% of the individuals living in census tract 7 at day 20. For more information, see OutbreakIndividual parameters. You can see the complete campaign.json file in the in the ScenariosHINT05_SeattleCommuting directory. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Campaign_Name\" : \"Generic Seattle Regression Campaign\" , \"Event_Coordinator_Config\" : { \"Coverage\" : 1 , \"Demographic_Coverage\" : 0.01 , \"Intervention_Config\" : { \"Antigen\" : 0 , \"Event_Name\" : \"Outbreak\" , \"Genome\" : 0 , \"outbreak_source\" : \"ImportCases\" , \"class\" : \"OutbreakIndividual\" }, \"Number_Distributions\" : 1 , \"Number_Repetitions\" : 1 , \"Target_Age_Max\" : 100 , \"Target_Age_Min\" : 0 , \"Target_Demographic\" : \"Everyone\" , \"Property_Restrictions\" : [ \"Geographic:007\" ], \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 20 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Note Because the EMOD model is stochastic , your graphs may appear slightly different from those given below. Expected simulation output \u00b6 This tutorial shows the timing difference of disease spread between groups due to spatial coupling. It has 125 groups that represent the census tracts in Seattle and simulates a daily commute between the census tracts. The following Inset Charts shows the daily number of infected people for each census tract in the simulation sorted by census tract ID. The census tracts are sequentially ordered with the census tract with the lowest number at the top of the graph. the graph shows the difference in timing between census tracts in terms of the disease spread. Figure 1: Number of infected people per census tract Exploring the model \u00b6 Change the return rates to either increase or decrease the spatial coupling when generating the demographic input and transmission matrix. In general, when the return rates are high, there is less coupling between census tracts so the disease curves of the outbreak are less synchronized. The following example shows a less synchronized behavior of disease spread when the return rate = 24 (people spend 1 hour on average at their destination) with the outbreak targeted at \u201cGeographic: 7\u201d in the demographics file. Updating a demographics file \u00b6 You can manually modify the the Seattle_template provided in the ScenariosHINT05_SeattleCommutingscript directory. However, because of the size of the TransmissionMatrix, you might want to use the Python scripts (also in ScenariosHINT05_SeattleCommutingscript) to modify the Seattle_template and regenerate the demographics file. Update the value of returnRate in ConvertMultipleNodeDemographicsToSingleNodePools using an editor of your choice. Note All of the multi-node demographics files and Seattle_template must be in the same directory. For example, the following Inset Chart graph shows decreased spatial coupling after modifying the demographics file. Figure 2: Decreased coupling between census tracts", "tags": "", "title": "HINT 5: Seattle commuting \u2014 Training Tutorials  documentation"}, {"url": "./index.html", "text": "Welcome to EMOD training \u00b6 The Institute for Disease Modeling (IDM) develops complex software modeling. The primary software, Epidemiological MODeling software (EMOD), helps determine the combination of health policies and intervention strategies that can lead to disease eradication. We share this modeling software with the research community to advance the understanding of disease dynamics. However, if you are not already familiar with epidemiology and disease modeling, it can be challenging to get started with EMOD. This training documentation provides extensive tutorials that walk you through prepared modeling scenarios. In addition, you can easily run simulations with the Quick Start using the provided Windows batch files or within the familiar interface of the Excel Front End. Note The Quick Start can only be run locally on a 64-bit Windows computer and the Excel Front End is only supported on 32-bit Microsoft Excel, though there is a workaround if you do not want to reinstall your version of Excel. See Troubleshooting . The Quick Start is not updated as frequently as the EMOD source, so may not contain all parameters available in the latest version of EMOD. In addition, while you can modify the parameter values, you cannot add or subtract parameters from JSON files within the Excel Front End. Documentation structure \u00b6 IDM recently separated the documentation into this training set and disease-specific sets that provide guidance for researchers modeling particular diseases. If you already have disease modeling experience, we recommend using the documentation that corresponds to disease you study or, if you want to extend the software to study another disease, the general disease documentation set. Introduction to the software Input data files Configuration file Campaign file Running a simulation Output files Install the Quick Start How to run Quick Start simulations Introduction to disease modeling Generic simulation tutorials Generic 1: SIR - Simple single-node outbreak Generic 2: SIR - Targeted vaccination campaigns Generic 3: SIER - Incubation periods Generic 4: SEIR - Vital dynamics (birth and death) Generic 5: SIRS - Waning immunity Generic 6: SEIRS - Disease-specific mortality and endemics Generic 7: SI - Lifelong infection Generic 8: SIS - Reoccurring infections Generic 9: Density scaling of infectivity Generic 10: Animal reservoir zoonosis Heterogeneous Intra-Node Transmission (HINT) tutorials HINT 1: Risk groups HINT 2: School closure HINT 3: Age and accessibility HINT 4: Washington state pertussis HINT 5: Seattle commuting Vector and malaria tutorials Baseline simulation scenarios Vector and malaria 1: Simple vector model Vector and malaria 2: Full malaria model Vector and malaria 3: Impact of climate on seasonality Scenarios with campaigns Vector and malaria 5: Bednet distribution Vector and malaria 6: Bednets and pre-erythrocytic vaccines Vector and malaria 7: Gametocyte-blocking vaccines Vector and malaria 8: Garki project retrospective - IRS and MDA Vector and malaria 9: Sugar-baited traps Vector and malaria 10: Insect-killing fence Vector and malaria 11: Linear combination vector habitat EMOD parameter reference Demographics file structure and parameters Configuration parameters Campaign parameters Troubleshooting Glossary", "tags": "", "title": "Welcome to EMOD training \u2014 Training Tutorials  documentation"}, {"url": "./install-overview.html", "text": "Install the Quick Start \u00b6 The Quick Start contains a pre-built Epidemiological MODeling software (EMOD) executable called Eradication.exe . It also contains directories for many different real-world scenarios that you can model, each of which includes the necessary input data files , configuration file , and campaign file . You can easily run simulations with the provided Windows batch files or within the familiar interface of the Excel Front End. After you complete installation, we recommend that you walk through the tutorials that describe the underlying mathematics of EMOD, starting with the generic tutorials that demonstrate the basic SEIR model framework before moving on to the disease-specific tutorials. Note The Quick Start can only be run locally on a 64-bit Windows computer and the Excel Front End is only supported on 32-bit Microsoft Excel, though there is a workaround if you do not want to reinstall your version of Excel. See Troubleshooting . The Quick Start is not updated as frequently as the EMOD source, so may not contain all parameters available in the latest version of EMOD. In addition, while you can modify the parameter values, you cannot add or subtract parameters from JSON files within the Excel Front End. Download the Quick Start \u00b6 Go to the EMOD-QuickStart GitHub repository. From the Clone or download menu, select Download ZIP . On your local computer, download and unzip the file to a location of your choice. Warning You must not save the Quick Start to a network drive or include any spaces in the path to the location or the environment setup will fail. Within the EMOD-QuickStart-master directory, you will see Eradication.exe, the EMOD application, various files for setting up the environment and running simulations, and the Scenarios directory, which contains all required files for modeling various scenarios described in the tutorials. Within the Scenarios directory, each subdirectory corresponds to a tutorial section and contains the scenarios associated with those tutorials. The exceptions to this rule are EmbeddingPython, which does not yet have tutorials, InputFiles, which contains the input data files used by all scenarios, and Scripts, which contains the scripts used to plot the simulation output. Set up the Quick Start environment \u00b6 The Quick Start assumes all directories are located on the virtual Z drive, regardless of where you physically save them. The following setup script configures environment variables so that the tools, such as the Excel Front End, can access the correct location. Open the EMOD-QuickStart-master directory. Double-click the EMOD Environment shortcut to launch the command window in the virtual Z drive. If a Security Warning appears, click Open to ignore it. This runs the setenv.bat file, which temporarily sets up the Path and Python_Path environment variables. From the command line, enter dir . The directories and files will be listed if the environment is correctly set up. If the directories and files are not listed, this indicates that your virtual Z drive is already being used. Use the steps below to select another virtual drive and update some of the Quick Start files to use the other virtual drive. (Optional) Set up a different virtual drive \u00b6 In following example, the Quick Start is edited to use the virtual drive A. Modify the instructions as necessary if you want to use a different virtual drive. Open the EMOD-QuickStart-master directory. In the setenv.cmd script, replace all occurrences of \u201cZ\u201d with \u201cA\u201d. In the emod.bat script, replace all occurrences of \u201cZ\u201d with \u201cA\u201d. In each scenario directory, in each Excel configuration file, in cell B2 of the Commissioning sheet, replace \u201cZ\u201d with \u201cA\u201d. If you want to use embedded Python, in ScenariosEmbeddingPython, in the emod.bat script, replace all occurrences of \u201cZ\u201d with \u201cA\u201d. After updating the files to the virtual drive A, double-click the EMOD Environment shortcut to see if the command window opens in the virtual drive A and displays the Quick Start directories and files. (Optional) Install plotting software \u00b6 None of the following plotting software is required to run simulations with the Quick Start, but they are useful for creating graphs from and evaluating the model output. When you install the Quick Start, Python and and the package manager, pip, is also installed. When you run Quick Start simulations, the Quick Start automatically runs Python scripts to create graphs of the output. These scripts also require you to install the software packages listed below. However, most of these graphs are reproduced in the documentation if you do not want to install additional software to create the graphs yourself. Note that all batch files assume that Python is installed at C:Python27. If you already have Python installed in a different location, use a plain text editor to modify the path in the batch files. Note IDM does not provide support or guarantees for any third-party software, even software that we recommend you install. Send feedback if you encounter any issues, but any support must come from the makers of those software packages and their user communities. We recommended that you download some of the Python packages from http://www.lfd.uci.edu/~gohlke/pythonlibs , a page compiled by Christoph Gohlke, University of California, Irvine. The libraries there are compiled Windows binaries, including the 64-bit versions required by EMOD. Python utilities \u00b6 The Quick Start Python scripts require the Python utilities dateutil, six, and pyparsing. The Python utilities dateutil, six, and pyparsing provide text parsing and datetime functionality. Open a Command Prompt window. Enter the following commands: pip install python - dateutil pip install pyparsing NumPy \u00b6 The Quick Start Python scripts require the NumPy Python package. The NumPy package adds support for large, multi-dimensional arrays and matrices to Python. Go to http://www.lfd.uci.edu/~gohlke/pythonlibs/#numpy and select the WHL file for NumPy 1.11.3 (64-bit) that is compatible with Python 2.7.11 or 2.7.12. Save the file to your Python installation directory. Open a Command Prompt window and navigate to the Python installation directory, then enter the following, substituting the name of the specific NumPy WHL file you downloaded: pip install numpy - 1. x . x + mkl - cp27m - win_amd64 . whl Matplotlib \u00b6 The Quick Start Python scripts require the matplotlib Python library. Matplotlib is a Python library for making publication quality plots using syntax familiar to MATLAB users. Matplotlib also uses NumpPy for numeric manipulation. Output formats include PDF, Postscript, SVG, and PNG, as well as a screen display. Go to http://www.lfd.uci.edu/~gohlke/pythonlibs/#matplotlib and select the WHL file for Matplotlib 1.5.3 (64-bit) that is compatible with Python 2.7.11 or 2.7.12. Save the file to your Python installation directory. Open a Command Prompt window and navigate to the Python installation directory, then enter the following, substituting the name of the specific matplotlib WHL file you downloaded: pip install matplotlib - 1. x . x + mkl - cp27m - win_amd64 . whl", "tags": "", "title": "Install the Quick Start \u2014 Training Tutorials  documentation"}, {"url": "./intro-campaign.html", "text": "Campaign file \u00b6 The campaign file is an optional file that distributes outbreaks and contains all parameters that define the collection of interventions that make up a disease eradication campaign . For example, When, how, and who to test for a disease When, how, and who to distribute treatments to The vaccines or other preventative interventions to use The medications to use for treatment Like the configuration file , the campaign file is a JavaScript Object Notation (JSON) formatted file. It is hierarchically organized into logical groups of parameters that can have arbitrary levels of nesting. For some interventions, there can be a very complex hierarchical structure, including recursion. Typically, the file is named campaign.json. The relative path to this file is specified by Campaign_Filename in the configuration file. To distribute an intervention, you must configure the following: campaign event A JSON object that determines when and where an intervention is distributed during a campaign. event coordinator A JSON object that determines who will receive a particular intervention during a campaign. intervention A JSON object that determines what will be distributed to reduce the spread of a disease. An intervention can be distributed either to an individual (such as a vaccine, drug, or bednet) or to a node (such as a larvicide). Sometimes this can be an intermediate intervention that schedules another intervention. The Quick Start includes complete simulation scenarios in the Scenarios directory. Within each of the scenario subdirectories, there is a campaign.json file. Overlay file \u00b6 As you work more with EMOD, you may want to change the values of a view parameters of interest while keeping the rest constant, for example, when running simulation experiments or testing source code changes. You have the option of creating an overlay file to keep campaign parameters of interest in a separate file from the complete campaign file that contains default values. These files can be flattened into a single file and the values in the overlay file will override those in the default file. While some of the scenarios in the Quick Start include an overlay file, it is only to highlight certain campaign settings, not override any default values. See Campaign parameters for a comprehensive list and description of all parameters available to use in the campaign file for this simulation type.", "tags": "", "title": "Campaign file \u2014 Training Tutorials  documentation"}, {"url": "./intro-configuration.html", "text": "Configuration file \u00b6 The primary means of configuring the disease simulation is the configuration file . This required file is a JavaScript Object Notation (JSON) formatted file that is typically named config.json. The configuration file controls many different aspects of the simulation. For example, The names of the campaign file and input data files to use Simulation-wide demographics, climate, and migration data General disease attributes such as infectivity, immunity, mortality, and so on Attributes specific to the disease type being modeled, including treatment efficacy The reports to output from the simulation The Quick Start includes complete simulation scenarios in the Scenarios directory. Within each of the simulation subdirectories, there is a config.json file. The following is an example of configuration file. { \"parameters\" : { \"Acquisition_Blocking_Immunity_Decay_Rate\" : 0.1 , \"Acquisition_Blocking_Immunity_Duration_Before_Decay\" : 60 , \"Age_Initialization_Distribution_Type\" : \"DISTRIBUTION_SIMPLE\" , \"Animal_Reservoir_Type\" : \"NO_ZOONOSIS\" , \"Base_Incubation_Period\" : 0 , \"Base_Individual_Sample_Rate\" : 1 , \"Base_Infectious_Period\" : 4 , \"Base_Infectivity\" : 3.5 , \"Base_Mortality\" : 0 , \"Base_Population_Scale_Factor\" : 1 , \"Birth_Rate_Dependence\" : \"POPULATION_DEP_RATE\" , \"Birth_Rate_Time_Dependence\" : \"NONE\" , \"Burnin_Cache_Mode\" : \"none\" , \"Burnin_Cache_Period\" : 0 , \"Burnin_Name\" : \"\" , \"Campaign_Filename\" : \"campaign.json\" , \"Climate_Model\" : \"CLIMATE_OFF\" , \"Config_Name\" : \"00_DEFAULT\" , \"Custom_Reports_Filename\" : \"NoCustomReports\" , \"Death_Rate_Dependence\" : \"NONDISEASE_MORTALITY_OFF\" , \"Default_Geography_Initial_Node_Population\" : 1000 , \"Default_Geography_Torus_Size\" : 10 , \"Demographics_Filenames\" : [ \"../00_Default/demographics.json\" ], \"Enable_Absolute_Time\" : \"NO\" , \"Enable_Aging\" : 1 , \"Enable_Birth\" : 1 , \"Enable_Default_Reporting\" : 1 , \"Enable_Default_Shedding_Function\" : 1 , \"Enable_Demographics_Birth\" : 0 , \"Enable_Demographics_Builtin\" : 0 , \"Enable_Demographics_Gender\" : 1 , \"Enable_Demographics_Other\" : 0 , \"Enable_Demographics_Reporting\" : 1 , \"Enable_Disease_Mortality\" : 0 , \"Enable_Heterogeneous_Intranode_Transmission\" : 0 , \"Enable_Immune_Decay\" : 0 , \"Enable_Immunity\" : 1 , \"Enable_Interventions\" : 1 , \"Enable_Maternal_Transmission\" : 0 , \"Enable_Property_Output\" : 0 , \"Enable_Spatial_Output\" : 0 , \"Enable_Superinfection\" : 0 , \"Enable_Vital_Dynamics\" : 0 , \"Geography\" : \"\" , \"Immunity_Acquisition_Factor\" : 0 , \"Immunity_Initialization_Distribution_Type\" : \"DISTRIBUTION_OFF\" , \"Immunity_Mortality_Factor\" : 0 , \"Immunity_Transmission_Factor\" : 0 , \"Incubation_Period_Distribution\" : \"FIXED_DURATION\" , \"Individual_Sampling_Type\" : \"TRACK_ALL\" , \"Infection_Updates_Per_Timestep\" : 1 , \"Infectious_Period_Distribution\" : \"EXPONENTIAL_DURATION\" , \"Infectivity_Scale_Type\" : \"CONSTANT_INFECTIVITY\" , \"Job_Node_Groups\" : \"Chassis08\" , \"Job_Priority\" : \"BELOWNORMAL\" , \"Listed_Events\" : [], \"Load_Balance_Filename\" : \"\" , \"Local_Simulation\" : 0 , \"Maternal_Transmission_Probability\" : 0 , \"Max_Individual_Infections\" : 1 , \"Max_Node_Population_Samples\" : 40 , \"Migration_Model\" : \"NO_MIGRATION\" , \"Minimum_Adult_Age_Years\" : 15 , \"Mortality_Blocking_Immunity_Decay_Rate\" : 0.001 , \"Mortality_Blocking_Immunity_Duration_Before_Decay\" : 60 , \"Mortality_Time_Course\" : \"DAILY_MORTALITY\" , \"Node_Grid_Size\" : 0.042 , \"Num_Cores\" : 1 , \"Number_Basestrains\" : 1 , \"Number_Substrains\" : 1 , \"PKPD_Model\" : \"FIXED_DURATION_CONSTANT_EFFECT\" , \"Population_Density_C50\" : 30 , \"Population_Density_Infectivity_Correction\" : \"CONSTANT_INFECTIVITY\" , \"Population_Scale_Type\" : \"USE_INPUT_FILE\" , \"Report_Event_Recorder\" : 0 , \"Run_Number\" : 1 , \"Sample_Rate_0_18mo\" : 1 , \"Sample_Rate_10_14\" : 1 , \"Sample_Rate_15_19\" : 1 , \"Sample_Rate_18mo_4yr\" : 1 , \"Sample_Rate_20_Plus\" : 1 , \"Sample_Rate_5_9\" : 1 , \"Sample_Rate_Birth\" : 1 , \"Serialization_Test_Cycles\" : 0 , \"Simulation_Duration\" : 90 , \"Simulation_Timestep\" : 1 , \"Simulation_Type\" : \"GENERIC_SIM\" , \"Start_Time\" : 0 , \"Susceptibility_Scale_Type\" : \"CONSTANT_SUSCEPTIBILITY\" , \"Transmission_Blocking_Immunity_Decay_Rate\" : 0.1 , \"Transmission_Blocking_Immunity_Duration_Before_Decay\" : 60 , \"x_Air_Migration\" : 1 , \"x_Birth\" : 1 , \"x_Local_Migration\" : 1 , \"x_Other_Mortality\" : 1 , \"x_Population_Immunity\" : 1 , \"x_Regional_Migration\" : 1 , \"x_Sea_Migration\" : 1 , \"x_Temporary_Larval_Habitat\" : 1 } } Overlay file \u00b6 As you work more with EMOD, you may want to change the values of a view parameters of interest while keeping the rest constant, for example, when running simulation experiments or testing source code changes. You have the option of creating an overlay file to keep configuration parameters of interest in a separate file from the complete configuration file that contains default values. These files can be flattened into a single file and the values in the overlay file will override those in the default file. While some of the scenarios in the Quick Start include an overlay file, it is only to highlight certain configuration settings, not override any default values. See Configuration parameters for a comprehensive list and description of all parameters available to use in the configuration file for this simulation type.", "tags": "", "title": "Configuration file \u2014 Training Tutorials  documentation"}, {"url": "./intro-inputs.html", "text": "Input data files \u00b6 EMOD accepts the following categories of input data files that contain the relatively fixed information about the population within each geographic node. For example, the number of individuals, geographic data, climate, demographics, and migration data. This is in contrast to the demographic, geographic, and migration parameters in the configuration file that control simulation-wide qualities, such as enabling air migration across all nodes in the simulation. Although a demographics file is the only required input data file, additional files are generally needed for a realistic simulation. The demographics files use JavaScript Object Notation (JSON) . The other input data files use both a JSON file for metadata and an associated binary file that contains the actual data. The Quick Start includes all necessary input data files in the Scenarios/InputFiles directory. Demographics \u00b6 Demographics files are JSON formatted files containing information on the demographics of the population in a geographical region to simulate. For example, the number of individuals and the distribution for age, gender, immunity, risk, and mortality. In addition, demographics files are useful for creating heterogeneous groups within a population. You can define values for accessibility, age, geography, risk, and other properties and assign individuals to groups based on those property values. For example, you might want to divide a population up into different bins based on age so you can target interventions to individuals in a particular age range. Another common use is to configure treatment coverage to be higher for individuals in easily accessible regions and lower for individuals in areas that are difficult to access. EMOD assumes homogeneous mixing and disease transmission for the generic simulation type. You can use the HINT feature to add heterogeneous transmission to your generic model. You cannot manually configure heterogeneous transmission using HINT with other simulation types because the heterogeneity in transmission for specific diseases and disease classes is already configured by the simulation type. You can specify multiple demographics files, which function as a \u201cbase layer\u201d file and one or more \u201coverlay\u201d files that override the base layer configuration. Overlay files can change the value of parameters already specified in the base layer or add new parameters. Support for multiple demographics layers allows for the following scenarios: Separating different sets of parameters and values into individual layers (for example, to separate those that are useful for specific diseases into different layers) Adding new parameters for a simulation into a new layer for easier prototyping Overriding certain parameters of interest in a new layer Overriding certain parameters for a particular sub-region Simulating subsets of a larger region for which input data files have been constructed Migration \u00b6 Migration files describe the rate of migration of individuals in and out of a geographic node . There are four types of migration files that can be used by EMOD: local migration, regional migration, air migration, and sea migration. For all types, migration data is contained in a set of two files, a JSON metadata file with header information and a binary data file with the actual migration data. Both files are required. The basic file structure is identical for all types, with the only exception being the number of columns per row allowed to each type. Local \u00b6 Local migration describes the foot travel of people into and out of adjacent nodes. A local migration file is required for simulations that support more than one node. Regional \u00b6 Regional migration describes migration via a road or rail network. If a node is not part of the network, the regional migration of individuals to and from that node considers the closest road hub city. When you create the migration file, you must create a Voronoi tiling based on road hubs of the region, with each non-hub connected to the hub of its tile. Air \u00b6 Air migration describes migration via airplane travel. It is usually required for simulations of an entire country or larger geographies. Sea \u00b6 Sea migration describes migration via ship. Climate \u00b6 There are two general types of climate files usable by EMOD: climate files generated through actual data, referred to as \u201cclimate by data,\u201d and climate files generated from the Koppen classification system, referred to as \u201cclimate by Koppen.\u201d For both types, climate data is contained in a set of two files, a JSON metadata file with header information and a binary file that contains the actual climate data. Both files are required. Climate by data \u00b6 Climate by data files contain real data gathered from weather stations in the region to be simulated. This includes rainfall, temperature, relative humidity, and so on. Climate by Koppen \u00b6 Climate by Koppen files contain the Koppen classifier for the region. The Koppen classification system is one of the most widely used climate classification systems. It makes the assumption that native vegetation is the best reflection of climate.", "tags": "", "title": "Input data files \u2014 Training Tutorials  documentation"}, {"url": "./intro-modeling.html", "text": "Introduction to disease modeling \u00b6 To understand the complex dynamics underlying disease transmission, epidemiologists utilize a set of models called compartmental models. Developed in the early 20th century, these models stratify a population into groups, generally based on their risk or infection status. Underlying these models is a system of differential equations, so the model is able to track the population as a function of time. Further, the models track the number of people in each category, are able to predict the various properties of pathogen spread, can estimate the duration of epidemics, and can be used to understand how different situations or interventions can impact the outcome of pathogen spread. To see a typical plot of a population in SIR conditions, see the plot in Types of compartmental models . If you would like a more in-depth introduction to epidemiology and disease modeling, you may want to take the Epidemics course from The Pennsylvania State University through Coursera. An agent-based model (ABM) is another powerful tool used to help understand the complexity inherent in disease transmission systems. These models form a type of \u201cmicroscale model,\u201d where they simulate the simultaneous interactions of agents in an effort to re-crate complex phenomena. Each agent (such as a human, or vector) can be assigned a variety of \u201cproperties\u201d (for example, age, gender, etc), and their behavior and interactions with one another are determined by using decision rules. These models have strong predictive power, and are able to leverage spatial and temporal dynamics. Further, complex environments can be developed in which the agents act, and agents may \u201clearn\u201d from interactions or \u201cadapt\u201d to their environment. As a result, ABMs are excellent for identifying \u201cemerging properties\u201d of the system: patterns that are not explicitly modeled, but instead occur as a consequence of the rules that govern the agents. Agent-based models are extensively used in epidemiology due to their predictive power in modeling the spread (or conversely, control) of epidemics. A popular type of ABM for this is one in which each agent\u2019s rules follow the dynamics specified in the compartmental models, where each agent flows through the compartments as a function of both \u201cwithin-host\u201d rules (such as duration of infection) and interactions between agents (such as becoming infected when coming into contact with an infectious agent). By combining the epidemiological basis of compartmental models with the flexibility of an agent-based model, this type of ABM is quite powerful due to their ability to simultaneously address the ecology, epidemiology, and pathology of complex systems. Why use disease modeling \u00b6 Disease models play an important role in understanding and managing the transmission dynamics of various pathogens. We can use them to describe the spatial and temporal patterns of disease prevalence, as well as to explore or better understand the factors that influence infection incidence. Modeling is a key step in understanding what treatments and interventions can be most effective, how cost-effective these approaches may be, and what specific factors need to be considered when trying to eradicate disease. Types of compartmental models \u00b6 The following diagrams illustrate common compartmental models. See the Generic simulation tutorials in the Training documentation set for more information, as well as exercises on running EMOD simulations utilizing these models. SIR Plot, showing typical categorization of a population into Susceptible, Infectious, and Recovered states. Definitions \u00b6 Susceptible Individual is able to become infected. Exposed Individual has been infected with a pathogen, but due to the pathogen\u2019s incubation period, is not yet infectious. Infectious Individual is infected with a pathogen and is capable of transmitting the pathogen to others. Recovered Individual is either no longer infectious, or \u201cremoved\u201d from the population. SI (Susceptible - Infectious) model \u00b6 In this situation, people never leave the infectious state and have life-long infections. For example, herpes is a disease with life-long infectiousness. The dashed line shows how the model becomes an SIS (Susceptible - Infectious - Susceptible) model, where infection does not confer immunity (or there is waning immunity). Individuals have repeat or reoccurring infections, and infected individuals return to the susceptible state. For example, sexually transmitted diseases such as gonorrhea or chlamydia fall into this group. SI - SIS model SIR (Susceptible - Infectious - Recovered) model \u00b6 In this category, individuals in the recovered state gain immunity to the pathogen. For example, measles, mumps, rubella, and pertussis may be modeled using the SIR framework. The dashed line shows how the model becomes an SIRS (Susceptible - Infectious - Recovered - Susceptible) model, where recovery does not confer life-long immunity, and individuals may become susceptible again. SIR - SIRS model SEIR (Susceptible - Exposed - Infectious - Recovered) model \u00b6 In this category, individuals experience a long incubation duration (the \u201cexposed\u201d category), such that the individual is infected but not yet infectious . For example chicken pox, and even vector-borne diseases such as Dengue Hemorrhagic fever have a long incubation duration where the individual cannot yet transmit the pathogen to others. The dashed line shows how the model becomes an SEIRS (Susceptible - Exposed - Infectious - Recovered - Susceptible) model, where recovered people may become susceptible again (recovery does not confer life-long immunity). For example, rotovirus and malaria are diseases with long incubation durations, and where infection only confers temporary immunity. SEIR - SEIRS model Fundamental concepts in epidemiology and disease modeling \u00b6 Basic reproductive number (R 0 ) The average number of secondary infections generated by the first infectious individual in a population of completely susceptible individuals. R 0 =S*L* (where S = the number of susceptible hosts, L = length of infection, and = transmissibility). When R 0 > 1, disease will spread. It is essentially a measure of the expected or average outcome of transmission. This is the threshold parameter used to determine whether or not an epidemic will occur, and determines: The initial rate of increase of an epidemic (the exponential growth phase). The final size of an epidemic (what fraction of susceptibles will be infected). The endemic equilibrium fraction of susceptibles in a population (=1/ R 0 ). The critical vaccination threshold, which is equal to 1-(1/ R 0 ), and determines the number of people that must be vaccinated to prevent the spread of a pathogen. Disability-adjusted life years (DALY) The number of years of life lost due to premature mortality plus the years lost due to disability while infected. Used to quantify the burden of disease. Epidemic An outbreak of an infectious disease, such that a greater number of individuals than normal has the disease. Epidemics have very high R 0 (Recall R 0 >1 for a disease to spread) and are often associated with acute, highly transmissible pathogens that can be directly transmitted. Further, pathogens with lower infectious periods create more explosive epidemics. To control epidemics, it is necessary to reduce R 0 . This can be done by: Reducing transmissibility. Decreasing the number of susceptibles (by vaccination, for example). Decreasing the mean number of contacts or the transmissibility, such as by improving sanitation, or limiting the number of interactions sick people have with healthy people. Reducing length of infectious period. Force of infection (FoI) A measure of the degree to which an infected individual can spread infection; the per-capita rate at which susceptibles contract infection. Typically increases with transmissibility and prevalence of infection. Herd immunity Vaccines protect individuals, but also provide indirect protection to anyone those vaccinated people may have infected. The portion of the population that needs to be immunized in order to achieve herd immunity is P > 1 \u2013 (1/ R 0 ), where P = proportion vaccinated * vaccine efficacy. Incidence (of a pathogen) The number of new cases or infections in a given time period. Immune Unable to become infected/infectious Prevalence (of a pathogen) The proportion of a population that is infectious at any given time. Transmissibility ( ) Also known as the effective contact rate, is the product of the contact rate and the probability of transmission per contact. Virulence The capacity of a pathogen to produce disease. It is proportional to parasitemia, or the number of circulating copies of the pathogen in the host. The higher the virulence (given contact between S and I individuals), the more likely transmission is to occur. However, higher virulence means contact may be less likely as infected hosts show more symptoms of the disease. There is a trade-off that occurs between high transmissibility and disease-induced mortality.", "tags": "", "title": "Introduction to disease modeling \u2014 Training Tutorials  documentation"}, {"url": "./intro-outputs.html", "text": "Output files \u00b6 After running a simulation, the simulation data is extracted, aggregated, and saved as an output report under the output subdirectory within each scenario. The Quick Start produces the output reports InsetChart.json, which contains the simulation-wide average disease prevalence , incidence , and other measurements by time step , and BinnedReport.json, which contains the same information divided into bins. For example, prevalence for different age ranges. To visualize the output, you can then plot any measurement in the output data within the Excel Front End for the most flexibility. Or, you can run Windows batch files to plot the InsetChart.json data and additional data, such as campaign cost, births, deaths, recovered population, and more.", "tags": "", "title": "Output files \u2014 Training Tutorials  documentation"}, {"url": "./intro-overview.html", "text": "Introduction to the software \u00b6 The Institute for Disease Modeling (IDM) develops detailed simulations of disease transmission through the use of extensive and complex software modeling. The primary software, Epidemiological MODeling software (EMOD), helps determine the combination of health policies and intervention strategies that can lead to disease eradication. EMOD calculates how diseases may spread in particular areas and is used to analyze the effects of current and future health policies and intervention strategies. It supports infectious disease campaign planning, data gathering, new product development, and policy decisions. We share this modeling software with the research community to advance the understanding of disease dynamics. EMOD supports the following simulation types for modeling a variety of diseases: Generic disease (GENERIC_SIM) Vector-borne diseases (VECTOR_SIM) Malaria (MALARIA_SIM) Tuberculosis (TB_SIM) Sexually transmitted infections (STI_SIM) HIV (HIV_SIM) The illustration below shows how the simulation types are built upon one another. All parameters available in the generic simulation type are inherited by the vector simulation type. The vector simulation type adds additional parameters specific to the biology of vector-borne diseases, which in turn are inherited by the malaria simulation type and so on. Therefore, depending on the simulation type you select, different parameters are available for you to use. Simulation type inheritance EMOD is a stochastic , agent-based model that simulates the actions and interactions of individuals within geographic areas known as nodes to understand the disease dynamics in a population over time. EMOD can produce statistically significant results over a broad set of parameters and scenarios. Quantitative analysis of the simulated output enables disease eradication efforts to make more data-driven decisions. The IDM research team has published many articles related to modeling, as well as the modeling concepts underpinning EMOD. For a list of published articles, see IDM Publications . While EMOD is very powerful, it can be challenging to learn if you are not already familiar with epidemiology and disease modeling. If that is the case, IDM provides the Quick Start for you to learn more about modeling disease dynamics. It has a simpler interface for running simulations and extensive tutorials that walk you through how to model several prepared real- world scenarios. The Quick Start is installed separately from, and has fewer installation requirements than, the EMOD source. The Quick Start is built on the same modeling software, but has a few limitations that make it primarily a learning tool. It is not suitable for running large-scale simulations or running simulations using your own data. The Quick Start provides the following: All files needed to model many different prepared real-world scenarios Extensive tutorials that walk you through how to model the scenarios An Excel Front End and Windows batch files to quickly and easy run simulations Windows batch files to create graphs of the model output Built EMOD executable (Eradication.exe) Note The Quick Start can only be run locally on a 64-bit Windows computer and the Excel Front End is only supported on 32-bit Microsoft Excel, though there is a workaround if you do not want to reinstall your version of Excel. See Troubleshooting . The Quick Start is not updated as frequently as the EMOD source, so may not contain all parameters available in the latest version of EMOD. In addition, while you can modify the parameter values, you cannot add or subtract parameters from JSON files within the Excel Front End. This section provides the conceptual overview of EMOD. The architecture diagram below shows, at a high level, how the system functions. If you run simulations in parallel on a multi-node cluster, there is also a Message Passing Interface (MPI) component used to pass data between multiple instances of EMOD. High-level EMOD system architecture Input data files \u00b6 Not all files accepted as input for running a simulation are considered input data files. Rather, input data files contain the relatively fixed information about a population to model. For example, climate, geography, demographics, and migration data. Configuration file \u00b6 The configuration file contains parameters that control many different aspects of the simulation. EMOD provides hundreds of parameters for you to configure your simulation. For example, configuration parameters can control the following: The disease or disease class to simulate (simulation type) The name and location for the input data files and output files Whether to include births, deaths, and migration Disease attributes, such as infectivity, transmission, immunity, and mortality The computing resources to use Campaign file \u00b6 The campaign file contains parameters that distribute outbreaks and the interventions used to control the spread of disease. For example, campaign parameters can control the following: Target demographic (age, location, gender, etc.) for interventions Diagnostic tests to use The cost and timing of interventions Running a simulation \u00b6 The EMOD executable (Eradication.exe) accepts the input data, configuration, and campaign files and then simulates the susceptibility and infection of individuals within each geographic node . The simulation type controls the transmission mechanism of the disease. After the simulation is complete, the Eradication.exe produces output reports that describe various aspects of the disease dynamics within the modeled population. Output files \u00b6 After the simulation finishes, a reporter extracts simulation data, aggregates it, and outputs it to a file (known as an output report ). Most of the reports are also JSON files, the most important of which is InsetChart.json. The InsetChart.json file provides simulation-wide averages of disease prevalence at each time step . See the topics listed below for a more detailed explanation of each of these system components. Input data files Configuration file Campaign file Running a simulation Output files", "tags": "", "title": "Introduction to the software \u2014 Training Tutorials  documentation"}, {"url": "./intro-run-simulation.html", "text": "Running a simulation \u00b6 The Quick Start provides the simplest way to run simulations. You can run simulations within Microsoft Excel using the Excel Front End or with Windows command and batch files. Both of these methods will run a single simulation locally. Installing EMOD from the source provides many more options for running simulations. For example, you can run one or more simulations, run simulations locally or on a remote cluster (for large simulations or multiple simulations), or run simulations for debugging the source code.", "tags": "", "title": "Running a simulation \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-1-simple.html", "text": "Vector and malaria 1: Simple vector model \u00b6 In this tutorial you will configure and run a simple, three-species vector simulation in Namawala. You will calculate the baseline EIR, and by varying the parameter x_Temporary_Larval_Habitat , observe the dependence of EIR on larval habitat. (See Scaling Factors.) Use the config.json and campaign.json in the following directories for this tutorial. ScenariosVectorAndMalaria01_A_Namawala3Vectors_lowEIR ScenariosVectorAndMalaria01_B_Namawala3Vectors_highEIR Steps \u00b6 The config.json is set with the following default values. Optional settings are detailed in the various tutorial steps. Set the parameter Simulation_Type to the value \u201cVECTOR_SIM\u201d. This vector model inherits the same human-infection model structure from the generic simulation type: uninfected, latent incubation, infectious, multiple immune variables, and super-infection. However, the transmission of infections is no longer between individual humans, but rather via the human-to-vector and vector-to-human pathways. The vector life cycle and feeding cycle in the model are described in the article Eckhoff, Malaria Journal 2011, 10:303 . Also, see the Larval Habitat topic in the malaria documentation. Set the vector population. A separate VectorPopulation object is created for each vector species in the simulation. Modeling multiple vector species simultaneously allows for a mechanistic description of vector abundances and behavior through the effects of climate and weather on different preferred larval habitats. The three vector species present in this tutorial are Anopheles arabiensis , Anopheles gambiae , and Anopheles funestus , as these vectors drive most transmission in the Namawala region. An. funestus has more relative importance during the dry season, but An. arabiensis and An. gambiae combined carry most of the annual transmission. For more information, see Vector parameters. Set the vector population with the following parameters: Set Vector_Sampling_Type to \u201cVECTOR_COMPARTMENTS NUMBER\u201d Set Vector_Species_Names to \u201carabiensis,\u201d \u201cfunestus,\u201d \u201cgambiae\u201d Set Enable_Interventions to 0 to run a baseline simulation with no public health campaign. For more information, see Simulation setup parameters. Set the incubation period. The incubation period Incubation_Period_Distribution is set to \u201cFIXED_DURATION\u201d and Base_Incubation_Period is set to 25 days corresponding to the duration from an infectious mosquito bite to the development of sufficiently numerous gametocytes to infect mosquitoes. For more information, see General disease parameters. Set the infectious period with the following parameters: Set Base_Infectious_Period to 180. Set Base_Infectivity to 1. Set Base_Mortality to 0. Set Enable_Superinfection to 1. Set Infectious_Period_Distribution to \u201cEXPONENTIAL_DURATION.\u201d Configure the vector species parameters. The vector species parameters described in the article Eckhoff, Malaria Journal 2011, 10:303 are specified for each species present in the simulation. For more information, see Vector species and Scaling factors parameters. Set Acquire_Modifier to 0.2. Set Adult_Life_Expectancy to 10. Set Anthropophily to 0.95. Set Aquatic_Arrhenius1 to 84200000000. Set Aquatic_Arrhenius2 to 8328. Set Aquatic_Mortality_Rate to 0.1. Set Days_Between_Feeds to 3. Set Egg_Batch_Size to 100. Set Immature_Duration to 2. Set Indoor_Feeding_Fraction to 1. Set Infected_Arrhenius1 to 117000000000. Set Infected_Arrhenius2 to 8336. Set Infected_Egg_Batch_Factor to 0.8. Set Infectious_Human_Feed_Moratality_Factor to 1.5. Set Larval_Habitat_Types to \u201cTEMPORARY_RAINFALL\u201d: 11250000000. Set Transmission_Rate to 0.5. Set x_Temporary_Larval_Habitat to 0.1. Warning The above settings are for An. arabiensis vector population, but settings are different for the other species. These are noted in detail within the configuration file under the Vector_Species_Params parameter. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations .", "tags": "", "title": "Vector and malaria 1: Simple vector model \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-10-fences.html", "text": "Vector and malaria 10: Insect-killing fence \u00b6 The InsectKillingFence is a node-targeted intervention that is any sort of fence-like system designed to stop insects, such as mosquitoes. For example, you could create an insect-killing fence by stringing together a series of bednets. The mode-of-action is the killing of the specified fraction of all feeding mosquitoes both at the beginning and at the end of the feeding cycle. While the idea of an insect-killing fence (especially a photonic fence) is interesting in and of itself, the real focus of this tutorial concerns its use. Namely, this is an intervention that can kill mosquitoes outdoors both on their way in to feed and on their way out to oviposit. Use the config.json and campaign.json in the following directory for this tutorial. ScenariosVectorAndMalaria10_InsectKillingFence Steps \u00b6 In the campaign file, in Intervention_Config , set class to \u201cInsectKillingFence\u201d. Set Start_Day to 1460. For more information, see Campaign event parameters. Set the following parameters in the intervention. For more information, see InsectKillingFence and Waning effect parameters. Set Cost_To_Consumer to 3.75. In Killing_Config , set Initial_Effect to 0.28. Set Decay_Time_Constant to 2190. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations .", "tags": "", "title": "Vector and malaria 10: Insect-killing fence \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-11-habitat.html", "text": "Vector and malaria 11: Linear combination vector habitat \u00b6 In this tutorial, you will configure and run a species vector simulation in Namawala where the vectors reside in more than one habitat. This tutorial, investigating a linear-combination habitat, is important because in many cases, a mosquito species will not be characterized by a single type of habitat. Rather a mosquito species may predominantly use a single habitat type, but also make use of other habitat types as available. The linear-combination habitat models allow you much more control over local vector ecology, especially in dry seasons. Use the config.json and campaign.json in the following directory for this tutorial. ScenariosVectorAndMalaria11_MultipleLarvalHabitats Steps \u00b6 The configuration file is set with default values described below. Optional settings are detailed in the various tutorial steps. Set the habitat parameters as shown below. For more information, see Vector parameters. The following details the parameter configuration in the config.json file for the species that is present in this simulation, An. arabiensis. Set Larval_Habitat_Types to {\u201cTEMPORARY_RAINFALL\u201d: 11250000000, \u201cWATER_VEGETATION\u201d: 6000000000}. Set Vector_Species_Names to \u201carabiensis.\u201d The campaign uses oviposition traps that target only the water-vegetation habitat. Set the campaign file parameters as shown below. Set Initial_Effect to 0.95. Set Decay_Time_Constant to 2190. Set Habitat_Target to \u201cWATER_VEGETATION\u201d. { \"Use_Defaults\" : 1 , \"Events\" : [{ \"Event_Coordinator_Config\" : { \"Intervention_Config\" : { \"Secondary_Decay_Time_Constant\" : 3650 , \"Reduction\" : 0 , \"Target\" : \"Larvicides_WaterVegetation\" , \"Cost_To_Consumer\" : 3.75 , \"Primary_Decay_Time_Constant\" : 2190 , \"Durability_Time_Profile\" : \"DECAYDURABILITY\" , \"Killing\" : 0.95 , \"class\" : \"OvipositionTrap\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 140 , \"class\" : \"CampaignEvent\" }] } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations .", "tags": "", "title": "Vector and malaria 11: Linear combination vector habitat \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-2-full.html", "text": "Vector and malaria 2: Full malaria model \u00b6 The previous tutorial uses a simplified model for infections in the human population. It is based on a simple SEIRS model with enhancements. For example, the duration spent in the latent state can be set to be a fixed interval rather than an exponentially-distributed duration. This provides the EMOD agent-based framework with some advantages over the continuum approximations of an ordinary differential equation (ODE) -based model. In particular, the advanced user can go beyond these four states by using the built-in immunity functions for acquiring infection, transmitting infection, and dying from infection. These are all available in the simple disease model. Many users, though, will want to study aspects of malaria infections and population dynamics that do not fit into the SEIRS framework. We have developed a detailed microsolver using within-host parasite dynamics as described in Eckhoff, Malaria Journal 2011, 10:303 where development of clinical and parasitological immunity is tracked through innate and adaptive immune responses to specific antigens. A detailed parasite count over time is tracked for each infected individual, permitting simulation of measured prevalence over time by slide microscopy, Rapid Diagnostic Tests (RDTs), or other diagnostics. Gametocyte production and decay is included to study the human infectious reservoir. This tutorial provides a basic template for running the same simulation of the vector population, but with infections in the human population simulated with the detailed parasite microsolver. Start by running the provided simulation, and then vary different parameters to observe sensitivities. The gametocyte parameters are the most uncertain in the solver. Key outputs include the measured prevalence by different diagnostic methods, the prevalence of fever, and the human infectious reservoir over time. Use the config.json and campaign.json files in the following directory for this tutorial: ScenariosVectorAndMalaria02_Namawala_Full_Malaria_Model Steps \u00b6 The config.json file is all set with default values described below. Optional settings are detailed in the various tutorial steps. Set the parameter Simulation_Type to the value \u201cMALARIA_SIM\u201d. For more information, see Simulation setup parameters. The \u201cMALARIA_SIM\u201d setting maintains the vector transmission from \u201cVECTOR_SIM\u201d, but now replaces the human infection model with a detailed microsolver of parasite dynamics. Set the following parameters. For more information, see General disease and Malaria parameters. Set Base_Incubation_Period to 7. Set Base_Infectious_Period to 180. Set Infection_Updates_Per_Timestep to 24. Set Malaria_Model to \u201cMALARIA_MECHANISTIC_MODEL.\u201d Set Malaria_Strain_Model to \u201cFALCIPARUM_RANDOM_STRAIN.\u201d Set Parasite_Switch_Type to \u201cRATE_PER_PARASITE_7VARS\u201d Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . The duration of infection is solved dynamically. Experiment by varying parameters to observe sensitivities. The gametocyte parameters are the most uncertain in the solver. As noted in the overview, key outputs include the measured prevalence by different diagnostic methods, the prevalence of fever, and the human infectious reservoir over time.", "tags": "", "title": "Vector and malaria 2: Full malaria model \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-3-climate.html", "text": "Vector and malaria 3: Impact of climate on seasonality \u00b6 Compare impact of climate on seasonality and run a simple vector model in Namawala, Muheza, and Garki single-node. In this next tutorial, we now move out from Namawala to explore the impact of different locations and weather patterns. Use the config.json and campaign.json files in the following directories for this tutorial: ScenariosVectorAndMalaria03_A_Namawala_Vector ScenariosVectorAndMalaria03_B_Muheza_Vector ScenariosVectorAndMalaria03_C__Garki_single_node_Vector Steps \u00b6 The files in this tutorial are set up to use the demographic and weather files for Namawala, Tanzania in the 1990s, Muheza, Tanzania in the 2000s, and Garki, Nigeria in the early 1970s. The values are set in the config.json file. Follow the steps detailed in Vector and malaria 1: Simple vector model . Enable_Interventions remains set to 0 to run a baseline simulation with no public health campaign. For more information, see Simulation setup parameters. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Compare the impact of climate on seasonality with a simple vector model in Namawala, Muheza, and Garki single-node geographies. The differences among the three locations go beyond the differences in annual entomological inoculation rate (EIR) . The dry season is much drier in Garki compared to Namawala, and thus the depth of the low transmission season is lower. This causes a stronger variation in human infection in Garki, especially if the tutorial is re-run with the full malaria microsolvers.", "tags": "", "title": "Vector and malaria 3: Impact of climate on seasonality \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-5-bednets.html", "text": "Vector and malaria 5: Bednet distribution \u00b6 Beginning with this tutorial, you will be introduced to the procedure for layering campaign intervention events on top of the tutorials from the previous topics located in Baseline simulation scenarios . Use the config.json and campaign.json files in the following directory for this tutorial. ScenariosVectorAndMalaria05_Namawala_Vector_ITNs Steps \u00b6 The disease simulation parameters of PrimaryScenarios_6_Namawala_Vector_ITNs are identical to the ones used it the PrimaryScenarios_2B_Namawala3Vectors_highEIR scenario. However, in order to configure the simulation to dispense campaign events, two parameters are modified. { \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.7 , \"Intervention_Config\" : { \"Bednet_Type\" : \"ITN\" , \"Cost_To_Consumer\" : 3.75 , \"Killing_Config\" : { \"Initial_Effect\" : 0.6 , \"Decay_Time_Constant\" : 1460 , \"class\" : \"WaningEffectExponential\" }, \"Blocking_Config\" : { \"Initial_Effect\" : 0.9 , \"Decay_Time_Constant\" : 730 , \"class\" : \"WaningEffectExponential\" }, \"class\" : \"SimpleBednet\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" } ], \"Use_Defaults\" : 1 } Set the interventions. Change Enable_Interventions to 1 to enable the use of campaign interventions. You set this in the config.json file. This setting directs the simulation to look for the file specified in Campaign_Filename to define how campaign events are configured and distributed. For more information, see Simulation setup and Input files parameters. Set the fields in the campaign file as described below. For more information, see Campaign event parameters. Set class to \u201cCampaignEvent.\u201d Set Nodeset_Config to {\u201cclass\u201d: \u201cNodeSetAll\u201d}. Set Start_Day to 1460. Set the Event_Coordinator_Config , which specifies which event coordinator class will handle the event, and then configures the coordinator in the campaign file. Set Demographic_Coverage to 0.7. Set Target_Demographic to \u201cEveryone.\u201d Set the Intervention_Config object, which selects a class for the intervention and configures the parameters specific for that intervention class. Insecticide-Treated Nets (ITNs) are a key component of modern malaria control efforts and have recently been scaled up towards universal coverage in sub-Saharan Africa. Modern bednets are made of a polyethylene or polyester mesh which is impregnated with a slowly releasing pyrethroid insecticide. Mosquitoes that are seeking a blood meal will encounter a net, and if the net retains its physical integrity and has been correctly installed the feeding attempt is blocked. Blocked feeding attempts carry the possibility of killing the mosquito. Two efficacies are thus specified for bednets: the Blocking_Config , which sets the rate of blocking indoor feeds on individuals with an ITN, and Killing_Config , which sets the rate at which the mosquito dies conditional on a successfully blocked feed. The rate of blocking and killing both decay at a specified rate. For more information, see SimpleBednet and Waning effect parameters. Set the Bednet_Type as \u201cITN\u201d. In Blocking_Config , set the Initial_Effect to 0.9 and Decay_Time_Constant to 730. In Killing_Config , set the Initial_Effect to 0.6 and Decay_Time_Constant to 1460. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Then, experiment with changing the durability profiles and efficacies.", "tags": "", "title": "Vector and malaria 5: Bednet distribution \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-6-bednets-vaccines.html", "text": "Vector and malaria 6: Bednets and pre-erythrocytic vaccines \u00b6 This tutorial layers an additional campaign intervention, a pre-erythrocytic vaccine (PEV) , on top of the bednet distribution from the previous tutorial. Use the config.json and campaign.json files in the following directory for this tutorial. ScenariosVectorAndMalaria06_Namawala_Vector_ITNsPEV Steps \u00b6 Set the parameters for the SimpleBednet event in the campaign file as noted in Vector and malaria 5: Bednet distribution . Set the Event_Coordinator_Config parameters in the campaign file for the for the SimpleVaccine event. For more information, see SimpleBednet and Waning effect parameters. Set class to \u201cSimpleVaccine.\u201d Set Cost_To_Consumer to 10.0 Set Demographic_Coverage to 0.8. Set Box_Duration to 730. Set Reduced_Acquire to 0.9. Set Vaccine_Take to 1. Set Vaccine_Type to \u201cAcquisitionBlocking.\u201d { \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.7 , \"Intervention_Config\" : { \"Bednet_Type\" : \"ITN\" , \"Blocking_Config\" : { \"Decay_Time_Constant\" : 730 , \"Initial_Effect\" : 0.9 , \"class\" : \"WaningEffectExponential\" }, \"Cost_To_Consumer\" : 3.75 , \"Killing_Config\" : { \"Decay_Time_Constant\" : 1460 , \"Initial_Effect\" : 0.6 , \"class\" : \"WaningEffectExponential\" }, \"class\" : \"SimpleBednet\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" }, { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.8 , \"Intervention_Config\" : { \".Reduced_Acquire\" : 0.9 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"Cost_To_Consumer\" : 10.0 , \"class\" : \"SimpleVaccine\" , \"Waning_Config\" : { \"Initial_Effect\" : 0.9 , \"Box_Duration\" : 730 , \"class\" : \"WaningEffectBox\" } }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" } ], \"Use_Defaults\" : 1 } Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Compare to baseline and bednets alone after running the simulation.", "tags": "", "title": "Vector and malaria 6: Bednets and pre-erythrocytic vaccines \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-7-vaccines.html", "text": "Vector and malaria 7: Gametocyte-blocking vaccines \u00b6 This tutorial focuses on implementing a transmission-blocking vaccine (TBV) . Altogether the campaign template includes four events: insecticide-treated nets (ITN) , indoor residual spraying (IRS) , pre-erythrocytic vaccine (PEV) and TBV. Only the TBV event has a non-zero demographic coverage. The other events are provided as additional examples for experimentation. Use the config.json and campaign.json files in the following directory for this tutorial: 07_Namawala_Vector_SSV Steps \u00b6 This event is based on the Event_Coordinator_Config parameters as set in Vector and malaria 6: Bednets and pre-erythrocytic vaccines with the following changes in the campaign file: For the \u201cTransmissionBlocking\u201d SimpleVaccine event, set the Start_Day to 1460. Make the following changes in the Intervention_Config to configure the last event with the SimpleVaccine class. For more information, see SimpleVaccine parameters. Set Demographic_Coverage to 0.85. Set Decay_Time_Constant to 2190. Set Initial_Effect to 0.9. Set Vaccine_Type to \u201cTransmissionBlocking.\u201d Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Experiment with the four events provided in this tutorial to vary your results.", "tags": "", "title": "Vector and malaria 7: Gametocyte-blocking vaccines \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-8-garki.html", "text": "Vector and malaria 8: Garki project retrospective - IRS and MDA \u00b6 This tutorial examines the historical combination of indoor residual spraying (IRS) and mass drug administration (MDA) in the Garki District of Nigeria. This intervention contains two events: an IRSHousingModification event and an AntimalarialDrug event. Use the templates (config.json and campaign.json) in the following directory for this tutorial. ScenariosVectorAndMalaria08_Garki_IRSMDA Steps \u00b6 Set Demographics_Filename to \u201cGarki_single_demographics.compiled.json\u201d. For more information, see Input files . Set the disease parameters for \u201cMALARIA_SIM\u201d as noted in Vector and malaria 2: Full malaria model . Configure the IRSHousingModification event. For more information, see IRSHousingModification parameters. Set Start_Day to 2190. Set Demographic_Coverage to 0.8. Set Number_Repetitions to 4. Set Secondary_Decay_Time_Constant to 3650. Set Timesteps_Between_Repettions to 365. Configure the AntimalarialDrug event. For more information, see AntimalarialDrug parameters. Set the Start_Day to 2190. Set Cost_To_Customer to 3.75. Set Demograpchi_coverage to 0.8. Set Dosing_Type to \u201cFullTreatmentParasiteDetect.\u201d Set Drug_Type to \u201cChloroquine.\u201d Set Number_Repetitions to 9. Set Timesteps_Between_Repetitions to 60. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Compare to baseline and bednets alone after running the simulation.", "tags": "", "title": "Vector and malaria 8: Garki project retrospective - IRS and MDA \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-9-sugar-traps.html", "text": "Vector and malaria 9: Sugar-baited traps \u00b6 The sugar-baited traps tutorial introduces a new vector-life cycle targeting intervention, the sugar-baited trap or attracting toxic sugar bait. This is a node-level intervention of class SugarTrap . Use the config.json and campaign.json in the following directory for this tutorial. ScenariosVectorAndMalaria09_Namawala_Vector_SugarBaitedTraps Steps \u00b6 In the config.json file, set the following parameters. For more information, see Vector parameters. Set Vector_Sampling_Type to \u201cTRACK_ALL_VECTORS.\u201d Set Vector_Sugar_Feeding_Frequency to \u201cVECTOR_SUGAR_FEEDING_EVERY_DAY.\u201d In the campaign.json file, set the following parameters. For more information, see Sugartrap and Waning effect parameters. Set class to \u201cSugarTrap\u201d. Set Start_Day to 1460. Set Cost_To_Consumer to 3.75. In Killing_Config , set Initial_Effect to 0.1 and Decay_Time_Constant to 2190. Run the simulation \u00b6 Run the simulation and generate graphs of the simulation output. For more information, see How to run Quick Start simulations . Compare to baseline and bednets alone after running the simulation.", "tags": "", "title": "Vector and malaria 9: Sugar-baited traps \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-baseline-sim-scenarios.html", "text": "Baseline simulation scenarios \u00b6 Run the baseline scenario simulations to understand some of the basic features available in EMOD. The baseline set includes scenarios for each of the generic SEIRS model , vector-borne, and malaria disease types as well as those that explore the impact of climate and migration on spatially distributed simulations. In this sections, you focus on the simulation configuration file . The scenarios in Scenarios with campaigns focus on interventions and introduce the use of a campaign file . The following scenarios are baseline simulation scenarios: Vector and malaria 1: Simple vector model Vector and malaria 2: Full malaria model Vector and malaria 3: Impact of climate on seasonality", "tags": "", "title": "Baseline simulation scenarios \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-overview.html", "text": "Vector and malaria tutorials \u00b6 These tutorials are separated into two categories: baseline simulation scenarios and scenarios with campaigns. Baseline tutorials focus on simulation the spread of disease without the use of interventions. Scenarios with campaigns focus on campaigns aimed at stopping the spread of disease with one or more interventions. The tutorials are numbered in the order in which they should be read while learning. Baseline simulation scenarios Vector and malaria 1: Simple vector model Vector and malaria 2: Full malaria model Vector and malaria 3: Impact of climate on seasonality Scenarios with campaigns Vector and malaria 5: Bednet distribution Vector and malaria 6: Bednets and pre-erythrocytic vaccines Vector and malaria 7: Gametocyte-blocking vaccines Vector and malaria 8: Garki project retrospective - IRS and MDA Vector and malaria 9: Sugar-baited traps Vector and malaria 10: Insect-killing fence Vector and malaria 11: Linear combination vector habitat", "tags": "", "title": "Vector and malaria tutorials \u2014 Training Tutorials  documentation"}, {"url": "./malaria-VM-tutorials-scenarios-with-campaigns.html", "text": "Scenarios with campaigns \u00b6 This section includes a few examples that test a characteristic sample of the campaign features available in EMOD. See Campaign parameters for a comprehensive list of the parameters available for experimenting. In the Baseline simulation scenarios , you focus on the simulation configuration file . The scenarios here focus on interventions and introduce the use of a campaign file . Vector and malaria 5: Bednet distribution Vector and malaria 6: Bednets and pre-erythrocytic vaccines Vector and malaria 7: Gametocyte-blocking vaccines Vector and malaria 8: Garki project retrospective - IRS and MDA Vector and malaria 9: Sugar-baited traps Vector and malaria 10: Insect-killing fence Vector and malaria 11: Linear combination vector habitat", "tags": "", "title": "Scenarios with campaigns \u2014 Training Tutorials  documentation"}, {"url": "./parameter-campaign.html", "text": "Campaign parameters \u00b6 The parameters described in this reference section determine the interventions used as part of a campaign to control the spread of a disease and the outbreak of the disease itself. Interventions are configured in the campaign file . They are enabled in the configuration file by setting the parameter, Enable_Interventions , to 1. The configuration file also indicates the name of the campaign file. Warning The Outbreak must be the last event in the campaign file or none of the interventions will take place. EMOD supports a variety of supported campaign interventions. It is worth noting that each intervention contains multiple parameters used to configure that specific event. For example, the distribution of bednets behaves differently in the model than does the use of larvicides. It is therefore important to ensure that interventions are configured correctly and applied to the appropriate level target (individual versus node). This framework enables rigorous testing of possible control strategies to determine which events or combination of events will best aid in the elimination of disease for specific geographic locations. Warning Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false. The tables below contain only parameters available when using the training tutorials. Contents Events array Campaign event Event coordinators NodeSet_Config Interventions AnimalFeedKill AntimalarialDrug ArtificialDiet ArtificialDietHousingModification BirthTriggeredIv BroadcastEvent DelayedIntervention HealthSeekingBehaviorUpdateable HumanHostSeekingTrap InsectKillingFence InsectKillingFenceHousingModification InterpolatedValueMap IRSHousingModification InputEIR IVCalendar Ivermectin Larvicides MalariaChallenge MosquitoRelease MultiInterventionDistributor NodeLevelHealthTriggeredIV Outbreak OutbreakIndividual OutdoorRestKill OvipositionTrap PMTCT PropertyValueChanger RTSSVaccine ScaleLarvalHabitat ScreeningHousingModification SimpleBednet SimpleDiagnostic SimpleHealthSeekingBehavior SimpleIndividualRepellent SpaceSpraying SpatialRepellent SpatialRepellentHousingModification Sugartrap WaningEffect Events array \u00b6 The campaign file contains an Events array with one or more CampaignEvent elements and a Use_Defaults parameter. If Use_Defaults is set to 1, the simulation will use the default values for required parameters that are not configured in the campaign file. If it is set to 0, you must configure all required parameters. { \"Events\" : [{ \"Event_Name\" : \"Campaign Event 1\" , \"class\" : \"CampaignEvent\" }, { \"Event_Name\" : \"Campaign Event 2\" , \"class\" : \"CampaignEvent\" }], \"Use_Defaults\" : 1 } Campaign event \u00b6 To distribute an intervention, you must configure a campaign event, an event coordinator, and the intervention. The campaign event configures when the event occurs ( Start_Day ) and where it is distributed ( Nodeset_Config ). The event coordinator is a nested JSON object in Event_Coordinator_Config . For more information on how to structure this file, see howto-campaign . { \"Event_Name\" : \"Campaign Event Example\" , \"class\" : \"CampaignEvent\" , \"Start_Day\" : 1 , \"Nodeset_Config\" : { }, \"Event_Coordinator_Config\" : { } } Parameter Data type Minimum Maximum Default Description Example Event_Coordinator_Config EventCoordinator 0 3.40E+38 1 The configuration of the EventCoordinator to be instantiated. \u00a0 Nodeset_Config NodeSet 0 3.40E+38 1 The NodeSet describing the nodes covered by the event. \u00a0 Start_Day float 0 3.40E+38 1 Day of the simulation to activate the event\u2019s event coordinator. \u00a0 Event coordinators \u00b6 The Event_Coordinator_Config configures who can receive the intervention (for example, Target_Demographic and Demographic_Coverage ) and which intervention is distributed. The intervention is a nested JSON object in Intervention_Config . { \"Event_Coordinator_Config\" : { \"class\" : \"The event coordinator that will be used.\" , \"Target_Demographic\" : \"Everyone\" , \"Demographic_Coverage\" : 0.0005 , \"Intervention_Config\" : {} } } Parameter Data type Minimum Maximum Default Description Example Demographic_Coverage float 0 1 1 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Intervention_Config Intervention 0 3.40E+38 3.40E+38 The nested json of the actual intervention to be distributed by this event coordinator. \u00a0 Number_Repetitions integer -1 1000 -1 Number of times intervention given, used with tsteps_between_reps. \u00a0 Property_Restrictions Dynamic String Set NA NA [] Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Property_Restrictions_Within_Node PropertyRestrictions NA NA Everyone Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Target_Age_Max float 0 3.40E+38 3.40E+38 Upper end of age targeted for intervention, in years. \u00a0 Target_Age_Min float 0 3.40E+38 0 Lower end of age targeted for intervention, in years. \u00a0 Target_Demographic enum NA NA Everyone The demographic group (from a list of possibles) targeted by this intervention (e.g., Infants). \u00a0 Target_Gender enum NA NA All Specify the gender restriction for the intervention. Defaults to All. \u00a0 Target_Residents_Only boolean 0 1 0 If true, only distribute the intervention to those that claim the node as their residence - i.e. started in the node \u00a0 Timesteps_Between_Repetitions integer -1 10000 -1 Repetition interval. \u00a0 NodeSet_Config \u00b6 There are three configuration options: NodeSetAll , NodeSetNodeList and NodeSetPolygon . NodeSetAll \u00b6 If you are running a simulation that is in a single location (node) or you want the intervention to be distributed in all of the nodes, use NodeSetAll . { \"Nodeset_Config\" : \"NodeSetAll\" } NodeSetNodeList \u00b6 The intervention is distributed to nodes in the intervention list. Parameter Data type Minimum Maximum Default Description Example Node_List NodeListConfig NA NA SHAPE Comma-separated list of node IDs. \u00a0 NodeSetPolygon \u00b6 The intervention is distributed in nodes within the specified polygon. Parameter Data type Minimum Maximum Default Description Example Polygon_Format enum NA NA SHAPE Right now just SHAPE. \u00a0 Vertices string NA NA SHAPE Comma-separated list of polygon points. \u00a0 Interventions \u00b6 Commonly used vector-control interventions \u00b6 Name Intervention Level Target Life Stage Target Biting Preference Target Biting Location Effect Notes AnimalFeedKill node feeding cycle animal \u2013 killing \u00a0 ArtificialDiet node feeding cycle human all blocking Artificial_Diet_Target: AD_WithinVillage, AD_OutsideVillage HumanHostSeekingTrap individual feeding cycle human \u00a0 killing \u00a0 InsectKillingFence node feeding cycle all all killing Photonic fence. Two killing chances per feeding cycle: to and from oviposition site. InsectKillingFenceHousingModification individual \u00a0 \u00a0 \u00a0 killing \u00a0 IRSHousingModification individual feeding cycle human indoor killing, blocking \u00a0 Ivermectin individual feeding cycle human all killing \u00a0 Larvicides node larva all all killing, reduction Specify targeted habitat. MosquitoRelease node \u00a0 \u00a0 \u00a0 \u00a0 \u00a0 OutdoorRestKill node feeding cycle human outdoor killing \u00a0 OvipositionTrap node feeding cycle all all killing Specify targeted habitat. Requires individual vectors. ScaleLarvalHabitat node larva all all reduction Specify targeted habitat. ScreeningHousingModification individual feeding cycle human indoor killing, blocking \u00a0 SimpleBednet individual feeding cycle human indoor killing, blocking \u00a0 SimpleIndividualRepellent individual feeding cycle human all blocking \u00a0 SpaceSpraying ( SpaceSpray_Indoor ) node feeding cycle human indoor killing Node-based IRS for household. SpaceSpraying node feeding cycle human outdoor killing SpaceSpray_FemalesOnly, MalesOnly, FemalesAndMales SpatialRepellent node feeding cycle all outdoor blocking \u00a0 SpatialRepellentHousingModification individual feeding cycle human indoor killing, blocking \u00a0 SugarTrap node adults all all killing Requires individual vectors. Killing parameter specifies effective fraction of sugar feeds that occur on bait stations. AnimalFeedKill \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 43800 365 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 AntimalarialDrug \u00b6 When using antimalarial drugs in the model, the configuration file sets the main effects of the drug, while the intervention (configured in the campaign file) determines which drugs are used and to whom they are distributed. The following example provides the syntax to configure the AntiMalarialDrug intervention: { \"Intervention_Config\" : { \"class\" : \"AntimalarialDrug\" , \"Cost_To_Consumer\" : 3.75 , \"Drug_Type\" : \"Chloroquine\" , \"Dosing_Type\" : \"FullTreatmentParasiteDetect\" } } Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per drug (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Dosing_Type enum NA NA SingleDose Type of anti-malarial dosing to distribute in a drugs intervention. Possible values are: SingleDose, FullTreatmentCourse, Prophylaxis, SingleDoseWhenSymptom, FullTreatmentWhenSymptom, SingleDoseParasiteDetect, FullTreatmentParasiteDetect, SingleDoseNewDetectionTech, FullTreatmentNewDetectionTech. \u00a0 Drug_Type Constrained String 0 3.40E+38 3.40E+38 Type of anti-malarial drug to distribute in a drugs intervention. Possible values are: Artemisinin (AR), Chloroquine (CQ), Quinine (QN), Sulfadoxine-Pyramethamine (SP), Primaquine (PQ), Artemether_Lumefantrine (AR_LM), Generic Transmission Blocking (TBD), Generic Pre-erythrocytic (PED), Tafenoquine (TQ). \u00a0 Intervention_Name string 0 3.40E+38 3.40E+38 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 ArtificialDiet \u00b6 Parameter Data type Minimum Maximum Default Description Example Artificial_Diet_Target enum NA NA AD_WithinVillage The targeted deployment of artificial diet. Possible values are: AD_WithinVillage, AD_OutsideVillage. \u00a0 Attraction_Config WaningEffect NA NA AD_WithinVillage Fraction of feeds attracted to the artificial diet. A parameter that is decayed. \u00a0 Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 ArtificialDietHousingModification \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 BirthTriggeredIv \u00b6 Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention 0 1 0 The configuration of the actual individual-based intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Demographic_Coverage float 0 1 1 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Duration float -1 3.40E+38 -1 The number of days to continue this intervention. \u00a0 Intervention_Name string -1 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Property_Restrictions Dynamic String Set NA NA [] Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Property_Restrictions_Within_Node PropertyRestrictions NA NA Everyone Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Target_Age_Max float 0 3.40E+38 3.40E+38 Upper end of age targeted for intervention, in years. \u00a0 Target_Age_Min float 0 3.40E+38 0 Lower end of age targeted for intervention, in years. \u00a0 Target_Demographic enum NA NA Everyone The demographic group (from a list of possibles) targeted by this intervention (e.g., Infants). \u00a0 Target_Gender enum NA NA All Specify the gender restriction for the intervention. Defaults to All. \u00a0 Target_Residents_Only boolean 0 1 0 If true, only distribute the intervention to those that claim the node as their residence - i.e. started in the node \u00a0 BroadcastEvent \u00b6 Parameter Data type Minimum Maximum Default Description Example Broadcast_Event Constrained String 0 1 0 The event that should occur at the end of the delay period. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 DelayedIntervention \u00b6 Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Configs IndividualIntervention 0 3.40E+38 6 Array of nested interventions to be distributed at end of delay period to covered fraction. \u00a0 Coverage float 0 1 1 Fraction of individuals receiving delayed distribution of configured interventions. \u00a0 Delay_Distribution enum NA NA NOT_INITIALIZED Distribution of duration of delay period. Supported values are: FIXED_DURATION UNIFORM_DURATION GAUSSIAN_DURATION EXPONENTIAL_DURATION. \u00a0 Delay_Period float 0 3.40E+38 6 If a FIXED_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period to directly specify the time delay (in number of days). If EXPONENTIAL_DURATION is specified as the Delay_Distribution, Delay_Period, represents the exponential rate that describes the distribution of the time delay (in units of 1/days). \u00a0 Delay_Period_Max float 0.6 3.40E+38 0 If a UNIFORM_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Max to directly specify the maximum time delay (in number of days). \u00a0 Delay_Period_Mean float 0 3.40E+38 6 If a GAUSSIAN_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Mean to directly specify the mean time delay (in number of days). \u00a0 Delay_Period_Min float 0 3.40E+38 0 If a UNIFORM_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Min to directly specify the minimum time delay (in number of days). \u00a0 Delay_Period_Std_Dev float 0 3.40E+38 1 If a GAUSSIAN_DURATION is specified as the Delay_Distribution, add this parameter Delay_Period_Std_Dev to specify the standard deviation describing the Gaussian distribution (in number of days). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 HealthSeekingBehaviorUpdateable \u00b6 Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention 0 1 1 The configuration of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Actual_IndividualIntervention_Event Constrained String 1900 2200 2000 The event of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Event_Or_Config enum NA NA Config Specifies whether the current intervention distributes a nested intervention (the Config option) or an event will be broadcast which may trigger other interventions in the campaign file (the Event option). See event list. \u00a0 Intervention_Name string 1900 2200 2000 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Single_Use boolean 0 1 1 One-time, or persistent? \u00a0 Tendency float 0 1 1 The probability of seeking healthcare. \u00a0 HumanHostSeekingTrap \u00b6 Parameter Data type Minimum Maximum Default Description Example Attract_Config WaningEffect 0 999999 3.75 The configuration of killing efficacy and waning for human host-seeking trap. \u00a0 Cost_To_Consumer float 0 999999 3.75 Unit cost per trap (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 999999 10 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 3.75 The configuration of killing efficacy and waning for human host-seeking trap. \u00a0 InsectKillingFence \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 InsectKillingFenceHousingModification \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 InterpolatedValueMap \u00b6 Parameter Data type Minimum Maximum Default Description Example Times float 0 99999 NA An array of years. \u00a0 Values float 0 3.40282e+38 NA An array of values to match the defined Times. \u00a0 IRSHousingModification \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 InputEIR \u00b6 Parameter Data type Minimum Maximum Default Description Example Age_Dependence enum NA NA OFF Dependence of InputEIR on age. \u00a0 Intervention_Name string 0 3.40E+38 1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Monthly_EIR InputEIRConfig NA NA OFF An array of 12 elements that contain an entomological inoculation rate (EIR) for each month. \u00a0 IVCalendar \u00b6 Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Configs IndividualIntervention 0 1 0 An array of interventions distributed as specified in the calendar. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Calendar CalendarIV 0 1 0 An array of ages days and probability of receiving an intervention. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Dropout boolean 0 1 0 If an intervention in the series is missed, all subsequent interventions are also missed. If false (0), all calendar dates/doses are applied independently of each other. If true (1), a missed dose (by applying the probability) will result in no further doses, that is, a miss means you have dropped out altogether. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Ivermectin \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 8 Unit cost per Ivermectin dosing (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of drug-killing efficacy and waning. \u00a0 Larvicides \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect NA NA TEMPORARY_RAINFALL The configuration of larval habitat reduction and waning for targeted stage. \u00a0 Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Habitat_Target enum NA NA TEMPORARY_RAINFALL The larval habitat type targeted for larvicide. Possible values are: TEMPORARY_RAINFALL WATER_VEGETATION HUMAN_POPULATION CONSTANT BRACKISH_SWAMP ALL_HABITATS. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect NA NA TEMPORARY_RAINFALL The configuration of larval killing efficacy and waning for targeted stage. \u00a0 MalariaChallenge \u00b6 Parameter Data type Minimum Maximum Default Description Example Challenge_Type enum NA NA InfectiousBites The type of malaria challenge. Possible values are: InfectiousBites Sporozoites. \u00a0 Coverage float 0 1 1 Fraction of individuals receiving distribution. \u00a0 Infectious_Bite_Count integer 0 1000 1 Number of infectious bites. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Sporozoite_Count integer 0 1000 1 Number of sporozoites. \u00a0 MosquitoRelease \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 0 Cost of each mosquito release. \u00a0 Intervention_Name string NA NA WOLBACHIA_FREE The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Mated_Genetics ResistanceHegGenetics 1 100000000 10000 The genetic properties of the mate if released mosquitoes have mated, e.g. HEG and pesticide resistance. \u00a0 Released_Gender enum NA NA VECTOR_FEMALE Gender of released mosquitoes. Possible values are: VECTOR_FEMALE VECTOR_MALE. \u00a0 Released_Genetics ResistanceHegGenetics 1 100000000 10000 The genetic properties of the released mosquito, e.g. HEG and pesticide resistance. \u00a0 Released_Number integer 1 100000000 10000 Number of mosquitos released in the intervention. \u00a0 Released_Species Constrained String 1 100000000 10000 Name of released mosquito species. \u00a0 Released_Sterility enum NA NA VECTOR_FERTILE Sterility of released mosquitoes. Possible values are: VECTOR_FERTILE VECTOR_STERILE. \u00a0 Released_Wolbachia enum NA NA WOLBACHIA_FREE Wolbachia type of released mosquitoes. Possible values are: WOLBACHIA_FREE VECTOR_WOLBACHIA_A VECTOR_WOLBACHIA_B VECTOR_WOLBACHIA_AB. \u00a0 MultiInterventionDistributor \u00b6 Parameter Data type Minimum Maximum Default Description Example Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_List IndividualIntervention 0 1 0 The array of nested JSON parameters for the interventions to be distributed by this intervention. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 NodeLevelHealthTriggeredIV \u00b6 Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention -1 3.40E+38 -1 The configuration of the actual individual-based intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Actual_NodeIntervention_Config NodeIntervention NA NA [] The configuration of the actual node-level intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Blackout_Event_Trigger Constrained String 0 1 0 The event to broadcast if an intervention cannot be distributed due to the Blackout_Period. \u00a0 Blackout_Period float 0 3.40E+38 0 After the initial intervention distribution, the time, in days, to wait before distributing the intervention again. If it cannot distribute due to the blackout period, it will broadcast the user-defined Blackout_Event_Trigger. \u00a0 Demographic_Coverage float 0 1 1 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Duration float -1 3.40E+38 -1 The number of days to continue this intervention. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Property_Restrictions Dynamic String Set NA NA [] Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Property_Restrictions_Within_Node PropertyRestrictions 0 3.40E+38 3.40E+38 Fraction of individuals in the target demographic that will receive this intervention. \u00a0 Target_Age_Max float 0 3.40E+38 3.40E+38 Upper end of age targeted for intervention, in years. \u00a0 Target_Age_Min float 0 3.40E+38 0 Lower end of age targeted for intervention, in years. \u00a0 Target_Demographic enum NA NA Everyone The demographic group (from a list of possibles) targeted by this intervention (e.g., Infants). \u00a0 Target_Gender enum NA NA All Specify the gender restriction for the intervention. Defaults to All. \u00a0 Target_Residents_Only boolean 0 1 0 If true, only distribute the intervention to those that claim the node as their residence - i.e. started in the node \u00a0 Trigger_Condition Constrained String NA NA [] The condition for triggering a health seeking intervention. \u00a0 Trigger_Condition_List Vector String NA NA [] Included if Trigger_Condition is set to \u2018TriggerList\u2019. A list (JSON array) of triggers (strings) drawn from config.json \u2018Listed_Events\u2019 or from built-in events. \u00a0 Trigger_Condition_String Constrained String 0 1 0 Included if Trigger_Condition is set to \u2018TriggerString\u2019. Normally Trigger_Condition must specify an event from the built-in list of pre-defined events or triggers from the IndividualEventTriggerType enum. Trigger_Condition_String can be any string provided it is listed in the config.json\u2019s Listed_Events array. \u00a0 Outbreak \u00b6 Parameter Data type Minimum Maximum Default Description Example Antigen integer 0 10 0 The antigenic ID of the outbreak infection. \u00a0 Genome integer -1 16777200 0 The genetic ID of the outbreak infection. \u00a0 Import_Age float 0 43800 365 Age (in days) of infected import cases \u00a0 Incubation_Period_Override boolean 0 1 0 0 = outbreak will bypass incubation, 1 = outbreak infection will start from beginning. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Number_Cases_Per_Node integer 0 2147480000 1 Number of new cases of Outbreak to import (per node). This will increase population. There is no control over demographics of these individuals. \u00a0 OutbreakIndividual \u00b6 Parameter Data type Minimum Maximum Default Description Example Antigen integer 0 10 0 The antigenic ID of the outbreak infection. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Genome integer -1 16777200 0 The genetic ID of the outbreak infection. \u00a0 Incubation_Period_Override boolean 0 1 0 0 = outbreak will bypass incubation, 1 = outbreak infection will start from beginning. \u00a0 Intervention_Name string NA NA OFF The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 OutdoorRestKill \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 OvipositionTrap \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Habitat_Target enum NA NA TEMPORARY_RAINFALL The oviposition habitat type targeted by oviposition traps. Possible values are: TEMPORARY_RAINFALL WATER_VEGETATION HUMAN_POPULATION CONSTANT BRACKISH_SWAMP ALL_HABITATS. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect NA NA TEMPORARY_RAINFALL Killing of the targeted stage. A parameter that is decayed. \u00a0 PMTCT \u00b6 Parameter Data type Minimum Maximum Default Description Example Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Efficacy float 0 1 0.5 Represents the efficacy of a PMTCT intervention, defined as the rate ratio of MTCT between women receiving the intervention and women not receiving the intervention. A setting of 1 is equivalent to 100% blocking efficacy, and 0 reverts to the default probability of transmission, configured through the config.json parameter Maternal_Transmission_Probability. \u00a0 Intervention_Name string 0 3.40E+38 6 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 PropertyValueChanger \u00b6 Parameter Data type Minimum Maximum Default Description Example Daily_Probability float 0 1 1 The probability per day that an individual will move to the Target_Property_Value. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Maximum_Duration float -1 3.40E+38 3.40E+38 Maximum duration of \u2018delayed-release\u2019 property change, in days. \u00a0 Revert float 0 10000 0 Reversion rate, or the number of days until reversion, or a 2 if reversion is allowed. \u00a0 Target_Property_Key Constrained String 0 10000 0 The name of the EMOD-defined property of the group. \u00a0 Target_Property_Value Constrained String 0 10000 0 The value of the user-defined group where the individuals will be transitioned. \u00a0 RTSSVaccine \u00b6 Parameter Data type Minimum Maximum Default Description Example Antibody_Type enum NA NA CSP Antibody type. Possible values are: CSP MSP1 PfEMP1_minor PfEMP1_major. \u00a0 Antibody_Variant integer 0 100000 0 Antibody variant index. \u00a0 Boosted_Antibody_Concentration float 0 3.40E+38 1 Boosted antibody concentration (where unity equals maximum from natural exposure). \u00a0 Cost_To_Consumer float 0 999999 3.75 Unit cost of RTS,S vaccination (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 ScaleLarvalHabitat \u00b6 Parameter Data type Minimum Maximum Default Description Example Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Larval_Habitat_Multiplier LarvalHabitatMultiplier 0 1 0 The value by which to scale the larval habitat availability specified in Larval_Habitat_Types across all habitat types, for specific habitat types, or for specific mosquito species within each habitat type. \u00a0 ScreeningHousingModification \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string -1 3.40E+38 -1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 SimpleBednet \u00b6 The following example provides the syntax for configuring a SimpleBednet campaign. { \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.7 , \"Intervention_Config\" : { \"Bednet_Type\" : \"ITN\" , \"Cost_To_Consumer\" : 3.75 , \"Killing_Config\" : { \"Initial_Effect\" : 0.6 , \"Decay_Time_Constant\" : 1460 , \"class\" : \"WaningEffectExponential\" }, \"Blocking_Config\" : { \"Initial_Effect\" : 0.9 , \"Decay_Time_Constant\" : 730 , \"class\" : \"WaningEffectExponential\" }, \"class\" : \"SimpleBednet\" }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" } ], \"Use_Defaults\" : 1 } Parameter Data type Minimum Maximum Default Description Example Bednet_Type enum NA NA Barrier Type of bednet. Possible values are: Barrier, ITN, LLIN, Retreatment. Note: This enum does not affect the code at all. \u00a0 Blocking_Config WaningEffect 0 999999 3.75 A fraction of feed attempts that have been blocked. The rate of blocking indoor feeds on individuals with an ITN. A parameter that is decayed. \u00a0 Cost_To_Consumer float 0 999999 3.75 Unit cost per bednet (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 3.75 A fraction of feed attempts that kill the vector. The rate at which the mosquito dies conditional on a successfully blocked feed. A parameter that is decayed. \u00a0 SimpleDiagnostic \u00b6 Parameter Data type Minimum Maximum Default Description Example Base_Sensitivity float 0 1 1 The sensitivity of the diagnostic. This sets the proportion of the time that individuals with the condition being tested receive a positive diagnostic test. When set to 1, the diagnostic always accurately reflects the condition. When set to zero, then individuals who have the condition always receive a false-negative diagnostic test. \u00a0 Base_Specificity float 0 1 1 The specificity of the diagnostic. This sets the proportion of the time that individuals without the condition being tested receive a negative diagnostic test. When set to 1, the diagnostic always accurately reflects the lack of having the condition. When set to zero, then individuals who do not have the condition always receive a false-positive diagnostic test. \u00a0 Cost_To_Consumer float 0 3.40E+38 1 The unit \u2018cost\u2019 assigned to the intervention. Setting Cost_To_Consumer to zero for all other interventions, and to a non-zero amount for one intervention, provides a convenient way to track the number of times the intervention has been applied in a simulation. \u00a0 Days_To_Diagnosis float 0 3.40E+38 0 The number of days from test until diagnosis. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Event_Or_Config enum NA NA Config Specifies whether the current intervention distributes a nested intervention (the Config option) or an event will be broadcast which may trigger other interventions in the campaign file (the Event option). See event list. \u00a0 Intervention_Name string 0 1 1 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Positive_Diagnosis_Config IndividualIntervention 0 1 0 The intervention distributed to individuals if they test positive. This is only valid if Event_Or_Config is set to \u2018Config\u2019. \u00a0 Positive_Diagnosis_Event Constrained String 0 3.40E+38 0 The next health care event that occurs for individuals who test negative. This is only valid if Event_Or_Config is set to \u2018Event\u2019. \u00a0 Treatment_Fraction float 0 1 1 The fraction of positive diagnoses that are treated. \u00a0 SimpleHealthSeekingBehavior \u00b6 Parameter Data type Minimum Maximum Default Description Example Actual_IndividualIntervention_Config IndividualIntervention 0 1 1 The configuration of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Actual_IndividualIntervention_Event Constrained String 0 999999 10 The event of an actual intervention sought. Selects a class for the intervention and configures the parameters specific for that intervention class. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Event_Or_Config enum NA NA Config Specifies whether the current intervention distributes a nested intervention (the Config option) or an event will be broadcast which may trigger other interventions in the campaign file (the Event option). See event list. \u00a0 Intervention_Name string 0 999999 10 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Single_Use boolean 0 1 1 One-time, or persistent? \u00a0 Tendency float 0 1 1 The probability of seeking healthcare. \u00a0 SimpleIndividualRepellent \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of efficacy and waning for individual repellent. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per repellent (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string NA NA Config The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 SimpleVaccine \u00b6 The following example provides the syntax for configuring a SimpleVaccine campaign. { \"Events\" : [ { \"Event_Coordinator_Config\" : { \"Demographic_Coverage\" : 0.8 , \"Intervention_Config\" : { \".Reduced_Acquire\" : 0.9 , \"Vaccine_Take\" : 1 , \"Vaccine_Type\" : \"AcquisitionBlocking\" , \"Cost_To_Consumer\" : 10.0 , \"class\" : \"SimpleVaccine\" , \"Waning_Config\" : { \"Initial_Effect\" : 0.9 , \"Box_Duration\" : 730 , \"class\" : \"WaningEffectBox\" } }, \"Target_Demographic\" : \"Everyone\" , \"class\" : \"StandardInterventionDistributionEventCoordinator\" }, \"Nodeset_Config\" : { \"class\" : \"NodeSetAll\" }, \"Start_Day\" : 1460 , \"class\" : \"CampaignEvent\" } ], \"Use_Defaults\" : 1 } Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vaccine (unamortized). The unit \u2018cost\u2019 assigned to the intervention. Setting Cost_To_Consumer to zero for all other interventions, and to a non-zero amount for one intervention, provides a convenient way to track the number of times the intervention has been applied in a simulation. \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Vaccine_Take float 0 1 1 This specifies the rate at which delivered vaccines will successfully stimulate an immune response and achieve the desired efficacy. If vaccine_take is set to 0.9, then with a 90 percent chance, the vaccine will start with the specified efficacy, and with a 10 percent chance it will have no efficacy at all. \u00a0 Vaccine_Type enum NA NA Generic Type of vaccine to distribute in a vaccine intervention. Possible values are: - Generic - The vaccine can have all of reduced_transmit, reduced_acquire, and reduced_mortality fields and the vaccine will have both effects. - TransmissionBlocking - Specifies the fraction by which the infectivity of the infected and vaccinated Individual is reduced (reduced_transmit). - AcquisitionBlocking - Specifies the fraction by which the force of infection experienced by the vaccinated Individual is reduced (reduced_acqu \u00a0 Waning_Config WaningEffect 0 1 1 The configuration of drug-killing efficacy and waning. \u00a0 SpaceSpraying \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Habitat_Target enum NA NA TEMPORARY_RAINFALL The larval habitat type targeted for habitat reduction. Possible values are: TEMPORARY_RAINFALL WATER_VEGETATION HUMAN_POPULATION CONSTANT BRACKISH_SWAMP ALL_HABITATS. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect NA NA NA The configuration of killing efficacy and waning for space spaying. \u00a0 Reduction_Config WaningEffect NA NA NA The configuration of larval habitat reduction and waning for space spraying. \u00a0 Spray_Kill_Target enum NA NA SpaceSpray_FemalesOnly The gender kill target of space spraying. Possible values are: SpaceSpray_FemalesOnly SpaceSpray_MalesOnly SpaceSpray_FemalesAndMales. \u00a0 SpatialRepellent \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Repellency_Config WaningEffect 0 999999 10 The configuration of efficacy and waning for spatial repellent. \u00a0 SpatialRepellentHousingModification \u00b6 Parameter Data type Minimum Maximum Default Description Example Blocking_Config WaningEffect 0 999999 8 The configuration of pre-feed mosquito repellency and waning for housing modification. \u00a0 Cost_To_Consumer float 0 999999 8 Unit cost per housing modification (unamortized). \u00a0 Dont_Allow_Duplicates boolean 0 1 0 If an individual\u2019s container has an intervention, set to 1 to prevent them from receiving another copy of the intervention. Supported by all intervention types. \u00a0 Intervention_Name string 0 1 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 8 The configuration of killing efficacy and waning for housing modification. Killing is conditional on NOT blocking the mosquito before feeding. \u00a0 Sugartrap \u00b6 Parameter Data type Minimum Maximum Default Description Example Cost_To_Consumer float 0 999999 10 Unit cost per vector control (unamortized). \u00a0 Intervention_Name string 0 100000 0 The optional name used to refer to this intervention as a means to differentiate it from others that use the same class. \u00a0 Killing_Config WaningEffect 0 999999 10 Killing of the targeted stage. A parameter that is decayed. \u00a0 Waning effect \u00b6 Waning effect parameters are used with interventions such as vaccines, drugs, and bednets. It has six different classes. Each class specifies a different waning effect and uses a different combination of the waning effect parameters. The waning effect parameters are a nested JSON object and can be configured using several different parameters such as Killing_Config , Waning_Config , Blocking_Config , and Mortality_Config . The following example uses Killing_Config . See the specific intervention sections for the supported parameters. { \"Intervention_Config\" : { \"class\" : \"Ivermectin\" , \"Cost_To_Consumer\" : 1 , \"Killing_Config\" : { \"class\" : \"WaningEffectBox\" , \"Box_Duration\" : 3 , \"Initial_Effect\" : 1 } } } WaningEffectBox class \u00b6 Parameter Data type Minimum Maximum Default Description Example Box_Duration float 0 100000 100 Box duration of effect in days. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectBoxExponential class \u00b6 Parameter Data type Minimum Maximum Default Description Example Box_Duration float 0 100000 100 Box duration of effect in days. \u00a0 Decay_Time_Constant float 0 100000 100 Exponential decay length in days. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectConstant class \u00b6 Parameter Data type Minimum Maximum Default Description Example Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectExponential class \u00b6 Parameter Data type Minimum Maximum Default Description Example Decay_Time_Constant float 0 100000 100 Exponential decay length in days. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectLinear class \u00b6 Parameter Data type Minimum Maximum Default Description Example Expire_At_Durability_Map_End boolean 0 1 0 Set to 1 to let the intervention expire when the end of the map is reached. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect. \u00a0 WaningEffectPiecewise class \u00b6 Parameter Data type Minimum Maximum Default Description Example Expire_At_Durability_Map_End boolean 0 1 0 Set to 1 to let the intervention expire when the end of the map is reached. \u00a0 Initial_Effect float 0 1 1 Initial strength of effect.", "tags": "", "title": "Campaign parameters \u2014 Training Tutorials  documentation"}, {"url": "./parameter-configuration.html", "text": "Configuration parameters \u00b6 The parameters described in this reference section determine the behavior of a simulation. Simulations are configured in a configuration file . This file is a JavaScript Object Notation (JSON) formatted file that contains mostly a flat list of JSON key-value pairs. For information on JSON, see EMOD parameter reference . Warning Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false. The tables below contain only parameters available when using the training tutorials. Contents Cluster options Demographics Event recorder report settings General disease Geography and environment Input files Malaria Malaria drug efficacy Migration Output options Sampling Scaling factors Simulation setup Spatial output channels Vector Vector species Cluster options \u00b6 Parameter Data type Minimum Maximum Default Description Example Load_Balance_Filename string NA NA UNINITIALIZED STRING Path to input file used when a static load balancing scheme is selected. \u00a0 Num_Cores integer NA NA NA Number of cores used to run a simulation. This is used by the infrastructure that runs the job, such as an HPC cluster or the Regression scripts, and is not used by the DTK. \u00a0 Demographics \u00b6 Parameter Data type Minimum Maximum Default Description Example Age_Initialization_Distribution_Type enum NA NA DISTRIBUTION_OFF Method for initializing the age distribution in the simulated population. Possible values are: DISTRIBUTION_OFF - All individuals default to age 20-years-old. DISTRIBUTION_SIMPLE - Individual ages are drawn from a distribution whose functional form and parameters are specified in the IndividualAttributes demographics input layers: AgeDistributionFlag, AgeDistribution1, and AgeDistribution2. DISTRIBUTION_COMPLEX - Individual ages are drawn from a piecewise linear distribution specified in the IndividualAttributes demographics input layer AgeDistribution. \u00a0 Base_Population_Scale_Factor float 0 3.40E+38 1 If Population_Scale_Type is set to FIXED_SCALING, the initial simulation population is uniformly scaled over the entire area to adjust for historical or future population density. \u00a0 Birth_Rate_Dependence enum NA NA FIXED_BIRTH_RATE This parameter determines how the birth rate specified in the demographics file is used. Possible values are: FIXED_BIRTH_RATE- The absolute rate at which new individuals are born. POPULATION_DEP_RATE - Scales the node population to determine the birth rate. DEMOGRAPHIC_DEP_RATE - Scales the female population within fertility age ranges to determine the birth rate. INDIVIDUAL_PREGNANCIES - Results in a similar birth rate as DEMOGRAPHIC_DEP_RATE, but pregnancies are assigned on an individual basis and result in a 40-week pregnancy for a specific individual with a birth at the end, INDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE, and INDIVIDUAL_PREGNANCIES_BY_AGE_AND_YEAR. \u00a0 Death_Rate_Dependence enum NA NA NONDISEASE_MORTALITY_OFF This parameter determines how the death rate specified in the demographics file is used. Possible values are: NONDISEASE_MORTALITY_OFF, NONDISEASE_MORTALITY_BY_AGE_AND_GENDER and NONDISEASE_MORTALITY_BY_YEAR_AND_AGE_FOR_EACH_GENDER. \u00a0 Enable_Aging boolean 0 1 1 Set to 1 to account for aging in the population. Set to 0 to assume that the population does not age. \u00a0 Enable_Birth boolean 0 1 1 Set to 1 to enable individuals to be added to the simulation by birth. \u00a0 Enable_Demographics_Birth boolean 0 1 0 Assumes that newborn characteristics such as sickle-cell status are heterogeneous. Set to 0 to assume by default that all newborns have identical characteristics. \u00a0 Enable_Demographics_Builtin boolean 0 1 0 Uses demographics input files to configure the initial population. Enter 0 to use a standard testing sandbox which is useful for testing and validating code pathways. \u00a0 Enable_Demographics_Gender boolean 0 1 1 Enable_Demographics_Gender=0 assigns gender based on a 50/50 draw, while Enable_Demographics_Gender=1 draws from a male/female ratio that is randomly smeared by a Gaussian of width 1%. \u00a0 Enable_Demographics_Other boolean 0 1 0 Includes the impact of other relevant demographic factors. For example, fraction of individuals above poverty, urban/rural characteristics, heterogeneous initial immunity and risk. Set to 0 to run the simulation without other relevant demographic factors. \u00a0 Enable_Disease_Mortality boolean 0 1 1 Set to 1 to enable disease mortality. Set to 0 to disable disease mortality. \u00a0 Enable_Heterogeneous_Intranode_Transmission boolean 0 1 0 Set to 1 to enable heterogeneous intra-node disease transmission. Requires individual property definitions and beta matrix to be specified. Set to 0 to disable. \u00a0 Enable_Vital_Dynamics boolean 0 1 1 Set to 1 as a master switch to enable vital dynamics (births and deaths). Set to 0 to disable vital dynamics. Even when set to 1, the individual toggles for births and deaths can still disable these. \u00a0 Population_Scale_Type enum NA NA USE_INPUT_FILE Either use the initial population specified in the demographics input file or a fixed scaling of this value based on the population scaling factor parameter. Possible values are: USE_INPUT_FILE and FIXED_SCALING. \u00a0 Event recorder report settings \u00b6 Parameter Data type Minimum Maximum Default Description Example Report_Event_Recorder boolean 0 1 0 Enables or disables the ReportEventRecorder.csv report. \u00a0 Report_Event_Recorder_Ignore_Events_In_List boolean 0 1 0 f this flag is set to true, all events listed in Report_Event_Recorder_Events will be ignored. If it is set to false, all events will be recorded. \u00a0 Report_Event_Recorder_Individual_Properties Dynamic String Set NA NA NA Fraction of individuals in the target demographic that will receive this intervention. \u00a0 General disease \u00b6 Parameter Data type Minimum Maximum Default Description Example Acquisition_Blocking_Immunity_Duration_Before_Decay float 0 45000 0 Number of days after infection until acquisition-blocking immunity begins to decay. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Animal_Reservoir_Type enum NA NA NO_ZOONOSIS Configures whether there is an animal reservoir and how the risk of zoonosis in configured. Use of the animal reservoir sets a low constant baseline of infectivity beyond what is present in the human population. It allows a more random introduction of cases in continuous time, which is more applicable for various situations such as zoonosis. Possible values are: NO_ZOONOSIS, CONSTANT_ZOONOSIS where the daily rate of zoonotic infection is configured by the parameter Zoonosis_Rate, and ZOONOSIS_FROM_DEMOGRAPHICS where the zoonosis rate is additionally scaled by the node-specific Zoonosis value in the NodeAttributes section of a demographics overlay file. See Zoonosis_Rate. \u00a0 Base_Incubation_Period float 0 3.40E+38 6 Average duration, in days, of the incubation period before infected individuals becomes infectious. \u00a0 Base_Infectious_Period float 0 3.40E+38 6 Average duration, in days, of the infectious period before the infection is cleared. \u00a0 Base_Infectivity float 0 1000 0.3 The Base_Infectivity parameter determines the base infectiousness of individuals before accounting for transmission-blocking effects of acquired immunity and/or campaign interventions. It has a slightly different meaning for each of the disease types. GENERIC_SIM: The average number of individuals per time step who will be exposed to infection by one infectious individual. In the case of super-infection, the infectiousness is summed over all infections. VECTOR_SIM: The probability of infecting a mosquito during a successful blood meal (modulated by the vector parameter Acquire_Modifier). For vector simulations, the sum infectiousness of an individual is not allowed to exceed 100%. MALARIA_SIM: This simple scale factor is not used. Instead, gametocyte abundances and cytokine-mediated infectiousness are modeled explicitly. \u00a0 Base_Mortality float 0 1000 0.001 The base mortality of the infection before accounting for individual immune modification factors. Depending on the setting of Mortality_Time_Course, this is either the daily probability of the disease being fatal (DAILY_MORTALITY) or the probability of death at the end of the infection duration (MORTALITY_AFTER_INFECTIOUS). \u00a0 Enable_Immune_Decay boolean 0 1 1 Set to 1 if immunity decays after an infection clears. Set to 0 if recovery from the disease confers complete immunity for life. \u00a0 Enable_Immunity boolean 0 1 1 Set to 1 if an individual has protective immunity after an infection clears. Set to 0 if recovery from the infection does not affect the individual\u2019s future immunity. \u00a0 Enable_Maternal_Transmission boolean 0 1 0 Set to 1 to account for infection of infants at birth by infected mothers. Set to 0 to assume that infection is not transmitted by mothers to infants at birth. \u00a0 Enable_Superinfection boolean 0 1 0 Set to 1 if an individual can have multiple infections with the same agent simultaneously. Set to 0 if multiple infections are not possible. See the Max_Individual_Infections parameter. \u00a0 Immunity_Acquisition_Factor float 0 1000 0 Multiplicative reduction in probability of reacquiring the disease. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Immunity_Initialization_Distribution_Type enum NA NA DISTRIBUTION_OFF Method for initializing the immunity distribution in the simulated population. Possible values are: DISTRIBUTION_OFF - All individuals default to no immunity. DISTRIBUTION_SIMPLE - Individual immunities are drawn from a distribution whose functional form and parameters are specified in the IndividualAttributes demographics input layers: ImmunityDistributionFlag, ImmunityDistribution1, and ImmunityDistribution2. DISTRIBUTION_COMPLEX - Individual immnunities are drawn from an age-dependent piecewise l. \u00a0 Immunity_Mortality_Factor float 0 1000 0 Multiplicative reduction in probability of dying from infection after getting re-infected. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Immunity_Transmission_Factor float 0 1000 0 Multiplicative reduction in probability of transmitting infection after getting re-infected. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Incubation_Period_Distribution enum NA NA NOT_INITIALIZED Distribution of duration of incubation period. Can be fixed or exponentially distributed with an average duration from Base_Incubation_Period. Possible values are: FIXED_DURATION and EXPONENTIAL_DURATION. \u00a0 Incubation_Period_Max float 0.6 3.40E+38 0 The maximum length of the incubation period. Used when Incubation_Period_Distribution is UNIFORM_DURATION. \u00a0 Incubation_Period_Mean float 0 3.40E+38 6 The standard deviation used when Incubation_Period_Distribution is either GAUSSIAN_DURATION or POISSON_DURATION. \u00a0 Incubation_Period_Min float 0 3.40E+38 0 The minimum length of the incubation period. Used when the Incubation_Period_Distribution is UNIFORM_DURATION. \u00a0 Incubation_Period_Std_Dev float 0 3.40E+38 1 The standard deviation incubation period. Used when the Incubation_Period_Distribution is GAUSSIAN_DURATION. \u00a0 Infection_Updates_Per_Timestep integer 0 144 1 Number of infection updates executed during each timestep. A timestep defaults to one day. \u00a0 Infectious_Period_Distribution enum NA NA NOT_INITIALIZED Distribution of duration of infectious period. Can be fixed or exponentially distributed with an average duration from Base_Infectious_Period. Possible values are: FIXED_DURATION, EXPONENTIAL_DURATION. \u00a0 Infectious_Period_Max float 0.6 3.40E+38 0 The maximum length of the infectious period. Used when Infectious_Period_Distribution is UNIFORM_DURATION. \u00a0 Infectious_Period_Mean float 0 3.40E+38 6 The mean of the infectious period. Used when Infectious_Period_Distribution is either GAUSSIAN_DURATION or POISSON_DURATION. \u00a0 Infectious_Period_Min float 0 3.40E+38 0 The minimum length of the infectious period. Used when the Infectious_Period_Distribution is UNIFORM_DURATION. \u00a0 Infectious_Period_Std_Dev float 0 3.40E+38 1 The standard deviation of the infectious period. Used when the Infectious_Period_Distribution is GAUSSIAN_DURATION. \u00a0 Infectivity_Scale_Type enum NA NA CONSTANT_INFECTIVITY Alter infectivity by time or season. Possible values are: CONSTANT_INFECTIVITY - No infectivity correction is applied. FUNCTION_OF_TIME_AND_LATITUDE - Infectivity is corrected for approximate seasonal forcing. The use of a seasonal infectivity correction is a proxy for the effects of varying climate. From October through March, infectivity increases in the Northern Hemisphere and decreases in the Southern Hemisphere. From April through September, the trend reverses. Regions closer to the equator have reduced forcing compared to temperate regions. This is not a substitute for the weather-driven vector dynamics of vector-borne and malaria simulations. FUNCTION_OF_CLIMATE - Allows infectivity to be modulated by weather directly, for example, relative humidity in airborne simulations or rainfall in environmental simulations. There is no default climate dependence enabled for generic simulations. EXPONENTIAL_FUNCTION_OF_TIME - To facilitate certain burn-in scenarios, infectivity ramps up from zero at the beginning of the simulation according to the functional form, 1-exp(-rate*time), where the rate is specified by the parameter Infectivity_Scaling_Rate. \u00a0 Maternal_Transmission_Probability float 0 1 0 Probability of transmission of infection from mother to infant at birth. Relevant only if Enable_Maternal_Transmission is set to 1. Note: This parameter should be set to 0 and ignored for malaria and vector simulations. \u00a0 Max_Individual_Infections integer 0 1000 1 Limit on the number of infections that an individual can have simultaneously. Note: Relevant only if the Enable_Superinfection parameter is set to 1 to allow multiple infections. \u00a0 Mortality_Blocking_Immunity_Decay_Rate float 0 1000 0.001 Rate at which mortality-blocking immunity decays after the mortality-blocking immunity offset period. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Mortality_Time_Course enum NA NA DAILY_MORTALITY Determines whether disease deaths are calculated on every time step or once at the end of the disease duration. Possible values are: DAILY_MORTALITY and MORTALITY_AFTER_INFECTIOUS. \u00a0 Number_Basestrains integer 1 10 1 The number of base strains in the simulation, such as antigenic variants. \u00a0 Number_Substrains integer 1 16777200 256 The number of disease substrains for each base strain, such as genetic variants. \u00a0 Population_Density_Infectivity_Correction enum NA NA CONSTANT_INFECTIVITY Correction to alter infectivity by population density set in the Population_Density_C50 parameter. Measured in people per sq. km. Possible values are: CONSTANT_INFECTIVITY and SATURATING_FUNCTION_OF_DENSITY. Note: Sparsely populated areas have a lower infectivity, while densely populated areas have a higher infectivity which rises to saturate at the Base_Infectivity value. \u00a0 Transmission_Blocking_Immunity_Decay_Rate float 0 1000 0.001 Rate at which transmission-blocking immunity decays after the base transmission-blocking immunity offset period. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Transmission_Blocking_Immunity_Duration_Before_Decay float 0 45000 0 Number of days after infection until transmission-blocking immunity begins to decay. Relevant only when Enable_Immunity and Enable_Immune_Decay parameters are set to 1. \u00a0 Zoonosis_Rate float 0 1 0 The daily rate of zoonotic infection per individual when the Animal_Reservoir_Type parameter is set to either CONSTANT_ZOONOSIS or ZOONOSIS_FROM_DEMOGRAPHICS. \u00a0 Geography and environment \u00b6 Parameter Data type Minimum Maximum Default Description Example Air_Temperature_Variance float 0 5 2 The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily air temperature values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA. Only used if the Enable_Climate_Stochasticity is set to 1. \u00a0 Base_Air_Temperature float -55 45 22 The value of air temperature where Climate_Model is set to CLIMATE_CONSTANT. \u00a0 Base_Land_Temperature float -55 60 26 The value of land temperature where Climate_Model is set to CLIMATE_CONSTANT. \u00a0 Base_Rainfall float 0 150 10 The value of rainfall per day in mm when Climate_Model is set to CLIMATE_CONSTANT. \u00a0 Base_Relative_Humidity float 0 1 0.75 The value of humidity where Climate_Model is set to CLIMATE_CONSTANT. \u00a0 Climate_Model enum NA NA CLIMATE_OFF Determines how the climate of a simulation is configured and from what file(s). Possible values are: CLIMATE_OFF, CLIMATE_CONSTANT - Uses the conditional parameters that give the fixed values of temperature or rain for land temperature, air temperature, rainfall and humidity. CLIMATE_KOPPEN - Uses an input file that decodes Koppen codes by geographic location. CLIMATE_BY_DATA - Reads everything out of several input files with additional parameters that allow the addition of stochasticity or scale offsets. \u00a0 Climate_Update_Resolution enum NA NA CLIMATE_UPDATE_YEAR Update resolution for data in climate files. Possible values are: CLIMATE_UPDATE_YEAR, CLIMATE_UPDATE_MONTH, CLIMATE_UPDATE_WEEK, CLIMATE_UPDATE_DAY and CLIMATE_UPDATE_HOUR. \u00a0 Default_Geography_Initial_Node_Population integer 0 1000000 1000 When using the default geography (i.e. Enable_Demographics_Initial = 0), this is the initial number of individuals in each node. \u00a0 Default_Geography_Torus_Size integer 3 100 10 When using the default geography (i.e. Enable_Demographics_Initial = 0), this is the square root of the number of nodes in the simulation. For migration, the nodes are assumed to be a torus. \u00a0 Enable_Climate_Stochasticity boolean 0 1 0 Controls overall stochasticity for climate. Use value 0 to disable all additional climate stochasticity. Use value 1 to enable additional variation as specified by the parameters: Air_Temperature_Variance, Land_Temperature_Variance, Enable_Rainfall_Stochasticity and Relative_Humidity_Variance. \u00a0 Enable_Rainfall_Stochasticity boolean 0 1 1 Set to 0 to disable rainfall stochasticity. Set to 1 to enable stochastic variation of rainfall drawn from an exponential distribution (with a mean value as the daily rainfall from the Climate_Model values, CLIMATE_CONSTANT or CLIMATE_BY_DATA). \u00a0 Land_Temperature_Offset float -20 20 0 The linear shift of land surface temperature in degrees Celsius. This is only used or needed when Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Land_Temperature_Variance float 0 7 2 The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily land temperature values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA. Only used if the Enable_Climate_Stochasticity is set to 1. \u00a0 Node_Grid_Size float 0.004167 90 0.004167 Spatial resolution indicating the node grid size for a simulation in degrees. \u00a0 Rainfall_In_mm_To_Fill_Swamp float 1 10000 1000 Millimeters of rain to fill larval habitat to capacity. Only used for vector species with Habitat_Type set to BRACKISH_SWAMP. \u00a0 Rainfall_Scale_Factor float 0.1 10 1 The scalar used in multiplying rainfall value(s). This is only used or needed when Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Relative_Humidity_Scale_Factor float 0.1 10 1 The scalar used in multiplying relative humidity value(s). This is only used or needed when Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Relative_Humidity_Variance float 0 0.12 0.05 The standard deviation (in percentage, %) for normally distributed noise applied to the daily relative humidity values when Climate_Model is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA. Only used if the Enable_Climate_Stochasticity is set to 1. \u00a0 Input files \u00b6 Parameter Data type Minimum Maximum Default Description Example Air_Migration_Filename string NA NA \u00a0 Path of input file defining patterns of migration by airplane. \u00a0 Air_Temperature_Filename string NA NA air_temp.json Path of input file defining air temperature data measured 2 meters above ground. Relevant only if Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Campaign_Filename string NA NA \u00a0 Path of the simulation campaign file. This configuration file defines campaign related settings and thus, is an optional file, only required when interventions are part of the simulation. \u00a0 Demographics_Filenames Vector String NA NA \u00a0 Array of demographics files containing information on the identity and demographics of the region to simulate. \u00a0 Koppen_Filename string NA NA UNINITIALIZED STRING Path to input file used to specify Koppen climate classifications. This is only used or needed when Climate_Model is set to CLIMATE_KOPPEN. \u00a0 Land_Temperature_Filename string NA NA land_temp.json Path of input file defining temperature data measured at land surface. Relevant only if Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Local_Migration_Filename string NA NA \u00a0 Path of input file defining patterns of migration to adjacent nodes by foot travel. \u00a0 Rainfall_Filename string NA NA rainfall.json Path of input file defining rainfall data. Relevant only if Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Regional_Migration_Filename string NA NA \u00a0 Path of input file defining patterns of migration by vehicle via road or rail network. If the node is not on a road or rail network, regional migration focuses on the closest hub city in the network. \u00a0 Relative_Humidity_Filename string NA NA rel_hum.json Path of input file defining relative humidity data measured 2 meters above ground. Relevant only if Climate_Model is set to CLIMATE_BY_DATA. \u00a0 Sea_Migration_Filename string NA NA \u00a0 Path of input file defining patterns of migration by ship. \u00a0 Malaria \u00b6 Parameter Data type Minimum Maximum Default Description Example Anemia_Mortality_Inverse_Width float 0.1 1000000 10 The inverse width relative to threshold value of severe disease turn-on around threshold for anemia. \u00a0 Anemia_Mortality_Threshold double NA NA 3 The threshold value on hemoglobin count in grams per deciliter (g/dL) at which 50% of individuals die per day. \u00a0 Anemia_Severe_Inverse_Width float 0.1 1000000 10 The inverse width relative to threshold value of severe disease turn-on around threshold for anemia. \u00a0 Anemia_Severe_Threshold float 0 100 5 The severe disease threshold level for anemia. Threshold units are in grams per deciliter (g/dL). \u00a0 Antibody_Capacity_Growth_Rate float 0 1 0.1 Maximum daily rate of specific antibody capacity increase. Antibody production begins at capacity of 0.3. Hyperimmunity results at 0.4. The actual growth rate will tend to be lower and is a function of antigen concentrations. \u00a0 Antibody_CSP_Decay_Days float 1 3.40E+38 90 Exponential decay time for CSP antibody concentration (in days). \u00a0 Antibody_CSP_Killing_Inverse_Width float 1.00E-06 1000000 1.5 Inverse width of sigmoidal sporozoite killing function of CSP antibody concentration. \u00a0 Antibody_CSP_Killing_Threshold float 1.00E-06 1000000 10 Threshold value on CSP antibody concentration for sporozoite killing. \u00a0 Antibody_IRBC_Kill_Rate double NA NA 2 Factor multiplied by antibody level to produce the rate of clearance of the infected red blood cell (IRBC) population. \u00a0 Antibody_Memory_Level float 0 0.35 0.2 Limiting level of antibody capacity that remains after a prolonged absence of a specific antigen. The antibody capacity decays to this level gradually after infection is cleared. The decay rate in antibody capacity is set so that hyperimmunity is lost within 4 months, and capacity continues to decay to this level. The antibody memory level is relevant for year-scale dynamics, but not for long dynamics (10-20 years). \u00a0 Antibody_Stimulation_C50 float 0.1 10000 10 Concentration of an antigen, measured in IRBC/ul at which growth in antibody capacity against the antigen increases at half the maximum rate specified by Antibody_Capacity_Growth_Rate. \u00a0 Antigen_Switch_Rate double NA NA 2.00E-09 Antigenic switching rate per infected red blood cell per asexual cycle. See Parasite_Switch_Type for different switching patterns. \u00a0 Base_Gametocyte_Fraction_Male double NA NA 0.2 Fraction of gametocytes that are male. \u00a0 Base_Gametocyte_Mosquito_Survival_Rate float 0 1 0.01 Average fraction of gametocytes in blood meal that are successful in infecting mosquito in the absence of other modulating effects, such as fever. \u00a0 Base_Gametocyte_Production_Rate double NA NA 0.02 Fraction of IRBCs producing gametocytes. \u00a0 Base_Sporozoite_Survival_Fraction float 0 1 0.25 Fraction of sporozoites that survive to infect a hepatocyte in the absence of anti-CSP protection. \u00a0 Clinical_Fever_Threshold_High float 0 15 1 The fever threshold (in degrees Celsius above normal) to start a clinical incident. \u00a0 Clinical_Fever_Threshold_Low float 0 5 1 The fever threshold (in degrees Celsius above normal) to end a clinical incident. \u00a0 Cytokine_Gametocyte_Inactivation float 0 1 0.02 The strength of inflammatory response inactivation of gametocytes. \u00a0 Enable_Sexual_Combination boolean 0 1 0 Set to 1 to enable sexual combination of male and female gametocytes. Set to 0 for transmission of each strain only. Note: This parameter is currently not in use. \u00a0 Erythropoiesis_Anemia_Effect float 0 1000 3.5 Exponential rate of increased red-blood-cell production from reduced red-blood-cell availability. \u00a0 Falciparum_MSP_Variants integer 0 1000 100 The number of distinct MSP variants for P. falciparum malaria in the overall parasite population in the simulation, not necessarily in an individual. \u00a0 Falciparum_Nonspecific_Types integer 0 1000 20 The number of distinct non-specific types of P. falciparum malaria. \u00a0 Falciparum_PfEMP1_Variants integer 0 100000 1000 The number of distinct PfEMP1 variants for P. falciparum malaria in the overall parasite population in the simulation. \u00a0 Fever_Detection_Threshold float 0.5 5 1 Level of body temperature above normal, in degrees Celsius, corresponding to detectable fever. Normal body temperature is defined as 37 degrees Celsius. \u00a0 Fever_IRBC_Kill_Rate float 0 1000 0.15 Factor multiplied by fever level to produce a decay rate of infected red blood cell population due to inflammatory innate immune response. \u00a0 Fever_Mortality_Inverse_Width float 0.1 1000000 10 The inverse width relative to threshold value of mortality turn-on around threshold for fever. \u00a0 Fever_Mortality_Threshold double NA NA 3 The fever mortality threshold for a simulation in units of degrees Celsius above normal body temperature. Sets the midpoint in the sigmoidal response for mortality rate associated with fever. \u00a0 Fever_Severe_Inverse_Width float 0.1 1000000 10 The inverse width relative to threshold value of severe disease turn-on around threshold for fever. \u00a0 Fever_Severe_Threshold float 0 10 1.5 The severe disease threshold level for fever. \u00a0 Gametocyte_Stage_Survival_Rate double NA NA 1 Rate of gametocyte survival from one development stage to the next in the absence of drugs or inflammatory immune response. \u00a0 Innate_Immune_Variation_Type enum NA NA NONE Type of individual innate immune variation. Possible values are: NONE (homogeneous), PYROGENIC_THRESHOLD (parasite threshold resulting in cytokine production), PYROGENIC_THRESHOLD_VS_AGE (age-dependent threshold), CYTOKINE_KILLING (effectiveness of cytokines in killing infected red-blood cells). \u00a0 Malaria_Model enum NA NA MALARIA_MECHANISTIC_MODEL The type of malaria model used in a simulation. Possible values are: MALARIA_FIXED_DURATION, MALARIA_EXPONENTIAL_DURATION, MALARIA_REDUCEDSTATE_MODEL and MALARIA_MECHANISTIC_MODEL. Important: All malaria simulations should use MALARIA_MECHANISTIC_MODEL. The others are placeholder values. \u00a0 Malaria_Strain_Model enum NA NA FALCIPARUM_NONRANDOM_STRAIN The generator that is used to construct the antigenic repertoire of a malaria infection. Possible values are: FALCIPARUM_NONRANDOM_STRAIN, FALCIPARUM_RANDOM50_STRAIN, FALCIPARUM_RANDOM_STRAIN and FALCIPARUM_STRAIN_GENERATOR. \u00a0 Maternal_Antibodies_Type enum NA NA OFF Type of maternal antibody protection. Possible values are: OFF, SIMPLE_WANING, CONSTANT_INITIAL_IMMUNITY (independent of acquired immunity in possible mothers). \u00a0 Maternal_Antibody_Decay_Rate float 0 3.40E+38 0.01 Decay rate per day in protection from maternal antibodies. \u00a0 Maternal_Antibody_Protection float 0 1 0.1 Strength of protection from maternal antibodies as a multiple of full antibody killing effect. \u00a0 Max_MSP1_Antibody_Growthrate float 0 1 0.02 Maximum increase in MSP1 antibody capacity during each asexual cycle. The higher this value, the sooner early clearances are observed and the earlier the parasite density envelope decreases. \u00a0 Mean_Sporozoites_Per_Bite float 0 1000 11 Mean number of sporozoites per infectious mosquito bite. \u00a0 Merozoites_Per_Hepatocyte double NA NA 15000 Number of IRBCs caused by a single infected hepatocyte at the start of infection. \u00a0 Merozoites_Per_Schizont double NA NA 16 Number of merozoites released by a single infected schizont after each asexual cycle. The number of resulting IRBC\u2019s depends on the RBC availability and merozoite-specific immunity. \u00a0 Min_Adapted_Response float 0 1 0.02 Minimum level of antibody stimulation to novel antigen. The value sets the low-range asymptote for antibody capacity growth, which is calculated from Antibody_Capacity_Growth_Rate and antigen density, in the presence of any nonzero antigen level. \u00a0 Min_Days_Between_Clinical_Incidents float 0 1000000 3 The number of days with fever below the low threshold before a new incident can start when the high fever threshold is again exceeded. \u00a0 MSP1_Merozoite_Kill_Fraction double NA NA 0.5 Fraction of merozoites inhibited from invading new erythrocytes when MSP1-specific antibody level is 1. \u00a0 New_Diagnostic_Sensitivity float 0.0001 100000 0.01 The number of microliters of blood tested to find single parasites in a new diagnostic (corresponds to inverse parasites/microliters sensitivity). \u00a0 Nonspecific_Antibody_Growth_Rate_Factor float 0 1000 0.5 Factor that adjusts Antibody_Capacity_Growth_Rate for less immunogenic surface proteins called the minor epitopes. \u00a0 Nonspecific_Antigenicity_Factor double NA NA 0.2 Nonspecific antigenicity factor that adjusts antibody IRBC killrate to account for IRBCs caused by antibody responses to antigenically weak surface proteins. \u00a0 Number_Of_Asexual_Cycles_Without_Gametocytes integer 0 500 1 Number of asexual reproduction cycles that do not produce gametocytes. All later cycles will produce gametocytes according to Base_Gametocyte_Production_Rate. \u00a0 Parasite_Mortality_Inverse_Width float 0.1 1000000 10 The inverse width relative to threshold value of mortality turn-on around threshold for parasite density. \u00a0 Parasite_Mortality_Threshold double NA NA 2000000 The parasite-density mortality threshold for a simulation. \u00a0 Parasite_Severe_Inverse_Width float 0.1 1000000 10 The inverse width relative to threshold value of severe disease turn-on around threshold for parasite density. \u00a0 Parasite_Severe_Threshold float 0 3.40E+38 1000000 The severe disease threshold level for parasite density. \u00a0 Parasite_Smear_Sensitivity float 0.0001 100 0.1 The number of microliters of blood tested to find single parasites in a traditional smear (corresponds to inverse parasites/microliters sensitivity). \u00a0 Parasite_Switch_Type enum NA NA CONSTANT_SWITCH_RATE_2VARS Parasite switch type for erythrocyte surface antigens. Possible values are: CONSTANT_SWITCH_RATE_2VARS, RATE_PER_PARASITE_7VARS - this is the one that is best parameterized - and finally, RATE_PER_PARASITE_5VARS_DECAYING. \u00a0 PKPD_Model enum NA NA FIXED_DURATION_CONSTANT_EFFECT Determines which pharmacokinetic pharmacodynamic module to use. Possible values are: FIXED_DURATION_CONSTANT_EFFECT and CONCENTRATION_VERSUS_TIME. \u00a0 Pyrogenic_Threshold float 0.1 20000 1000 Level of bloodstream infection, measured in IRBC per microliter, at which stimulation of the innate inflammatory immune response is half its maximum value. \u00a0 RBC_Destruction_Multiplier double NA NA 9.5 Number of total RBCs destroyed per infected rupturing schizont. \u00a0 Malaria drug efficacy \u00b6 Parameter Data type Minimum Maximum Default Description Example Bodyweight_Exponent float 0 100000 0 Effect of body weight on maximum drug concentration in an individual patient. \u00a0 Drug_Cmax float 0 100000 1000 Maximum drug concentration in the same units as Drug_PKPD_C50. Note: Since drug concentrations and C50 values are often specified with different unit systems, these two parameters have been designed to be flexible, provided that the units are the same for each. \u00a0 Drug_Decay_T1 float 0 100000 1 Primary drug decay rate, in days. \u00a0 Drug_Decay_T2 float 0 100000 1 Secondary drug decay rate, in days. \u00a0 Drug_Dose_Interval float 0 100000 1 Interval between doses, in days. \u00a0 Drug_Fulltreatment_Doses float 1 100000 3 Number of doses for a full treatment. \u00a0 Drug_Gametocyte02_Killrate float 0 100000 0 Log reduction per day in early-stage gametocytes. \u00a0 Drug_Gametocyte34_Killrate float 0 100000 0 Log reduction per day in late-stage gametocytes. \u00a0 Drug_GametocyteM_Killrate float 0 100000 0 Log reduction per day in mature gametocytes at saturated drug concentration. \u00a0 Drug_Hepatocyte_Killrate float 0 100000 0 Log reduction in hepatocytes per day. \u00a0 Drug_PKPD_C50 float 0 100000 100 Concentration at which drug killing rates are half of maximum. Note: This parameter and the Drug_Cmax parameter must use the same units. \u00a0 Drug_Vd float 0 100000 10 Volume of distribution. This value is the ratio of the volume of the second compartment to the volume of the first compartment in a two-compartment model, and is dimensionless. \u00a0 Fraction_Of_Adult_Dose DoseMap 0 1 NA The fraction of an adult drug dose given to children below the corresponding age (in years). \u00a0 Fractional_Dose_By_Upper_Age array NA NA NA An array of doses that contain Upper_Age_In_Years and Fraction_Of_Adult_Dose values, in days. \u00a0 Malaria_Drug_Params JSON object NA NA NA This JSON structure defines the name of an anti-malaria drug and the parameters that define it. { \"Malaria_Drug_Params\" : { \"Artemether\" : { \"Bodyweight_Exponent\" : 0 , \"Drug_Cmax\" : 114 , \"Drug_Decay_T1\" : 0.12 , \"Drug_Decay_T2\" : 0.12 , \"Drug_Vd\" : 1 , \"Drug_PKPD_C50\" : 0.6 , \"Drug_Dose_Interval\" : 0.5 , \"Drug_Fulltreatment_Doses\" : 6 , \"Drug_Gametocyte02_Killrate\" : 2.5 , \"Drug_Gametocyte34_Killrate\" : 1.5 , \"Drug_GametocyteM_Killrate\" : 0.7 , \"Drug_Hepatocyte_Killrate\" : 0.0 , \"Max_Drug_IRBC_Kill\" : 8.9 , \"Fractional_Dose_By_Upper_Age\" : [ { \"Fraction_Of_Adult_Dose\" : 0.25 , \"Upper_Age_In_Years\" : 3 }, { \"Fraction_Of_Adult_Dose\" : 0.5 , \"Upper_Age_In_Years\" : 6 }, { \"Fraction_Of_Adult_Dose\" : 0.75 , \"Upper_Age_In_Years\" : 10 }, ] } } } \u00a0 \u00a0 \u00a0 \u00a0 \u00a0 \u00a0 \u00a0 Max_Drug_IRBC_Kill float 5 100000 5 Maximum log reduction in IRBCs per day. \u00a0 Upper_Age_In_Years DoseMap 0 125 NA The age (in years) below which children are given the corresponding fraction of an adult drug dose. \u00a0 Migration \u00b6 Parameter Data type Minimum Maximum Default Description Example Air_Migration_Roundtrip_Duration float 0 10000 1 The average time spent at the destination node during a round-trip migration by airplane. \u00a0 Air_Migration_Roundtrip_Probability float 0 1 0.8 Likelihood that an individual who flies to another cell will return to the cell of origin during the next migration. Relevant only when Enable_Air_Migration is selected. \u00a0 Enable_Air_Migration boolean 0 1 0 Set to 1 to account for migration by airplane in and out of cities with airports. Set to 0 to assume by default that no migration occurs by air travel. \u00a0 Enable_Local_Migration boolean 0 1 0 Set to 1 to enable local migration, the diffusion of people in and out of nearby nodes by foot travel. Set to 0 to disable local migration. \u00a0 Enable_Migration_Heterogeneity boolean 0 1 1 Set to 1 to use migration rate distribution in the demographics file. Set to 0 to assume by default that the same migration rate applies for all individuals. \u00a0 Enable_Regional_Migration boolean 0 1 0 Set to 1 to account for migration by road vehicle in and out of nodal cities in the road network. Set to 0 to assume by default that no migration occurs by road travel. \u00a0 Enable_Sea_Migration boolean 0 1 0 Set to 1 to account for migration on ships in and out of coastal cities with seaports. Set to 0 to assume by default that no migration occurs by sea travel. \u00a0 Local_Migration_Roundtrip_Duration float 0 10000 1 The average time spent at the destination node during a round-trip migration by foot travel. \u00a0 Local_Migration_Roundtrip_Probability float 0 1 0.95 Likelihood that an individual who walks into a neighboring cell will return to the cell of origin during the next migration. Relevant only when Enable_Local_Migration is selected. \u00a0 Migration_Model enum NA NA NO_MIGRATION Model to use for migration. Possible values are: NO_MIGRATION, FIXED_RATE_MIGRATION, VARIABLE_RATE_MIGRATION and LEVY_FLIGHTS. Note: VARIABLE_RATE_MIGRATION and LEVY FLIGHTS are currently not supported. \u00a0 Migration_Pattern enum NA NA RANDOM_WALK_DIFFUSION The type of roundtrip. For example, a single roundtrip. Possible values are: RANDOM_WALK_DIFFUSION, SINGLE_ROUND_TRIPS and WAYPOINTS_HOME. \u00a0 Regional_Migration_Roundtrip_Duration float 0 10000 1 The average time spent at the destination node during a round-trip migration by road network. \u00a0 Regional_Migration_Roundtrip_Probability float 0 1 0.1 Likelihood that an individual who travels by vehicle to another cell will return to the cell of origin during the next migration. Relevant only when Enable_Regional_Migration is checked. \u00a0 Roundtrip_Waypoints integer 0 1000 10 The maximum number of points reached during a trip before steps are retraced on the return trip home. \u00a0 Sea_Migration_Roundtrip_Duration float 0 10000 1 The average time spent at the destination node during a round-trip migration by ship. \u00a0 Sea_Migration_Roundtrip_Probability float 0 1 0.25 Likelihood that an individual who travels by ship into a neighboring cell will return to the cell of origin during the next migration. Relevant only when Enable_Sea_Migration is checked. \u00a0 Output options \u00b6 Parameter Data type Minimum Maximum Default Description Example Enable_Default_Reporting boolean 0 1 1 Enables and disables the default InsetChart.json report. \u00a0 Enable_Demographics_Reporting boolean 0 1 1 Outputs demographic summary data and age-binned reports to file. \u00a0 Enable_Property_Output boolean 0 1 0 Use value 0 to disable Individual_Property reports (separate file). Use value 1 to enable. \u00a0 Enable_Spatial_Output boolean 0 1 0 Use value 0 to disable all spatial output channels. Use value 1 to enable spatial output of all channels listed in the array parameter: Spatial_Output_Channels. For full documentation of spatial output channels, please see the online documentation. \u00a0 Report_Event_Recorder boolean 0 1 1 Enables or disables the ReportEventRecorder.csv report. \u00a0 Sampling \u00b6 Parameter Data type Minimum Maximum Default Description Example Base_Individual_Sample_Rate float 0 1 1 Base rate of sampling for individuals. This rate equals the fraction of individuals in each node being sampled. \u00a0 Individual_Sampling_Type enum NA NA TRACK_ALL Type of individual human sampling. Possible values are: TRACK_ALL, FIXED_SAMPLING, ADAPTED_SAMPLING_BY_POPULATION_SIZE, ADAPTED_SAMPLING_BY_AGE_GROUP, and ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE. \u00a0 Max_Node_Population_Samples float 1 3.40E+38 30 Number of individuals when the sampling rate starts dropping to the rate of adapted sampling by population size. \u00a0 Scaling factors \u00b6 Parameter Data type Minimum Maximum Default Description Example x_Air_Migration float 0 3.40E+38 1 Multiplier for rate of migration by air. \u00a0 x_Birth float 0 3.40E+38 1 Multiplier of birth rate from input demographics file. \u00a0 x_Local_Migration float 0 3.40E+38 1 Multiplier for rate of migration by foot travel. \u00a0 x_Other_Mortality float 0 3.40E+38 1 Multiplier for scaling mortality from causes other than disease being simulated. \u00a0 x_Regional_Migration float 0 3.40E+38 1 Multiplier for rate of migration by road vehicle. \u00a0 x_Sea_Migration float 0 3.40E+38 1 Multiplier for rate of migration by sea. \u00a0 x_Temporary_Larval_Habitat double NA NA 1 Scales the habitat size for all mosquito populations. \u00a0 Simulation setup \u00b6 Parameter Data type Minimum Maximum Default Description Example Config_Name string NA NA UNINITIALIZED STRING User-supplied title naming a configuration. \u00a0 Enable_Interventions boolean 0 1 0 Set to 1 to load campaign interventions from file. Set to 0 to run a baseline simulation without campaigns. \u00a0 Run_Number integer 0 2147480000 1 Sets the random number seed through a bit manipulation process for USE_PSEUDO_DES. When running a multi-core simulation, combines with processor rank to produce independent random number streams for each process. \u00a0 Simulation_Duration float 0 1000000 1 Elapsed time noted in days from the start to the end of a simulation. \u00a0 Simulation_Timestep float 0 1000000 1 Value indicating simulation time step in days. \u00a0 Simulation_Type enum NA NA GENERIC_SIM Type of disease being simulated. Supported values include GENERIC_SIM, VECTOR_SIM, MALARIA_SIM, TB_SIM, STI_SIM, HIV_SIM, and PY_SIM. To query the values supported for your specific build, use the \u2013version option. \u00a0 Start_Time float 0 1000000 1 Time noted in days when the simulation begins. This time influences the point in the temporal input data, such as where in the climate input the DTK starts running the simulation. Note: The Start_Day of campaign events is in absolute time. So, time relative to the beginning of the simulation depends on this parameter. \u00a0 Spatial output channels \u00b6 Parameter Data type Minimum Maximum Default Description Example Spatial_Output_Channels Fixed String Set NA NA [] An array of channel names for spatial output by Node and time step. The data from each channel will be written to a separate binary file. Channel names include Air_Temperature, Births, Campaign_Cost, Disease_Deaths, Human_Infectious_Reservoir, Infection_Rate, Land_Temperature, New_Infections, New_Reported_Infections, Population, Rainfall and Relative_Humidity. \u00a0 Vector \u00b6 Parameter Data type Minimum Maximum Default Description Example Age_Dependent_Biting_Risk_Type enum NA NA OFF Type of functional form for age-dependent biting-risk. Possible values are: OFF, LINEAR, SURFACE_AREA_DEPENDENT. \u00a0 Egg_Hatch_Delay_Distribution enum NA NA NO_DELAY Is there a delay in egg hatching from oviposition? If so, how is that delay distributed? Possible values are: NO_DELAY and EXPONENTIAL_DURATION. \u00a0 Egg_Saturation_At_Oviposition enum NA NA NO_SATURATION Is there saturation in the number of viable eggs at oviposition depending on available larval habitat? Possible values are: NO_SATURATION and SATURATION_AT_OVIPOSITION. \u00a0 Enable_Temperature_Dependent_Feeding_Cycle boolean 0 1 0 Set to 1 to have the duration between blood-feeds depend on the temperature. Otherwise, if set to 0, days between feeds will be constant and specified by the parameter, Days_Between_Feeds, for each species. \u00a0 Enable_Vector_Aging boolean 0 1 0 Set to 1 to simulate vector senescence. Set to 0 to assume that vector behavior does not change with age. \u00a0 Enable_Vector_Migration boolean 0 1 0 Set to 1 to enable vector migration. Set to 0 to disable vector migration. Note: Only supported for individual-vector model with Vector_Sampling_Type value of TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. \u00a0 Enable_Vector_Migration_Local boolean 0 1 0 Set to 1 to consider local vector migration by vectors to adjacent nodes. Set to 0 to not consider local vector migration. \u00a0 HEG_Fecundity_Limiting float 0 1 0 The fractional reduction in the number of eggs laid by a female mosquito who is homozygous in the homing endonuclease gene (HEG). \u00a0 HEG_Homing_Rate float 0 1 0 The fractional redistribution from heterozygous offspring to the homozygous homing-endonuclease-gene (HEG) behavior. \u00a0 HEG_Model enum NA NA OFF An enum governing the homing endonuclease gene (HEG) dynamics. Possible values are: OFF, GERMLINE_HOMING (early), EGG_HOMING (late). \u00a0 Human_Feeding_Mortality float 0 1 0.1 Fraction (dimensionless) of mosquitoes that die while feeding on a human. As of version 1.5, this is now a one-value-per-simulation quantity. \u00a0 Larval_Density_Dependence enum NA NA UNIFORM_WHEN_OVERPOPULATION Set to UNIFORM_WHEN_OVERPOPULATION to apply mortality uniformly to all larvae when the population exceeds the specified carrying capacity for that habitat. Set to GRADUAL_INSTAR_SPECIFIC to apply mortality and delayed growth in an instar-specific fashion where the younger larvae are more susceptible to predation and competition from older larvae. The LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY and DENSITY_DELAYED_GROWTH_NOT_MORTALITY options configure only the age-dependent mortality component or only the delayed-growth component, respectively. For the NO_DENSITY_DEPENDENCE option, there is no additional larval density-dependent mortality factor. Possible values are: UNIFORM_WHEN_OVERPOPULATION, GRADUAL_INSTAR_SPECIFIC, LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY, DENSITY_DELAYED_GROWTH_NOT_MORTALITY, NO_DENSITY_DEPENDENCE. \u00a0 Larval_Density_Mortality_Offset float 0.0001 1000 0.1 An offset factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models. \u00a0 Larval_Density_Mortality_Scalar float 0.01 1000 10 A scale factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models. \u00a0 Larval_Rainfall_Mortality_Threshold float 0.01 1000 100 The threshold value on daily rainfall in millimeters, above which larval mortality is applied when Vector_Larval_Rainfall_Mortality is set to either SIGMOID or SIGMOID_HABITAT_SHIFTING. \u00a0 Mean_Egg_Hatch_Delay float 0 120 0 Mean delay in egg hatch time noted in days from the time of oviposition. \u00a0 Mosquito_Weight integer 1 10000 1 Value indicating how many mosquitoes are represented by a sample mosquito. This parameter is only used for Vector_Sampling_Type value SAMPLE_IND_VECTORS. \u00a0 Rainfall_In_mm_To_Fill_Swamp float 1 10000 1000 Millimeters of rain to fill larval habitat to capacity. Only used for vector species with Habitat_Type set to BRACKISH_SWAMP. \u00a0 Semipermanent_Habitat_Decay_Rate float 0.0001 100 0.01 Daily rate of larval habitat loss for semi-permanent habitats with Habitat_Type parameter of WATER_VEGETATION or BRACKISH_SWAMP. \u00a0 Temporary_Habitat_Decay_Factor float 0.001 100 0.05 Factor to convert raw evaporation rate (ignoring boundary layer effects) to the daily rate of larval habitat loss for temporary habitats with Habitat_Type parameter of TEMPORARY_RAINFALL. Units are (larval carrying capacity per day) / (kg per square meter per second). \u00a0 Vector_Larval_Rainfall_Mortality enum NA NA NONE Type of vector larval mortality function due to rainfall. Possible values are NONE, SIGMOID, where the mortality rate grows linearly from 0 at the threshold to 1 at twice the threshold value, and SIGMOID_HABITAT_SHIFTING, where the threshold value is reduced by a factor proportional to how full the larval habitat is. \u00a0 Vector_Sampling_Type enum NA NA TRACK_ALL_VECTORS Type of vector module used in a simulation. Possible values are: TRACK_ALL_VECTORS, SAMPLE_IND_VECTORS, VECTOR_COMPARTMENTS_NUMBER and VECTOR_COMPARTMENTS_PERCENT. \u00a0 Vector_Species_Names Dynamic String Set NA NA [] An array of vector species names for all species that are present in the simulation. For example, [\u201carabiensis\u201d, \u201cfunestus\u201d, \u201cgambiae\u201d]. Any vector species name may be used. However, you must specify the parameters as listed in Vector_Species_Params for each species. See Vector Species Parameters for a usage example. \u00a0 Vector_Sugar_Feeding_Frequency enum NA NA VECTOR_SUGAR_FEEDING_NONE How often does a female mosquito take a sugar feed? Possible values are: VECTOR_SUGAR_FEEDING_NONE - No sugar feeding. VECTOR_SUGAR_FEEDING_ON_EMERGENCE_ONLY - Sugar feeding once at emergence. VECTOR_SUGAR_FEEDING_EVERY_FEED - Sugar feeding occurs once per blood meal. VECTOR_SUGAR_FEEDING_EVERY_DAY - Sugar feeding occurs everyday. Note: This parameter is used in conjunction with the SugarTrap and OvipositionTrap interventions. These features require Vector_Sampling_Type to be set to TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. \u00a0 Wolbachia_Infection_Modification float 0 100 1 The change in vector susceptibility to infection due to a Wolbachia infection. \u00a0 Wolbachia_Mortality_Modification float 0 100 1 The change in vector mortality due to a Wolbachia infection. \u00a0 Vector species \u00b6 Parameter Data type Minimum Maximum Default Description Example Acquire_Modifier float 0 1 1 Modifier of the probability of successful infection of a mosquito by a malaria-infected individual, given the individual\u2019s infectiousness. { \"Acquire_Modifier\" : 0.2 } Adult_Life_Expectancy float 0 730 10 Number of days an average mosquito survives. The daily adult mortality rate is 1 / (value of this parameter). \u00a0 Anthropophily float 0 1 1 Propensity (dimensionless) of the mosquito to feed on humans instead of animals, measured as a fraction of blood meals from human hosts. This propensity is important in differentiating the mosquito species. \u00a0 Aquatic_Arrhenius_1 float 0 1.00E+15 8.42E+10 The duration of aquatic development is a decreasing function of temperature. The daily rate of fractional progression from egg-hatching to emergence is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a1, is a temperature-independent scale factor on the development rate. Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111. \u00a0 Aquatic_Arrhenius_2 float 0 1.00E+15 8328 The duration of aquatic development is a decreasing function of temperature. The daily rate of fractional progression from egg-hatching to emergence is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a2, governs how quickly the rate changes with temperature. Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111. \u00a0 Aquatic_Mortality_Rate float 0 1 0.1 Base aquatic mortality per day for the species before adjustment for effects of overpopulation and drying. The rate is locally adjusted for other considerations, including overpopulation and drying out of the aquatic habitat. \u00a0 Days_Between_Feeds float 0 730 2 Factor (in days) used to calculate the rate at which female mosquitoes attempt to feed. The rate equals 1 / (value of this parameter). The Days_Between_Feeds parameter represents a 3-day feeding cycle at normal tropical temperatures (30 C). If it is colder, the cycle is 4 days at 24 C, and if the temperature is warmer, the cycle is shorter, increasing the fraction of 2-day cycles from 30 C to 48 C. Whole temperature-dependent distribution can be shifted by changing Days_Between_Feeds which is always the value at 30 C. See Enable_Temperature_Dependent_Feeding_Cycle for weather-dependent feeding. \u00a0 Egg_Batch_Size float 0 10000 100 Size of egg batch laid by one female mosquito that has fed successfully. \u00a0 Immature_Duration float 0 730 2 Factor (in days) used to calculate the rate at which immature organisms develop into adult mosquitoes, which equals 1 / (value of this parameter). Note that development from immature individual to adult is not dependent on temperature. \u00a0 Indoor_Feeding_Fraction float 0 1 1 Fraction (dimensionless) of feeds on humans that occur indoors. Thus the fraction of feeds on humans that occur outdoors equals 1 - (value of this parameter).\u201d \u00a0 Infected_Arrhenius_1 float 0 1.00E+15 1.17E+11 The duration of sporogony is a decreasing function of temperature. The daily rate of fractional progression of infected mosquitoes to an infectious state is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a1, is a temperature-independent scale factor on the development rate. Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111. \u00a0 Infected_Arrhenius_2 float 0 1.00E+15 8340 The duration of sporogony is a decreasing function of temperature. The daily rate of fractional progression of infected mosquitoes to an infectious state is parameterized by the Arrhenius equation, a1 * exp( -a2 / T ), with T in degrees Kelvin. This parameter, a2, governs how quickly the rate changes with temperature. Reference: Craig M.H., Snow R.W., et al. (1999). A climate-based distribution model of malaria transmission in sub-Saharan Africa. Parasitology Today 15(3) 105-111. \u00a0 Infected_Egg_Batch_Factor float 0 10 0.8 Factor (dimensionless) used to modify egg batch size to account for effects of infection in a female mosquito. Female mosquitoes undergoing sporogony have reduced fertility.\u201d \u00a0 Infectious_Human_Feed_Mortality_Factor float 0 1000 1.5 Correction factor (dimensionless) used to modify the death rate of mosquitoes when feeding on humans, to account for the increase caused by infections in mosquitoes. Infected mosquitoes have a higher mortality rate while feeding on humans than uninfected mosquitoes. References: Wekesa J. W., Copeland R. S. , et al. (1992). Effect of Plasmodium Falciparum on Blood Feeding Behavior of Naturally Infected Anopheles Mosquitoes in Western Kenya. Am J Trop Med Hyg 47(4): 484-488.Anderson R. A., Knols B. G. J., et al. (2000). Plasmodium falciparum sporozoites increase feeding-associated mortality of their mosquito hosts Anopheles gambiae s.l. Parasitology 120(04): 329-333. \u00a0 Larval_Habitat_Types nested JSON object 0 1.00E+15 1.25E+10 A measure of the habitat type and scaling factors to use to estimate larval population. You can specify this with a dictionary specifying habitat type with a simple numeric scaling factor or, for LINEAR_SPLINE, with a more detailed configuration for scaling. Possible habitat values are: \u201cCONSTANT\u201d \u201cTEMPORARY_RAINFALL\u201d \u201cWATER_VEGETATION\u201d \u201cHUMAN_POPULATION\u201d \u201cBRACKISH_SWAMP\u201d \u201cLINEAR_SPLINE\u201d The numeric scaling value represents larval density with the number of larvae in a 1x1-degree area. The factor multiplicatively scales the resulting weather or population dependent functional form. The LINEAR_SPLINE configuration specifies the day of year, larval value, and larval capacity scaling number. The model linearly interpolates the values to estimate the habitat availability for each vector species without requiring climatological data. The following example shows how to specify the larval habitat using one of the climatological habitat types. { \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 1000000 }} The following example shows how to specify larval habitat using LINEAR_SPLINE. { \"Larval_Habitat_Types\" : { \"LINEAR_SPLINE\" : { \"Capacity_Distribution_Per_Year\" : { \"Times\" : [ 0.0 , 60.833 , 121.667 , 182.5 , 243.333 , 204.167 ], \"Values\" : [ 0.0 , 0.0 , 0.2 , 1.0 , 0.5 , 0.0 ] }, \"Max_Larval_Capacity\" : 10000000000 } } } Vector_Species_Params JSON object NA NA NA For each of the species specified in Vector_Species_Name, there is a set of required species-parameters. The code example, using the arabiensis parameter value, configures the vector Anopheles arbiensis . The parameter name values for the other species are: farauti, fenestus and bambiae. { \"Vector_Species_Params\" : { \"arabiensis\" : { \"Acquire_Modifier\" : 1 , \"Adult_Life_Expectancy\" : 10 , \"Anthropophily\" : 0.95 , \"Aquatic_Arrhenius_1\" : 84200000000 , \"Aquatic_Arrhenius_2\" : 8328 , \"Aquatic_Mortality_Rate\" : 0.1 , \"Days_Between_Feeds\" : 3 , \"Egg_Batch_Size\" : 100 , \"Immature_Duration\" : 4 , \"Indoor_Feeding_Fraction\" : 0.5 , \"Infected_Arrhenius_1\" : 117000000000 , \"Infected_Arrhenius_2\" : 8336 , \"Infected_Egg_Batch_Factor\" : 0.8 , \"Infectious_Human_Feed_Mortality_Factor\" : 1.5 , \"Larval_Habitat_Types\" : { \"TEMPORARY_RAINFALL\" : 1e10 , \"WATER_VEGETATION\" : 1e6 }, \"Transmission_Rate\" : 0.5 } } }", "tags": "", "title": "Configuration parameters \u2014 Training Tutorials  documentation"}, {"url": "./parameter-demographics.html", "text": "Demographics file structure and parameters \u00b6 The parameters described in this reference section can be added to the JavaScript Object Notation (JSON) formatted demographics file to determine the demographics of the population within each geographic node in a simulation. For example, the number of individuals and the distribution for age, gender, immunity, risk, and mortality. This is in contrast to the configuration parameters related to demographics that are simulation-wide and generally control whether certain events, such as births or deaths, are enabled in a simulation. At least one demographics file is required for every simulation unless you set the parameter Enable_Demographics_Builtin to 0 (zero) in the configuration file . This is generally only used for testing and validating code pathways by using a standard testing sandbox instead of actual demographics information. If Migration_Model is set to \u201cFIXED_RATE_MIGRATION\u201d, built-in demographics will use a form of local migration where individuals migrate only to adjacent nodes. Demographics files are usually the only input data file you will modify. Demographics files are named using the name of the region, appended with \u201c_demographics.json\u201d. Additionally, you can use more than one demographics file, with one serving as the base layer and the one or more others acting as overlays that override the values in the base layer. Contents Demographic file structure Metadata Defaults Nodes Parameters NodeAttributes IndividualAttributes IndividualProperties Example Demographic file structure \u00b6 Demographics files are organized into three sections: Metadata Information such as data source, created date or region name. Metadata is used to provide provenance information. Defaults Default parameter values applied to all nodes. Nodes Parameter values specific to individual geographic nodes, which override the default values. There is one entry for each node in the simulation. The following example shows the skeletal format of a demographics file. { \"Metadata\" : { \"DateCreated\" : \"dateTime\" , \"Tool\" : \"scriptUsedToGenerate\" , \"Author\" : \"author\" , \"IdReference\" : \"ID reference\" , \"NodeCount\" : 2 }, \"Defaults\" : { \"NodeAttributes\" : {}, \"IndividualAttributes\" : {}, \"IndividualProperties\" : {} }, \"Nodes\" : [{ \"NodeID\" : 1 , \"NodeAttributes\" : {}, \"IndividualAttributes\" : {}, \"IndividualProperties\" : {} }, { \"NodeID\" : 2 , \"NodeAttributes\" : {}, \"IndividualAttributes\" : {}, \"IndividualProperties\" : {} }] } Metadata \u00b6 Most of the items in the Metadata section informational only and are not used by EMOD. This includes information such as author, date created, and tool used to create the file. However, the following two values are used by EMOD: NodeCount The number of nodes to expect in the demographics file. IdReference The unique string that indicates the method used for generating the NodeID , the identifier used for each node in the simulation. Defaults \u00b6 Parameter values in the Defaults section is applied to all nodes. However, any node-level information provided in the Nodes section will take precedence for that node. The Defaults section is optional and can be completely omitted, as is often the case of single- node simulations. The Defaults section can contain any of the parameters listed below. Nodes \u00b6 Parameter values in the Nodes section is applied to specific nodes. The intent of the Nodes section is to have only the information that is unique to each node, such as identifier and location (longitude and latitude). The section contains an array with each element in the array representing a node that is identified by its NodeID . The NodeID is a unique integer value. If a parameter appears in both the Defaults and Nodes sections, the value in the Nodes section will take precedence. The Nodes section can contain any of the parameters listed below, except for the Society parameters that can only be applied in the Defaults section. Parameters \u00b6 The demographic parameters are divided into the following broad categories. The parameters will be contained in a nested JSON object of the same name. Warning Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false. NodeAttributes \u00b6 NodeAttributes are a JSON object that contains parameters that add or modify information regarding the location, migration, habitat, and population of a simulation. Some NodeAttributes depend on values set in the configuration parameters. Parameter Data type Minimum Maximum Default Description Example AbovePoverty float 0 1 1 The proportion of the population that is living above the poverty line. { \"NodeAttributes\" : { \"AbovePoverty\" : 0.5 } } Airport bool 0 1 0 Set to 1 to indicate that the node is connected by air migration. { \"NodeAttributes\" : { \"Airport\" : 0 } } Altitude float -3.40282e+038 3.40E+38 0 Required. Only used when Climate_Model is set to \u201cCLIMATE_KOPPEN\u201d. The unit of measurement is in meters. { \"NodeAttributes\" : { \"Altitude\" : 250 } } BirthRate float 0 1 0.25 The birth rate. In the configuration file, Enable_Birth must be set to 1 and Birth_Rate_Dependence will affect how this rate is used. { \"NodeAttributes\" : { \"BirthRate\" : 0.0001 } } InitialPopulation integer 0 2.15E+09 \u00a0 The number of people that will be populated into the node at the beginning of the simulation. You can scale this number using Base_Population_Scale_Factor in the configuration file. { \"NodeAttributes\" : { \"InitialPopulation\" : 1000 } } Latitude float -3.40282e+038 3.40E+38 \u00a0 Latitude of the node in decimal degrees. This can be used for several things including determining infectiousness by latitude and defining the size of grid cells. { \"NodeAttributes\" : { \"Latitude\" : 12.4 } } Longitude float -3.40282e+038 3.40E+38 \u00a0 Longitude of the node in decimal degrees. This can be used for several things including defining the size of grid cells. { \"NodeAttributes\" : { \"Longitude\" : 9.1 } } Region bool 0 1 0 Set to 1 to indicate that the node is connected by regional migration. { \"NodeAttributes\" : { \"Region\" : 1 } } Seaport bool 0 1 0 Set to 1 to indicate that the node is connected by sea migration. { \"NodeAttributes\" : { \"Seaport\" : 1 } } Urban bool 0 1 0 Set to 1 to enable urban settings. In the configuration file, Enable_Demographics_Other must be set to 1 and Birth_Rate_Dependence must be set to \u201cINDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE\u201d. { \"NodeAttributes\" : { \"Urban\" : 1 } } Zoonosis float 0 1 0 The daily rate of zoonotic infection per individual. If Animal_Reservoir_Type is set to \u201cZOONOSIS_FROM_DEMOGRAPHICS\u201d, the value for Zoonosis overrides the value set for Zoonosis_Rate in the configuration file. { \"NodeAttributes\" : { \"Zoonosis\" : 0 } } Parameter Data type Minimum Maximum Default Description Example AbovePoverty float 0 1 1 The proportion of the population that is living above the poverty line. { \"AbovePoverty\" : 0.5 } Airport bool 0 1 0 Set to 1 to indicate that the node is connected by air migration. { \"Airport\" : 0 } Altitude float -3.40282e+038 3.40E+38 0 Required. Only used when Climate_Model is set to \u201cCLIMATE_KOPPEN\u201d. The unit of measurement is in meters. { \"Altitude\" : 250 } BirthRate float 0 1 0.25 The birth rate. In the configuration file, Enable_Birth must be set to 1 and Birth_Rate_Dependence will affect how this rate is used. { \"BirthRate\" : 0.0001 } InitialPopulation integer 0 2.15E+09 \u00a0 The number of people that will be populated into the node at the beginning of the simulation. You can scale this number using Base_Population_Scale_Factor in the configuration file. { \"InitialPopulation\" : 1000 } LarvalHabitatMultiplier float or nested JSON object \u00a0 \u00a0 \u00a0 The value by which to scale the larval habitat availability specified in the configuration file with Larval_Habitat_Types across all habitat types, for specific habitat types, or for specific mosquito species within each habitat type. The following example scales the larval habitat equally across all habitat types and mosquito species. { \"LarvalHabitatMultiplier\" : 2.0 } The following example scales the larval habitat only in the temporary rainfall habitat for all mosquito species. { \"LarvalHabitatMultiplier\" : { \"TEMPORARY_RAINFALL\" : 2.0 } } The following example scales the larval habitat independently for A. gambiae in the temporary rainfall habitat and for A. arabiensis in the brackish swamp habitat. { \"LarvalHabitatMultiplier\" : { \"TEMPORARY_RAINFALL\" : { \"gambiae\" : 2.0 }, \"BRACKISH_SWAMP\" : { \"arabiensis\" : 2.5 } } } Latitude float -3.40282e+038 3.40E+38 \u00a0 Latitude of the node in decimal degrees. This can be used for several things including determining infectiousness by latitude and defining the size of grid cells. { \"Latitude\" : 12.4 } Longitude float -3.40282e+038 3.40E+38 \u00a0 Longitude of the node in decimal degrees. This can be used for several things including defining the size of grid cells. { \"Longitude\" : 9.1 } Region bool 0 1 0 Set to 1 to indicate that the node is connected by regional migration. { \"Region\" : 1 } Seaport bool 0 1 0 Set to 1 to indicate that the node is connected by sea migration. { \"Seaport\" : 1 } Urban bool 0 1 0 Set to 1 to enable urban settings. In the configuration file, Enable_Demographics_Other must be set to 1 and Birth_Rate_Dependence must be set to \u201cINDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE\u201d. { \"Urban\" : 1 } Zoonosis float 0 1 0 The daily rate of zoonotic infection per individual. If Animal_Reservoir_Type is set to \u201cZOONOSIS_FROM_DEMOGRAPHICS\u201d, the value for Zoonosis overrides the value set for Zoonosis_Rate in the configuration file. { \"Zoonosis\" : 0 } IndividualAttributes \u00b6 IndividualAttributes are a JSON object that contains parameters that add or modify the distribution of attributes across individuals in a simulation. For example, the age or immunity distribution. An initial value for an individual is a randomly selected value from a distribution. For example, if you use a uniform distribution to initialize age, the initial ages of individuals in the simulation will be evenly distributed between some minimum and maximum value. There are two types of distributions: simple and complex. Simple distributions are defined by three parameters where one is the distribution type and the other two are used to further define the distribution. You can use complex distributions to define a distribution that does not fit some standard. For example, a complex distribution is useful when trying to represent real-world data. The table below shows the simple distributions and their associated parameters. Distribution Demographic parameters Related configuration parameters Age AgeDistributionFlag AgeDistribution1 AgeDistribution2 Age_Initialization_Distribution_Type must be set to \u201cDISTRIBUTION_SIMPLE\u201d. Prevalence PrevalenceDistributionFlag PrevalenceDistribution1 PrevalenceDistribution2 Enable_Demographics_Other must be set to 1. Immunity ImmunityDistributionFlag ImmunityDistribution1 ImmunityDistribution2 \u00a0 Risk RiskDistributionFlag RiskDistribution1 RiskDistribution2 Enable_Demographics_Other must be set to 1. Migration heterogeneity MigrationHeterogeneityDistributionFlag MigrationHeterogeneityDistribution1 MigrationHeterogeneityDistribution2 Enable_Migration_Heterogeneity must be set to 1. The table below shows the available probability distribution types to use with simple distributions and how to set the other two demographic parameters to initialize the distribution. Distribution type Flag value Distribution1 value Distribution2 value Constant 0 Value N/A (set to 0) Uniform 1 Minimum Maximum Gaussian 2 Mean Standard deviation Exponential 3 Mean N/A (set to 0) Poisson 4 Mean N/A (set to 0) Log normal 5 Mean Log (standard deviation) Bimodal 6 Proportion of time to return the value specified by Distribution2. Value must be between 0 and 1. A positive value to be returned. If the value is not positive, a value of 1 is returned. Weibull 7 Scale parameter Shape parameter EMOD also supports the following complex distributions: AgeDistribution FertilityDistribution MortalityDistribution MortalityDistributionMale MortalityDistributionFemale HIVCoinfectionDistribution HIVMortalityDistribution MSP_mean_antibody_distribution nonspec_mean_antibody_distribution PfEMP1_mean_antibody_distribution MSP_variance_antibody_distribution nonspec_variance_antibody_distribution Malaria PfEMP1_variance_antibody_distribution IndividualProperties \u00b6 IndividualProperties are a JSON array that contains parameters that add properties to individuals in a simulation as a means of setting up groups. For example, you can define values for accessibility, age, geography, risk, and other properties and assign individuals to different groups based on those property values. Parameter Data type Description Example Property string The property type for which you will assign values to create groups. Accepted values are: Accessibility Geographic Place Risk QualityofCare HasActiveTB Age_Bin { \"IndividualProperties\" : [{ \"Property\" : \"Risk\" }] } Values array An array of the user-defined values that can be assigned to individuals for this property. You can have up to 125 values for the Geographic property type and up to 5 values for all other types. For \u201cAge_Bin\u201d property types, omit and use Age_Bin_In_Years instead. { \"IndividualProperties\" : [{ \"Values\" : [ \"Low\" , \"Medium\" , \"High\" ] }] } Initial_Distribution array An array of numbers that define the proportion of individuals to assign to each group at the beginning of the simulation. Their total must equal 1 and the number of members in this array must match the number of members in Values . For \u201cAge_Bin\u201d property types, omit this parameter as the demographics file controls the age distribution. { \"IndividualProperties\" : [{ \"Initial_Distribution\" : [ 0.2 , 0.4 , 0.4 ] }] } Age_Bin_In_Years array An array of integers that represents the ages, in years, at which to demarcate the age groups. Used only with the \u201cAge_Bin\u201d property type. The first number must be 0, the last must be -1, and they must be listed in ascending order. The following example creates three age groups: 0 to 5, older than 5 to 13, and older than 13 to the maximum age. { \"IndividualProperties\" : [{ \"Age_Bin_In_Years\" : [ 0 , 5 , 13 , -1 ] }] } Transitions array An array contains multiple JSON objects that each define how an individual transitions from one property group to another. See the following table for information about the parameters to include in the object. For \u201cAge_Bin\u201d property types, set to an empty array, as individuals will transition to the next age bin based on the passing of time. { \"IndividualProperties\" : [{ \"Transitions\" : [] }] } TransmissionMatrix array An object that contains Route and Matrix parameters that define how to scale the base infectivity from individuals in one group to individuals in another. { \"IndividualProperties\" : [{ \"TransmissionMatrix\" : { \"Route\" : \"Contact\" , \"Matrix\" : [ [ 10 , 0.1 ], [ 0.1 , 1 ] ] } }] } The following table contains the parameters that are available to use in the Transitions array. Parameter Data type Description Example From string Which group an individual will transition from. { \"Transitions\" : [{ \"From\" : \"Low\" }] } To string Which group an individual will transition to. { \"Transitions\" : [{ \"To\" : \"High\" }] } Type string The type of condition that starts transitioning individuals. Set to \u201cAt_Age\u201d or \u201cAt_Timestep\u201d. The other parameters you must set depend on the condition type set here. { \"Transitions\" : [{ \"Type\" : \"At_Age\" }] } Age_In_Years_Restriction nested JSON object The age when an individual is eligible to transition. Min is optional and Max is required. Do not use this with \u201cAt_Age\u201d types because it will conflict with other age parameters in Age_In_Years . This is required for \u201cAt_Timestep\u201d types, though it can be empty for no age restriction. { \"Transitions\" : [{ \"Type\" : \"At_Timestep\" , \"Age_In_Years_Restriction\" : { \"Min\" : 5 , \"Max\" : 40 } }] } Timestep_Restriction nested JSON object The time step when transitioning starts and stops. Required for both \u201cAt_Age\u201d and \u201cAt_Timestep\u201d types. Start is required and End is optional. { \"Transitions\" : [{ \"Timestep_Restriction\" : { \"Start\" : 20 } }] } Coverage float A value between 0 and 1 that indicates the proportion of the population that EMOD will attempt to transition. The calculation used with Probability_Per_Timeset only uses the proportion of the population specified in Cover to calculate the probability of an individual transitioning at a given time step. Coverage has no effect when the value is set to 1. Required for both \u201cAt_Age\u201d and \u201cAt_Timestep\u201d types. { \"Transitions\" : [{ \"Coverage\" : 0.5 }] } Probability_Per_Timestep float A value between 0 and 1 that is used to calculate the probability of an individual transitioning at the current time step given the number of individuals remaining in the group. Required for both \u201cAt_Age\u201d and \u201cAt_Timestep\u201d types. { \"Transitions\" : [{ \"Probability_Per_Timestep\" : 0.25 }] } Age_In_Years integer The age at which individuals are eligible to transition. Do not use for \u201cAt_Timestep\u201d types. Required for \u201cAt_Age\u201d types. { \"Transitions\" : [{ \"Type\" : \"At_Age\" , \"Age_In_Years\" : 10 }] } Timesteps_Until_Reversion integer The number of time steps after the start of transitioning when individuals revert back to their original group. The start of transitioning is specified by the Start parameter in Timestep_Restriction . { \"Transitions\" : [{ \"Timestep_Restriction\" : { \"Start\" : 20 }, \"Timesteps_Until_Reversion\" : 100 }] } Example \u00b6 The example below shows a complete multi-node demographics file that uses a gridded IdReference type. { \"Metadata\" : { \"DateCreated\" : \"03/03/2016\" , \"Tool\" : \"DemographicsWorker\" , \"Author\" : \"Institute for Disease Modeling\" , \"NodeType\" : \"Grid\" , \"ProjectName\" : \"IDM-Madagascar\" , \"RegionName\" : \"Madagascar\" , \"IdReference\" : \"Gridded world grump2.5arcmin\" , \"NodeCount\" : 15 , \"DataProvenance\" : \"Generated by internally-developed demographic tools\" , \"Resolution\" : 150 , \"UpperLatitude\" : -10.6666666666667 , \"LeftLongitude\" : 41.6666666666667 , \"BottomLatitude\" : -27.3333333333333 , \"RightLongitude\" : 51.3333333333333 }, \"Defaults\" : { \"NodeAttributes\" : { \"Altitude\" : 0 , \"Airport\" : 0 , \"Region\" : 1 , \"Seaport\" : 0 , \"BirthRate\" : 0.000102 , \"AbovePoverty\" : 0 }, \"IndividualAttributes\" : { \"AgeDistributionFlag\" : 3 , \"AgeDistribution1\" : 0.00015 , \"AgeDistribution2\" : 0 , \"PrevalenceDistributionFlag\" : 1 , \"PrevalenceDistribution1\" : 0.1 , \"PrevalenceDistribution2\" : 0.2 , \"ImmunityDistributionFlag\" : 0 , \"ImmunityDistribution1\" : 1 , \"ImmunityDistribution2\" : 0 , \"RiskDistributionFlag\" : 0 , \"RiskDistribution1\" : 1 , \"RiskDistribution2\" : 0 , \"MigrationHeterogeneityDistributionFlag\" : 0 , \"MigrationHeterogeneityDistribution1\" : 1 , \"MigrationHeterogeneityDistribution2\" : 0 , \"MortalityDistribution\" : { \"NumDistributionAxes\" : 2 , \"AxisNames\" : [ \"gender\" , \"age\" ], \"AxisUnits\" : [ \"male=0,female=1\" , \"years\" ], \"AxisScaleFactors\" : [ 1 , 365 ], \"NumPopulationGroups\" : [ 2 , 3 ], \"PopulationGroups\" : [ [ 0 , 1 ], [ 0 , 100 , 2000 ] ], \"ResultUnits\" : \"annual deaths per 1000 individuals\" , \"ResultScaleFactor\" : 0.00000273972602739726027397260273973 , \"ResultValues\" : [ [ 0 , 20.0000035 , 400.00007 ], [ 0 , 20.0000035 , 400.00007 ] ] } } }, \"Nodes\" : [{ \"NodeID\" : 358876977 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.1875 , \"Altitude\" : 34 , \"Area\" : 25 , \"InitialPopulation\" : 2088 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358876978 , \"NodeAttributes\" : { \"Latitude\" : -13.2708330154419 , \"Longitude\" : 48.1875 , \"Altitude\" : 19 , \"Area\" : 25 , \"InitialPopulation\" : 1682 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358942512 , \"NodeAttributes\" : { \"Latitude\" : -13.3541669845581 , \"Longitude\" : 48.2291679382324 , \"Altitude\" : 127 , \"Area\" : 25 , \"InitialPopulation\" : 5160 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358942513 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.2291679382324 , \"Altitude\" : 128 , \"Area\" : 25 , \"InitialPopulation\" : 2860 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 358942514 , \"NodeAttributes\" : { \"Latitude\" : -13.2708330154419 , \"Longitude\" : 48.2291679382324 , \"Altitude\" : 26 , \"Area\" : 25 , \"InitialPopulation\" : 790 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008047 , \"NodeAttributes\" : { \"Latitude\" : -13.3958330154419 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 48 , \"Area\" : 25 , \"InitialPopulation\" : 6436 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008048 , \"NodeAttributes\" : { \"Latitude\" : -13.3541669845581 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 111 , \"Area\" : 25 , \"InitialPopulation\" : 4983 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008049 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 124 , \"Area\" : 25 , \"InitialPopulation\" : 1245 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008050 , \"NodeAttributes\" : { \"Latitude\" : -13.2708330154419 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 57 , \"Area\" : 25 , \"InitialPopulation\" : 1387 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359008051 , \"NodeAttributes\" : { \"Latitude\" : -13.2291669845581 , \"Longitude\" : 48.2708320617676 , \"Altitude\" : 66 , \"Area\" : 25 , \"InitialPopulation\" : 1516 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073583 , \"NodeAttributes\" : { \"Latitude\" : -13.3958330154419 , \"Longitude\" : 48.3125 , \"Altitude\" : 132 , \"Area\" : 25 , \"InitialPopulation\" : 5957 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073584 , \"NodeAttributes\" : { \"Latitude\" : -13.3541669845581 , \"Longitude\" : 48.3125 , \"Altitude\" : 44 , \"Area\" : 25 , \"InitialPopulation\" : 643 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073585 , \"NodeAttributes\" : { \"Latitude\" : -13.3125 , \"Longitude\" : 48.3125 , \"Altitude\" : 55 , \"Area\" : 25 , \"InitialPopulation\" : 1171 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359073587 , \"NodeAttributes\" : { \"Latitude\" : -13.2291669845581 , \"Longitude\" : 48.3125 , \"Altitude\" : 36 , \"Area\" : 25 , \"InitialPopulation\" : 378 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }, { \"NodeID\" : 359139117 , \"NodeAttributes\" : { \"Latitude\" : -13.4791669845581 , \"Longitude\" : 48.3541679382324 , \"Altitude\" : 282 , \"Area\" : 25 , \"InitialPopulation\" : 674 , \"BirthRate\" : 0.000113117808219178 , \"AbovePoverty\" : 0 , \"Urban\" : 0 } }] }", "tags": "", "title": "Demographics file structure and parameters \u2014 Training Tutorials  documentation"}, {"url": "./parameter-overview.html", "text": "EMOD parameter reference \u00b6 The configurable parameters described in this reference section determine the behavior of a particular simulation and interventions. The parameter descriptions are intended to guide you in correctly configuring simulations to match your disease scenario of interest. These parameters include three major groups: parameters for the demographics file , parameters for the configuration file , and parameters for the campaign file . JSON formatting overview \u00b6 All of these parameters are contained in JavaScript Object Notation (JSON) formatted files. JSON is a data format that is human-readable and easy for software to read and generate. It can be created and edited using a simple text editor. JSON has two basic structures: a collection of key-value pairs or an ordered list of values (an array). These structures are within a JSON object, which is enclosed in a set of braces. You will notice that each of these files begins with a left brace ({) and ends with a right brace (}). A value can be a string, number, Boolean, array, or an object. The campaign and data input files often use nested objects and arrays. See www.json.org for more information on JSON. A few important details to call out when creating JSON files: Add a comma after every key-value pair or array except for the last key-value pair in an array or object. The keys (parameters) are case-sensitive. For example, \u201cNodeID\u201d is not the same as \u201cNodeId\u201d. Decimals require a 0 (zero) before the decimal point. EMOD does not support Booleans (\u201cTrue\u201d, \u201cFalse\u201d). Instead, EMOD use the integer 1 for \u201cTrue\u201d and 0 for \u201cFalse\u201d. Every opening brace or bracket ({ or [) requires a corresponding closing brace or bracket (} or ]). The following is an example of a JSON formatted file. { \"A_Complex_Key\" : { \"An_Array_with_a_Nested_Object_Value\" : [ { \"A_Simple_Key\" : \"Value\" , \"A_Simple_Array\" : [ \"Value1\" , \"Value2\" ], \"An_Array_with_Number_Values\" : [ 0.1 , 0.2 ], \"A_Nested_Object\" : { \"Another_Simple_Key\" : \"Value\" , \"Nested_Arrays\" : [ [ 10 , 0.1 ], [ 0.1 , 1 ] ] } } ] } } Demographics file structure and parameters Configuration parameters Campaign parameters", "tags": "", "title": "EMOD parameter reference \u2014 Training Tutorials  documentation"}, {"url": "./search-local.html", "text": "Search results \u00b6", "tags": "", "title": "Search results \u2014 Training Tutorials  documentation"}, {"url": "./search.html", "text": "Please activate JavaScript to enable the search functionality.", "tags": "", "title": "Search \u2014 Training Tutorials  documentation"}, {"url": "./troubleshooting.html", "text": "Troubleshooting \u00b6 If you encounter any of the following errors or problems when attempting to run simulations with the Quick Start, see the information below to resolve the issue. Setenv.cmd is not recognized \u00b6 When you double-click the EMOD Environment shortcut to launch the command window, you may see the following error with user-specific information displayed: ' \\\\ file.myRegion.myCompany\\user2$\\myUserName\\EMOD-QuickStart-master' CMD . EXE was started with the above path as the current directory . UNC paths are not supported . Defaulting to Windows directory . 'setenv.cmd' is not recognized as an internal or external command , operable program or batch file . This error indicates that the Quick Start was saved to a network directory. Move the Quick Start to a locally stored directory on your computer. Code must be updated for 64-bit systems \u00b6 If you have a 64-bit version of Microsoft Excel installed, when you try to run the Excel Front End, you will see the following error: \u201cThe code in this project must be updated for use on 64-bit systems. Please review and update Declare Statements and then mark them with PtrSafe attribute.\u201d Follow the steps below to work around this issue: Open EMOD_Macros.xlsm in Excel. On the View menu, in the Macros section, select View Macros . In the Macro window, select the TestingHelloWorld macro, and then click Edit . In the Microsoft Visual Basic for Applications window that opens, locate the lines that contain \u201cPrivate Declare Function\u201d. Add the PtrSafe attribute by modifying these lines to \u201cPrivate Declare PtrSafe Function\u201d. Cannot open config.xlsm file in the Excel Front End \u00b6 Excel will not open more than one file with the same name. In all of the training tutorials, the configuration files for the Excel Front End are named config.xlsm. If Excel will not open your config.xlsm, make sure that you don\u2019t have a configuration file already open.", "tags": "", "title": "Troubleshooting \u2014 Training Tutorials  documentation"}, {"url": "./using-quickstart.html", "text": "How to run Quick Start simulations \u00b6 This topic describes how to run simulations to model the scenarios provided with the Quick Start. Each of the tutorials in the Quick Start documentation describes in more detail how to modify various parameters to model disease and the rationale and mathematics behind the EMOD model. This topic describes in a broader sense how to run simulations. Outside of the Quick Start, simulations are generally run using command-line options and the JSON files that control the configuration and campaign are modified directly. While there\u2019s nothing preventing you from running simulations using the scenarios here at the command line, this process is not documented here because it is more complicated. If you would like to learn more about running simulations this way so you can leverage the full power of EMOD, see any of the other documentation sets. In general, we assume you will be running simulations using the Excel Front End because this interface allows you to easily view and modify the parameters associated with each of the scenarios and greater control in plotting the output. However, you also have the option of running scenarios using the provided Windows command and batch scripts. Run a simulation using the Excel Front End \u00b6 Open Windows Explorer. Navigate to the Quick Start on the virtual Z drive (or other virtual drive you set up during installation). If you get an error, re-run the environment setup script. See Install the Quick Start . In the Scenarios directory, open the directory of the scenario you want to simulate. The tutorials will guide you through the scenarios in greater detail. Double-click config.xlsm. If a warning appears, click Update to dismiss it. If you\u2019ve made any changes, save the workbook. The Commissioning sheet contains parameters that control how and where the simulation runs. You generally will not need to modify this sheet. The other sheets contain configuration and campaign parameters. Follow the instructions in the tutorial to modify these parameters as necessary. Select the CONTROL sheet tab. Click EXECUTE . If a security notice appears, click Enable Macros to dismiss it. This will open a Command Prompt window for simulation output and logging. Ignore the EMOD_Macros.xslm file that opens. In the Command Prompt window, verify that you see the following output: \u201cController executed successfully.\u201d Close the Command Prompt window to open the InsetChart workbook. This workbook contains the simulation-wide averages of prevalence , incidence , and other common metrics at each time step . If desired, use Save as to save the results of the simulation with a different name. By default, files will be saved in the testing directory. To plot the output, press Ctrl and select the columns of interest, then on the Insert tab, select a 2-D line chart. For example, press Ctrl and select columns B, J, and L to plot the recovered, susceptible, and infectious populations at each time step. Of course, you can use any of the Microsoft Excel plotting features to plot the output data in different ways. Run a simulation using scripts \u00b6 Open Windows Explorer. Navigate to the Quick Start on the virtual Z drive (or other virtual drive you set up during installation). If you get an error, re-run the environment setup script. See Install the Quick Start . In the Scenarios directory, open the directory of the scenario you want to simulate. If you want to make changes to the campaign or configuration files, open the JSON file in a text editor, make your changes, and save. The tutorials will guide you through the scenarios in greater detail. Double-click runEmod.cmd. This will open a Command Prompt window for simulation output and logging. In the Command Prompt window, verify that you see the following output: \u201cController executed successfully.\u201d You may then close the window. By default, output files will be saved in the output directory. To plot the output, double-click plotInsetChart.bat or plotAllCharts.bat. These batch files use Python to create plots of campaign cost, exposed and recovered population, new infections, and more. Note that all batch files assume that Python is installed at C:Python27. If Python is installed in a different location, use a plain text editor to modify the path in the batch files.", "tags": "", "title": "How to run Quick Start simulations \u2014 Training Tutorials  documentation"}]};
