

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Configuration parameters &mdash; Malaria Model  documentation</title>
  

  
  
    <link rel="shortcut icon" href="_static/favicon.ico"/>
  

  

  
  
    
      <link rel="search" type="application/opensearchdescription+xml" title="Search within Malaria Model  documentation" href="_static/opensearch.xml"/>
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/tipuesearch/tipuesearch.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Malaria Model  documentation" href="index.html"/>
        <link rel="up" title="EMOD parameter reference" href="parameter-overview.html"/>
        <link rel="next" title="Campaign parameters" href="parameter-campaign.html"/>
        <link rel="prev" title="Demographics parameters" href="parameter-demographics.html"/>

<link href="https://fonts.googleapis.com/css?family=Lato|Montserrat|Open+Sans" rel="stylesheet">



  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Malaria Model
          

          
            
            <img src="_static/IDM_white.png" class="logo" />
          
          </a>

          
            
            
              <div class="version">
                2.13
              </div>
            
          

          <div role="search">
  <form id='rtd-search-form' class='wy-form' action="search-local.html" method="get">
    <input type="text" name="q" id="tipue_search_input" autocomplete="off"
           placeholder="Search docs" required>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Using the model</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="whats-new.html">What’s new</a></li>
<li class="toctree-l1"><a class="reference internal" href="install-overview.html">EMOD installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="intro-overview.html">Introduction to the software</a></li>
<li class="toctree-l1"><a class="reference internal" href="intro-modeling.html">Introduction to disease modeling</a></li>
<li class="toctree-l1"><a class="reference internal" href="howto-overview.html">How to use the software</a></li>
<li class="toctree-l1"><a class="reference internal" href="malaria-overview.html">Malaria disease overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="malaria-model-overview.html">Malaria model overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="malaria-IDM.html">Malaria research at IDM</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="parameter-overview.html">EMOD parameter reference</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="parameter-demographics.html">Demographics parameters</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Configuration parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="parameter-campaign.html">Campaign parameters</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="file-reference-overview.html">Input and output file structure</a></li>
<li class="toctree-l1"><a class="reference internal" href="troubleshooting.html">Troubleshooting EMOD simulations</a></li>
<li class="toctree-l1"><a class="reference internal" href="glossary.html">Glossary</a></li>
</ul>
<p class="caption"><span class="caption-text">Advancing the model</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="dev-install-overview.html">EMOD source code installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="dev-architecture-overview.html">EMOD architecture</a></li>
<li class="toctree-l1"><a class="reference internal" href="dev-debugging-overview.html">Debugging and testing</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Malaria Model</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          





<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
  <li><a href="../index.html">IDM docs</a> &raquo;</li>
    <li><a href="index.html">Malaria</a> &raquo;</li>
      
          <li><a href="parameter-overview.html">EMOD parameter reference</a> &raquo;</li>
      
    <li>Configuration parameters</li>
      <li class="wy-breadcrumbs-aside">
        
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
<!-- From _templates/page.html -->

  <div class="section" id="configuration-parameters">
<h1>Configuration parameters<a class="headerlink" href="#configuration-parameters" title="Permalink to this headline">¶</a></h1>
<p>The parameters described in this reference section can be added to the <a class="reference internal" href="glossary.html#term-javascript-object-notation-json"><span class="xref std std-term">JavaScript Object
Notation (JSON)</span></a> formatted configuration file to determine the core behavior of a simulation
including the computing environment, functionality to enable, additional files to use, and
characteristics of the disease being modeled. This file contains mostly a flat list of JSON key-
value pairs.</p>
<p>The tables below contain only parameters available when using the malaria <a class="reference internal" href="glossary.html#term-simulation-type"><span class="xref std std-term">simulation type</span></a>.
Some parameters may appear in multiple categories.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Parameters are case-sensitive. For Boolean parameters, set to 1 for true or 0 for false.</p>
</div>
<button class="collapse-table-button btn btn-info">Collapse all parameter tables</button><div class="contents local topic" id="contents">
<p class="topic-title first">Contents</p>
<ul class="simple">
<li><a class="reference internal" href="#drugs-and-treatments" id="id1">Drugs and treatments</a></li>
<li><a class="reference internal" href="#enable-or-disable-features" id="id2">Enable or disable features</a></li>
<li><a class="reference internal" href="#general-disease" id="id3">General disease</a></li>
<li><a class="reference internal" href="#geography-and-environment" id="id4">Geography and environment</a></li>
<li><a class="reference internal" href="#immunity" id="id5">Immunity</a></li>
<li><a class="reference internal" href="#incubation" id="id6">Incubation</a></li>
<li><a class="reference internal" href="#infectivity-and-transmission" id="id7">Infectivity and transmission</a></li>
<li><a class="reference internal" href="#input-data-files" id="id8">Input data files</a></li>
<li><a class="reference internal" href="#larval-habitat" id="id9">Larval habitat</a></li>
<li><a class="reference internal" href="#migration" id="id10">Migration</a></li>
<li><a class="reference internal" href="#mortality-and-survival" id="id11">Mortality and survival</a></li>
<li><a class="reference internal" href="#output-settings" id="id12">Output settings</a></li>
<li><a class="reference internal" href="#parasite-dynamics" id="id13">Parasite dynamics</a></li>
<li><a class="reference internal" href="#population-dynamics" id="id14">Population dynamics</a></li>
<li><a class="reference internal" href="#sampling" id="id15">Sampling</a></li>
<li><a class="reference internal" href="#scalars-and-multipliers" id="id16">Scalars and multipliers</a></li>
<li><a class="reference internal" href="#simulation-setup" id="id17">Simulation setup</a></li>
<li><a class="reference internal" href="#symptoms-and-diagnosis" id="id18">Symptoms and diagnosis</a></li>
<li><a class="reference internal" href="#vector-control" id="id19">Vector control</a></li>
<li><a class="reference internal" href="#vector-life-cycle" id="id20">Vector life cycle</a></li>
</ul>
</div>
<div class="section" id="drugs-and-treatments">
<span id="config-drugs"></span><h2><a class="toc-backref" href="#id1">Drugs and treatments</a><a class="headerlink" href="#drugs-and-treatments" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the efficacy of drugs and other treatments.</p>
<p>For more information on the drugs used to treat malaria, see <a class="reference internal" href="malaria-antimalarial-drugs.html"><span class="doc">Antimalarial drugs</span></a>.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Bodyweight_Exponent</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>The effect of bodyweight on maximum drug concentration in an individual patient. <strong>Drug_Cmax</strong> is divided by patient bodyweight raised to the power of <strong>Bodyweight_Exponent</strong> to account for the influence of body size on volume of distribution.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether_Lumefantrine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Bodyweight_Exponent&quot;</span><span class="p">:</span> <span class="mi">2</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drug_Cmax</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1000</td>
<td><p class="first">The maximum drug concentration used.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This parameter and the <strong>Drug_PKPD_C50</strong> parameter must use the same units.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether_Lumefantrine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Cmax&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_PKPD_C50&quot;</span><span class="p">:</span> <span class="mi">100</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Drug_Decay_T1</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1</td>
<td>The primary drug decay rate, in days.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemisinin&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Decay_T1&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T2&quot;</span><span class="p">:</span> <span class="mi">1</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drug_Decay_T2</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1</td>
<td>The secondary drug decay rate, in days.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemisinin&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Decay_T1&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T2&quot;</span><span class="p">:</span> <span class="mi">1</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Drug_Dose_Interval</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>1</td>
<td>The interval between doses of drugs, in days.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemisinin&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Dose_Interval&quot;</span><span class="p">:</span> <span class="mi">1</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drug_Fulltreatment_Doses</td>
<td>float</td>
<td>1</td>
<td>100000</td>
<td>3</td>
<td>The number of doses for a full treatment of a drug.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Chloroquine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Fulltreatment_Doses&quot;</span><span class="p">:</span> <span class="mi">3</span>
        <span class="p">}</span>
    <span class="p">}</span>  
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Drug_Gametocyte02_Killrate</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>The log reduction per day in early-stage gametocytes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Chloroquine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span> 
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drug_Gametocyte34_Killrate</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>The log reduction per day in late-stage gametocytes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Chloroquine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span> 
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Drug_GametocyteM_Killrate</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>The log reduction per day in mature <a class="reference internal" href="glossary.html#term-gametocyte"><span class="xref std std-term">gametocyte</span></a> numbers at saturated drug concentrations.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Chloroquine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span> 
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drug_Hepatocyte_Killrate</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>0</td>
<td>The log reduction in <a class="reference internal" href="glossary.html#term-hepatocyte"><span class="xref std std-term">hepatocyte</span></a> numbers per day.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Chloroquine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Hepatocyte_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span> 
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Drug_PKPD_C50</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>100</td>
<td><p class="first">The concentration at which drug killing rates are half of the maximum.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This parameter and the <strong>Drug_Cmax</strong> parameter must use the same units.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether_Lumefantrine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Cmax&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_PKPD_C50&quot;</span><span class="p">:</span> <span class="mi">100</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drug_Vd</td>
<td>float</td>
<td>0</td>
<td>100000</td>
<td>10</td>
<td>The volume of drug distribution in a pharmacokinetic two compartment model. The first compartment comprises central organs and tissues and the second compartment comprises peripheral tissues. This value is the ratio of the volume of the second compartment to that of the first.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>  
        <span class="nt">&quot;Chloroquine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Drug_Vd&quot;</span><span class="p">:</span> <span class="mi">3</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Fractional_Dose_By_Upper_Age</td>
<td>array of JSON objects</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>An array to specify the fraction of the adult dose to use for children at various ages. Contains <strong>Upper_Age_In_Years</strong> and <strong>Fraction_Of_Adult_Dose</strong> values.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Fractional_Dose_By_Upper_Age&quot;</span><span class="p">:</span> <span class="p">[{</span>
         	    <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">3</span>
            <span class="p">},</span> <span class="p">{</span>
                <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">6</span>
            <span class="p">},</span> <span class="p">{</span>
                <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.75</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">10</span>
            <span class="p">}]</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Fraction_Of_Adult_Dose</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>NA</td>
<td>The fraction of the adult drug dose given to children below the age defined in <strong>Upper_Age_In_Years</strong>. Set in the <strong>Fractional_Dose_By_Upper_Age</strong> array.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Fractional_Dose_By_Upper_Age&quot;</span><span class="p">:</span> <span class="p">[{</span>
         	    <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">3</span>
            <span class="p">},</span> <span class="p">{</span>
                <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">6</span>
            <span class="p">},</span> <span class="p">{</span>
                <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.75</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">10</span>
            <span class="p">}]</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Malaria_Drug_Params</td>
<td>JSON object</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>This JSON structure contains the names of anti-malarial drugs and the parameters that define them.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether_Lumefantrine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Bodyweight_Exponent&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Cmax&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T1&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T2&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Dose_Interval&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Fulltreatment_Doses&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Hepatocyte_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_PKPD_C50&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Vd&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Max_Drug_IRBC_Kill&quot;</span><span class="p">:</span> <span class="mf">4.6</span>
        <span class="p">},</span>
        <span class="nt">&quot;Artemisinin&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Bodyweight_Exponent&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Cmax&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T1&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T2&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Dose_Interval&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Fulltreatment_Doses&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Hepatocyte_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_PKPD_C50&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Vd&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Max_Drug_IRBC_Kill&quot;</span><span class="p">:</span> <span class="mf">4.61</span>
        <span class="p">}</span>
    <span class="p">}</span>	
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Max_Drug_IRBC_Kill</td>
<td>float</td>
<td>5</td>
<td>100000</td>
<td>5</td>
<td>The maximum log reduction in IRBCs per day due to treatment.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether_Lumefantrine&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Bodyweight_Exponent&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Cmax&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T1&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Decay_T2&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Dose_Interval&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Fulltreatment_Doses&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte02_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Gametocyte34_Killrate&quot;</span><span class="p">:</span> <span class="mf">2.3</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_GametocyteM_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Hepatocyte_Killrate&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_PKPD_C50&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Drug_Vd&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Max_Drug_IRBC_Kill&quot;</span><span class="p">:</span> <span class="mf">4.61</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>		
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>PKPD_Model</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>FIXED_DURATION_CONSTANT_EFFECT</td>
<td><p class="first">Determines which pharmacokinetic pharmacodynamic module to use. 
Possible values are:</p>
<dl class="last docutils">
<dt>FIXED_DURATION_CONSTANT_EFFECT</dt>
<dd>This is the simplified model of drug action.  Each dose of the drug has a constant effect for a fixed duration of time, after which the effect is zero.</dd>
<dt>CONCENTRATION_VERSUS_TIME</dt>
<dd>This model of drug action uses continuous pharmacokinetics with two-compartment decay.  Effects of each drug depend on the concentration at that point in time.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;PKPD_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_DURATION_CONSTANT_EFFECT&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Upper_Age_In_Years</td>
<td>float</td>
<td>0</td>
<td>125</td>
<td>NA</td>
<td>The age, in years, below which children are given a fraction of the adult drug dose, as defined in <strong>Fraction_Of_Adult_Dose</strong>. Note that this parameter can be specified for each drug included in the configuration file, and different fractional doses may be used for different age groups.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Drug_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Artemether&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Fractional_Dose_By_Upper_Age&quot;</span><span class="p">:</span> <span class="p">[{</span>
         	    <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">3</span>
            <span class="p">},</span> <span class="p">{</span>
                <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">6</span>
            <span class="p">},</span> <span class="p">{</span>
                <span class="nt">&quot;Fraction_Of_Adult_Dose&quot;</span><span class="p">:</span> <span class="mf">0.75</span><span class="p">,</span>
                <span class="nt">&quot;Upper_Age_In_Years&quot;</span><span class="p">:</span> <span class="mi">10</span>
            <span class="p">}]</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="enable-or-disable-features">
<span id="config-enable"></span><h2><a class="toc-backref" href="#id2">Enable or disable features</a><a class="headerlink" href="#enable-or-disable-features" title="Permalink to this headline">¶</a></h2>
<p>The following parameters enable or disable features of the model, such as allowing births, deaths,
or aging. Set to false (0) to disable; set to true (1) to enable.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Enable_Aging</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not individuals in a population age during the simulation. <strong>Enable_Vital_Dynamics</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Aging&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Air_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not migration by air travel will occur. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
     <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
     <span class="nt">&quot;Enable_Air_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
     <span class="nt">&quot;Air_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/air_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Birth</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not individuals will be added to the simulation by birth. <strong>Enable_Vital_Dynamics</strong> must be set to true (1). If you want new individuals to have the same intervention coverage as existing individuals, you must add a <strong>BirthTriggeredIV</strong> to the campaign file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Climate_Stochasticity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not the climate has stochasticity. <strong>Climate_Model</strong> must be set to CLIMATE_CONSTANT or CLIMATE_BY_DATA. Set the variance using the parameters <strong>Air_Temperature_Variance</strong>,  <strong>Land_Temperature_Variance</strong>, <strong>Enable_Rainfall_Stochasticity</strong>, and <strong>Relative_Humidity_Variance</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_BY_DATA&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Climate_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Air_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Rainfall_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Land_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Relative_Humidity_Variance&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Default_Reporting</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not the default InsetChart.json report is created.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Default_Reporting&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Demographics_Birth</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not newborns have identical or heterogeneous characteristics. Set to false (0) to give all newborns identical characteristics; set to true (1) to allow for heterogeneity in traits such as sickle-cell status. <strong>Enable_Birth</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Demographics_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Demographics_Builtin</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not built-in demographics for default geography will be used. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. Set to true (1) to define the initial population and number of nodes using <strong>Default_Geography_Initial_Node_Population</strong> and <strong>Default_Geography_Torus_Size</strong>. Set to false (0) to use demographics input files defined in <strong>Demographics_Filenames</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Builtin&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Initial_Node_Population&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Torus_Size&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Demographics_Gender</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not gender ratios are drawn from a Gaussian or 50/50 draw. Set to true (1) to create gender ratios drawn from a male/female ratio that is randomly smeared by a Gaussian of width 1%; set to false (0) to assign a gender ratio based on a 50/50 draw.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Gender&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Demographics_Other</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not other demographic factors are included in the simulation, such as the fraction of individuals above poverty, urban/rural characteristics, heterogeneous initial immunity, or risk. These factors are set in the demographics file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Other&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Demographics_Reporting</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not demographic summary data and age-binned reports are outputted to file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Reporting&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Disease_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not individuals die due to disease.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Disease_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Egg_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not to include a daily mortality rate on the egg population, which is independent of climatic factors.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Egg_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Family_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not all members of a household can migrate together. All residents must be home before they can leave on the trip. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Migration&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Family_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Family_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/family_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Heterogeneous_Intranode_Transmission</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not individuals experience heterogeneous disease transmission within a node. When set to true (1), individual property definitions and the <img class="math" src="_images/math/410a9d0df9c135dd73b269cba7ef04dcfd932b1f.png" alt="\beta"/> matrix must be specified in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-properties"><span class="std std-ref">NodeProperties and IndividualProperties</span></a> parameters). The <img class="math" src="_images/math/410a9d0df9c135dd73b269cba7ef04dcfd932b1f.png" alt="\beta"/> values are multiplied with the <img class="math" src="_images/math/410a9d0df9c135dd73b269cba7ef04dcfd932b1f.png" alt="\beta"/> <sub>0</sub> value configured by <strong>Base_Infectivity</strong>.</p>
<p class="last">This is used only in generic simulations, but must be set to false (0) for all other simulation types. Heterogeneous transmission for other diseases uses other mechanistic parameters included with the simulation type.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Heterogeneous_Intranode_Transmission&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Immune_Decay</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not immunity decays after an infection clears. Set to true (1) if immunity decays; set to false (0) if recovery from the disease confers complete immunity for life. <strong>Enable_Immunity</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Immune_Decay&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Immunity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not an individual has protective immunity after an infection clears.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Interventions</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not campaign interventions will be used in the simulation. Set <strong>Campaign_Filename</strong> to the path of the file that contains the campaign interventions.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Interventions&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Campaign_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;campaign.json&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Local_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not local migration (the diffusion of people in and out of nearby nodes by foot travel) occurs. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Local_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Local_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/local_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Maternal_Transmission</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not infectious mothers infect infants at birth. <strong>Enable_Birth</strong> must be set to true (1).</p>
<p class="last">Set to false (0) for this simulation type.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Maternal_Transmission&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Migration_Heterogeneity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not migration rate is heterogeneous among individuals. Set to true (1) to use a migration rate distribution in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters); set to false (0) to use the same migration rate applied to all individuals. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Migration_Heterogeneity&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Property_Output</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not to create property output reports, which detail groups as defined in <strong>IndividualProperties</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-properties"><span class="std std-ref">NodeProperties and IndividualProperties</span></a> parameters).  When there is more than one property type, the report will display the channel information for all combinations of the property type groups.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Property_Output&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Rainfall_Stochasticity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not there is stochastic variation in rainfall; set to  true (1) for stochastic variation of rainfall that is drawn from an exponential distribution (with a mean value as the daily rainfall from the  <strong>Climate_Model</strong> values CLIMATE_CONSTANT or CLIMATE_BY_DATA), or set to false (0) to disable rainfall stochasticity.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Climate_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Air_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Rainfall_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Land_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Relative_Humidity_Variance&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Regional_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not there is migration by road vehicle into and out of nodes in the road network. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Regional_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Regional_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/regional_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Sea_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not there is migration on ships into and out of coastal cities with seaports. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Sea_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Sea_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/sea_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Sexual_Combination</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not male and female gametocytes undergo sexual combination. Set to true (1) to allow for sexual combination; set to false (0) for transmission of each strain only.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This parameter is currently not in use.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Sexual_Combination&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Spatial_Output</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not spatial output reports are created. If set to true (1), spatial output reports include all channels listed in the parameter <strong>Spatial_Output_Channels</strong>.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Spatial output files require significant processing time and disk space.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Spatial_Output&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Spatial_Output_Channels&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;Prevalence&quot;</span><span class="p">,</span>
        <span class="s2">&quot;New_Infections&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Superinfection</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not an individual can have multiple infections simultaneously. Set to true (1) to allow for multiple simultaneous infections; set to false (0) if multiple infections are not possible. Set the <strong>Max_Individual_Infections</strong> parameter.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Superinfection&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Max_Individual_Infections&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vector_Aging</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors undergo senescence as they age.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vector_Aging&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Vector_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors can migrate. <strong>Vector_Sampling_Type</strong> must be set to  TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. Specific migration types must be enabled or disabled.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration_Local&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Migration_Filename_Local&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/vector_local.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vector_Migration_Local</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors may migrate to adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration_Local&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Migration_Filename_Local&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/vector_local.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Vector_Migration_Regional</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors can migration to non-adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration_Regional&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Migration_Filename_Regional&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/vector_regional.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vector_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not vectors can die. <strong>Vector_Sampling_Type</strong> must be set to TRACK_ALL_VECTORS.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Enable_Vector_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Vital_Dynamics</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not births and deaths occur in the simulation. Births and deaths must be individually enabled and set.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Death_Rate_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;NONDISEASE_MORTALITY_OFF&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Base_Mortality&quot;</span><span class="p">:</span> <span class="mf">0.002</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="general-disease">
<span id="config-general-disease"></span><h2><a class="toc-backref" href="#id3">General disease</a><a class="headerlink" href="#general-disease" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine general disease characteristics.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Animal_Reservoir_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_ZOONOSIS</td>
<td><p class="first">The type of animal reservoir and configuration for zoonosis. Use of the animal reservoir sets a low constant baseline of infectivity beyond what is present in the human population. It allows a more random introduction of cases in continuous time, which is more applicable for various situations such as zoonosis. Possible values are:</p>
<dl class="last docutils">
<dt>NO_ZOONOSIS</dt>
<dd>There is no animal reservoir.</dd>
<dt>CONSTANT_ZOONOSIS</dt>
<dd>The daily rate of zoonotic infection is configured by the parameter <strong>Zoonosis_Rate</strong>.</dd>
<dt>ZOONOSIS_FROM_DEMOGRAPHICS</dt>
<dd>The zoonosis rate is scaled first by the node-specific <strong>Zoonosis</strong> value in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters). If <strong>Zoonosis</strong> is not configured, the simulation will then use the value for <strong>Zoonosis_Rate</strong>.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Animal_Reservoir_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;CONSTANT_ZOONOSIS&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Number_Basestrains</td>
<td>integer</td>
<td>1</td>
<td>10</td>
<td>1</td>
<td>The number of base strains in the simulation, such as antigenic variants.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Number_Basestrains&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Number_Substrains</td>
<td>integer</td>
<td>1</td>
<td>16777200</td>
<td>256</td>
<td>The number of disease substrains for each base strain, such as genetic variants.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Number_Substrains&quot;</span><span class="p">:</span> <span class="mi">16777216</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Zoonosis_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The daily rate of zoonotic infection per individual.  <strong>Animal_Reservoir_Type</strong> must be set to CONSTANT_ZOONOSIS or ZOONOSIS_FROM_DEMOGRAPHICS. If  <strong>Animal_Reservoir_Type</strong> is set to ZOONOSIS_FROM_DEMOGRAPHICS, the value for the <strong>Zoonosis</strong> <strong>NodeAttribute</strong> in the demographics file will override the value set for <strong>Zoonosis_Rate</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Zoonosis_Rate&quot;</span><span class="p">:</span> <span class="mf">0.005</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="geography-and-environment">
<span id="config-env"></span><h2><a class="toc-backref" href="#id4">Geography and environment</a><a class="headerlink" href="#geography-and-environment" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine characteristics of the geography and environment of the simulation.
For example, how to use the temperature or rainfall data in the climate files and the size of the
nodes in the simulation.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Air_Temperature_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>air_temp.json</td>
<td>The path to the input data file that defines air temperature data measured two meters above ground. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_BY_DATA&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Air_Temperature_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_air_temperature_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Air_Temperature_Offset</td>
<td>float</td>
<td>-20</td>
<td>20</td>
<td>0</td>
<td>The linear shift of air temperature in degrees Celsius. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Air_Temperature_Offset&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Air_Temperature_Variance</td>
<td>float</td>
<td>0</td>
<td>5</td>
<td>2</td>
<td>The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily air temperature values when <strong>Climate_Model</strong> is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA. <strong>Enable_Climate_Stochasticity</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Climate_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Air_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Base_Air_Temperature</td>
<td>float</td>
<td>-55</td>
<td>45</td>
<td>22</td>
<td>The air temperature, in degrees Celsius, where <strong>Climate_Model</strong> is set to CLIMATE_CONSTANT.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_CONSTANT&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Base_Air_Temperature&quot;</span><span class="p">:</span> <span class="mi">30</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Base_Land_Temperature</td>
<td>float</td>
<td>-55</td>
<td>60</td>
<td>26</td>
<td>The land temperature, in degrees Celsius, where <strong>Climate_Model</strong> is set to CLIMATE_CONSTANT.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_CONSTANT&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Base_Land_Temperature&quot;</span><span class="p">:</span> <span class="mi">20</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Base_Rainfall</td>
<td>float</td>
<td>0</td>
<td>150</td>
<td>10</td>
<td>The value of rainfall per day in millimeters when <strong>Climate_Model</strong> is set to CLIMATE_CONSTANT.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
     <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_CONSTANT&quot;</span><span class="p">,</span> 
     <span class="nt">&quot;Base_Rainfall&quot;</span><span class="p">:</span> <span class="mi">20</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Base_Relative_Humidity</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.75</td>
<td>The value of humidity where <strong>Climate_Model</strong> is set to CLIMATE_CONSTANT.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
    <span class="nt">&quot;Base_Relative_Humidity&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Climate_Model</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CLIMATE_OFF</td>
<td><p class="first">How and from what files the climate of a simulation is configured. The possible values are:</p>
<dl class="last docutils">
<dt>CLIMATE_OFF</dt>
<dd>No climate files used.</dd>
<dt>CLIMATE_CONSTANT</dt>
<dd>Uses the conditional parameters that give the fixed values of temperature or rain for land temperature, air temperature, rainfall, and humidity.</dd>
<dt>CLIMATE_KOPPEN</dt>
<dd>Uses an input data file that decodes Koppen codes by geographic location.</dd>
<dt>CLIMATE_BY_DATA</dt>
<dd>Reads everything out of several input data files with additional parameters that allow the addition of stochasticity or scale offsets.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>   
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_CONSTANT&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Climate_Update_Resolution</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CLIMATE_UPDATE_YEAR</td>
<td><p class="first">The  resolution of data in climate files. <strong>Climate_Model</strong> must be set to CLIMATE_CONSTANT, CLIMATE_BY_DATA, or CLIMATE_KOPPEN.
Possible values are:</p>
<p class="last">CLIMATE_UPDATE_YEAR
CLIMATE_UPDATE_MONTH
CLIMATE_UPDATE_WEEK
CLIMATE_UPDATE_DAY
CLIMATE_UPDATE_HOUR</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Update_Resolution&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_UPDATE_DAY&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Default_Geography_Initial_Node_Population</td>
<td>integer</td>
<td>0</td>
<td>1000000</td>
<td>1000</td>
<td>When using the built-in demographics for default geography, the initial number of individuals in each node. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. <strong>Enable_Demographics_Builtin</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Builtin&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Initial_Node_Population&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Torus_Size&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Default_Geography_Torus_Size</td>
<td>integer</td>
<td>3</td>
<td>100</td>
<td>10</td>
<td><p class="first">When using the built-in demographics for default geography, the square root of the number of nodes in the simulation. The simulation uses an N x N square grid of nodes with N specified by this parameter. If migration is enabled, the N x N nodes are assumed to be a torus and individuals can migrate from any node to all four adjacent nodes.</p>
<p>To enable migration, set <strong>Migration_Model</strong> to FIXED_RATE_MIGRATION. Built-in migration is a form of “local” migration where individuals only migrate to the adjacent nodes. You can use the <strong>x_Local_Migration</strong> parameter to control the rate of migration. The other migration parameters are ignored. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing.</p>
<p class="last"><strong>Enable_Demographics_Builtin</strong> must be set to true (1).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Builtin&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Initial_Node_Population&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Torus_Size&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Climate_Stochasticity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not the climate has stochasticity. <strong>Climate_Model</strong> must be set to CLIMATE_CONSTANT or CLIMATE_BY_DATA. Set the variance using the parameters <strong>Air_Temperature_Variance</strong>,  <strong>Land_Temperature_Variance</strong>, <strong>Enable_Rainfall_Stochasticity</strong>, and <strong>Relative_Humidity_Variance</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_BY_DATA&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Climate_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Air_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Rainfall_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Land_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Relative_Humidity_Variance&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Rainfall_Stochasticity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not there is stochastic variation in rainfall; set to  true (1) for stochastic variation of rainfall that is drawn from an exponential distribution (with a mean value as the daily rainfall from the  <strong>Climate_Model</strong> values CLIMATE_CONSTANT or CLIMATE_BY_DATA), or set to false (0) to disable rainfall stochasticity.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Climate_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Air_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Rainfall_Stochasticity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Land_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Relative_Humidity_Variance&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Koppen_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>The path to the input file used to specify Koppen climate classifications; only used  when <strong>Climate_Model</strong> is set to CLIMATE_KOPPEN. The file must be in .dat format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Koppen_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Mad_2_5arcminute_koppen.dat&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Land_Temperature_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>land_temp.json</td>
<td>The path of the input file defining temperature data measured at land surface; used only when <strong>Climate_Model</strong> is set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Land_Temperature_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_land_temperature_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Land_Temperature_Offset</td>
<td>float</td>
<td>-20</td>
<td>20</td>
<td>0</td>
<td>The linear shift of land surface temperature in degrees Celsius; only used  when <strong>Climate_Model</strong> is set to CLIMATE_BY_DATA.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Land_Temperature_Offset&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Land_Temperature_Variance</td>
<td>float</td>
<td>0</td>
<td>7</td>
<td>2</td>
<td>The standard deviation (in degrees Celsius) for normally distributed noise applied to the daily land temperature values when <strong>Climate_Model</strong> is configured to CLIMATE_CONSTANT or CLIMATE_BY_DATA; only used if the <strong>Enable_Climate_Stochasticity</strong> is set to true (1).</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Land_Temperature_Variance&quot;</span><span class="p">:</span> <span class="mf">1.5</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Node_Grid_Size</td>
<td>float</td>
<td>0.00416</td>
<td>90</td>
<td>0.004167</td>
<td>The spatial resolution indicating the node grid size for a simulation in degrees.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Node_Grid_Size&quot;</span><span class="p">:</span> <span class="mf">0.042</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Rainfall_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>rainfall.json</td>
<td>The path of the input file which defines rainfall data. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Rainfall_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_rainfall_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Rainfall_In_mm_To_Fill_Swamp</td>
<td>float</td>
<td>1</td>
<td>10000</td>
<td>1000</td>
<td>Millimeters of rain to fill larval habitat to capacity.  This is only used for vector species with <strong>Larval_Habitat_Types</strong> set to BRACKISH_SWAMP.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Rainfall_In_mm_To_Fill_Swamp&quot;</span><span class="p">:</span> <span class="mf">1000.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Rainfall_Scale_Factor</td>
<td>float</td>
<td>0.1</td>
<td>10</td>
<td>1</td>
<td>The scale factor used in multiplying rainfall value(s). <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Rainfall_Scale_Factor&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Relative_Humidity_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>rel_hum.json</td>
<td>The path of the input file which defines relative humidity data measured 2 meters above ground. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Relative_Humidity_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_relative_humidity_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Relative_Humidity_Scale_Factor</td>
<td>float</td>
<td>0.1</td>
<td>10</td>
<td>1</td>
<td>The scale factor used in multiplying relative humidity values. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Relative_Humidity_Scale_Factor&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Relative_Humidity_Variance</td>
<td>float</td>
<td>0</td>
<td>0.12</td>
<td>0.05</td>
<td>The standard deviation (in percentage) for normally distributed noise applied to the daily relative humidity values when <strong>Climate_Model</strong> is configured as CLIMATE_CONSTANT or CLIMATE_BY_DATA.   <strong>Enable_Climate_Stochasticity</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Relative_Humidity_Variance&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="immunity">
<span id="config-immunity"></span><h2><a class="toc-backref" href="#id5">Immunity</a><a class="headerlink" href="#immunity" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the immune system response for the disease being modeled, including
waning immunity after an infection clears.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Acquisition_Blocking_Immunity_Decay_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>The rate at which acquisition-blocking immunity decays after the initial period indicated by the base acquisition-blocking immunity offset. Only used when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> parameters are set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Acquisition_Blocking_Immunity_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Acquisition_Blocking_Immunity_Duration_Before_Decay</td>
<td>float</td>
<td>0</td>
<td>45000</td>
<td>0</td>
<td>The number of days after infection until acquisition-blocking immunity begins to decay. <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Acquisition_Blocking_Immunity_Duration_Before_Decay&quot;</span><span class="p">:</span> <span class="mi">10</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Antibody_Capacity_Growth_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>The maximum daily rate of specific antibody capacity increase. Antibody production begins at a capacity of 0.3., and hyperimmunity results at 0.4. The actual growth rate tends to be lower and is a function of antigen concentrations.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_Capacity_Growth_Rate&quot;</span><span class="p">:</span> <span class="mf">0.09</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Antibody_CSP_Decay_Days</td>
<td>float</td>
<td>1</td>
<td>3.40E+38</td>
<td>90</td>
<td>Exponential decay time for the circumsporozoite protein (CSP) antibody concentration (in days) when boosted above natural maximum concentrations.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_CSP_Decay_Days&quot;</span><span class="p">:</span> <span class="mi">80</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Antibody_CSP_Killing_Inverse_Width</td>
<td>float</td>
<td>1.00E-0</td>
<td>1000000</td>
<td>1.5</td>
<td>The inverse width of the sigmoidal sporozoite killing function of circumsporozoite protein (CSP) antibody concentration.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_CSP_Killing_Inverse_Width&quot;</span><span class="p">:</span> <span class="mf">1.7</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Antibody_CSP_Killing_Threshold</td>
<td>float</td>
<td>1.00E-0</td>
<td>1000000</td>
<td>10</td>
<td>The threshold value on circumsporozoite protein (CSP) antibody concentration for sporozoite killing.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_CSP_Killing_Threshold&quot;</span><span class="p">:</span> <span class="mi">20</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Antibody_IRBC_Kill_Rate</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2</td>
<td>The scale factor multiplied by antibody level to produce the rate of clearance of the infected red blood cell (IRBC) population.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_IRBC_Kill_Rate&quot;</span><span class="p">:</span> <span class="mf">1.595</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Antibody_Memory_Level</td>
<td>float</td>
<td>0</td>
<td>0.35</td>
<td>0.2</td>
<td>The limiting level of antibody capacity that remains after a prolonged absence of a specific antigen. The antibody capacity decays to this level gradually after infection is cleared. The decay rate in antibody capacity is set so that hyperimmunity is lost within 4 months, and capacity continues to decay to this level. The antibody memory level is relevant for year-scale dynamics, but not for long-term dynamics (10-20 years).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_Memory_Level&quot;</span><span class="p">:</span> <span class="mf">0.3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Antibody_Stimulation_C50</td>
<td>float</td>
<td>0.1</td>
<td>10000</td>
<td>10</td>
<td>The concentration of an antigen, measured in IRBC/ul at which growth in antibody capacity against the antigen increases at half the maximum rate specified by <strong>Antibody_Capacity_Growth_Rate</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_Stimulation_C50&quot;</span><span class="p">:</span> <span class="mi">30</span><span class="p">,</span>
    <span class="nt">&quot;Antibody_Capacity_Growth_Rate&quot;</span><span class="p">:</span> <span class="mf">0.09</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Antigen_Switch_Rate</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2.00E-09</td>
<td>The antigenic switching rate per infected red blood cell per asexual cycle. See <strong>Parasite_Switch_Type</strong> for different switching patterns.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antigen_Switch_Rate&quot;</span><span class="p">:</span> <span class="mf">5e-09</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Cytokine_Gametocyte_Inactivation</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>The strength of inflammatory response inactivation of gametocytes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Cytokine_Gametocyte_Inactivation&quot;</span><span class="p">:</span> <span class="mf">0.0167</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Immune_Decay</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not immunity decays after an infection clears. Set to true (1) if immunity decays; set to false (0) if recovery from the disease confers complete immunity for life. <strong>Enable_Immunity</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Immune_Decay&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Immunity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not an individual has protective immunity after an infection clears.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Erythropoiesis_Anemia_Effect</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>3.5</td>
<td>The exponential rate of increased red-blood-cell production from reduced red-blood-cell availability.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Erythropoiesis_Anemia_Effect&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Falciparum_MSP_Variants</td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>100</td>
<td>The number of distinct <a class="reference internal" href="glossary.html#term-merozoite-surface-protein-msp"><span class="xref std std-term">merozoite surface protein (MSP)</span></a> variants for <em>P. falciparum</em> malaria in the overall parasite population in the simulation, not necessarily in an individual.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Falciparum_MSP_Variants&quot;</span><span class="p">:</span> <span class="mi">4</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Falciparum_Nonspecific_Types</td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>20</td>
<td>The number of distinct non-specific types of <em>P. falciparum</em> malaria.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Falciparum_Nonspecific_Types&quot;</span><span class="p">:</span> <span class="mi">92</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Falciparum_PfEMP1_Variants</td>
<td>integer</td>
<td>0</td>
<td>100000</td>
<td>1000</td>
<td>The number of distinct <a class="reference internal" href="glossary.html#term-plasmodium-falciparum-erythrocyte-membrane-protein-1-pfemp1"><span class="xref std std-term">Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1)</span></a> variants for <em>P. falciparum</em> malaria in the overall parasite population in the simulation.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
    <span class="nt">&quot;Falciparum_PfEMP1_Variants&quot;</span><span class="p">:</span> <span class="mi">300</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Fever_IRBC_Kill_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.15</td>
<td>The maximum kill rate for infected red blood cells due to the inflammatory innate immune response. As fever increases above 38.5 degrees Celsius, the kill rate becomes successively higher along a sigmoidal curve approaching this rate.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
    <span class="nt">&quot;Fever_IRBC_Kill_Rate&quot;</span><span class="p">:</span> <span class="mf">1.4</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Immune_Threshold_For_Downsampling</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Threshold on acquisition immunity at which to apply immunity dependent downsampling. <strong>Individual_Sampling_Type</strong> must set to ADAPTED_SAMPLING_BY_IMMUNE_STATE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Individual_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ADAPTED_SAMPLING_BY_IMMUNE_STATE&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Immune_Threshold_For_Downsampling&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Immunity_Acquisition_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>The multiplicative reduction in the probability of reacquiring disease. Only used when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> are set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Immune_Decay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Immunity_Acquisition_Factor&quot;</span><span class="p">:</span> <span class="mf">0.9</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Immunity_Initialization_Distribution_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>DISTRIBUTION_OFF</td>
<td><p class="first">The method for initializing the immunity distribution in the simulated population. <strong>Enable_Immunity</strong> must be set to true (1). Possible values are:</p>
<dl class="last docutils">
<dt>DISTRIBUTION_OFF</dt>
<dd>All individuals default to no immunity.</dd>
<dt>DISTRIBUTION_SIMPLE</dt>
<dd>Individual immunities are drawn from a distribution whose functional form and parameters are specified in the demographics file in <strong>IndividualAttributes</strong> using  <strong>ImmunityDistributionFlag</strong>, <strong>ImmunityDistribution1</strong>, and <strong>ImmunityDistribution2</strong> (see <a class="reference internal" href="parameter-demographics.html#demo-simpledistro"><span class="std std-ref">Simple distributions</span></a> parameters).</dd>
<dt>DISTRIBUTION_COMPLEX</dt>
<dd>Individual immunities are drawn from an age-dependent piecewise linear function for each specific antibody in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-complexdistro"><span class="std std-ref">Complex distributions</span></a> parameters).</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Immunity_Initialization_Distribution_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;DISTRIBUTION_COMPLEX&quot;</span> 
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Immunity_Mortality_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>The multiplicative reduction in the probability of dying from infection after getting re-infected. <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Immune_Decay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Immunity_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Immunity_Transmission_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>The multiplicative reduction in the probability of transmitting infection after getting re-infected. Only used when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> are set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Immunity_Decay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Immunity_Transmission_Factor&quot;</span><span class="p">:</span> <span class="mf">0.9</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Innate_Immune_Variation_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONE</td>
<td><p class="first">The type of innate immunity on which to apply individual-level variation. 
Possible values are:</p>
<dl class="last docutils">
<dt>NONE</dt>
<dd>No variation in innate immunity between individuals.</dd>
<dt>PYROGENIC_THRESHOLD</dt>
<dd>Applies individual variation using the immunity distribution values set in <strong>ImmunityDistributionFlag</strong>, <strong>ImmunityDistribution1</strong>, and <strong>ImmunityDistribution2</strong> in the demographics file to the average threshold for infected red blood cells resulting in fever, as set in <strong>Pyrogenic_Threshold</strong>.</dd>
<dt>PYROGENIC_THRESHOLD_VS_AGE</dt>
<dd>Ignores the demographic file settings and applies an age-dependent threshold for infected red blood cells resulting in fever.</dd>
<dt>CYTOKINE_KILLING</dt>
<dd>Applies individual variation using the immunity distribution values set in <strong>ImmunityDistributionFlag</strong>, <strong>ImmunityDistribution1</strong>, and <strong>ImmunityDistribution2</strong> in the demographics file to the effectiveness of cytokines in killing infected red blood cells, as set in <strong>Fever_IRBC_Kill_Rate</strong>.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Innate_Immune_Variation_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;PYROGENIC_THRESHOLD&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Maternal_Antibodies_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>OFF</td>
<td><p class="first">The type of maternal antibody protection.  
Possible values are:</p>
<dl class="last docutils">
<dt>OFF</dt>
<dd>No maternal antibody protection provided.</dd>
<dt>SIMPLE_WANING</dt>
<dd>The protection from maternal antibodies is a function of the configured maximum fraction protection (<strong>Maternal_Antibody_Protection</strong>) and the age of the child (through <strong>Maternal_Antibody_Decay_Rate</strong>), but also of the transmission-intensity in the simulation. In particular, the protection is modified by a factor corresponding to the fraction of PfEMP1 antigenic variants to which possible mothers (14 to 45-year-old females) have been exposed in their lifetimes.</dd>
<dt>CONSTANT_INITIAL_IMMUNITY</dt>
<dd>Protection is independent of acquired immunity in possible mothers.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Maternal_Antibodies_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;CONSTANT_INITIAL_IMMUNITY&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Maternal_Antibody_Decay_Rate</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0.01</td>
<td>The decay rate per day in protection from maternal antibodies. <strong>Maternal_Antibodies_Type</strong> must be set to SIMPLE_WANING or CONSTANT_INITIAL_IMMUNITY.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Maternal_Antibodies_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;SIMPLE_WANING&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Maternal_Antibody_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span><span class="p">,</span> 
    <span class="nt">&quot;Maternal_Antibody_Protection&quot;</span><span class="p">:</span> <span class="mf">0.1327</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Maternal_Antibody_Protection</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>The strength of protection from maternal antibodies as a multiple of full antibody killing effect. <strong>Maternal_Antibodies_Type</strong> must be set to SIMPLE_WANING or CONSTANT_INITIAL_IMMUNITY.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Maternal_Antibodies_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;SIMPLE_WANING&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Maternal_Antibody_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span><span class="p">,</span> 
    <span class="nt">&quot;Maternal_Antibody_Protection&quot;</span><span class="p">:</span> <span class="mf">0.1327</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Max_MSP1_Antibody_Growthrate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>The maximum increase in MSP1 antibody capacity during each asexual cycle. The higher this value, the sooner early clearances are observed and the earlier the parasite density envelope decreases.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Max_MSP1_Antibody_Growthrate&quot;</span><span class="p">:</span> <span class="mf">0.045</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Min_Adapted_Response</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>The minimum level of antibody stimulation to a novel antigen. The value sets the low-range asymptote for antibody capacity growth, which is calculated from <strong>Antibody_Capacity_Growth_Rate</strong> and antigen density, in the presence of any nonzero antigen level.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Min_Adapted_Response&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Mortality_Blocking_Immunity_Decay_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>The rate at which mortality-blocking immunity decays after the mortality-blocking immunity offset period. <strong>Enable_Immune_Decay</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Mortality_Blocking_Immunity_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Mortality_Blocking_Immunity_Duration_Before_Decay</td>
<td>float</td>
<td>0</td>
<td>45000</td>
<td>0</td>
<td>The number of days after infection until mortality-blocking immunity begins to decay. <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Mortality_Blocking_Immunity_Duration_Before_Decay&quot;</span><span class="p">:</span> <span class="mi">270</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>MSP1_Merozoite_Kill_Fraction</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.5</td>
<td>The fraction of merozoites inhibited from invading new erythrocytes when MSP1-specific antibody level is 1.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;MSP1_Merozoite_Kill_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.4364623975644405</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Nonspecific_Antibody_Growth_Rate_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.5</td>
<td>The factor that adjusts <strong>Antibody_Capacity_Growth_Rate</strong> for less immunogenic surface proteins, called minor epitopes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Nonspecific_Antibody_Growth_Rate_Factor&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Nonspecific_Antigenicity_Factor</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.2</td>
<td>The nonspecific antigenicity factor that adjusts antibody IRBC kill rate to account for IRBCs caused by antibody responses to antigenically weak surface proteins.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Nonspecific_Antigenicity_Factor&quot;</span><span class="p">:</span> <span class="mf">0.39192559432597257</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>RBC_Destruction_Multiplier</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>9.5</td>
<td>The number of total red blood cells destroyed per infected rupturing <a class="reference internal" href="glossary.html#term-schizont"><span class="xref std std-term">schizont</span></a>.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;RBC_Destruction_Multiplier&quot;</span><span class="p">:</span> <span class="mf">3.291048711</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Transmission_Blocking_Immunity_Decay_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>The rate at which transmission-blocking immunity decays after the base transmission-blocking immunity offset period. Used only when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> parameters are set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Transmission_Blocking_Immunity_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Transmission_Blocking_Immunity_Duration_Before_Decay</td>
<td>float</td>
<td>0</td>
<td>45000</td>
<td>0</td>
<td>The number of days after infection until transmission-blocking immunity begins to decay. Only used when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> parameters are set to true (1).</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Transmission_Blocking_Immunity_Duration_Before_Decay&quot;</span><span class="p">:</span> <span class="mi">90</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="incubation">
<span id="config-incubation"></span><h2><a class="toc-backref" href="#id6">Incubation</a><a class="headerlink" href="#incubation" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the characteristics of the incubation period.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Base_Incubation_Period</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>Average duration, in days, of the incubation period before infected individuals become infectious. <strong>Incubation_Period_Distribution</strong> must be set to either FIXED_DURATION or EXPONENTIAL_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Base_Incubation_Period&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Incubation_Period_Distribution</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NOT_INITIALIZED</td>
<td><p class="first">The distribution for the duration of the incubation period. Possible values are:</p>
<dl class="last docutils">
<dt>NOT_INITIALIZED</dt>
<dd>No distribution set.</dd>
<dt>FIXED_DURATION</dt>
<dd><strong>Base_Incubation_Period</strong> is the constant duration incubation period.</dd>
<dt>UNIFORM_DURATION</dt>
<dd><strong>Incubation_Period_Min</strong> and <strong>Incubation_Period_Max</strong> define the ranges of a uniform random draw for the duration of the incubation period.</dd>
<dt>GAUSSIAN_DURATION</dt>
<dd><strong>Incubation_Period_Mean</strong> and <strong>Incubation_Period_Std_Dev</strong> define the mean and standard deviation of the Gaussian from which the incubation period is drawn. Negative values are truncated at zero.</dd>
<dt>EXPONENTIAL_DURATION</dt>
<dd><strong>Base_Incubation_Period</strong> is the mean of the exponential random draw.</dd>
<dt>POISSON_DURATION</dt>
<dd><strong>Incubation_Period_Mean</strong> is the mean of the random Poisson draw.</dd>
<dt>LOG_NORMAL_DURATION</dt>
<dd><strong>Incubation_Period_Log_Mean</strong> is the mean of log normal and <strong>Incubation_Period_Log_Width</strong> is the log width of log normal.</dd>
<dt>BIMODAL_DURATION</dt>
<dd>The distribution is bimodal, the duration is a fraction of <strong>Base_Incubation_Period</strong> for a specified period of time and equal to <strong>Base_Incubation_Period</strong> otherwise.</dd>
<dt>PIECEWISE_CONSTANT</dt>
<dd>The distribution is specified with a list of years and a matching list of values. The duration at a given year is that specified for the nearest previous year.</dd>
<dt>PIECEWISE_LINEAR</dt>
<dd>The distribution is specified with a list of years and matching list of values. The duration at a given year is a linear interpolation of the specified values.</dd>
<dt>WEIBULL_DURATION</dt>
<dd>The duration is a <a class="reference internal" href="glossary.html#term-weibull-distribution"><span class="xref std std-term">Weibull distribution</span></a> with a given scale and shape.</dd>
<dt>DUAL_TIMESCALE_DURATION</dt>
<dd>The duration is two exponential distributions with given means.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Incubation_Period&quot;</span><span class="p">:</span> <span class="mi">5</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_DURATION&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Incubation_Period_Log_Mean</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>The mean of log normal for the incubation period distribution. <strong>Incubation_Period_Distribution</strong> must be set to LOG_NORMAL_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;LOG_NORMAL_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Log_Mean&quot;</span><span class="p">:</span> <span class="mf">5.758</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Log_Width&quot;</span><span class="p">:</span> <span class="mf">0.27</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Incubation_Period_Log_Width</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>The log width of log normal for the incubation period distribution. <strong>Incubation_Period_Distribution</strong> must be set to LOG_NORMAL_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;LOG_NORMAL_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Log_Mean&quot;</span><span class="p">:</span> <span class="mf">5.758</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Log_Width&quot;</span><span class="p">:</span> <span class="mf">0.27</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Incubation_Period_Max</td>
<td>float</td>
<td>0.6</td>
<td>3.40E+38</td>
<td>0</td>
<td>The maximum length of the incubation period. <strong>Incubation_Period_Distribution</strong> must be set to UNIFORM_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;UNIFORM_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Min&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Max&quot;</span><span class="p">:</span> <span class="mi">6</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Incubation_Period_Mean</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>The mean of the incubation period. <strong>Incubation_Period_Distribution</strong> must be set to either GAUSSIAN_DURATION or POISSON_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;GAUSSIAN_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Mean&quot;</span><span class="p">:</span> <span class="mi">7</span><span class="p">,</span>
    <span class="nt">&quot;Infectious_Period_Std_Dev&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Incubation_Period_Min</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The minimum length of the incubation period. <strong>Incubation_Period_Distribution</strong> must be set to UNIFORM_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;UNIFORM_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Min&quot;</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Max&quot;</span><span class="p">:</span> <span class="mi">6</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Incubation_Period_Std_Dev</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>The standard deviation incubation period. <strong>Incubation_Period_Distribution</strong> must be set to GAUSSIAN_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Incubation_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;GAUSSIAN_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Incubation_Period_Mean&quot;</span><span class="p">:</span> <span class="mi">7</span><span class="p">,</span>
    <span class="nt">&quot;Infectious_Period_Std_Dev&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="infectivity-and-transmission">
<span id="config-infectivity"></span><h2><a class="toc-backref" href="#id7">Infectivity and transmission</a><a class="headerlink" href="#infectivity-and-transmission" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine aspects of infectivity and disease transmission. For example,
how infectious individuals are and the length of time for which they remain infectious, whether the
disease can be maternally transmitted, and how population density affects infectivity.</p>
<p>The vector transmission model does not use many of the parameters provided by the generic simulation
type. Instead, <a class="reference internal" href="glossary.html#term-gametocyte"><span class="xref std std-term">gametocyte</span></a> abundances and <a class="reference internal" href="glossary.html#term-cytokine"><span class="xref std std-term">cytokine</span></a> mediated infectiousness are
modeled explicitly. See <a class="reference internal" href="malaria-vector-transmission-model.html"><span class="doc">Vector transmission model</span></a> for more information.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Acquire_Modifier</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Modifier of the probability of successful infection of a mosquito by a malaria-infected individual, given the individual&#8217;s infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;aegypti&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mi">1</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Age_Dependent_Biting_Risk_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>OFF</td>
<td><p class="first">The type of functional form for age-dependent risk. Possible values are:</p>
<dl class="last docutils">
<dt>OFF</dt>
<dd>This is the default value.</dd>
<dt>LINEAR</dt>
<dd>The biting risk is 20% of the adult exposure rising linearly until age 20.</dd>
<dt>SURFACE_AREA_DEPENDENT</dt>
<dd>The biting risk rises linearly from 7% to 23% over the first two years of life and then rises with a shallower linear slope to the adult value at age 20.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Age_Dependent_Biting_Risk_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;SURFACE_AREA_DEPENDENT&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Base_Infectious_Period</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td>Average duration, in days, of the infectious period before the infection is cleared. <strong>Infectious_Period_Distribution</strong> must be set to either FIXED_DURATION or EXPONENTIAL_DURATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Infectious_Period&quot;</span><span class="p">:</span> <span class="mi">4</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Base_Infectivity</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.3</td>
<td><p class="first">The base infectiousness of individuals before accounting for transmission-blocking effects of acquired immunity and/or campaign interventions.</p>
<p>For vector simulations, this is the probability of infecting a mosquito during a successful blood meal (modulated by the vector parameter <strong>Acquire_Modifier</strong>). The sum infectiousness of an individual is not allowed to exceed 100%.</p>
<p class="last">For malaria simulations, you do not use scale factors to modify this value. Instead, <a class="reference internal" href="glossary.html#term-gametocyte"><span class="xref std std-term">gametocyte</span></a> abundances and <a class="reference internal" href="glossary.html#term-cytokine"><span class="xref std std-term">cytokine</span></a> mediated infectiousness are modeled explicitly.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Infectivity&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Heterogeneous_Intranode_Transmission</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not individuals experience heterogeneous disease transmission within a node. When set to true (1), individual property definitions and the <img class="math" src="_images/math/410a9d0df9c135dd73b269cba7ef04dcfd932b1f.png" alt="\beta"/> matrix must be specified in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-properties"><span class="std std-ref">NodeProperties and IndividualProperties</span></a> parameters). The <img class="math" src="_images/math/410a9d0df9c135dd73b269cba7ef04dcfd932b1f.png" alt="\beta"/> values are multiplied with the <img class="math" src="_images/math/410a9d0df9c135dd73b269cba7ef04dcfd932b1f.png" alt="\beta"/> <sub>0</sub> value configured by <strong>Base_Infectivity</strong>.</p>
<p class="last">This is used only in generic simulations, but must be set to false (0) for all other simulation types. Heterogeneous transmission for other diseases uses other mechanistic parameters included with the simulation type.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Heterogeneous_Intranode_Transmission&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Maternal_Transmission</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not infectious mothers infect infants at birth. <strong>Enable_Birth</strong> must be set to true (1).</p>
<p class="last">Set to false (0) for this simulation type.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Maternal_Transmission&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Superinfection</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not an individual can have multiple infections simultaneously. Set to true (1) to allow for multiple simultaneous infections; set to false (0) if multiple infections are not possible. Set the <strong>Max_Individual_Infections</strong> parameter.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Superinfection&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Max_Individual_Infections&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infected_Arrhenius_1</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>1.17E+11</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of infected mosquitoes to an infectious state. The duration of sporogony is a decreasing function of temperature. The variable a1 is a temperature-independent scale factor on the progression rate to infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infected_Arrhenius_2</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8340</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of infected mosquitoes to an infectious state. The duration of sporogony is a decreasing function of temperature. The variable a2 is a temperature-dependent scale factor on the progression rate to infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infected_Egg_Batch_Factor</td>
<td>float</td>
<td>0</td>
<td>10</td>
<td>0.8</td>
<td>The dimensionless factor used to modify mosquito egg batch size  in order to account for reduced fertility effects arising due to infection (e.g. when females undergo sporogony).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infection_Updates_Per_Timestep</td>
<td>integer</td>
<td>0</td>
<td>144</td>
<td>1</td>
<td>The number of infection updates executed during each timestep; note that a timestep defaults to one day.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infection_Updates_Per_Timestep&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectious_Human_Feed_Mortality_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1.5</td>
<td>The (dimensionless) factor used to modify the death rate of mosquitoes when feeding on humans, to account for the higher mortality rate infected mosquitoes experience during human feeds versus uninfected mosquitoes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectious_Period_Distribution</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NOT_INITIALIZED</td>
<td><p class="first">The distribution of the duration of the infectious period.</p>
<p>For malaria simulations, this parameter is not used.  Instead, <a class="reference internal" href="glossary.html#term-gametocyte"><span class="xref std std-term">gametocyte</span></a> abundances and <a class="reference internal" href="glossary.html#term-cytokine"><span class="xref std std-term">cytokine</span></a> mediated infectiousness are modeled explicitly.</p>
<p>Possible values are:</p>
<dl class="last docutils">
<dt>NOT_INITIALIZED</dt>
<dd>No distribution set.</dd>
<dt>FIXED_DURATION</dt>
<dd><strong>Base_Infectious_Period</strong> is the constant-duration infectious period.</dd>
<dt>UNIFORM_DURATION</dt>
<dd><strong>Infectious_Period_Min</strong> and <strong>Infectious_Period_Max</strong> define the ranges of a uniform random draw for the duration of the infectious period.</dd>
<dt>GAUSSIAN_DURATION</dt>
<dd><strong>Infectious_Period_Mean</strong> and <strong>Infectious_Period_Std_Dev</strong> define the mean and standard deviation of the Gaussian from which the infectious period is drawn. Negative values are truncated at zero.</dd>
<dt>EXPONENTIAL_DURATION</dt>
<dd><strong>Base_Infectious_Period</strong> is the mean of the exponential random draw.</dd>
<dt>POISSON_DURATION</dt>
<dd><strong>Infectious_Period_Mean</strong> is the mean of the random Poisson draw.</dd>
<dt>LOG_NORMAL_DURATION</dt>
<dd>The duration is a log normal distribution defined by a specified mean and log width.</dd>
<dt>BIMODAL_DURATION</dt>
<dd>The distribution is bimodal, the duration is a fraction of <strong>Base_Incubation_Period</strong> for a specified period of time and equal to <strong>Base_Incubation_Period</strong> otherwise.</dd>
<dt>PIECEWISE_CONSTANT</dt>
<dd>The distribution is specified with a list of years and a matching list of values. The duration at a given year is that specified for the nearest previous year.</dd>
<dt>PIECEWISE_LINEAR</dt>
<dd>The distribution is specified with a list of years and matching list of values. The duration at a given year is a linear interpolation of the specified values.</dd>
<dt>WEIBULL_DURATION</dt>
<dd>The duration is a <a class="reference internal" href="glossary.html#term-weibull-distribution"><span class="xref std std-term">Weibull distribution</span></a> with a given scale and shape.</dd>
<dt>DUAL_TIMESCALE_DURATION</dt>
<dd>The duration is two exponential distributions with given means.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectious_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_DURATION&quot;</span> 
    
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectious_Period_Max</td>
<td>float</td>
<td>0.6</td>
<td>3.40E+38</td>
<td>0</td>
<td><p class="first">The maximum length of the infectious period; used when <strong>Infectious_Period_Distribution</strong> is set to UNIFORM_DURATION.</p>
<p class="last">For malaria simulations, this parameter is not used.  Instead, gametocyte abundances and cytokine-mediated infectiousness are modeled explicitly.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectious_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;UNIFORM_DURATION&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Max&quot;</span><span class="p">:</span> <span class="mi">15</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Min&quot;</span><span class="p">:</span> <span class="mi">5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectious_Period_Mean</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>6</td>
<td><p class="first">The mean of the infectious period; used when <strong>Infectious_Period_Distribution</strong> is set to either GAUSSIAN_DURATION or POISSON_DURATION.</p>
<p class="last">For malaria simulations, this parameter is not used.  Instead, gametocyte abundances and cytokine-mediated infectiousness are modeled explicitly.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectious_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;GAUSSIAN_DURATION&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Mean&quot;</span><span class="p">:</span> <span class="mi">12</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Std_Dev&quot;</span><span class="p">:</span> <span class="mi">10</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectious_Period_Min</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td><p class="first">The minimum length of the infectious period; used when  <strong>Infectious_Period_Distribution</strong> is set to UNIFORM_DURATION.</p>
<p class="last">For malaria simulations, this parameter is not used.  Instead, gametocyte abundances and cytokine-mediated infectiousness are modeled explicitly.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectious_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;UNIFORM_DURATION&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Max&quot;</span><span class="p">:</span> <span class="mi">15</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Min&quot;</span><span class="p">:</span> <span class="mi">5</span>
<span class="p">}</span>			
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectious_Period_Std_Dev</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td><p class="first">The standard deviation of the infectious period; used when <strong>Infectious_Period_Distribution</strong> is set to GAUSSIAN_DURATION.</p>
<p class="last">For malaria simulations, this parameter is not used.  Instead, gametocyte abundances and cytokine-mediated infectiousness are modeled explicitly.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectious_Period_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;GAUSSIAN_DURATION&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Mean&quot;</span><span class="p">:</span> <span class="mi">12</span><span class="p">,</span> 
    <span class="nt">&quot;Infectious_Period_Std_Dev&quot;</span><span class="p">:</span> <span class="mi">10</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectivity_Boxcar_Forcing_Amplitude</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The fractional increase in R0 during the high-infectivity season when <strong>Infectivity_Scale_Type</strong> is equal to ANNUAL_BOXCAR_FUNCTION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_End_Time&quot;</span><span class="p">:</span> <span class="mi">270</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_Start_Time&quot;</span><span class="p">:</span> <span class="mi">90</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span>
<span class="p">}</span>		
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectivity_Boxcar_Forcing_End_Time</td>
<td>float</td>
<td>0</td>
<td>365</td>
<td>0</td>
<td>The end of the high-infectivity season when <strong>Infectivity_Scale_Type</strong> is equal to ANNUAL_BOXCAR_FUNCTION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_End_Time&quot;</span><span class="p">:</span> <span class="mi">270</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_Start_Time&quot;</span><span class="p">:</span> <span class="mi">90</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span>
<span class="p">}</span>		
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectivity_Boxcar_Forcing_Start_Time</td>
<td>float</td>
<td>0</td>
<td>365</td>
<td>0</td>
<td>The beginning of the high-infectivity season, in days, when <strong>Infectivity_Scale_Type</strong> is equal to ANNUAL_BOXCAR_FUNCTION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.25</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_End_Time&quot;</span><span class="p">:</span> <span class="mi">270</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Boxcar_Forcing_Start_Time&quot;</span><span class="p">:</span> <span class="mi">90</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span>
<span class="p">}</span>		
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectivity_Exponential_Baseline</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The scale factor applied to <strong>Base_Infectivity</strong> at the beginning of a simulation, before the infectivity begins to grow exponentially. <strong>Infectivity_Scale_Type</strong> must be set to EXPONENTIAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Exponential_Baseline&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Exponential_Delay&quot;</span><span class="p">:</span> <span class="mi">90</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Exponential_Rate&quot;</span><span class="p">:</span> <span class="mi">45</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_FUNCTION_OF_TIME&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectivity_Exponential_Delay</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The number of days before infectivity begins to ramp up exponentially. <strong>Infectivity_Scale_Type</strong> must be set to EXPONENTIAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Exponential_Baseline&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Exponential_Delay&quot;</span><span class="p">:</span> <span class="mi">90</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Exponential_Rate&quot;</span><span class="p">:</span> <span class="mi">45</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_FUNCTION_OF_TIME&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectivity_Exponential_Rate</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The daily rate of exponential growth to approach to full infectivity after the delay set by <strong>Infectivity_Exponential_Delay</strong> has passed. <strong>Infectivity_Scale_Type</strong> must be set to EXPONENTIAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Exponential_Rate&quot;</span><span class="p">:</span> <span class="mi">45</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectivity_Scale_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_INFECTIVITY</td>
<td><p class="first">A scale factor that allows infectivity to be altered by time or season.  
Possible values are:</p>
<dl class="last docutils">
<dt>CONSTANT_INFECTIVITY</dt>
<dd>No infectivity correction is applied.</dd>
<dt>FUNCTION_OF_TIME_AND_LATITUDE</dt>
<dd><p class="first">Infectivity is corrected for approximate seasonal forcing. The use of a seasonal infectivity correction is a proxy for the effects of varying climate. From October through March, infectivity increases in the Northern Hemisphere and decreases in the Southern Hemisphere. From April through September, the trend reverses: regions closer to the equator have reduced forcing compared to temperate regions.</p>
<p class="last">This is not a substitute for the weather-driven vector dynamics of vector-borne and malaria simulations.</p>
</dd>
<dt>FUNCTION_OF_CLIMATE</dt>
<dd>Allows infectivity to be modulated by weather directly, for example, relative humidity in airborne simulations or rainfall in waterborne simulations. There is no default climate dependence enabled for generic simulations.</dd>
<dt>EXPONENTIAL_FUNCTION_OF_TIME</dt>
<dd>To facilitate certain burn-in scenarios, infectivity ramps up from zero at the beginning of the simulation according to the functional form, 1-exp(-rate*time), where the rate is specified by the parameter <strong>Infectivity_Scaling_Rate</strong>.</dd>
<dt>SINUSOIDAL_FUNCTION_OF_TIME</dt>
<dd>Allows infectivity to be time-dependent, following a sinusoidal shape.</dd>
<dt>ANNUAL_BOXCAR_FUNCTION</dt>
<dd>Allows infectivity to follow a boxcar function, such that it will be equal to zero for an entire time period (e.g. year) except for a single interval in which it is equal to a constant.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;FUNCTION_OF_CLIMATE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectivity_Sinusoidal_Forcing_Amplitude</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Sets the amplitude of sinusoidal variations in <strong>Base_Infectivity</strong>. Only used when <strong>Infectivity_Scale_Type</strong> is set to SINUSOIDAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;SINUSOIDAL_FUNCTION_OF_TIME&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Sinusoidal_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Sinusoidal_Forcing_Phase&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectivity_Sinusoidal_Forcing_Phase</td>
<td>float</td>
<td>0</td>
<td>365</td>
<td>0</td>
<td>Sets the phase of sinusoidal variations in <strong>Base_Infectivity</strong>. Only used when <strong>Infectivity_Scale_Type</strong> is set to SINUSOIDAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;SINUSOIDAL_FUNCTION_OF_TIME&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Sinusoidal_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Sinusoidal_Forcing_Phase&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Malaria_Strain_Model</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>FALCIPARUM_NONRANDOM_STRAIN</td>
<td><p class="first">The generator that is used to construct the antigenic repertoire of a malaria infection. To create parasite diversity, various antigenic strains are created by conducting draws for <a class="reference internal" href="glossary.html#term-merozoite-surface-protein-msp"><span class="xref std std-term">merozoite surface protein (MSP)</span></a> variants, <a class="reference internal" href="glossary.html#term-plasmodium-falciparum-erythrocyte-membrane-protein-1-pfemp1"><span class="xref std std-term">Plasmodium falciparum erythrocyte membrane protein 1 (PfEMP1)</span></a> variants, and minor surface epitopes, out of available populations. Each possible value for this parameter utilizes different settings for the available population draws.
Possible values are:</p>
<dl class="last docutils">
<dt>FALCIPARUM_NONRANDOM_STRAIN</dt>
<dd>The strain created uses the exact same MSP variant, non-specific PfEMP1 variant, and ordered repertoire of PfEMP1 major epitopes. Minor surface protein epitopes are randomly drawn from a set of 5.</dd>
<dt>FALCIPARUM_RANDOM50_STRAIN</dt>
<dd>Strain creation extends the logic used for FALCIPARUM_NONRANDOM_STRAIN, but increases variation by randomly drawing PfEMP1 variants from a population of 50, and allowing the variants to have a random switching order. Random draws are done with replacement.</dd>
<dt>FALCIPARUM_RANDOM_STRAIN</dt>
<dd>The MSP variant is drawn from the population set by the parameter <strong>Falciparum_MSP_Variants</strong>, a repertoire of 50 PfEMP1 variants is drawn from the population set by the parameter <strong>Falciparum_PfEMP1_Variants</strong>, and the set of five minor surface protein epitope variants is drawn from the population set by the parameter <strong>Falciparum_Nonspecific_Types</strong>.</dd>
<dt>FALCIPARUM_STRAIN_GENERATOR</dt>
<dd>This is an exploratory effort created to provide a pseudo-random generation of strains similar to FALCIPARUM_RANDOM_STRAIN, but with variant indices deterministically assigned according to the strain ID.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Strain_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FALCIPARUM_STRAIN_GENERATOR&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Maternal_Transmission_Probability</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The probability of transmission of infection from mother to infant at birth. <strong>Enable_Maternal_Transmission</strong> must be set to 1. Note: For malaria and vector simulations, set this to 0. Instead, use the <strong>Maternal_Antibody_Protection</strong>, <strong>Maternal_Antibody_Decay_Rate</strong>, and <strong>Maternal_Antibodies_Type</strong> parameters.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Maternal_Transmission_Probability&quot;</span><span class="p">:</span> <span class="mf">0.3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Max_Individual_Infections</td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>The limit on the number of infections that an individual can have simultaneously. <strong>Enable_Superinfection</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Max_Individual_Infections&quot;</span><span class="p">:</span> <span class="mi">5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Population_Density_C50</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>10</td>
<td>The population density at which R0 for a 2.5-arc minute square reaches half of its initial value. <strong>Population_Density_Infectivity_Correction</strong> must be set to SATURATING_FUNCTION_OF_DENSITY.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Population_Density_C50&quot;</span><span class="p">:</span> <span class="mi">30</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Population_Density_Infectivity_Correction</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_INFECTIVITY</td>
<td><p class="first">Correction to alter infectivity by population density set in the <strong>Population_Density_C50</strong> parameter. Measured in people per square kilometer. 
Possible values are:</p>
<ul class="simple">
<li>CONSTANT_INFECTIVITY</li>
<li>SATURATING_FUNCTION_OF_DENSITY</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Sparsely populated areas have a lower infectivity, while densely populated areas have a higher infectivity, which rises to saturate at the <strong>Base_Infectivity</strong> value.</p>
</div>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Population_Density_Infectivity_Correction&quot;</span><span class="p">:</span> <span class="s2">&quot;SATURATING_FUNCTION_OF_DENSITY&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Pyrogenic_Threshold</td>
<td>float</td>
<td>0.1</td>
<td>20000</td>
<td>1000</td>
<td>The level of bloodstream infection, measured in IRBC per microliter, at which stimulation of the innate inflammatory immune response is half its maximum value.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Pyrogenic_Threshold&quot;</span><span class="p">:</span> <span class="mi">15000</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Relative_Sample_Rate_Immune</td>
<td>float</td>
<td>0.001</td>
<td>1</td>
<td>0.1</td>
<td>The relative sampling rate for people who have acquired immunity through recovery or vaccination.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Relative_Sample_Rate_Immune&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Susceptibility_Scale_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_SUSCEPTIBILITY</td>
<td><p class="first">The effect of time or season on infectivity. 
Possible values are:</p>
<p class="last">CONSTANT_SUSCEPTIBILITY
LOG_LINEAR_FUNCTION_OF_TIME
LINEAR_FUNCTION_OF_AGE
LOG_LINEAR_FUNCTION_OF_AGE</p>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Susceptibility_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;CONSTANT_SUSCEPTIBILITY&quot;</span>
<span class="p">}</span>	
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Transmission_Blocking_Immunity_Decay_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>The rate at which transmission-blocking immunity decays after the base transmission-blocking immunity offset period. Used only when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> parameters are set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Transmission_Blocking_Immunity_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Transmission_Blocking_Immunity_Duration_Before_Decay</td>
<td>float</td>
<td>0</td>
<td>45000</td>
<td>0</td>
<td>The number of days after infection until transmission-blocking immunity begins to decay. Only used when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> parameters are set to true (1).</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Transmission_Blocking_Immunity_Duration_Before_Decay&quot;</span><span class="p">:</span> <span class="mi">90</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Transmission_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.5</td>
<td>The probability that the bite of an infected mosquito establishes a new infection in an immunologically naive and uninfected individual, or the modifier of the probability of success for an individual with pre-erythrocytic immunity. Note that each mosquito species will have their own <strong>Transmission_Rate</strong> parameter.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span> <span class="p">:</span> <span class="mf">99.0</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span> <span class="p">:</span> <span class="mf">88.0</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span> <span class="p">:</span> <span class="mf">0.77</span><span class="p">,</span>
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span> <span class="p">:</span> <span class="mf">1.0</span><span class="p">,</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span><span class="p">,</span>
                <span class="nt">&quot;BRACKISH_SWAMP&quot;</span><span class="p">:</span> <span class="mi">10000000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Zoonosis_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The daily rate of zoonotic infection per individual.  <strong>Animal_Reservoir_Type</strong> must be set to CONSTANT_ZOONOSIS or ZOONOSIS_FROM_DEMOGRAPHICS. If  <strong>Animal_Reservoir_Type</strong> is set to ZOONOSIS_FROM_DEMOGRAPHICS, the value for the <strong>Zoonosis</strong> <strong>NodeAttribute</strong> in the demographics file will override the value set for <strong>Zoonosis_Rate</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Zoonosis_Rate&quot;</span><span class="p">:</span> <span class="mf">0.005</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="input-data-files">
<span id="config-input"></span><h2><a class="toc-backref" href="#id8">Input data files</a><a class="headerlink" href="#input-data-files" title="Permalink to this headline">¶</a></h2>
<p>The following parameters set the paths to the the <a class="reference internal" href="glossary.html#term-campaign-file"><span class="xref std std-term">campaign file</span></a> and the <a class="reference internal" href="glossary.html#term-input-data-files"><span class="xref std std-term">input data
files</span></a> for climate, migration, demographics, and load-balancing.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Air_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path to the input data file that defines patterns of migration by airplane. <strong>Enable_Air_Migration</strong> must be set to true (1). The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Air_Migration&quot;</span> <span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Air_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../Global_1degree_air_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Air_Temperature_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>air_temp.json</td>
<td>The path to the input data file that defines air temperature data measured two meters above ground. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Climate_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;CLIMATE_BY_DATA&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Air_Temperature_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_air_temperature_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Campaign_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path to the campaign file. It is required when interventions are part of the simulation and <strong>Enable_Interventions</strong> is set to true (1). The file must be in .json format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Interventions&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Campaign_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;campaign.json&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Demographics_Filenames</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>An array of the paths to demographics files containing information on the identity and demographics of the region to simulate. The files must be in .json format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
	<span class="nt">&quot;Demographics_Filenames&quot;</span><span class="p">:</span> <span class="p">[</span>    		
        <span class="s2">&quot;Namawala_single_node_demographics.json&quot;</span><span class="p">,</span>
	    <span class="s2">&quot;Namawala_demographics_overlay.json&quot;</span>
	<span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Family_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The name of the binary file to use to configure family migration. <strong>Enable_Family_Migration</strong> must be set to true (1). The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Family_Migration&quot;</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Family_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/family_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Koppen_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>The path to the input file used to specify Koppen climate classifications; only used  when <strong>Climate_Model</strong> is set to CLIMATE_KOPPEN. The file must be in .dat format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Koppen_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Mad_2_5arcminute_koppen.dat&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Land_Temperature_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>land_temp.json</td>
<td>The path of the input file defining temperature data measured at land surface; used only when <strong>Climate_Model</strong> is set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Land_Temperature_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_land_temperature_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Load_Balance_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>The path to the input file used when a static load balancing scheme is selected. The file must be in .json format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Load_Balance_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;GitHub_426_LoadBalance.json&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Local_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path of the input file which defines patterns of migration to adjacent nodes by foot travel. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Local_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Local_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Rainfall_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>rainfall.json</td>
<td>The path of the input file which defines rainfall data. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Rainfall_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_rainfall_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Regional_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path of the input file which defines patterns of migration by vehicle via road or rail network. If the node is not on a road or rail network, regional migration focuses on the closest hub city in the network. The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Regional_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Regional_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Relative_Humidity_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>rel_hum.json</td>
<td>The path of the input file which defines relative humidity data measured 2 meters above ground. <strong>Climate_Model</strong> must be set to CLIMATE_BY_DATA. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Relative_Humidity_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Namawala_single_node_relative_humidity_daily.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Sea_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path of the input file which defines patterns of migration by ship. Only used when <strong>Enable_Sea_Migration</strong> is set to true (1). The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sea_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;5x5_Households_Work_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Serialized_Population_Filenames</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>Array of filenames with serialized population data. The number of filenames must match the number of cores used for the simulation. The file must be in .dtk format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Serialized_Population_Filenames&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;state-00010.dtk&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Serialized_Population_Path</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>.</td>
<td>The root path for the serialized population files.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Serialized_Population_Path&quot;</span><span class="p">:</span> <span class="s2">&quot;../00_Generic_Version_1_save/output&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Filename_Local</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNSPECIFIED_FILE</td>
<td>The path to the vector migration file which defines patterns of vector migration to adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to 1. The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Filename_Local&quot;</span><span class="p">:</span> <span class="s2">&quot;5x5_Households_Local_Vector_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Filename_Regional</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNSPECIFIED_FILE</td>
<td>The path to the vector migration file which defines patterns of vector migration to non-adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to 1. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Filename_Regional&quot;</span><span class="p">:</span> <span class="s2">&quot;5x5_Households_Regional_Vector_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="larval-habitat">
<span id="config-larval"></span><h2><a class="toc-backref" href="#id9">Larval habitat</a><a class="headerlink" href="#larval-habitat" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine mosquito larval development related to habitat and climate. For
more information, see <a class="reference internal" href="malaria-larval-habitat.html"><span class="doc">Larval habitat</span></a>. Parameters for the vector life cycle more
broadly are described in <a class="reference internal" href="#config-vector-lifecycle"><span class="std std-ref">Vector life cycle</span></a>.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Aquatic_Arrhenius_1</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8.42E+10</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of mosquito aquatic development (egg-hatching through emergence). This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on development rate.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;aegypti&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mf">9752291.727</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Aquatic_Arrhenius_2</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8328</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of mosquito aquatic development (egg-hatching through emergence). This duration is a decreasing function of temperature.  The variable a2 governs how quickly the rate changes with temperature.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Aquatic_Mortality_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>The base aquatic mortality per day for the species before adjusting for effects of overpopulation and drying out of aquatic habitat.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Drought_Egg_Hatch_Delay</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.33</td>
<td>Proportion of regular egg hatching that still occurs when habitat dries up. <strong>Enable_Drought_Egg_Hatch_Delay</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
    <span class="nt">&quot;Enable_Drought_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Drought_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mf">0.33</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Egg_Arrhenius1</td>
<td>float</td>
<td>0</td>
<td>1.00E+10</td>
<td>6.16E+07</td>
<td>The Arrhenius equation, math:<cite>a_1^{-a_2/T}</cite>, with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on hatching rate. <strong>Enable_Temperature_Dependent_Egg_Hatching</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>     
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mi">61599956</span><span class="p">,</span>
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mi">5754</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Egg_Arrhenius2</td>
<td>float</td>
<td>0</td>
<td>1.00E+10</td>
<td>5754.03</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature.  The variable a2 is a temperature-dependent scale factor on hatching rate. <strong>Enable_Temperature_Dependent_Egg_Hatching</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>     
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mi">61599956</span><span class="p">,</span>
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mi">5754</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Egg_Hatch_Density_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_DENSITY_DEPENDENCE</td>
<td><p class="first">The effect of larval density on egg hatching rate. Possible values are:</p>
<dl class="last docutils">
<dt>DENSITY_DEPENDENCE </dt>
<dd>Egg hatching is reduced when the habitat is nearing its carrying capacity.</dd>
<dt>NO_DENSITY_DEPENDENCE</dt>
<dd>Egg hatching is not dependent upon larval density.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Egg_Hatch_Density_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;NO_DENSITY_DEPENDENCE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Egg_Saturation_At_Oviposition</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_SATURATION</td>
<td><p class="first">If laying all eggs from ovipositing females would overflow the larval habitat capacity on a given day, the means by which the viable eggs become saturated.  Possible values are:</p>
<dl class="last docutils">
<dt>NO_SATURATION</dt>
<dd>Egg number does not saturate; all eggs are laid.</dd>
<dt>SATURATION_AT_OVIPOSITION</dt>
<dd>Eggs are saturated at oviposition; habitat is filled to capacity and excess eggs are discarded.</dd>
<dt>SIGMOIDAL_SATURATION </dt>
<dd>Eggs are saturated along a sigmoidal curve; proportionately fewer eggs are laid depending on how oversubscribed the habitat would be.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Egg_Saturation_At_Oviposition&quot;</span><span class="p">:</span> <span class="s2">&quot;SATURATION_AT_OVIPOSITION&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Drought_Egg_Hatch_Delay</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not eggs hatch at delayed rates, set by <strong>Drought_Egg_Hatch_Delay</strong>, when the habitat dries up completely.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Drought_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Drought_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mf">0.33</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Egg_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not to include a daily mortality rate on the egg population, which is independent of climatic factors.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Egg_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Temperature_Dependent_Egg_Hatching</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not temperature has an effect on egg hatching, defined by <strong>Egg_Arrhenius_1</strong> and <strong>Egg_Arrhenius_2</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>  
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>  
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mf">61599956.864</span><span class="p">,</span> 
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mf">5754.033</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Larval_Density_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>UNIFORM_WHEN_OVERPOPULATION</td>
<td><p class="first">The functional form of mortality and growth delay for mosquito larvae based on population density.  
Possible values are:</p>
<dl class="last docutils">
<dt>UNIFORM_WHEN_OVERPOPULATION</dt>
<dd>Mortality is uniformly applied to all larvae when the population exceeds the specified carrying capacity for that habitat.</dd>
<dt>GRADUAL_INSTAR_SPECIFIC</dt>
<dd>Mortality and delayed growth are instar-specific, where the younger larvae are more susceptible to predation and competition from older larvae.</dd>
<dt>LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY</dt>
<dd>Mortality is based only on larval age.</dd>
<dt>DENSITY_DELAYED_GROWTH_NOT_MORTALITY</dt>
<dd>There is no mortality, only delayed growth in larvae.</dd>
<dt>NO_DENSITY_DEPENDENCE</dt>
<dd>There is no additional larval density-dependent mortality factor.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Density_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;GRADUAL_INSTAR_SPECIFIC&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Larval_Habitat_Types</td>
<td>JSON object</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td><p class="first">A measure of the habitat type and scale factors used to estimate larval population. This parameter is a dictionary that specifies habitat type with a simple numeric scale factor or, for LINEAR_SPLINE, with a more detailed configuration for scaling.  The numeric scaling value represents larval density with the number of larvae in a 1x1-degree area. The factor multiplicatively scales the resulting weather or population-dependent functional form. 
Possible habitat values are:</p>
<dl class="docutils">
<dt>CONSTANT</dt>
<dd>Larval carrying capacity is constant throughout the year and does not depend on weather. However, mosquito abundance will exhibit a seasonal signal due to the effects of temperature on aquatic developmental rates.</dd>
<dt>TEMPORARY_RAINFALL</dt>
<dd>This habitat type corresponds to temporary puddles which are replenished by rainfall and drained through evaporation and infiltration. Habitat availability is proportional to temperature and humidity, and decays over time as configured by the parameter <strong>Temporary_Habitat_Decay_Factor</strong>.</dd>
<dt>WATER_VEGATATION</dt>
<dd>This habitat type corresponds to semi-permanent habitats, such as developing vegetation on the edges of small water sources such as swamps or rice cultivation settings. Development of habitat lags behind rainfall, and typically peaks near the end of the rainy season. Habitat decays daily, as specified by the parameter <strong>Semipermanent_Habitat_Decay_Rate</strong>.</dd>
<dt>HUMAN_POPULATION</dt>
<dd>This habitat type scales with correlates of human development, such as available water in pots in urban areas. It is configured by multiplying the number of people in a node’s population times the capacity value set for this parameter, and is not climate-dependent.</dd>
<dt>BRACKISH_SWAMP</dt>
<dd>This habitat type deals with the dynamics of how rain fills brackish swamps, how this habitat availability decays, and includes a rainfall-driven mortality threshold. Habitat decay rate is specified by the parameter <strong>Semipermanent_Habitat_Decay_Rate</strong>.</dd>
<dt>LINEAR_SPLINE </dt>
<dd>The LINEAR_SPLINE configuration specifies the day of year, larval value, and larval capacity scaling number. The model linearly interpolates the values to estimate the habitat availability for each vector species without requiring climatological data.</dd>
</dl>
<p class="last">The values set here can be scaled per node using <strong>LarvalHabitatMultiplier</strong> in the demographics file (see NodeAttributes) or per intervention using <strong>Larval_Habitat_Multiplier</strong> in the campaign file (see ScaleLarvalHabitat).</p>
</td>
<td><p class="first">The following example shows how to specify larval habitat using LINEAR_SPLINE.</p>
<div class="highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;LINEAR_SPLINE&quot;</span><span class="p">:</span> <span class="p">{</span>
                    <span class="nt">&quot;Capacity_Distribution_Per_Year&quot;</span><span class="p">:</span> <span class="p">{</span>
                        <span class="nt">&quot;Times&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mf">0.0</span><span class="p">,</span> <span class="mf">60.833</span><span class="p">,</span> <span class="mf">121.667</span><span class="p">,</span> <span class="mf">182.5</span><span class="p">,</span> <span class="mf">243.333</span><span class="p">,</span> <span class="mf">204.167</span><span class="p">],</span>
                        <span class="nt">&quot;Values&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mf">0.0</span><span class="p">,</span> <span class="mf">0.0</span><span class="p">,</span> <span class="mf">0.2</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.0</span><span class="p">]</span>
                    <span class="p">},</span>
                    <span class="nt">&quot;Max_Larval_Capacity&quot;</span><span class="p">:</span> <span class="mf">10000000000.0</span>
                <span class="p">}</span>
            <span class="p">}</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
<p>The following example shows how to specify the larval habitat using climatological habitat types (you may use one or more for each vector species).</p>
<div class="last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span><span class="p">,</span>
            <span class="nt">&quot;WATER_VEGETATION&quot;</span><span class="p">:</span> <span class="mi">6000000000</span><span class="p">,</span>
            <span class="nt">&quot;BRACKISH_SWAMP&quot;</span><span class="p">:</span> <span class="mi">30000000</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Larval_Rainfall_Mortality_Threshold</td>
<td>float</td>
<td>0.01</td>
<td>1000</td>
<td>100</td>
<td>The threshold value on daily rainfall in millimeters, above which larval mortality is applied when <strong>Vector_Larval_Rainfall_Mortality</strong> is set to either SIGMOID or SIGMOID_HABITAT_SHIFTING.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Rainfall_Mortality_Threshold&quot;</span><span class="p">:</span> <span class="mf">30.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Rainfall_In_mm_To_Fill_Swamp</td>
<td>float</td>
<td>1</td>
<td>10000</td>
<td>1000</td>
<td>Millimeters of rain to fill larval habitat to capacity.  This is only used for vector species with <strong>Larval_Habitat_Types</strong> set to BRACKISH_SWAMP.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Rainfall_In_mm_To_Fill_Swamp&quot;</span><span class="p">:</span> <span class="mf">1000.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Semipermanent_Habitat_Decay_Rate</td>
<td>float</td>
<td>0.0001</td>
<td>100</td>
<td>0.01</td>
<td>Daily rate of larval habitat loss for semi-permanent habitats with <strong>Larval_Habitat_Types</strong> parameter value of WATER_VEGETATION or BRACKISH_SWAMP.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Semipermanent_Habitat_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Temporary_Habitat_Decay_Factor</td>
<td>float</td>
<td>0.001</td>
<td>100</td>
<td>0.05</td>
<td>The factor to convert raw evaporation rate (ignoring boundary layer effects) to the daily rate of larval habitat loss for temporary habitats (<strong>Larval_Habitat_Types</strong> set to TEMPORARY_RAINFALL). Units are (larval carrying capacity per day) / (kg per square meter per second).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temporary_Habitat_Decay_Factor&quot;</span><span class="p">:</span> <span class="mf">0.05</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Vector_Larval_Rainfall_Mortality</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONE</td>
<td><p class="first">The type of vector larval mortality function due to rainfall.  
Possible values are:</p>
<dl class="last docutils">
<dt>NONE</dt>
<dd>No larval mortality due to rainfall.</dd>
<dt>SIGMOID</dt>
<dd>The mortality rate grows linearly from 0 at the threshold to 1 at twice the threshold value.</dd>
<dt>SIGMOID_HABITAT_SHIFTING</dt>
<dd>The threshold value is reduced by a factor proportional to how full the larval habitat is.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Larval_Rainfall_Mortality&quot;</span><span class="p">:</span> <span class="s2">&quot;SIGMOID&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Habitat_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to migrate toward a node with more habitat. Only used when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Habitat_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Temporary_Larval_Habitat</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>1</td>
<td>Scale factor for the habitat size for all mosquito populations.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Temporary_Larval_Habitat&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="migration">
<span id="config-migration"></span><h2><a class="toc-backref" href="#id10">Migration</a><a class="headerlink" href="#migration" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine aspects of population migration into and outside of a <a class="reference internal" href="glossary.html#term-node"><span class="xref std std-term">node</span></a>,
including daily commutes, seasonal migration, and one-way moves. Modes of transport includes travel
by foot, automobile, sea, or air. Migration can also be configured to move all individuals in a
family at the same time.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Air_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path to the input data file that defines patterns of migration by airplane. <strong>Enable_Air_Migration</strong> must be set to true (1). The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Air_Migration&quot;</span> <span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Air_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../Global_1degree_air_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Air_Migration_Roundtrip_Duration</td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent (in days) at the destination node during a round-trip migration by airplane. <strong>Enable_Air_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Air_Migration&quot;</span> <span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Air_Migration_Roundtrip_Duration&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Air_Migration_Roundtrip_Probability</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.8</td>
<td>The likelihood that an individual who flies to another node will return to the node of origin during the next migration. <strong>Enable_Air_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Air_Migration&quot;</span> <span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Air_Migration_Roundtrip_Probability&quot;</span><span class="p">:</span> <span class="mf">0.9</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Air_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not migration by air travel will occur. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
     <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
     <span class="nt">&quot;Enable_Air_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
     <span class="nt">&quot;Air_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/air_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Family_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not all members of a household can migrate together. All residents must be home before they can leave on the trip. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Migration&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Family_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Family_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/family_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Local_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not local migration (the diffusion of people in and out of nearby nodes by foot travel) occurs. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Local_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Local_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/local_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Migration_Heterogeneity</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not migration rate is heterogeneous among individuals. Set to true (1) to use a migration rate distribution in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters); set to false (0) to use the same migration rate applied to all individuals. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Migration_Heterogeneity&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Regional_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not there is migration by road vehicle into and out of nodes in the road network. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Regional_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Regional_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/regional_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Sea_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not there is migration on ships into and out of coastal cities with seaports. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Sea_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Sea_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/sea_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vector_Migration</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors can migrate. <strong>Vector_Sampling_Type</strong> must be set to  TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. Specific migration types must be enabled or disabled.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration_Local&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Migration_Filename_Local&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/vector_local.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Vector_Migration_Local</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors may migrate to adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration_Local&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Migration_Filename_Local&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/vector_local.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vector_Migration_Regional</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors can migration to non-adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Vector_Migration_Regional&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Migration_Filename_Regional&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/vector_regional.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Family_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The name of the binary file to use to configure family migration. <strong>Enable_Family_Migration</strong> must be set to true (1). The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Family_Migration&quot;</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Family_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/family_migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Family_Migration_Roundtrip_Duration</td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The number of days to complete the trip and return to the original node. <strong>Migration_Pattern</strong> must be set to SINGLE_ROUND_TRIPS.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Migration_Pattern&quot;</span><span class="p">:</span> <span class="s2">&quot;SINGLE_ROUND_TRIPS&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Family_Migration_Roundtrip_Duration&quot;</span><span class="p">:</span> <span class="mi">100</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Local_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path of the input file which defines patterns of migration to adjacent nodes by foot travel. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Local_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Local_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Local_Migration_Roundtrip_Duration</td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent (in days) at the destination node during a round-trip migration by foot travel. Only used if <strong>Enable_Local_Migration</strong> is set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Local_Migration_Roundtrip_Duration&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Local_Migration_Roundtrip_Probability</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.95</td>
<td>The likelihood that an individual who walks into a neighboring cell will return to the cell of origin during the next migration. Only used when <strong>Enable_Local_Migration</strong> is set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Local_Migration_Roundtrip_Probability&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Migration_Model</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_MIGRATION</td>
<td><p class="first">Model to use for migration. Possible values are:</p>
<dl class="last docutils">
<dt>NO_MIGRATION</dt>
<dd>Migration into and out of nodes will not occur.</dd>
<dt>FIXED_RATE_MIGRATION</dt>
<dd>Migration into and out of nodes will occur at a fixed rate as defined in the migration files. At the beginning of the simulation or whenever an individual has just moved, they pick their next destination and the time and type of the migration. If an individual is on an outbound leg of their journey, they will query the node’s <strong>MigrationInfo</strong> object and, through probability, pick a new destination; if the individual is inbound, they will travel back to their previous location.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Local_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;../inputs/local_migration.bin&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Local_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Migration_Pattern</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>RANDOM_WALK_DIFFUSION</td>
<td><p class="first">Describes the type of roundtrip used during migration. <strong>Migration_Model</strong> must be set to FIXED_RATE_MIGRATION.
Possible values are:</p>
<dl class="last docutils">
<dt>RANDOM_WALK_DIFFUSION</dt>
<dd>Individuals retain no memory of where they came from; every move is to a new destination with no thought of returning home.</dd>
<dt>SINGLE_ROUND_TRIPS</dt>
<dd>There is a certain probability that an individual&#8217;s move will be a roundtrip (determined <strong>Local_Migration_Roundtrip_Probability</strong>, <strong>Air_Migration_Roundtrip_Probability</strong>, etc.). Otherwise, the departure point is forgotten and the individual does not return to their original location.</dd>
<dt>WAYPOINTS_HOME </dt>
<dd>Individuals go on a multi-step journey along several waypoints and then retrace their steps back along their path once they have reached a maximum number of waypoints from their home.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Migration_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_RATE_MIGRATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Migration_Pattern&quot;</span><span class="p">:</span> <span class="s2">&quot;SINGLE_ROUND_TRIPS&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Regional_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path of the input file which defines patterns of migration by vehicle via road or rail network. If the node is not on a road or rail network, regional migration focuses on the closest hub city in the network. The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Regional_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;Regional_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Regional_Migration_Roundtrip_Duration</td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent (in days) at the destination node during a round-trip migration by road network. <strong>Enable_Regional_Migration</strong> must be set to true (1).</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Regional_Migration_Roundtrip_Duration&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Regional_Migration_Roundtrip_Probability</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>The likelihood that an individual who travels by vehicle to another cell will return to the cell of origin during the next migration. <strong>Migration_Pattern</strong> must be set to SINGLE_ROUND_TRIPS.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Regional_Migration_Roundtrip_Probability&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Roundtrip_Waypoints</td>
<td>integer</td>
<td>0</td>
<td>1000</td>
<td>10</td>
<td>The maximum number of points reached during a trip before steps are retraced on the return trip home. <strong>Migration_Pattern</strong> must be set to WAYPOINTS_HOME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Roundtrip_Waypoints&quot;</span><span class="p">:</span> <span class="mi">5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Sea_Migration_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The path of the input file which defines patterns of migration by ship. Only used when <strong>Enable_Sea_Migration</strong> is set to true (1). The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sea_Migration_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;5x5_Households_Work_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Sea_Migration_Roundtrip_Duration</td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>1</td>
<td>The average time spent at the destination node during a round-trip migration by ship. Used only when <strong>Enable_Sea_Migration</strong> is set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sea_Migration_Roundtrip_Duration&quot;</span><span class="p">:</span> <span class="mi">10000</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Sea_Migration_Roundtrip_Probability</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.25</td>
<td>The likelihood that an individual who travels by ship into a neighboring cell will return to the cell of origin during the next migration. Used only when <strong>Enable_Sea_Migration</strong> is set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sea_Migration_Roundtrip_Probability&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Filename_Local</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNSPECIFIED_FILE</td>
<td>The path to the vector migration file which defines patterns of vector migration to adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to 1. The file must be in .bin format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Filename_Local&quot;</span><span class="p">:</span> <span class="s2">&quot;5x5_Households_Local_Vector_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Filename_Regional</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNSPECIFIED_FILE</td>
<td>The path to the vector migration file which defines patterns of vector migration to non-adjacent nodes. <strong>Enable_Vector_Migration</strong> must be set to 1. The file must be in .bin format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Filename_Regional&quot;</span><span class="p">:</span> <span class="s2">&quot;5x5_Households_Regional_Vector_Migration.bin&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Food_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to migrate toward a node currently occupied by humans, independent of the number of humans in the node. Used only when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Food_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Habitat_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to migrate toward a node with more habitat. Only used when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Habitat_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Modifier_Equation</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>LINEAR</td>
<td>The functional form of vector migration modifiers. <strong>Enable_Vector_Migration</strong> must be set to 1. 
Possible values are: 
LINEAR 
EXPONENTIAL</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Modifier_Equation&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Stay_Put_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to remain in its current node rather than migrate to another node. Used only when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Stay_Put_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>x_Air_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by air, as provided by the migration file. <strong>Enable_Air_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Air_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Family_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by families, as provided by the migration file. <strong>Enable_Family_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Family_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Local_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for rate of migration by foot travel, as provided by the migration file. <strong>Enable_Local_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Local_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Regional_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by road vehicle, as provided by the migration file. <strong>Enable_Regional_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Regional_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Sea_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by sea, as provided by the migration file. <strong>Enable_Sea_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Sea_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Vector_Migration_Local</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of vector migration to adjacent nodes, as provided by the vector migration file. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Vector_Migration_Local&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Vector_Migration_Regional</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of vector migration to non-adjacent nodes, as provided by the vector migration file. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Vector_Migration_Regional&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="mortality-and-survival">
<span id="config-mortality"></span><h2><a class="toc-backref" href="#id11">Mortality and survival</a><a class="headerlink" href="#mortality-and-survival" title="Permalink to this headline">¶</a></h2>
<p>The following parameter determine mortality and survival characteristics of the disease being modeled
and the population in general (non-disease mortality).</p>
<p>See <a class="reference internal" href="malaria-symptoms-diagnosis.html#malaria-severe-mortality"><span class="std std-ref">Severe disease and mortality</span></a> for more information about how fever, anemia, and
parasite counts lead to severe malaria and mortality.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Anemia_Mortality_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for anemia.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Anemia_Mortality_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">150</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Anemia_Mortality_Threshold</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>3</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for hemoglobin count in grams per deciliter (g/dL) at which 50% of individuals die per day.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Anemia_Mortality_Threshold&quot;</span><span class="p">:</span> <span class="mf">1.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Anemia_Severe_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for anemia.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>    
    <span class="nt">&quot;Anemia_Severe_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">20</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Anemia_Severe_Threshold</td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>5</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the severe disease threshold level for anemia. Threshold units are in grams per deciliter (g/dL).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Anemia_Severe_Threshold&quot;</span><span class="p">:</span> <span class="mf">3.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Base_Mortality</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td><p class="first">The base mortality of the infection before accounting for individual immune modification factors. Depending on the setting of <strong>Mortality_Time_Course</strong>, this is either the daily probability of the disease being fatal (DAILY_MORTALITY) or the probability of death at the end of the infection duration (MORTALITY_AFTER_INFECTIOUS). <strong>Enable_Vital_Dynamics</strong> must be set to true (1).</p>
<p class="last">This parameter is not used in this simulation type and any values set here will be overridden.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Mortality_Time_Course&quot;</span><span class="p">:</span> <span class="s2">&quot;DAILY_MORTALITY&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Base_Mortality&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Death_Rate_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONDISEASE_MORTALITY_OFF</td>
<td><p class="first">Determines how likely individuals are to die from natural, non-disease causes. <strong>Enable_Vital_Dynamics</strong> must be set to 1. Possible values are:</p>
<dl class="docutils">
<dt>NONDISEASE_MORTALITY_OFF</dt>
<dd>The daily mortality rate is 0, and no one dies from non-disease related causes.</dd>
<dt>NONDISEASE_MORTALITY_BY_AGE_AND_GENDER</dt>
<dd>The individual&#8217;s age and gender are taken into account to determine the daily mortality rate.</dd>
<dt>NONDISEASE_MORTALITY_BY_YEAR_AND_AGE_FOR_EACH_GENDER</dt>
<dd>Gender, age, and year, are all taken into account to determine the daily mortality rate.</dd>
</dl>
<p class="last">Properties, rates, and bin sizes can be set for non-disease mortality for each gender in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-complexdistro"><span class="std std-ref">Complex distributions</span></a> parameters).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Death_Rate_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;NONDISEASE_MORTALITY_OFF&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Fever_Mortality_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for fever.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Mortality_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">1000</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Fever_Mortality_Threshold</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>3</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the fever mortality threshold, in units of degrees Celsius above normal body temperature (defined as 37 C).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Mortality_Threshold&quot;</span><span class="p">:</span> <span class="mi">10</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Mortality_Blocking_Immunity_Decay_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.001</td>
<td>The rate at which mortality-blocking immunity decays after the mortality-blocking immunity offset period. <strong>Enable_Immune_Decay</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Mortality_Blocking_Immunity_Decay_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Mortality_Time_Course</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>DAILY_MORTALITY</td>
<td><p class="first">The method used to calculate disease deaths. 
Possible values are:</p>
<dl class="last docutils">
<dt>DAILY_MORTALITY </dt>
<dd>Calculated at every time step.</dd>
<dt>MORTALITY_AFTER_INFECTIOUS </dt>
<dd>Calculated once at the end of the disease duration.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Mortality_Time_Course&quot;</span><span class="p">:</span> <span class="s2">&quot;MORTALITY_AFTER_INFECTIOUS&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Other_Mortality</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for mortality from causes other than the disease being simulated, as provided by the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-complexdistro"><span class="std std-ref">Complex distributions</span></a> parameters). <strong>Enable_Vital_Dynamics</strong> must be set to 1.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Other_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="output-settings">
<span id="config-output"></span><h2><a class="toc-backref" href="#id12">Output settings</a><a class="headerlink" href="#output-settings" title="Permalink to this headline">¶</a></h2>
<p>The following parameters configure whether or not output reports are created for the simulation,
such as reports detailing spatial or demographic data at each time step. By default, the
<a class="reference internal" href="file-inset-chart.html"><span class="doc">Inset chart output report</span></a> is always created.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Custom_Reports_Filename</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>The name of the file containing custom report configuration parameters. Omitting this parameter or setting it to RunAllCustomReports will load all reporters found that are valid for the given simulation type. The file must be in JSON format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>  
    <span class="nt">&quot;Custom_Reports_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;custom_reports.json&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Default_Reporting</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not the default InsetChart.json report is created.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Default_Reporting&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Demographics_Reporting</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not demographic summary data and age-binned reports are outputted to file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Reporting&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Property_Output</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not to create property output reports, which detail groups as defined in <strong>IndividualProperties</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-properties"><span class="std std-ref">NodeProperties and IndividualProperties</span></a> parameters).  When there is more than one property type, the report will display the channel information for all combinations of the property type groups.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Property_Output&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Spatial_Output</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not spatial output reports are created. If set to true (1), spatial output reports include all channels listed in the parameter <strong>Spatial_Output_Channels</strong>.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Spatial output files require significant processing time and disk space.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Spatial_Output&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Spatial_Output_Channels&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;Prevalence&quot;</span><span class="p">,</span>
        <span class="s2">&quot;New_Infections&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Report_Event_Recorder</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Set to true (1) to enable or to false (0) to disable the ReportEventRecorder.csv output report.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Report_Event_Recorder&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Report_Event_Recorder_Events&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;VaccinatedA&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccineExpiredA&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccinatedB&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccineExpiredB&quot;</span>
    <span class="p">],</span> 
    <span class="nt">&quot;Report_Event_Recorder_Ignore_Events_In_List&quot;</span><span class="p">:</span> <span class="mi">0</span> 
<span class="p">}</span>			
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Report_Event_Recorder_Events</td>
<td>array</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>The list of events to include or exclude in the ReportEventRecorder.csv output report, based on how <strong>Report_Event_Recorder_Ignore_Events_In_List</strong> is set.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Report_Event_Recorder&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Report_Event_Recorder_Events&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;VaccinatedA&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccineExpiredA&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccinatedB&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccineExpiredB&quot;</span>
    <span class="p">],</span> 
    <span class="nt">&quot;Report_Event_Recorder_Ignore_Events_In_List&quot;</span><span class="p">:</span> <span class="mi">0</span> 
<span class="p">}</span>				
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Report_Event_Recorder_Ignore_Events_In_List</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">If set to false (0), only the events listed in the Report_Event_Recorder_Events array will be included in the ReportEventRecorder.csv output report. If set to true (1), only the events listed in the array will be excluded, and all other events will be included. If you want to return all events from the simulation, leave the events array empty.</p>
<table border="1" class="last docutils">
<colgroup>
<col width="20%" />
<col width="40%" />
<col width="40%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Value</th>
<th class="head">Events array</th>
<th class="head">Output file</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>0</td>
<td>No events</td>
<td>No events</td>
</tr>
<tr class="row-odd"><td>0</td>
<td>One or more events</td>
<td>Only the listed events.</td>
</tr>
<tr class="row-even"><td>1</td>
<td>No events</td>
<td>All events occurring in the simulation.</td>
</tr>
<tr class="row-odd"><td>1</td>
<td>One or more events</td>
<td>All simulation events occurring in the simulation, except for those listed.</td>
</tr>
</tbody>
</table>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Report_Event_Recorder&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Report_Event_Recorder_Events&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;VaccinatedA&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccineExpiredA&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccinatedB&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;VaccineExpiredB&quot;</span>
    <span class="p">],</span> 
    <span class="nt">&quot;Report_Event_Recorder_Ignore_Events_In_List&quot;</span><span class="p">:</span> <span class="mi">0</span> 
<span class="p">}</span>			
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Report_Event_Recorder_Individual_Properties</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>[]</td>
<td>Specifies an array of events that will be excluded from the property output report; all events NOT listed in the array will be included in the report. To report all events from the simulation, leave the events array empty.</td>
<td><p class="first">This example demonstrates reporting all individual property events:</p>
<div class="highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Report_Event_Recorder_Individual_Properties&quot;</span><span class="p">:</span> <span class="p">[]</span>
<span class="p">}</span>	
</pre></div>
</div>
<p>The following example demonstrates the syntax for excluding particular properties from the report:</p>
<div class="last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Report_Event_Recorder_Individual_Properties&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;Accessibility&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;Risk&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Spatial_Output_Channels</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>[]</td>
<td><p class="first">An array of channel names for spatial output by node and time step. The data from each channel will be written to a separate binary file. <strong>Enable_Spatial_Output</strong> must be set to true (1). Possible values are:</p>
<dl class="docutils">
<dt>Air_Temperature</dt>
<dd>Data related to air temperature.</dd>
<dt>Births</dt>
<dd>Data related to the number of births.</dd>
<dt>Campaign_Cost</dt>
<dd>Data related to the costs of a campaign.</dd>
<dt>Disease_Deaths</dt>
<dd>Data related to the number of deaths due to disease.</dd>
<dt>Human_Infectious_Reservoir</dt>
<dd>Data related to the total infectiousness of the population.</dd>
<dt>Infection_Rate</dt>
<dd>Data related to infection rates.</dd>
<dt>Land_Temperature</dt>
<dd>Data related to the average land temperature over all nodes.</dd>
<dt>New_Infections</dt>
<dd>Data related to the presence of new infections.</dd>
<dt>New_Reported_Infections</dt>
<dd>Data related to the presence of reported new infections.</dd>
<dt>Population</dt>
<dd>Data related to the total population in the simulation.</dd>
<dt>Prevalence</dt>
<dd>Data related to the fraction of the population that is infected.</dd>
<dt>Rainfall</dt>
<dd>Data related to the presence of rainfall.</dd>
<dt>Relative_Humidity</dt>
<dd>Data related to the presence of relative humidity.</dd>
</dl>
<dl class="docutils">
<dt>Fever_Prevalence</dt>
<dd>Data related to the presence of fever.</dd>
<dt>Mean_Parasitemia</dt>
<dd>Data related to the presence of mean parasitemia.</dd>
<dt>New_Clinical_Cases</dt>
<dd>Data related to the presence of new clinical cases.</dd>
<dt>New_Diagnostic_Prevalence</dt>
<dd>Data related to the presence of a new diagnostic.</dd>
<dt>New_Severe_Cases</dt>
<dd>Data related to the presence of new severe cases.</dd>
<dt>Parasite_Prevalence</dt>
<dd>Data related to the presence of parasites.</dd>
</dl>
<dl class="last docutils">
<dt>Adult_Vectors</dt>
<dd>Data related to the presence of adult vectors.</dd>
<dt>Daily_EIR</dt>
<dd>Data related to entomological inoculation rate (EIR) for mosquitoes.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Spatial_Output_Channels&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;Prevalence&quot;</span><span class="p">,</span>
        <span class="s2">&quot;New_Infections&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>        
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="parasite-dynamics">
<span id="config-parasite"></span><h2><a class="toc-backref" href="#id13">Parasite dynamics</a><a class="headerlink" href="#parasite-dynamics" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the dynamics of the <em>Plasmodium falciparum</em> parasite life cycle,
including dynamics within the host and human population. For more information, see
<a class="reference internal" href="malaria-infection-immune-model.html"><span class="doc">Malaria infection and immune model</span></a>.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Base_Gametocyte_Fraction_Male</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.2</td>
<td>Fraction of gametocytes that are male.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Gametocyte_Fraction_Male&quot;</span><span class="p">:</span> <span class="mf">0.25</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Base_Gametocyte_Mosquito_Survival_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.01</td>
<td>Average fraction of gametocytes in a blood meal that are successful in infecting the mosquito in the absence of other modulating effects, such as fever.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Gametocyte_Mosquito_Survival_Rate&quot;</span><span class="p">:</span> <span class="mf">0.015</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Base_Gametocyte_Production_Rate</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>0.02</td>
<td>The fraction of infected red blood cells (IRBCs) producing gametocytes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Gametocyte_Mosquito_Survival_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Base_Sporozoite_Survival_Fraction</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.25</td>
<td>The fraction of sporozoites that survive to infect a <a class="reference internal" href="glossary.html#term-hepatocyte"><span class="xref std std-term">hepatocyte</span></a> in the absence of anti-CSP protection.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Sporozoite_Survival_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.25</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Cytokine_Gametocyte_Inactivation</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.02</td>
<td>The strength of inflammatory response inactivation of gametocytes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Cytokine_Gametocyte_Inactivation&quot;</span><span class="p">:</span> <span class="mf">0.0167</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Sexual_Combination</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td><p class="first">Controls whether or not male and female gametocytes undergo sexual combination. Set to true (1) to allow for sexual combination; set to false (0) for transmission of each strain only.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This parameter is currently not in use.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Sexual_Combination&quot;</span><span class="p">:</span> <span class="mi">0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Gametocyte_Stage_Survival_Rate</td>
<td>double</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>The rate of gametocyte survival from one development stage to the next in the absence of drugs or inflammatory immune response.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Gametocyte_Stage_Survival_Rate&quot;</span><span class="p">:</span> <span class="mf">0.8</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Mean_Sporozoites_Per_Bite</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>11</td>
<td>The mean number of sporozoites per infectious mosquito bite.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Mean_Sporozoites_Per_Bite&quot;</span><span class="p">:</span> <span class="mi">11</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Merozoites_Per_Hepatocyte</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>15000</td>
<td>The number of IRBCs caused by a single infected <a class="reference internal" href="glossary.html#term-hepatocyte"><span class="xref std std-term">hepatocyte</span></a> at the start of infection.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Merozoites_Per_Hepatocyte&quot;</span><span class="p">:</span> <span class="mi">15000</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Merozoites_Per_Schizont</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>16</td>
<td>The number of merozoites released by a single infected <a class="reference internal" href="glossary.html#term-schizont"><span class="xref std std-term">schizont</span></a> after each asexual cycle. The number of resulting IRBC&#8217;s depends on the RBC availability and merozoite-specific immunity.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Merozoites_Per_Schizont&quot;</span><span class="p">:</span> <span class="mi">16</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Number_Of_Asexual_Cycles_Without_Gametocytes</td>
<td>integer</td>
<td>0</td>
<td>500</td>
<td>1</td>
<td>The number of asexual reproduction cycles that do not produce gametocytes. All later cycles will produce gametocytes according to <strong>Base_Gametocyte_Production_Rate</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Number_Of_Asexual_Cycles_Without_Gametocytes&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Parasite_Mortality_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) ).</p>
<p class="last">This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for parasite density.</p>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Parasite_Mortality_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">100</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Parasite_Mortality_Threshold</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2000000</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the parasite-density mortality threshold for a simulation.</p>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Parasite_Mortality_Threshold&quot;</span><span class="p">:</span> <span class="mi">3000000</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Parasite_Severe_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for parasite density.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Parasite_Severe_Inverse_Width&quot;</span><span class="p">:</span> <span class="mf">48.66825295</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Parasite_Severe_Threshold</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1000000</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures parasite density threshold level that results in severe disease.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Parasite_Severe_Threshold&quot;</span><span class="p">:</span> <span class="mi">253871</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Parasite_Smear_Sensitivity</td>
<td>float</td>
<td>0.0001</td>
<td>100</td>
<td>0.1</td>
<td>The number of microliters of blood tested to find single parasites in a traditional smear (corresponds to inverse parasites/microliters sensitivity).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Parasite_Smear_Sensitivity&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Parasite_Switch_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_SWITCH_RATE_2VARS</td>
<td><p class="first">The parasite switch type for erythrocyte surface antigens. 
Possible values are:</p>
<dl class="last docutils">
<dt>CONSTANT_SWITCH_RATE_2VARS</dt>
<dd>Introduction of new variants has a fixed probability which does not depend on the level of parasitemia.</dd>
<dt>RATE_PER_PARASITE_7VARS</dt>
<dd>Each IRBC has a fixed rate of switching to each of seven possible variants.</dd>
<dt>RATE_PER_PARASITE_5VARS_DECAYING</dt>
<dd>Each IRBC has a rate of switching to several possible variants, and the rate decreases with distance from the original variant.</dd>
</dl>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Parasite_Switch_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;RATE_PER_PARASITE_7VARS&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="population-dynamics">
<span id="config-population"></span><h2><a class="toc-backref" href="#id14">Population dynamics</a><a class="headerlink" href="#population-dynamics" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine characteristics related to population dynamics, such as age
distribution, births, deaths, and gender. The values set here generally interact closely with values
in the <a class="reference internal" href="glossary.html#term-demographics-file"><span class="xref std std-term">demographics file</span></a>.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Age_Initialization_Distribution_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>DISTRIBUTION_OFF</td>
<td><p class="first">The method for initializing the age distribution in the simulated population. Possible values are:</p>
<dl class="last docutils">
<dt>DISTRIBUTION_OFF</dt>
<dd>All individuals default to age 20 years old.</dd>
<dt>DISTRIBUTION_SIMPLE</dt>
<dd>Individual ages are drawn from a distribution whose functional form is specified in the demographics file using the <strong>IndividualAttributes</strong> simple <strong>Age</strong> distribution parameters.</dd>
<dt>DISTRIBUTION_COMPLEX</dt>
<dd>Individual ages are drawn from a piecewise linear function specified in the demographics file complex distribution parameters.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Age_Initialization_Distribution_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;DISTRIBUTION_SIMPLE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Base_Population_Scale_Factor</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>The scale factor for <strong>InitialPopulation</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters). If <strong>Population_Scale_Type</strong> is set to FIXED_SCALING, the initial simulation population is uniformly scaled over the entire area to adjust for historical or future population density.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Population_Scale_Factor&quot;</span><span class="p">:</span> <span class="mf">0.0001</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Birth_Rate_Boxcar_Forcing_Amplitude</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>Fractional increase in birth rate during high birth season when <strong>Birth_Rate_Time_Dependence</strong> is set to ANNUAL_BOXCAR_FUNCTION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Time_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Boxcar_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Birth_Rate_Boxcar_Forcing_End_Time</td>
<td>float</td>
<td>0</td>
<td>365</td>
<td>0</td>
<td>Day of the year when the high birth rate season ends when <strong>Birth_Rate_Time_Dependence</strong> is set to ANNUAL_BOXCAR_FUNCTION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Time_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Birth_Rate_Boxcar_Forcing_End_Time&quot;</span><span class="p">:</span> <span class="mi">220</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Birth_Rate_Boxcar_Forcing_Start_Time</td>
<td>float</td>
<td>0</td>
<td>365</td>
<td>0</td>
<td>Day of the year when the high birth rate season begins when <strong>Birth_Rate_Time_Dependence</strong> is set to ANNUAL_BOXCAR_FUNCTION.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Time_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Boxcar_Forcing_Start_Time&quot;</span><span class="p">:</span> <span class="mi">130</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Birth_Rate_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>FIXED_BIRTH_RATE</td>
<td><p class="first">The method used to modify the value set in <strong>BirthRate</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters). Possible values are:</p>
<dl class="last docutils">
<dt>NONE</dt>
<dd>Births are not allowed during the simulation, even if <strong>Enable_Birth</strong> is set to true (1).</dd>
<dt>FIXED_BIRTH_RATE</dt>
<dd>The absolute rate at which new individuals are born, as set by <strong>BirthRate</strong>.</dd>
<dt>POPULATION_DEP_RATE</dt>
<dd>Scales the node population to determine the birth rate. If <strong>BirthRate</strong> is greater than 0.005, a value of 2% per year (0.02/365) is used instead.</dd>
<dt>DEMOGRAPHIC_DEP_RATE</dt>
<dd>Scales the female population within fertility age ranges to determine the birth rate. If <strong>BirthRate</strong> is greater than 0.005, a value of 1 child every 8 years of fertility [1/8/365(~0.000342)] is used instead.</dd>
<dt>INDIVIDUAL_PREGNANCIES</dt>
<dd>Scales the female population within fertility age ranges to determine the birth rate, but pregnancies are assigned on an individual basis and result in a 40-week pregnancy for a specific individual with a birth at the end.</dd>
<dt>INDIVIDUAL_PREGNANCIES_BY_URBAN_AND_AGE</dt>
<dd>Similar to INDIVIDUAL_PREGNANCIES, but determines the rate based on the <strong>FertilityDistribution</strong> (in <strong>IndividualAttributes</strong>) and <strong>Urban</strong> (in <strong>NodeAttributes</strong>), as well as the individual&#8217;s age.</dd>
<dt>INDIVIDUAL_PREGNANCIES_BY_AGE_AND_YEAR</dt>
<dd>Similar to INDIVIDUAL_PREGNANCIES, but determines the rate based on the <strong>FertilityDistribution</strong> (in <strong>IndividualAttributes</strong>), using the individual&#8217;s age and the year of the simulation.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;POPULATION_DEP_RATE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Birth_Rate_Sinusoidal_Forcing_Amplitude</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The amplitude of sinusoidal variations in birth rate when <strong>Birth_Rate_Time_Dependence</strong> is set to SINUSOIDAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Time_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;SINUSOIDAL_FUNCTION_OF_TIME&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Sinusoidal_Forcing_Amplitude&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Birth_Rate_Sinusoidal_Forcing_Phase</td>
<td>float</td>
<td>0</td>
<td>365</td>
<td>0</td>
<td>The phase of sinusoidal variations in birth rate. <strong>Birth_Rate_Time_Dependence</strong> must be set to SINUSOIDAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Birth_Rate_Sinusoidal_Forcing_Phase&quot;</span><span class="p">:</span> <span class="mi">20</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Birth_Rate_Time_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONE</td>
<td><p class="first">A scale factor that allows the birth rate to be altered by time or season. <strong>Enable_Birth</strong> must be set to true (1). Possible values are:</p>
<dl class="last docutils">
<dt>NONE</dt>
<dd>Birth rate does not vary by time.</dd>
<dt>SINUSOIDAL_FUNCTION_OF_TIME</dt>
<dd>Allows birth rate to be time-dependent, following a sinusoidal shape. Set <strong>Birth_Rate_Sinusoidal_Forcing_Amplitude</strong> and <strong>Birth_Rate_Sinusoidal_Forcing_Phase</strong>.</dd>
<dt>ANNUAL_BOXCAR_FUNCTION</dt>
<dd>Allows birth rate to follow a boxcar function, such that it will be equal to zero for an entire time period (e.g. year) except for a single interval in which it is equal to a constant.  Set <strong>Birth_Rate_Boxcar_Forcing_Amplitude</strong>, <strong>Birth_Rate_Boxcar_Forcing_End_Time</strong>, and <strong>Birth_Rate_Boxcar_Forcing_Start_Time</strong>.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Time_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Death_Rate_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONDISEASE_MORTALITY_OFF</td>
<td><p class="first">Determines how likely individuals are to die from natural, non-disease causes. <strong>Enable_Vital_Dynamics</strong> must be set to 1. Possible values are:</p>
<dl class="docutils">
<dt>NONDISEASE_MORTALITY_OFF</dt>
<dd>The daily mortality rate is 0, and no one dies from non-disease related causes.</dd>
<dt>NONDISEASE_MORTALITY_BY_AGE_AND_GENDER</dt>
<dd>The individual&#8217;s age and gender are taken into account to determine the daily mortality rate.</dd>
<dt>NONDISEASE_MORTALITY_BY_YEAR_AND_AGE_FOR_EACH_GENDER</dt>
<dd>Gender, age, and year, are all taken into account to determine the daily mortality rate.</dd>
</dl>
<p class="last">Properties, rates, and bin sizes can be set for non-disease mortality for each gender in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-complexdistro"><span class="std std-ref">Complex distributions</span></a> parameters).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Death_Rate_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;NONDISEASE_MORTALITY_OFF&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Default_Geography_Initial_Node_Population</td>
<td>integer</td>
<td>0</td>
<td>1000000</td>
<td>1000</td>
<td>When using the built-in demographics for default geography, the initial number of individuals in each node. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. <strong>Enable_Demographics_Builtin</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Builtin&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Initial_Node_Population&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Torus_Size&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Demographics_Filenames</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>&nbsp;</td>
<td>An array of the paths to demographics files containing information on the identity and demographics of the region to simulate. The files must be in .json format.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
	<span class="nt">&quot;Demographics_Filenames&quot;</span><span class="p">:</span> <span class="p">[</span>    		
        <span class="s2">&quot;Namawala_single_node_demographics.json&quot;</span><span class="p">,</span>
	    <span class="s2">&quot;Namawala_demographics_overlay.json&quot;</span>
	<span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Aging</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not individuals in a population age during the simulation. <strong>Enable_Vital_Dynamics</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Aging&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Birth</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not individuals will be added to the simulation by birth. <strong>Enable_Vital_Dynamics</strong> must be set to true (1). If you want new individuals to have the same intervention coverage as existing individuals, you must add a <strong>BirthTriggeredIV</strong> to the campaign file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Demographics_Birth</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not newborns have identical or heterogeneous characteristics. Set to false (0) to give all newborns identical characteristics; set to true (1) to allow for heterogeneity in traits such as sickle-cell status. <strong>Enable_Birth</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Demographics_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Demographics_Builtin</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not built-in demographics for default geography will be used. Note that the built-in demographics feature does not represent a real geographical location and is mostly used for testing. Set to true (1) to define the initial population and number of nodes using <strong>Default_Geography_Initial_Node_Population</strong> and <strong>Default_Geography_Torus_Size</strong>. Set to false (0) to use demographics input files defined in <strong>Demographics_Filenames</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Builtin&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Initial_Node_Population&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span>
    <span class="nt">&quot;Default_Geography_Torus_Size&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Demographics_Gender</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not gender ratios are drawn from a Gaussian or 50/50 draw. Set to true (1) to create gender ratios drawn from a male/female ratio that is randomly smeared by a Gaussian of width 1%; set to false (0) to assign a gender ratio based on a 50/50 draw.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Gender&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Demographics_Other</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not other demographic factors are included in the simulation, such as the fraction of individuals above poverty, urban/rural characteristics, heterogeneous initial immunity, or risk. These factors are set in the demographics file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Demographics_Other&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Disease_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not individuals die due to disease.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Disease_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vital_Dynamics</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not births and deaths occur in the simulation. Births and deaths must be individually enabled and set.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Death_Rate_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;NONDISEASE_MORTALITY_OFF&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Base_Mortality&quot;</span><span class="p">:</span> <span class="mf">0.002</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Minimum_Adult_Age_Years</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>15</td>
<td>The age, in years, after which an individual is considered an adult. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Minimum_Adult_Age_Years&quot;</span><span class="p">:</span> <span class="mi">17</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Population_Density_Infectivity_Correction</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_INFECTIVITY</td>
<td><p class="first">Correction to alter infectivity by population density set in the <strong>Population_Density_C50</strong> parameter. Measured in people per square kilometer. 
Possible values are:</p>
<ul class="simple">
<li>CONSTANT_INFECTIVITY</li>
<li>SATURATING_FUNCTION_OF_DENSITY</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Sparsely populated areas have a lower infectivity, while densely populated areas have a higher infectivity, which rises to saturate at the <strong>Base_Infectivity</strong> value.</p>
</div>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Population_Density_Infectivity_Correction&quot;</span><span class="p">:</span> <span class="s2">&quot;SATURATING_FUNCTION_OF_DENSITY&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Population_Scale_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>USE_INPUT_FILE</td>
<td><p class="first">The method to use for scaling the initial population specified in the demographics input file.
Possible values are:</p>
<dl class="last docutils">
<dt>USE_INPUT_FILE</dt>
<dd>Turns off population scaling and uses <strong>InitialPopulation</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters).</dd>
<dt>FIXED_SCALING</dt>
<dd>Enables <strong>Base_Population_Scale_Factor</strong>.</dd>
</dl>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Population_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_SCALING&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>x_Birth</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for birth rate, as provided by the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters). <strong>Enable_Birth</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="sampling">
<span id="config-sampling"></span><h2><a class="toc-backref" href="#id15">Sampling</a><a class="headerlink" href="#sampling" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine how a population is sampled in the simulation. While you may
want every agent (individual object) to represent a single person, you can often optimize CPU
time with without degrading the accuracy of the simulation but having an agent represent
multiple people. The sampling rate may be adapted to have a higher or lower sampling rate for
particular regions or age groups.</p>
<p>For vector and malaria simulations, the same concepts apply to sampling the vector population.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Base_Individual_Sample_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>The base rate of sampling for individuals, equal to the fraction of individuals in each node being sampled. Reducing the sampling rate will reduce the time needed to run simulations. <strong>Individual_Sampling_Type</strong> must be set to FIXED_SAMPLING or ADAPTED_SAMPLING_BY_IMMUNE_STATE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Individual_Sample_Rate&quot;</span><span class="p">:</span> <span class="mf">0.01</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Immune_Threshold_For_Downsampling</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Threshold on acquisition immunity at which to apply immunity dependent downsampling. <strong>Individual_Sampling_Type</strong> must set to ADAPTED_SAMPLING_BY_IMMUNE_STATE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Individual_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ADAPTED_SAMPLING_BY_IMMUNE_STATE&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Immune_Threshold_For_Downsampling&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Individual_Sampling_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>TRACK_ALL</td>
<td><p class="first">The type of individual human sampling. Possible values are:</p>
<dl class="last docutils">
<dt>TRACK_ALL</dt>
<dd>Each person in the population is tracked as a single Individual object with a sampling weight of 1.</dd>
<dt>FIXED_SAMPLING</dt>
<dd>Members of the population are added to the simulation with a probability of <strong>Base_Individual_Sample_Rate</strong> and sampling weight of 1 / <strong>Base_Individual_Sample_Rate</strong>. This allows a smaller set of Individual objects to represent the overall population.</dd>
<dt>ADAPTED_SAMPLING_BY_POPULATION_SIZE</dt>
<dd>For each node, a maximum number of Individual objects is tracked (set in <strong>Max_Node_Population_Samples</strong>). If the population is smaller than this number, each person is tracked with a sampling rate of 1; if the population is larger, members of the population are added to the simulation with a probability of <strong>Max_Node_Population_Samples</strong> / population and sampling weight of population / <strong>Max_Node_Population_Samples</strong>. This setting is useful for simulations over large geographic areas with large heterogeneities in population density.</dd>
<dt>ADAPTED_SAMPLING_BY_AGE_GROUP</dt>
<dd>The population is binned by age and each age bin is sampled at a different rate as defined by <strong>Sample_Rate</strong> parameters. The setting is useful for diseases in which only infants or children are relevant by allowing full sampling of these age groups while using fewer objects to represent the rest of the population.</dd>
<dt>ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE</dt>
<dd>Each node has a maximum number of Individual objects as described in ADAPTED_SAMPLING_BY_POPULATION_SIZE, but when the population grows beyond that threshold, sampling is not done at a fixed rate, but varies by age as described in ADAPTED_SAMPLING_BY_AGE_GROUP.</dd>
<dt>ADAPTED_SAMPLING_BY_IMMUNE_STATE</dt>
<dd>Individuals who have acquired immunity are sampled at a lower rate as defined by <strong>Relative_Sample_Rate_Immune</strong> and <strong>Immune_Threshold_For_Downsampling</strong> parameters.</dd>
</dl>
</td>
<td><p class="first">The following example shows how to sampling immune individuals at a lower rate.</p>
<div class="highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Individual_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ADAPTED_SAMPLING_BY_IMMUNE_STATE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Immune_Threshold_For_Downsampling&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>  
    <span class="nt">&quot;Relative_Sample_Rate_Immune&quot;</span><span class="p">:</span> <span class="mf">0.1</span> 
<span class="p">}</span>
</pre></div>
</div>
<p>The following example shows how to sampling older individuals at a lower rate.</p>
<div class="last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Individual_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ADAPTED_SAMPLING_BY_AGE_GROUP&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Sample_Rate_0_18mo&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Sample_Rate_10_14&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
    <span class="nt">&quot;Sample_Rate_15_19&quot;</span><span class="p">:</span> <span class="mf">0.3</span><span class="p">,</span> 
    <span class="nt">&quot;Sample_Rate_18mo_4yr&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Sample_Rate_20_Plus&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
    <span class="nt">&quot;Sample_Rate_5_9&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
    <span class="nt">&quot;Sample_Rate_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Max_Node_Population_Samples</td>
<td>float</td>
<td>1</td>
<td>3.40E+38</td>
<td>30</td>
<td>The maximum number of individuals to track in a node. When the population exceeds this number, the sampling rate will drop according to the value set in <strong>Individual_Sampling_Type</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Individual_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;ADAPTED_SAMPLING_BY_POPULATION_SIZE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Max_Node_Population_Samples&quot;</span><span class="p">:</span> <span class="mi">100000</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Mosquito_Weight</td>
<td>integer</td>
<td>1</td>
<td>10000</td>
<td>1</td>
<td>The value indicating how many mosquitoes are represented by a sample mosquito in the simulation. <strong>Vector_Sampling_Type</strong> must be set to SAMPLE_IND_VECTORS.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Mosquito_Weight&quot;</span><span class="p">:</span> <span class="mi">10</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Relative_Sample_Rate_Immune</td>
<td>float</td>
<td>0.001</td>
<td>1</td>
<td>0.1</td>
<td>The relative sampling rate for people who have acquired immunity through recovery or vaccination.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Relative_Sample_Rate_Immune&quot;</span><span class="p">:</span> <span class="mf">0.1</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Sample_Rate_0_18mo</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for infants age 0 to 18 months. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_0_18mo&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Sample_Rate_10_14</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for children age 10 to 14 years. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_10_14&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Sample_Rate_15_19</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for adolescents age 15 to 19 years. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_15_19&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Sample_Rate_18mo_4yr</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for children age 18 months to 4 years. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_18mo_4yr&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Sample_Rate_20_plus</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for adults age 20 and older. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_20_plus&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Sample_Rate_5_9</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for children age 5 to 9 years. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_5_9&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Sample_Rate_Birth</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1</td>
<td>For age-adapted sampling, the relative sampling rate for births. <strong>Individual_Sampling_Type</strong> must be set to ADAPTED_SAMPLING_BY_AGE_GROUP or ADAPTED_SAMPLING_BY_AGE_GROUP_AND_POP_SIZE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Sample_Rate_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Vector_Sampling_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>TRACK_ALL_VECTORS</td>
<td><p class="first">The type of vector sampling used. 
Possible values are:</p>
<dl class="last docutils">
<dt>TRACK_ALL_VECTORS </dt>
<dd>Each vector in a node is part of the simulation as a separate object.</dd>
<dt>SAMPLE_IND_VECTORS</dt>
<dd>Each vector object in the simulation represents multiple vectors in the node population. <strong>Mosquito_Weight</strong> sets the number of mosquitos represented by each object.</dd>
<dt>VECTOR_COMPARTMENTS_NUMBER</dt>
<dd>Does not track every vector individually, but provides a compartment for every possible entry in state space and tracks the number of vectors in each compartment.  This successfully accounts for each vector, but does not distinguish between vectors that have identical states, just counts how many identical vectors of each state are present at a location.</dd>
<dt>VECTOR_COMPARTMENTS_PERCENT </dt>
<dd>Similar to VECTOR_COMPARTMENTS_NUMBER, but it only tracks the percentage of the population in each state.  Less useful, but included for comparison to other models.</dd>
</dl>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="scalars-and-multipliers">
<span id="config-scalars"></span><h2><a class="toc-backref" href="#id16">Scalars and multipliers</a><a class="headerlink" href="#scalars-and-multipliers" title="Permalink to this headline">¶</a></h2>
<p>The following parameters scale or multiply values set in other areas of the configuration file or
<a class="reference internal" href="glossary.html#term-input-data-files"><span class="xref std std-term">input data files</span></a>. This can be especially useful for understanding the sensitivities of
disease dynamics to input data without requiring modifications to those base values. For example,
one might set <strong>x_Birth</strong> to a value less than 1 to simulate a lower future birth rate due to increased
economic prosperity and available medical technology.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Acquire_Modifier</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Modifier of the probability of successful infection of a mosquito by a malaria-infected individual, given the individual&#8217;s infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;aegypti&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mi">1</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Antibody_IRBC_Kill_Rate</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>2</td>
<td>The scale factor multiplied by antibody level to produce the rate of clearance of the infected red blood cell (IRBC) population.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Antibody_IRBC_Kill_Rate&quot;</span><span class="p">:</span> <span class="mf">1.595</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Aquatic_Arrhenius_1</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8.42E+10</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of mosquito aquatic development (egg-hatching through emergence). This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on development rate.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;aegypti&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mf">9752291.727</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Aquatic_Arrhenius_2</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8328</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of mosquito aquatic development (egg-hatching through emergence). This duration is a decreasing function of temperature.  The variable a2 governs how quickly the rate changes with temperature.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Base_Population_Scale_Factor</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>The scale factor for <strong>InitialPopulation</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters). If <strong>Population_Scale_Type</strong> is set to FIXED_SCALING, the initial simulation population is uniformly scaled over the entire area to adjust for historical or future population density.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Population_Scale_Factor&quot;</span><span class="p">:</span> <span class="mf">0.0001</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Birth_Rate_Time_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NONE</td>
<td><p class="first">A scale factor that allows the birth rate to be altered by time or season. <strong>Enable_Birth</strong> must be set to true (1). Possible values are:</p>
<dl class="last docutils">
<dt>NONE</dt>
<dd>Birth rate does not vary by time.</dd>
<dt>SINUSOIDAL_FUNCTION_OF_TIME</dt>
<dd>Allows birth rate to be time-dependent, following a sinusoidal shape. Set <strong>Birth_Rate_Sinusoidal_Forcing_Amplitude</strong> and <strong>Birth_Rate_Sinusoidal_Forcing_Phase</strong>.</dd>
<dt>ANNUAL_BOXCAR_FUNCTION</dt>
<dd>Allows birth rate to follow a boxcar function, such that it will be equal to zero for an entire time period (e.g. year) except for a single interval in which it is equal to a constant.  Set <strong>Birth_Rate_Boxcar_Forcing_Amplitude</strong>, <strong>Birth_Rate_Boxcar_Forcing_End_Time</strong>, and <strong>Birth_Rate_Boxcar_Forcing_Start_Time</strong>.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vital_Dynamics&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Birth_Rate_Time_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;ANNUAL_BOXCAR_FUNCTION&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Cycle_Arrhenius_1</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>4.09E+10</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the mosquito feeding cycle rate. This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on feeding rate. <strong>Temperature_Dependent_Feeding_Cycle</strong> must be set to ARRHENIUS_DEPENDENCE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;ARRHENIUS_DEPENDENCE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">99</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">88</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Cycle_Arrhenius_2</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>7740</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the mosquito feeding cycle rate. This duration is a decreasing function of temperature.  The variable a2 is a temperature-independent scale factor on feeding rate. <strong>Temperature_Dependent_Feeding_Cycle</strong> must be set to ARRHENIUS_DEPENDENCE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;ARRHENIUS_DEPENDENCE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">99</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">88</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Cycle_Arrhenius_Reduction_Factor</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>The scale factor applied to cycle duration (from oviposition to oviposition) to reduce the duration when primary follicles are at stage II rather than I in the case of newly emerged females. <strong>Temperature_Dependent_Feeding_Cycle</strong> must be set to ARRHENIUS_DEPENDENCE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;ARRHENIUS_DEPENDENCE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;funestus&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mf">0.44</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Egg_Arrhenius1</td>
<td>float</td>
<td>0</td>
<td>1.00E+10</td>
<td>6.16E+07</td>
<td>The Arrhenius equation, math:<cite>a_1^{-a_2/T}</cite>, with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on hatching rate. <strong>Enable_Temperature_Dependent_Egg_Hatching</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>     
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mi">61599956</span><span class="p">,</span>
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mi">5754</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Egg_Arrhenius2</td>
<td>float</td>
<td>0</td>
<td>1.00E+10</td>
<td>5754.03</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature.  The variable a2 is a temperature-dependent scale factor on hatching rate. <strong>Enable_Temperature_Dependent_Egg_Hatching</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>     
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mi">61599956</span><span class="p">,</span>
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mi">5754</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Immunity_Acquisition_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0</td>
<td>The multiplicative reduction in the probability of reacquiring disease. Only used when <strong>Enable_Immunity</strong> and <strong>Enable_Immune_Decay</strong> are set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Immunity&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Enable_Immune_Decay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Immunity_Acquisition_Factor&quot;</span><span class="p">:</span> <span class="mf">0.9</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infected_Arrhenius_1</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>1.17E+11</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of infected mosquitoes to an infectious state. The duration of sporogony is a decreasing function of temperature. The variable a1 is a temperature-independent scale factor on the progression rate to infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infected_Arrhenius_2</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>8340</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of fractional progression of infected mosquitoes to an infectious state. The duration of sporogony is a decreasing function of temperature. The variable a2 is a temperature-dependent scale factor on the progression rate to infectiousness.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infected_Egg_Batch_Factor</td>
<td>float</td>
<td>0</td>
<td>10</td>
<td>0.8</td>
<td>The dimensionless factor used to modify mosquito egg batch size  in order to account for reduced fertility effects arising due to infection (e.g. when females undergo sporogony).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Infectious_Human_Feed_Mortality_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>1.5</td>
<td>The (dimensionless) factor used to modify the death rate of mosquitoes when feeding on humans, to account for the higher mortality rate infected mosquitoes experience during human feeds versus uninfected mosquitoes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infectivity_Exponential_Baseline</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The scale factor applied to <strong>Base_Infectivity</strong> at the beginning of a simulation, before the infectivity begins to grow exponentially. <strong>Infectivity_Scale_Type</strong> must be set to EXPONENTIAL_FUNCTION_OF_TIME.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Infectivity_Exponential_Baseline&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Exponential_Delay&quot;</span><span class="p">:</span> <span class="mi">90</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Exponential_Rate&quot;</span><span class="p">:</span> <span class="mi">45</span><span class="p">,</span> 
    <span class="nt">&quot;Infectivity_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_FUNCTION_OF_TIME&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Larval_Density_Mortality_Scalar</td>
<td>float</td>
<td>0.01</td>
<td>1000</td>
<td>10</td>
<td>A scale factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Density_Mortality_Scalar&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Newborn_Biting_Risk_Multiplier</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.2</td>
<td>The scale factor that defines the y-intercept of the linear portion of the biting risk curve when <strong>Age_Dependent_Biting_Risk_Type</strong> is set to LINEAR.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Newborn_Biting_Risk_Multiplier&quot;</span><span class="p">:</span> <span class="mf">0.2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Nonspecific_Antibody_Growth_Rate_Factor</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.5</td>
<td>The factor that adjusts <strong>Antibody_Capacity_Growth_Rate</strong> for less immunogenic surface proteins, called minor epitopes.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Nonspecific_Antibody_Growth_Rate_Factor&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Population_Scale_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>USE_INPUT_FILE</td>
<td><p class="first">The method to use for scaling the initial population specified in the demographics input file.
Possible values are:</p>
<dl class="last docutils">
<dt>USE_INPUT_FILE</dt>
<dd>Turns off population scaling and uses <strong>InitialPopulation</strong> in the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters).</dd>
<dt>FIXED_SCALING</dt>
<dd>Enables <strong>Base_Population_Scale_Factor</strong>.</dd>
</dl>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Population_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;FIXED_SCALING&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Susceptibility_Scale_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>CONSTANT_SUSCEPTIBILITY</td>
<td><p class="first">The effect of time or season on infectivity. 
Possible values are:</p>
<p class="last">CONSTANT_SUSCEPTIBILITY
LOG_LINEAR_FUNCTION_OF_TIME
LINEAR_FUNCTION_OF_AGE
LOG_LINEAR_FUNCTION_OF_AGE</p>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Susceptibility_Scale_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;CONSTANT_SUSCEPTIBILITY&quot;</span>
<span class="p">}</span>	
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Food_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to migrate toward a node currently occupied by humans, independent of the number of humans in the node. Used only when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Food_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Habitat_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to migrate toward a node with more habitat. Only used when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Habitat_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Modifier_Equation</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>LINEAR</td>
<td>The functional form of vector migration modifiers. <strong>Enable_Vector_Migration</strong> must be set to 1. 
Possible values are: 
LINEAR 
EXPONENTIAL</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Modifier_Equation&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Stay_Put_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to remain in its current node rather than migrate to another node. Used only when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Stay_Put_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>x_Air_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by air, as provided by the migration file. <strong>Enable_Air_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Air_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Birth</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for birth rate, as provided by the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-nodeattributes"><span class="std std-ref">NodeAttributes</span></a> parameters). <strong>Enable_Birth</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Birth&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Family_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by families, as provided by the migration file. <strong>Enable_Family_Migration</strong> must be set to true (1).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Family_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Local_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for rate of migration by foot travel, as provided by the migration file. <strong>Enable_Local_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Local_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Other_Mortality</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for mortality from causes other than the disease being simulated, as provided by the demographics file (see <a class="reference internal" href="parameter-demographics.html#demo-complexdistro"><span class="std std-ref">Complex distributions</span></a> parameters). <strong>Enable_Vital_Dynamics</strong> must be set to 1.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Other_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>x_Regional_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by road vehicle, as provided by the migration file. <strong>Enable_Regional_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Regional_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Sea_Migration</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of migration by sea, as provided by the migration file. <strong>Enable_Sea_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Sea_Migration&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Temporary_Larval_Habitat</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>1</td>
<td>Scale factor for the habitat size for all mosquito populations.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Temporary_Larval_Habitat&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>x_Vector_Migration_Local</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of vector migration to adjacent nodes, as provided by the vector migration file. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Vector_Migration_Local&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>x_Vector_Migration_Regional</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>1</td>
<td>Scale factor for the rate of vector migration to non-adjacent nodes, as provided by the vector migration file. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;x_Vector_Migration_Regional&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="simulation-setup">
<span id="config-setup"></span><h2><a class="toc-backref" href="#id17">Simulation setup</a><a class="headerlink" href="#simulation-setup" title="Permalink to this headline">¶</a></h2>
<p>These parameters determine the basic setup of a simulation including the type of simulation you are
running, such as “GENERIC_SIM” or “MALARIA_SIM”, the simulation duration, and the time step
duration.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Base_Year</td>
<td>float</td>
<td>1900</td>
<td>2200</td>
<td>2015</td>
<td><p class="first">The absolute time in years when the simulation begins. This can be combined with <strong>CampaignEventByYear</strong> to trigger campaign events.</p>
<p class="last">This parameter is not used in this simulation type.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Base_Year&quot;</span><span class="p">:</span> <span class="mi">1960</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Config_Name</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>UNINITIALIZED STRING</td>
<td>The optional, user-supplied title naming a configuration file.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
     <span class="nt">&quot;Config_Name&quot;</span><span class="p">:</span> <span class="s2">&quot;My_First_Config&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Interventions</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not campaign interventions will be used in the simulation. Set <strong>Campaign_Filename</strong> to the path of the file that contains the campaign interventions.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Interventions&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Campaign_Filename&quot;</span><span class="p">:</span> <span class="s2">&quot;campaign.json&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Listed_Events</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>[]</td>
<td>The list of valid, user-defined events that will be included in the campaign. Any event used in the campaign must either be one of the built-in events or in this list.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Listed_Events&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;Vaccinated&quot;</span><span class="p">,</span>
        <span class="s2">&quot;VaccineExpired&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Malaria_Model</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>MALARIA_MECHANISTIC_MODEL</td>
<td><p class="first">The type of malaria model used in a simulation. 
Possible values are:</p>
<ul class="simple">
<li>MALARIA_FIXED_DURATION</li>
<li>MALARIA_EXPONENTIAL_DURATION</li>
<li>MALARIA_REDUCEDSTATE_MODEL</li>
<li>MALARIA_MECHANISTIC_MODEL</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">All malaria simulations should use MALARIA_MECHANISTIC_MODEL. The others are placeholder values.</p>
</div>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Malaria_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;MALARIA_MECHANISTIC_MODEL&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Memory_Usage_Halting_Threshold_Working_Set_MB</td>
<td>integer</td>
<td>0</td>
<td>1.00E+06</td>
<td>8000</td>
<td>The maximum size (in MB) of working set memory before the system throws an exception and halts.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Memory_Usage_Halting_Threshold_Working_Set_MB&quot;</span><span class="p">:</span> <span class="mi">4000</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Memory_Usage_Warning_Threshold_Working_Set_MB</td>
<td>integer</td>
<td>0</td>
<td>1.00E+06</td>
<td>7000</td>
<td>The maximum size (in MB) of working set memory before memory usage statistics are written to the log regardless of log level.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Memory_Usage_Warning_Threshold_Working_Set_MB&quot;</span><span class="p">:</span> <span class="mi">3500</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Node_Grid_Size</td>
<td>float</td>
<td>0.00416</td>
<td>90</td>
<td>0.004167</td>
<td>The spatial resolution indicating the node grid size for a simulation in degrees.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Node_Grid_Size&quot;</span><span class="p">:</span> <span class="mf">0.042</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Number_Basestrains</td>
<td>integer</td>
<td>1</td>
<td>10</td>
<td>1</td>
<td>The number of base strains in the simulation, such as antigenic variants.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Number_Basestrains&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Num_Cores</td>
<td>integer</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>The number of cores used to run a simulation.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Num_Cores&quot;</span><span class="p">:</span> <span class="mi">4</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Run_Number</td>
<td>integer</td>
<td>0</td>
<td>2147480000</td>
<td>1</td>
<td>Sets the random number seed through a bit manipulation process. When running a multi-core simulation, combines with processor rank to produce independent random number streams for each process.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Run_Number&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Serialization_Time_Steps</td>
<td>array of integers</td>
<td>0</td>
<td>2.15E+09</td>
<td>&nbsp;</td>
<td>The list of time steps after which to save the serialized state. 0 (zero) indicates the initial state before simulation, n indicates after the nth time step. By default, no serialized state is saved.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Serialization_Time_Steps&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="mi">0</span><span class="p">,</span> 
        <span class="mi">10</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Serialized_Population_Filenames</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>Array of filenames with serialized population data. The number of filenames must match the number of cores used for the simulation. The file must be in .dtk format.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Serialized_Population_Filenames&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;state-00010.dtk&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Serialized_Population_Path</td>
<td>string</td>
<td>NA</td>
<td>NA</td>
<td>.</td>
<td>The root path for the serialized population files.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Serialized_Population_Path&quot;</span><span class="p">:</span> <span class="s2">&quot;../00_Generic_Version_1_save/output&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Simulation_Duration</td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>1</td>
<td>The elapsed time (in days) from the start to the end of a simulation.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Simulation_Duration&quot;</span><span class="p">:</span> <span class="mi">7300</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Simulation_Timestep</td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>1</td>
<td>The duration of a simulation time step, in days.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Simulation_Timestep&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>	
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Simulation_Type</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>GENERIC_SIM</td>
<td><p class="first">The type of disease being simulated. Possible IDM-supported values are:</p>
<ul class="last simple">
<li>GENERIC_SIM</li>
<li>VECTOR_SIM</li>
<li>MALARIA_SIM</li>
<li>TB_SIM</li>
<li>STI_SIM</li>
<li>HIV_SIM</li>
</ul>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Simulation_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;STI_SIM&quot;</span>
<span class="p">}</span>	
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Start_Time</td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>1</td>
<td><p class="first">The time, in days, when the simulation begins. This time is used to identify the starting values of the temporal input data, such as specifying which day’s climate values should be used for the first day of the simulation.</p>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <strong>Start_Day</strong> of campaign events is in absolute time, so time relative to the beginning of the simulation depends on this parameter.</p>
</div>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Start_Time&quot;</span><span class="p">:</span> <span class="mi">365</span>
<span class="p">}</span>	
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="symptoms-and-diagnosis">
<span id="config-symptoms"></span><h2><a class="toc-backref" href="#id18">Symptoms and diagnosis</a><a class="headerlink" href="#symptoms-and-diagnosis" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the characteristics of malaria diagnosis and symptoms as the
disease progresses, such as anemia and fever. See <a class="reference internal" href="malaria-symptoms-diagnosis.html"><span class="doc">Malaria symptoms and diagnostics</span></a> for more
information on malaria diagnostics and symptoms.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Anemia_Mortality_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for anemia.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Anemia_Mortality_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">150</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Anemia_Mortality_Threshold</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>3</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for hemoglobin count in grams per deciliter (g/dL) at which 50% of individuals die per day.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Anemia_Mortality_Threshold&quot;</span><span class="p">:</span> <span class="mf">1.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Anemia_Severe_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of severe disease turn-on around threshold for anemia.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>    
    <span class="nt">&quot;Anemia_Severe_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">20</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Anemia_Severe_Threshold</td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>5</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the severe disease threshold level for anemia. Threshold units are in grams per deciliter (g/dL).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Anemia_Severe_Threshold&quot;</span><span class="p">:</span> <span class="mf">3.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Clinical_Fever_Threshold_High</td>
<td>float</td>
<td>0</td>
<td>15</td>
<td>1</td>
<td>The fever threshold (in degrees Celsius above normal) to start a clinical incident.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Clinical_Fever_Threshold_High&quot;</span><span class="p">:</span> <span class="mf">1.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Clinical_Fever_Threshold_Low</td>
<td>float</td>
<td>0</td>
<td>5</td>
<td>1</td>
<td>The fever threshold (in degrees Celsius above normal) to end a clinical incident.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Clinical_Fever_Threshold_Low&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Erythropoiesis_Anemia_Effect</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>3.5</td>
<td>The exponential rate of increased red-blood-cell production from reduced red-blood-cell availability.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Erythropoiesis_Anemia_Effect&quot;</span><span class="p">:</span> <span class="mi">3</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Fever_Detection_Threshold</td>
<td>float</td>
<td>0.5</td>
<td>5</td>
<td>1</td>
<td>The level of body temperature above normal (defined as 37 C), in degrees Celsius, corresponding to detectable fever.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Detection_Threshold&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Fever_IRBC_Kill_Rate</td>
<td>float</td>
<td>0</td>
<td>1000</td>
<td>0.15</td>
<td>The maximum kill rate for infected red blood cells due to the inflammatory innate immune response. As fever increases above 38.5 degrees Celsius, the kill rate becomes successively higher along a sigmoidal curve approaching this rate.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
    <span class="nt">&quot;Fever_IRBC_Kill_Rate&quot;</span><span class="p">:</span> <span class="mf">1.4</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Fever_Mortality_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for fever.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Mortality_Inverse_Width&quot;</span><span class="p">:</span> <span class="mi">1000</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Fever_Mortality_Threshold</td>
<td>double</td>
<td>NA</td>
<td>NA</td>
<td>3</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the fever mortality threshold, in units of degrees Celsius above normal body temperature (defined as 37 C).</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Mortality_Threshold&quot;</span><span class="p">:</span> <span class="mi">10</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Fever_Severe_Inverse_Width</td>
<td>float</td>
<td>0.1</td>
<td>1000000</td>
<td>10</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the inverse width relative to threshold value of mortality turn-on around threshold for fever.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Severe_Inverse_Width&quot;</span><span class="p">:</span> <span class="mf">18.7</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Fever_Severe_Threshold</td>
<td>float</td>
<td>0</td>
<td>10</td>
<td>1.5</td>
<td><p class="first">The probability of having severe or fatal malaria is calculated according to three causes: anemia, fever as a pro-inflammatory correlate of cerebral malaria, and total parasite density. The probability associated with each factor is configured with two parameters of a sigmoidal function:</p>
<p>probability = 1.0 / ( 1.0 + exp( (threshold-variable) / (threshold/invwidth) ) )</p>
<p class="last">This parameter configures the threshold (in degrees Celsius above normal) for fever indicating severe disease.</p>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Fever_Severe_Threshold&quot;</span><span class="p">:</span> <span class="mf">2.5</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Min_Days_Between_Clinical_Incidents</td>
<td>float</td>
<td>0</td>
<td>1000000</td>
<td>3</td>
<td>The number of days with fever below the low-threshold before a new incident can start, when fever again exceeds the high-threshold.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Min_Days_Between_Clinical_Incidents&quot;</span><span class="p">:</span> <span class="mi">14</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>New_Diagnostic_Sensitivity</td>
<td>float</td>
<td>0.0001</td>
<td>100000</td>
<td>0.01</td>
<td>The number of microliters of blood tested to find single parasites in a new diagnostic (corresponds to inverse parasites/microliters sensitivity).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;New_Diagnostic_Sensitivity&quot;</span><span class="p">:</span> <span class="mf">0.025</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="vector-control">
<span id="config-vector-control"></span><h2><a class="toc-backref" href="#id19">Vector control</a><a class="headerlink" href="#vector-control" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the characteristics of campaign interventions aimed at vector
control, such as the <a class="reference internal" href="glossary.html#term-homing-endonuclease-gene-heg"><span class="xref std std-term">homing endonuclease gene (HEG)</span></a>, sugar feeding, and <a class="reference internal" href="glossary.html#term-wolbachia"><span class="xref std std-term">Wolbachia</span></a>
infection. When, to whom, and how those interventions are distributed is determined by the
<a class="reference internal" href="glossary.html#term-campaign-file"><span class="xref std std-term">campaign file</span></a>.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>HEG_Fecundity_Limiting</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The fractional reduction in the number of eggs laid by a female mosquito who is homozygous in the <a class="reference internal" href="glossary.html#term-homing-endonuclease-gene-heg"><span class="xref std std-term">homing endonuclease gene (HEG)</span></a>. <strong>HEG_Model</strong> must be set to GERMLINE_HOMING, EGG_HOMING, DUAL_GERMLINE_HOMING, or DRIVING_Y.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;HEG_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;EGG_HOMING&quot;</span><span class="p">,</span>
    <span class="nt">&quot;HEG_Fecundity_Limiting&quot;</span><span class="p">:</span> <span class="mf">0.7</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>HEG_Homing_Rate</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>The fractional redistribution from heterozygous offspring to the homozygous <a class="reference internal" href="glossary.html#term-homing-endonuclease-gene-heg"><span class="xref std std-term">homing endonuclease gene (HEG)</span></a> behavior. <strong>HEG_Model&#8221;</strong> must be set to GERMLINE_HOMING, EGG_HOMING, DUAL_GERMLINE_HOMING, or DRIVING_Y.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;HEG_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;EGG_HOMING&quot;</span><span class="p">,</span>
    <span class="nt">&quot;HEG_Homing_Rate&quot;</span><span class="p">:</span> <span class="mf">0.4</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>HEG_Model</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>OFF</td>
<td><p class="first">This parameter determines the structure of <a class="reference internal" href="glossary.html#term-homing-endonuclease-gene-heg"><span class="xref std std-term">homing endonuclease gene (HEG)</span></a> dynamics. Possible values are:</p>
<dl class="last docutils">
<dt>OFF</dt>
<dd>No HEG dynamics occur.</dd>
<dt>GERMLINE_HOMING</dt>
<dd>Female mosquito allele applies homing and then the fraction of wild/half/full HEG progeny are generated from combining with the male allele.</dd>
<dt>EGG_HOMING</dt>
<dd>The fraction of wild/half/full HEG progeny are calculated and then homing is applied to bias half towards full.</dd>
<dt>DUAL_GERMLINE_HOMING</dt>
<dd>Both male and female alleles apply germline bias before calculating progeny.</dd>
<dt>DRIVING_Y</dt>
<dd>Homing is applied as a bias of otherwise heterozygous progeny to be Y-drive male eggs.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;HEG_Model&quot;</span><span class="p">:</span> <span class="s2">&quot;EGG_HOMING&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;HEG_Homing_Rate&quot;</span><span class="p">:</span> <span class="mf">0.4</span><span class="p">,</span>
    <span class="nt">&quot;HEG_Fecundity_Limiting&quot;</span><span class="p">:</span> <span class="mf">0.9</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Vector_Sugar_Feeding_Frequency</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>VECTOR_SUGAR_FEEDING_NONE</td>
<td><p class="first">The frequency of sugar feeding by a female mosquito. Used is used in conjunction with the <strong>SugarTrap</strong> and <strong>OvipositionTrap</strong> interventions. <strong>Vector_Sampling_Type</strong> must be set to TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. Possible values are:</p>
<dl class="last docutils">
<dt>VECTOR_SUGAR_FEEDING_NONE</dt>
<dd>No sugar feeding.</dd>
<dt>VECTOR_SUGAR_FEEDING_ON_EMERGENCE_ONLY</dt>
<dd>Sugar feeding once at emergence.</dd>
<dt>VECTOR_SUGAR_FEEDING_EVERY_FEED</dt>
<dd>Sugar feeding occurs once per blood meal.</dd>
<dt>VECTOR_SUGAR_FEEDING_EVERY_DAY</dt>
<dd>Sugar feeding occurs every day.</dd>
</dl>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sugar_Feeding_Frequency&quot;</span><span class="p">:</span> <span class="s2">&quot;VECTOR_SUGAR_FEEDING_NONE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Wolbachia_Infection_Modification</td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>1</td>
<td>The change in vector susceptibility to infection due to a Wolbachia infection.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Wolbachia_Infection_Modification&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Wolbachia_Mortality_Modification</td>
<td>float</td>
<td>0</td>
<td>100</td>
<td>1</td>
<td>The change in vector mortality due to a Wolbachia infection.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Wolbachia_Infection_Modification&quot;</span><span class="p">:</span> <span class="mf">0.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="vector-life-cycle">
<span id="config-vector-lifecycle"></span><h2><a class="toc-backref" href="#id20">Vector life cycle</a><a class="headerlink" href="#vector-life-cycle" title="Permalink to this headline">¶</a></h2>
<p>The following parameters determine the characteristics of the vector life cycle. Set the vector species
to include in the simulation and the feeding, egg-laying, migration, and larval development habits
of each using these parameters. The parameters for larval development related to habitat and
climate are described in <a class="reference internal" href="#config-larval"><span class="std std-ref">Larval habitat</span></a>.</p>
<button class="toggle-button btn btn-info">Toggle parameter table</button><table border="1" class="toggle-table docutils">
<colgroup>
<col width="18%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="36%" />
<col width="9%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Parameter</th>
<th class="head">Data type</th>
<th class="head">Minimum</th>
<th class="head">Maximum</th>
<th class="head">Default</th>
<th class="head">Description</th>
<th class="head">Example</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Adult_Life_Expectancy</td>
<td>float</td>
<td>0</td>
<td>730</td>
<td>10</td>
<td>The number of days an adult mosquito survives. The daily adult mortality rate is 1 / (value of this parameter).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mf">5.9</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Anthropophily</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Mosquito preference for humans over animals, measured as a fraction of blood meals from human hosts. This dimensionless propensity is important in differentiating between mosquito species.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;aegypti&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Cycle_Arrhenius_1</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>4.09E+10</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the mosquito feeding cycle rate. This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on feeding rate. <strong>Temperature_Dependent_Feeding_Cycle</strong> must be set to ARRHENIUS_DEPENDENCE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;ARRHENIUS_DEPENDENCE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">99</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">88</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Cycle_Arrhenius_2</td>
<td>float</td>
<td>0</td>
<td>1.00E+15</td>
<td>7740</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the mosquito feeding cycle rate. This duration is a decreasing function of temperature.  The variable a2 is a temperature-independent scale factor on feeding rate. <strong>Temperature_Dependent_Feeding_Cycle</strong> must be set to ARRHENIUS_DEPENDENCE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;ARRHENIUS_DEPENDENCE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">99</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">88</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Cycle_Arrhenius_Reduction_Factor</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>The scale factor applied to cycle duration (from oviposition to oviposition) to reduce the duration when primary follicles are at stage II rather than I in the case of newly emerged females. <strong>Temperature_Dependent_Feeding_Cycle</strong> must be set to ARRHENIUS_DEPENDENCE.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;ARRHENIUS_DEPENDENCE&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;funestus&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mf">0.44</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Days_Between_Feeds</td>
<td>float</td>
<td>0</td>
<td>730</td>
<td>2</td>
<td>The number of days between female mosquito feeding attempts. In the cohort model, this value is converted to a feeding rate, equal to  1/(value of this parameter), and is used to determine the probability that one of each of the various feeding modes (animal host, human host, indoor, outdoor, etc) occurs. In the individual model, this parameter determines the number of days a vector waits between feeds.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;gambiae&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Drought_Egg_Hatch_Delay</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.33</td>
<td>Proportion of regular egg hatching that still occurs when habitat dries up. <strong>Enable_Drought_Egg_Hatch_Delay</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span> 
    <span class="nt">&quot;Enable_Drought_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="nt">&quot;Drought_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mf">0.33</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Egg_Arrhenius1</td>
<td>float</td>
<td>0</td>
<td>1.00E+10</td>
<td>6.16E+07</td>
<td>The Arrhenius equation, math:<cite>a_1^{-a_2/T}</cite>, with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature.  The variable a1 is a temperature-independent scale factor on hatching rate. <strong>Enable_Temperature_Dependent_Egg_Hatching</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>     
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mi">61599956</span><span class="p">,</span>
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mi">5754</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Egg_Arrhenius2</td>
<td>float</td>
<td>0</td>
<td>1.00E+10</td>
<td>5754.03</td>
<td>The Arrhenius equation, <img class="math" src="_images/math/1c4e98d68901ee4b4b9782c91fd5eb546328b6c0.png" alt="a_1^{-a_2/T}"/>, with T in degrees Kelvin, parameterizes the daily rate of mosquito egg hatching. This duration is a decreasing function of temperature.  The variable a2 is a temperature-dependent scale factor on hatching rate. <strong>Enable_Temperature_Dependent_Egg_Hatching</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>     
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mi">61599956</span><span class="p">,</span>
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mi">5754</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Egg_Batch_Size</td>
<td>float</td>
<td>0</td>
<td>10000</td>
<td>100</td>
<td>Number of eggs laid by one female mosquito that has fed successfully. The value of <strong>Egg_Batch_Size</strong> is used for both the number of female eggs and for the number of male eggs. For example, the default value of 100 means that there are 100 female eggs and 100 male eggs.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Egg_Hatch_Delay_Distribution</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_DELAY</td>
<td><p class="first">The distribution of the delay from oviposition to egg hatching. Possible values are:</p>
<dl class="last docutils">
<dt>NO_DELAY</dt>
<dd>No delay between oviposition and egg hatching.</dd>
<dt>EXPONENTIAL_DURATION</dt>
<dd>There is a delay that follows an exponential distribution. Set the mean delay in <strong>Mean_Egg_Hatch_Delay</strong>.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Egg_Hatch_Delay_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Mean_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Egg_Hatch_Density_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_DENSITY_DEPENDENCE</td>
<td><p class="first">The effect of larval density on egg hatching rate. Possible values are:</p>
<dl class="last docutils">
<dt>DENSITY_DEPENDENCE </dt>
<dd>Egg hatching is reduced when the habitat is nearing its carrying capacity.</dd>
<dt>NO_DENSITY_DEPENDENCE</dt>
<dd>Egg hatching is not dependent upon larval density.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Egg_Hatch_Density_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;NO_DENSITY_DEPENDENCE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Egg_Saturation_At_Oviposition</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_SATURATION</td>
<td><p class="first">If laying all eggs from ovipositing females would overflow the larval habitat capacity on a given day, the means by which the viable eggs become saturated.  Possible values are:</p>
<dl class="last docutils">
<dt>NO_SATURATION</dt>
<dd>Egg number does not saturate; all eggs are laid.</dd>
<dt>SATURATION_AT_OVIPOSITION</dt>
<dd>Eggs are saturated at oviposition; habitat is filled to capacity and excess eggs are discarded.</dd>
<dt>SIGMOIDAL_SATURATION </dt>
<dd>Eggs are saturated along a sigmoidal curve; proportionately fewer eggs are laid depending on how oversubscribed the habitat would be.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Egg_Saturation_At_Oviposition&quot;</span><span class="p">:</span> <span class="s2">&quot;SATURATION_AT_OVIPOSITION&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Egg_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not to include a daily mortality rate on the egg population, which is independent of climatic factors.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Egg_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Temperature_Dependent_Egg_Hatching</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not temperature has an effect on egg hatching, defined by <strong>Egg_Arrhenius_1</strong> and <strong>Egg_Arrhenius_2</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>  
    <span class="nt">&quot;Enable_Temperature_Dependent_Egg_Hatching&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>  
    <span class="nt">&quot;Egg_Arrhenius1&quot;</span><span class="p">:</span> <span class="mf">61599956.864</span><span class="p">,</span> 
    <span class="nt">&quot;Egg_Arrhenius2&quot;</span><span class="p">:</span> <span class="mf">5754.033</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Enable_Vector_Aging</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>Controls whether or not vectors undergo senescence as they age.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Enable_Vector_Aging&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Enable_Vector_Mortality</td>
<td>boolean</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>Controls whether or not vectors can die. <strong>Vector_Sampling_Type</strong> must be set to TRACK_ALL_VECTORS.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sampling_Type&quot;</span><span class="p">:</span> <span class="s2">&quot;TRACK_ALL_VECTORS&quot;</span><span class="p">,</span> 
    <span class="nt">&quot;Enable_Vector_Mortality&quot;</span><span class="p">:</span> <span class="mi">1</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Human_Feeding_Mortality</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>0.1</td>
<td>The fraction of mosquitoes, for all species, that die while feeding on a human.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Human_Feeding_Mortality&quot;</span><span class="p">:</span> <span class="mf">0.15</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Immature_Duration</td>
<td>float</td>
<td>0</td>
<td>730</td>
<td>2</td>
<td>The number of days for larvae to develop into adult mosquitoes. The value is used to calculate mosquito development rate, which equals 1 / (value of this parameter). Development from immature to adult is not dependent on temperature.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;funestus&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Indoor_Feeding_Fraction</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>The fraction (dimensionless) of feeds in which mosquitoes feed on humans indoors;  the fraction of feeds on humans that occur outdoors equals 1 – (value of this parameter).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span>
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span>
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span>
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span>
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span>
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span>
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span>
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span>
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span>
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span>
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span>
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span>
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span>
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Infected_Egg_Batch_Factor</td>
<td>float</td>
<td>0</td>
<td>10</td>
<td>0.8</td>
<td>The dimensionless factor used to modify mosquito egg batch size  in order to account for reduced fertility effects arising due to infection (e.g. when females undergo sporogony).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>	
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Larval_Density_Dependence</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>UNIFORM_WHEN_OVERPOPULATION</td>
<td><p class="first">The functional form of mortality and growth delay for mosquito larvae based on population density.  
Possible values are:</p>
<dl class="last docutils">
<dt>UNIFORM_WHEN_OVERPOPULATION</dt>
<dd>Mortality is uniformly applied to all larvae when the population exceeds the specified carrying capacity for that habitat.</dd>
<dt>GRADUAL_INSTAR_SPECIFIC</dt>
<dd>Mortality and delayed growth are instar-specific, where the younger larvae are more susceptible to predation and competition from older larvae.</dd>
<dt>LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY</dt>
<dd>Mortality is based only on larval age.</dd>
<dt>DENSITY_DELAYED_GROWTH_NOT_MORTALITY</dt>
<dd>There is no mortality, only delayed growth in larvae.</dd>
<dt>NO_DENSITY_DEPENDENCE</dt>
<dd>There is no additional larval density-dependent mortality factor.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Density_Dependence&quot;</span><span class="p">:</span> <span class="s2">&quot;GRADUAL_INSTAR_SPECIFIC&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Larval_Density_Mortality_Offset</td>
<td>float</td>
<td>0.0001</td>
<td>1000</td>
<td>0.1</td>
<td>An offset factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Density_Mortality_Offset&quot;</span><span class="p">:</span> <span class="mf">0.001</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Larval_Density_Mortality_Scalar</td>
<td>float</td>
<td>0.01</td>
<td>1000</td>
<td>10</td>
<td>A scale factor in the formula determining the larval-age-dependent mortality for the GRADUAL_INSTAR_SPECIFIC and LARVAL_AGE_DENSITY_DEPENDENT_MORTALITY_ONLY models.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Larval_Density_Mortality_Scalar&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Mean_Egg_Hatch_Delay</td>
<td>float</td>
<td>0</td>
<td>120</td>
<td>0</td>
<td>The mean delay in egg hatch time, in days, from the time of oviposition. Set the delay distribution with <strong>Egg_Hatch_Delay_Distribution</strong>.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Egg_Hatch_Delay_Distribution&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL_DURATION&quot;</span><span class="p">,</span>
    <span class="nt">&quot;Mean_Egg_Hatch_Delay&quot;</span><span class="p">:</span> <span class="mi">2</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Nighttime_Feeding_Fraction</td>
<td>float</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>The fraction of feeds on humans (for a specific mosquito species) that occur during the nighttime. Thus the fraction of feeds on humans that occur during the day equals 1 - (value of this parameter).</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mf">0.2</span><span class="p">,</span> 
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span> 
            <span class="nt">&quot;Anthropophily&quot;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span> 
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Cycle_Arrhenius_Reduction_Factor&quot;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span> 
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> 
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span> 
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span> 
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span> 
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span> 
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span> 
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mi">11250000000</span>
            <span class="p">},</span> 
            <span class="nt">&quot;Nighttime_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> 
            <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>   
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Temperature_Dependent_Feeding_Cycle</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>NO_TEMPERATURE_DEPENDENCE</td>
<td><p class="first">The effect of temperature on the duration between blood feeds. 
Possible values are:</p>
<dl class="last docutils">
<dt>NO_TEMPERATURE_DEPENDENCE </dt>
<dd>No relation to temperature; days between feeds will be constant and specified by <strong>Days_Between_Feeds</strong> for each species.</dd>
<dt>ARRHENIUS_DEPENDENCE</dt>
<dd>Use the Arrhenius equation with parameters <strong>Cycle_Arrhenius_1</strong> and <strong>Cycle_Arrhenius_2</strong>.</dd>
<dt>BOUNDED_DEPENDENCE</dt>
<dd>Use an equation bounded at 10 days at 15C and use <strong>Days_Between_Feeds</strong> to set the duration at 30C.</dd>
</dl>
</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Temperature_Dependent_Feeding_Cycle&quot;</span><span class="p">:</span> <span class="s2">&quot;BOUNDED_DEPENDENCE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Food_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to migrate toward a node currently occupied by humans, independent of the number of humans in the node. Used only when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Food_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Migration_Modifier_Equation</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>LINEAR</td>
<td>The functional form of vector migration modifiers. <strong>Enable_Vector_Migration</strong> must be set to 1. 
Possible values are: 
LINEAR 
EXPONENTIAL</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Modifier_Equation&quot;</span><span class="p">:</span> <span class="s2">&quot;EXPONENTIAL&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Vector_Migration_Stay_Put_Modifier</td>
<td>float</td>
<td>0</td>
<td>3.40E+38</td>
<td>0</td>
<td>The preference of a vector to remain in its current node rather than migrate to another node. Used only when <strong>Vector_Sampling_Type</strong> is set to TRACK_ALL_VECTORS. <strong>Enable_Vector_Migration</strong> must be set to 1.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Migration_Stay_Put_Modifier&quot;</span><span class="p">:</span> <span class="mf">1.0</span>
<span class="p">}</span>
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-even"><td>Vector_Species_Names</td>
<td>array of strings</td>
<td>NA</td>
<td>NA</td>
<td>[]</td>
<td>An array of vector species names for all species that are present in the simulation. You may provide any species names you desire, but the same names must be used in <strong>Vector_Species_Params</strong>.</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>   
    <span class="nt">&quot;Vector_Species_Names&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="s2">&quot;arabiensis&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;funestus&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;gambiae&quot;</span>
    <span class="p">]</span>   
<span class="p">}</span>   
</pre></div>
</div>
<p class="last"></p>
</td>
</tr>
<tr class="row-odd"><td>Vector_Species_Params</td>
<td>JSON object</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>A JSON object that lists each vector species, as defined in <strong>Vector_Species_Names</strong>, and the parameters that define them.</td>
<td><div class="first last highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Species_Params&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="nt">&quot;arabiensis&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="nt">&quot;Acquire_Modifier&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
            <span class="nt">&quot;Adult_Life_Expectancy&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span>
            <span class="nt">&quot;Anthropophily&quot;</span>
            <span class="s2">&quot;Aquatic_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">84200000000</span><span class="p">,</span>
            <span class="nt">&quot;Aquatic_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8328</span><span class="p">,</span>
            <span class="nt">&quot;Aquatic_Mortality_Rate&quot;</span><span class="p">:</span> <span class="mf">0.1</span><span class="p">,</span>
            <span class="nt">&quot;Days_Between_Feeds&quot;</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span>
            <span class="nt">&quot;Egg_Batch_Size&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span>
            <span class="nt">&quot;Immature_Duration&quot;</span><span class="p">:</span> <span class="mi">4</span><span class="p">,</span>
            <span class="nt">&quot;Indoor_Feeding_Fraction&quot;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
            <span class="nt">&quot;Infected_Arrhenius_1&quot;</span><span class="p">:</span> <span class="mi">117000000000</span><span class="p">,</span>
            <span class="nt">&quot;Infected_Arrhenius_2&quot;</span><span class="p">:</span> <span class="mi">8336</span><span class="p">,</span>
            <span class="nt">&quot;Infected_Egg_Batch_Factor&quot;</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">,</span>
            <span class="nt">&quot;Infectious_Human_Feed_Mortality_Factor&quot;</span><span class="p">:</span> <span class="mf">1.5</span><span class="p">,</span>
            <span class="nt">&quot;Larval_Habitat_Types&quot;</span><span class="p">:</span> <span class="p">{</span>
                <span class="nt">&quot;TEMPORARY_RAINFALL&quot;</span><span class="p">:</span> <span class="mf">1e10</span><span class="p">,</span>
                <span class="nt">&quot;WATER_VEGETATION&quot;</span><span class="p">:</span> <span class="mf">1e6</span>   
            <span class="p">},</span>
        <span class="nt">&quot;Transmission_Rate&quot;</span><span class="p">:</span> <span class="mf">0.5</span>
    <span class="p">}</span>   
<span class="p">}</span>
</pre></div>
</div>
</td>
</tr>
<tr class="row-even"><td>Vector_Sugar_Feeding_Frequency</td>
<td>enum</td>
<td>NA</td>
<td>NA</td>
<td>VECTOR_SUGAR_FEEDING_NONE</td>
<td><p class="first">The frequency of sugar feeding by a female mosquito. Used is used in conjunction with the <strong>SugarTrap</strong> and <strong>OvipositionTrap</strong> interventions. <strong>Vector_Sampling_Type</strong> must be set to TRACK_ALL_VECTORS or SAMPLE_IND_VECTORS. Possible values are:</p>
<dl class="last docutils">
<dt>VECTOR_SUGAR_FEEDING_NONE</dt>
<dd>No sugar feeding.</dd>
<dt>VECTOR_SUGAR_FEEDING_ON_EMERGENCE_ONLY</dt>
<dd>Sugar feeding once at emergence.</dd>
<dt>VECTOR_SUGAR_FEEDING_EVERY_FEED</dt>
<dd>Sugar feeding occurs once per blood meal.</dd>
<dt>VECTOR_SUGAR_FEEDING_EVERY_DAY</dt>
<dd>Sugar feeding occurs every day.</dd>
</dl>
</td>
<td><div class="first highlight-json"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="nt">&quot;Vector_Sugar_Feeding_Frequency&quot;</span><span class="p">:</span> <span class="s2">&quot;VECTOR_SUGAR_FEEDING_NONE&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p></p>
<p class="last"></p>
</td>
</tr>
</tbody>
</table>
</div>
</div>


<div id="tipue_search_content"></div>

           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="parameter-campaign.html" class="btn btn-neutral float-right" title="Campaign parameters" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="parameter-demographics.html" class="btn btn-neutral" title="Demographics parameters" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Intellectual Ventures Management, LLC (IVM). All rights reserved.

    </p>
  </div>
<p>EMOD is licensed under the <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/legalcode">Creative Commons Attribution-Noncommercial-ShareAlike 4.0 License</a>.</p>
<p>Send documentation feedback to <a href="mailto:feedback@idmod.org" target="_top">feedback@idmod.org</a>.</p>


</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/tipuesearch/tipuesearch.js"></script>
      <script type="text/javascript" src="_static/tipuesearch/tipuesearch_content.js"></script>
      <script type="text/javascript" src="_static/tipuesearch/tipuesearch_set.js"></script>
      <script type="text/javascript" src="_static/runsearch.js"></script>
      <script type="text/javascript" src="_static/tablecollapse.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
  
<!-- Google analytics for IDM docs -->
 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-87656833-3', 'auto');
  ga('send', 'pageview');

</script>



</body>
</html>